{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c509222c-61b7-4039-a79b-61c8e831934a",
   "metadata": {},
   "source": [
    "# Calculate enhancer、eRNA、EPI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "af8d351e-a5a0-40b2-852c-0b795cea6c5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('display.max_columns', None)\n",
    "import subprocess\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import os\n",
    "import pyBigWig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "460152cb-6ca0-4309-9d3f-720aa21b370d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fuction define. if eRNA intersect with promoter, return False. else return True\n",
    "def ifFilter(row):\n",
    "    try:\n",
    "        if (row['eRNA_start'] >= row['promoter_start']) & (row['eRNA_start'] <= row['promoter_end']):\n",
    "            return True\n",
    "        elif (row['eRNA_end'] >= row['promoter_start']) & (row['eRNA_end'] <= row['promoter_end']):\n",
    "            return True\n",
    "        else:\n",
    "            return False\n",
    "    except:\n",
    "        try:\n",
    "            if (row['enhancer_start'] >= row['promoter_start']) & (row['enhancer_start'] <= row['promoter_end']):\n",
    "                return True\n",
    "            elif (row['enhancer_end'] >= row['promoter_start']) & (row['enhancer_end'] <= row['promoter_end']):\n",
    "                return True\n",
    "            else:\n",
    "                return False\n",
    "        except:\n",
    "            raise KeyError"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "daa7ce5b-333e-4926-84f0-14bf444dbc9f",
   "metadata": {},
   "source": [
    "## enhancer identification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fae1e895-e563-4404-bb00-0c869e354d48",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # sequence data download\n",
    "# cmd = f\"bash download.sh\"\n",
    "# subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85b4fb56-9db1-40b7-b2cf-ba70e780edd8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2020264 sequences (100000094 bp)...\n",
      "[M::process] read 2023230 sequences (100000020 bp)...\n",
      "[M::mem_process_seqs] Processed 2020264 reads in 280.157 CPU sec, 31.736 real sec\n",
      "[M::process] read 2021940 sequences (100000042 bp)...\n"
     ]
    }
   ],
   "source": [
    "# # Align from original reads\n",
    "# cmd = f\"bash align.sh 10 /home/limh/Reference_Genome/Gallus/BWA/galGal6.fa\"\n",
    "# subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b08002fb-63e0-4f2e-b114-6b5cbff9da83",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: CPM\n",
      "normalization: CPM\n",
      "bamFilesList: ['H3K4me1_r1_sort_rmdup.bam', 'H3K4me1_r1_sort_rmdup.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: 0\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: CPM\n",
      "normalization: CPM\n",
      "bamFilesList: ['H3K4me3_r1_sort_rmdup.bam', 'H3K4me3_r2_sort_rmdup.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: 0\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    }
   ],
   "source": [
    "# Histone modification ChIP-seq call peak\n",
    "files = ['H3K27ac_r1_sort_rmdup.bam', 'H3K27ac_r2_sort_rmdup.bam', 'H3K4me1_r1_sort_rmdup.bam']\n",
    "for file in files:\n",
    "    inputFile = 'input_r1_sort_rmdup.bam' if 'r1' in file else 'input_r2_sort_rmdup.bam'\n",
    "    cmd = f\"sicer -t {file} -c {inputFile} -s galGal6 -fdr 0.05 -cpu 20 -o .\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    \n",
    "# Get standard-format broadPeak file from sicer result\n",
    "files = [x for x in os.listdir() if (x.endswith('-W200-G600.scoreisland') and 'H3K' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df.insert(3, 'peakName', [f'peak{x}' for x in range(1, df.shape[0]+1)])\n",
    "    df['strand'] = '.'\n",
    "    df['signalValue'] = df[3]\n",
    "    df['pValue'] = df[3]\n",
    "    df['qValue'] = df[3]\n",
    "    df.to_csv(file.replace('-W200-G600.scoreisland', '.broadPeak'), sep='\\t', header=None, index=False)\n",
    "    \n",
    "# Use idr to combine different replicates\n",
    "files = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'H3K' in x and 'r1' in x)]\n",
    "for file in files:\n",
    "    file_rep2 = file.replace('r1', 'r2')\n",
    "    outfile = file.replace('r1',  'idr')\n",
    "    cmd = f'''idr --samples {file} {file_rep2} --input-file-type broadPeak \\\n",
    "        --rank q.value --output-file {outfile} --output-file-type broadPeak \\\n",
    "            --plot --log-output-file {outfile}.log'''\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    \n",
    "# Filter idr result with threshold as 0.1\n",
    "files = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'H3K' in x and 'idr' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df = df[df[4]> int(-125 *  np.log2(0.1))]\n",
    "    df.to_csv(file.replace('.broadPeak', '_filter.broadPeak'), sep='\\t', header=None, index=False)\n",
    "\n",
    "# H3K4me1 H3K27ac intersection calculation\n",
    "H3K4me1_file = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'H3K4me1' in x)][0]\n",
    "H3K27ac_file = [x for x in os.listdir() if (x.endswith('_filter.broadPeak') and 'H3K27ac' in x and 'idr' in x)][0]\n",
    "cmd = f\"bedtools intersect -a {H3K4me1_file} -b {H3K27ac_file} > H3K4me1_H3K27ac_intersect.bed\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "# Retain H3K4me1 H3K27ac intersection part which show lower H3K4me3 histone modification signal. Get primary enhancer.\n",
    "H3K4me1_bam = [x for x in os.listdir() if (x.endswith('rmdup.bam') and 'H3K4me1' in x)]\n",
    "for file in H3K4me1_bam:\n",
    "    if f\"{file}.bai\" in os.listdir():\n",
    "        continue\n",
    "    cmd = f\"samtools index -@ 20 {file}\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "cmd = f'''bamCompare -b1 {H3K4me1_bam[0]} -b2 {H3K4me1_bam[0]} --outFileName H3K4me1_all.bw \\\n",
    "          --normalizeUsing CPM --binSize 10 --operation add --numberOfProcessors 20 --scaleFactorsMethod None'''\n",
    "subprocess.run(cmd, shell=True)\n",
    "H3K4me3_bam = [x for x in os.listdir() if (x.endswith('rmdup.bam') and 'H3K4me3' in x)]\n",
    "for file in H3K4me3_bam:\n",
    "    if f\"{file}.bai\" in os.listdir():\n",
    "        continue\n",
    "    cmd = f\"samtools index -@ 20 {file}\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "cmd = f'''bamCompare -b1 {H3K4me3_bam[0]} -b2 {H3K4me3_bam[1]} --outFileName H3K4me3_all.bw \\\n",
    "            --normalizeUsing CPM --binSize 10 --operation add --numberOfProcessors 20 --scaleFactorsMethod None'''\n",
    "subprocess.run(cmd, shell=True)\n",
    "me1BW = pyBigWig.open('H3K4me1_all.bw')\n",
    "me3BW = pyBigWig.open('H3K4me3_all.bw')\n",
    "def ifRetain(row, me1BW, me3BW, me1ScoreAll=1, me3ScoreAll=1):\n",
    "    chr = row[0]\n",
    "    start = row[1]\n",
    "    end = row[2]\n",
    "    me1Score = np.array(me1BW.stats(chr, start, end, type='sum'))[0] / me1ScoreAll * pow(10, 6)\n",
    "    me3Score = np.array(me3BW.stats(chr, start, end, type='sum'))[0] / me3ScoreAll * pow(10, 6)\n",
    "    if me1Score > 1.5 * me3Score:\n",
    "        return True, me1Score, me3Score\n",
    "    else:\n",
    "        return False, me1Score, me3Score\n",
    "df = pd.read_csv('H3K4me1_H3K27ac_intersect.bed', sep='\\t', header=None)\n",
    "df['result'] = df.apply(ifRetain, axis=1, args=(me1BW, me3BW))\n",
    "df[['retain', 'me1Score', 'me3Score']] = df['result'].apply(pd.Series)\n",
    "me1BW.close()\n",
    "me3BW.close()\n",
    "df[df['retain']][[0,1,2]].to_csv('enhancer_primary.bed', sep='\\t', header=None, index=False)\n",
    "\n",
    "# ATAC-seq data process to get broadPeak.\n",
    "files = [x for x in os.listdir() if ((x.endswith('rmdup.bam')) and ('ATAC' in x))]\n",
    "for file in files:\n",
    "    cmd = f\"sicer -t {file} -s galGal6 -fdr 0.05 -cpu 20 -o .\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "files = [x for x in os.listdir() if (x.endswith('-W200-G600.scoreisland') and 'ATAC' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df.insert(3, 'peakName', [f'peak{x}' for x in range(1, df.shape[0]+1)])\n",
    "    df['strand'] = '.'\n",
    "    df['signalValue'] = df[3]\n",
    "    df['pValue'] = df[3]\n",
    "    df['qValue'] = df[3]\n",
    "    df.to_csv(file.replace('-W200-G600.scoreisland', '.broadPeak'), sep='\\t', header=None, index=False)\n",
    "\n",
    "# Filter primary enhancer with its DNase broadPeak. Obtain final enhancer.\n",
    "DNase_file = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'ATAC' in x)][0]\n",
    "cmd = f\"bedtools intersect -a enhancer_primary.bed -b {DNase_file} -u -f 0.5 -F 0.5 -e > enhancer.bed\"\n",
    "subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "45e4b4fe-4781-4967-972c-a726999783a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# process enhancer file\n",
    "df = pd.read_csv('enhancer.bed', sep='\\t', header=None)\n",
    "df[3] = [f\"enhancer{i}\" for i in range(df.shape[0])]\n",
    "df[4] = 1\n",
    "df[5] = '.'\n",
    "df.to_csv('enhancer.bg', sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26043beb-855d-4138-b5a7-4e15323c5e2a",
   "metadata": {},
   "source": [
    "## enhancer promoter interaction construction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "c5eb03ae-7335-4de3-b6a1-8dc652524fb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args='bedtools sort -i TAD.bed > TAD_sort.bed', returncode=0)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Use arrowhead to get TAD        \n",
    "cmd = f\"juicer_tools arrowhead GSE185768_galGal6_HH25_limb_bud_MAPQ30_reordered_without_contigs.hic -k KR ./arrowhead\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "# get TAD bed file\n",
    "df = pd.read_csv('arrowhead/5000_blocks.bedpe', sep='\\t', header=None, skiprows=2)\n",
    "df[0] = 'chr' + df[0]\n",
    "df[[0,1,2]].to_csv('TAD.bed', sep='\\t', header=None, index=None)\n",
    "\n",
    "# Sort TAD, the result is: TAD_sort.bed\n",
    "cmd = f\"bedtools sort -i TAD.bed > TAD_sort.bed\"\n",
    "subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "2aadff93-eeec-4ad4-a581-09936d428a8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter TAD, the result is: TAD_filter_sort.bed\n",
    "def ifIntersect(x:tuple, y:tuple, f=0., F=0.)->bool:\n",
    "    '''If two section overlap'''\n",
    "    if int(x[0])<int(y[0])<int(x[1]):\n",
    "        if int(y[1])<int(x[1]):\n",
    "            return 1\n",
    "        else:\n",
    "            if int(x[1])-int(y[0]) / int(x[1])-int(x[0]) > f or int(x[1])-int(y[0]) / int(y[1])-int(y[0]) > F:\n",
    "                return 1\n",
    "            else:\n",
    "                return 0\n",
    "    elif int(y[0])<int(x[0])<int(y[1]):\n",
    "        if int(y[1]) > int(x[1]):\n",
    "            return 1\n",
    "        else:\n",
    "            if int(y[1])-int(x[0]) / int(x[1])-int(x[0]) > f or int(y[1])-int(x[0]) / int(y[1])-int(y[0]) > F:\n",
    "                return 1\n",
    "            else:\n",
    "                return 0\n",
    "    return 0\n",
    "\n",
    "from collections import defaultdict as ddict\n",
    "tadInterval = ddict(list)\n",
    "with open('TAD_sort.bed', 'r') as infile:\n",
    "    for line in infile:\n",
    "        if len(tadInterval[line.split()[0]]) == 0:\n",
    "            tadInterval[line.split()[0]].append((int(line.split()[1]), int(line.split()[2])))\n",
    "        else:\n",
    "            tadInter = (int(line.split()[1]), int(line.split()[2]))\n",
    "            if ifIntersect(tadInterval[line.split()[0]][-1], tadInter):\n",
    "                lengthOld = tadInterval[line.split()[0]][-1][1] - tadInterval[line.split()[0]][-1][0]\n",
    "                lengthNew = tadInter[1] - tadInter[0]\n",
    "                if lengthNew > lengthOld:\n",
    "                    tadInterval[line.split()[0]].remove(tadInterval[line.split()[0]][-1])\n",
    "                    tadInterval[line.split()[0]].append(tadInter)\n",
    "            else:\n",
    "                tadInterval[line.split()[0]].append(tadInter)\n",
    "outfile = open('TAD_filter.bed', 'w')\n",
    "for chromosome in tadInterval.keys():\n",
    "    for i in tadInterval[chromosome]:\n",
    "        print(f'{chromosome}\\t{i[0]}\\t{i[1]}', file=outfile)\n",
    "outfile.close()\n",
    "!bedtools sort -i TAD_filter.bed > TAD_filter_sort.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f90cfe38-e7da-4b9f-ba49-48b772425ab8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/limh/miniconda3/envs/chipseq/lib/python3.7/site-packages/ipykernel_launcher.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    }
   ],
   "source": [
    "# build enhancer promoter interaction\n",
    "cmd = f\"bedtools intersect -a enhancer.bg -b TAD_filter_sort.bed -wo -f 0.8 -F 0.8 -e > enhancer_TAD_intersectWo.txt\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "cmd = f\"bedtools intersect -a promoter.bed -b TAD_filter_sort.bed -wo -f 0.8 -F 0.8  -e > promoter_TAD_intersectWo.txt\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "promoter_TAD_intersectWo_df = pd.read_csv('promoter_TAD_intersectWo.txt', sep='\\t', header=None)\n",
    "promoter_TAD_intersectWo_df.columns = ['promoter_chr', 'promoter_start', 'promoter_end', 'promoter_name', 'promoter_score', 'promoter_strand',\n",
    "                                       'TAD_chr', 'TAD_start', 'TAD_end', 'intersect']\n",
    "eRNA_TAD_intersectWo_df = pd.read_csv('enhancer_TAD_intersectWo.txt', sep='\\t', header=None)\n",
    "eRNA_TAD_intersectWo_df.columns = ['eRNA_chr', 'eRNA_start', 'eRNA_end', 'eRNA_name', 'eRNA_score', 'eRNA_strand',\n",
    "                                   'TAD_chr', 'TAD_start', 'TAD_end', 'intersect']\n",
    "epi_df = pd.merge(eRNA_TAD_intersectWo_df, promoter_TAD_intersectWo_df,\n",
    "                  on = ['TAD_chr', 'TAD_start', 'TAD_end'], how='inner')\n",
    "\n",
    "# EPI filter\n",
    "epi_df['filter'] = epi_df.apply(ifFilter, axis=1)\n",
    "epi_filter_df = epi_df[epi_df['filter']==False]\n",
    "epi_filter_df['epi_name'] = epi_filter_df['eRNA_name'].str.cat(epi_filter_df['promoter_name'], sep=';')\n",
    "\n",
    "# EPI result save\n",
    "epi_filter_df[['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end',\n",
    "               'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']].to_csv('EPI.bedpe', sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6fe2288-d099-4ab3-be10-be6f9164b953",
   "metadata": {},
   "source": [
    "# EP similarity analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "755203b6-a372-45ab-9b94-727a87d15534",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import subprocess\n",
    "import os\n",
    "import numpy as np\n",
    "import pyBigWig\n",
    "pd.set_option('display.max_columns', None)\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "59094dc5-4265-4205-a6b8-d9f866ba22a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "index file /home/limh/Reference_Genome/Gallus/galGal6.fa.fai not found, generating...\n"
     ]
    }
   ],
   "source": [
    "# get enhancer、promoter from EPI\n",
    "referenceGenome = '/home/limh/Reference_Genome/Gallus/galGal6.fa'\n",
    "epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "epi_df[['eRNA_name', 'promoter_name']] = epi_df[6].str.split(';', expand=True)\n",
    "epi_df[[0, 1, 2, 'eRNA_name', 7, 8]].drop_duplicates('eRNA_name').to_csv('enhancer_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "epi_df[[3, 4, 5, 'promoter_name', 7, 9]].drop_duplicates('promoter_name').to_csv('promoter_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "\n",
    "# Get fasta of eRNAinEPI,promoterInEPI\n",
    "cmd = f\"bedtools getfasta -fi {referenceGenome} -fo enhancer_inEPI.fa -name -bed enhancer_inEPI.bed -s\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "cmd = f\"bedtools getfasta -fi {referenceGenome} -fo promoter_inEPI.fa -name -bed promoter_inEPI.bed -s\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "# Split fasta\n",
    "os.makedirs('EPI_fasta', exist_ok=True)\n",
    "with open('enhancer_inEPI.fa', 'r') as f:\n",
    "\tfor line in f:\n",
    "\t\tif line.startswith('>'):\n",
    "\t\t\tname = line.strip().split('::')[0][1:]\n",
    "\t\t\tout = open(f\"EPI_fasta/{name}.fa\", 'w')\n",
    "\t\t\t# out.write(f\">{name}\\n\")\n",
    "\t\t\t# out.write(line)\n",
    "\t\t\tprint('>' + line.strip().split('::')[1], file=out)\n",
    "\t\telse:\n",
    "\t\t\tout.write(line)\n",
    "\tout.close()\n",
    "with open('promoter_inEPI.fa', 'r') as f:\n",
    "\tfor line in f:\n",
    "\t\tif line.startswith('>'):\n",
    "\t\t\tname = line.strip().split('::')[0][1:]\n",
    "\t\t\tout = open(f\"EPI_fasta/{name}.fa\", 'w')\n",
    "\t\t\t# out.write(f\">{name}\\n\")\n",
    "\t\t\t# out.write(line)\n",
    "\t\t\tprint('>' + line.strip().split('::')[1], file=out)\n",
    "\t\telse:\n",
    "\t\t\tout.write(line)\n",
    "\tout.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "c7c54c93-24ae-4b2a-be0c-1bbb0bfa5478",
   "metadata": {},
   "outputs": [],
   "source": [
    "# enhancer\n",
    "# makeblastdb script\n",
    "genes = [x for x in os.listdir('EPI_fasta') if (x.endswith('.fa') and not x.startswith('enhancer'))]\n",
    "makedf_bash_file = open('EPI_makedb.sh', 'w')\n",
    "if os.path.exists('EPI_makedb.log'):\n",
    "    os.remove('EPI_makedb.log')\n",
    "for gene in enumerate(genes):\n",
    "    print(f\"makeblastdb -in EPI_fasta/{gene[1]} -dbtype nucl >> EPI_makedb.log\", file=makedf_bash_file)\n",
    "makedf_bash_file.close()\n",
    "\n",
    "# blast script\n",
    "epi_list = epi_df[['eRNA_name', 'promoter_name']].values.tolist()\n",
    "blast_bash_file = open('EPI_blast.sh', 'w')\n",
    "if os.path.exists('EPI_blastResult.txt'):\n",
    "    os.remove('EPI_blastResult.txt')\n",
    "for i in epi_list:\n",
    "    eRNA = i[0]\n",
    "    promoter = i[1]\n",
    "    print(f\"blastn -task blastn -query EPI_fasta/{eRNA}.fa -db EPI_fasta/{promoter}.fa -outfmt 6 -evalue 1e-4 >> EPI_blastResult.txt\", file=blast_bash_file)\n",
    "blast_bash_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "abd9157d-9862-491d-a74e-65c45394909d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "subprocess.run('bash EPI_makedb.sh', shell=True)\n",
    "subprocess.run('bash EPI_blast.sh', shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9cb2bb5b-662c-4ee7-a549-72f8e26a5251",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the sequence similar enhancer in EPI\n",
    "blastResultFile = 'EPI_blastResult.txt'\n",
    "blastResultBedFile = 'EPI_enhancerBlastHit.bed'\n",
    "with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "\tfor line in infile:\n",
    "\t\tchromosome = line.split()[0].split(':')[0]\n",
    "\t\tstartPre = int(line.split()[0].split(':')[1].split('-')[0])\n",
    "\t\tstopPre = int(line.split()[0].split(':')[1].split('-')[1].split('(')[0])\n",
    "\t\tname = line.split()[0]\n",
    "\t\totherName = line.split()[1]\n",
    "\t\tstrand = line.split()[0].split(':')[1].split('(')[1][0]\n",
    "\t\tif strand == '+' or strand == '.':\n",
    "\t\t\tif int(line.split()[6]) < int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[6])\n",
    "\t\t\t\tshiftEnd = int(line.split()[7])\n",
    "\t\t\telif int(line.split()[6]) > int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[7])\n",
    "\t\t\t\tshiftEnd = int(line.split()[6])\n",
    "\t\t\tstart = startPre + shiftStart - 1\n",
    "\t\t\tend = startPre + shiftEnd - 1\n",
    "\t\telif strand == '-':\n",
    "\t\t\tif int(line.split()[6]) < int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[6])\n",
    "\t\t\t\tshiftEnd = int(line.split()[7])\n",
    "\t\t\telif int(line.split()[6]) > int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[7])\n",
    "\t\t\t\tshiftEnd = int(line.split()[6])\n",
    "\t\t\tend = stopPre - shiftStart\n",
    "\t\t\tstart = stopPre - shiftEnd\n",
    "\t\tprint(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)\n",
    "        \n",
    "# Get the sequence similar promoter in EPI\n",
    "blastResultFile = 'EPI_blastResult.txt'\n",
    "blastResultBedFile = 'EPI_promoterBlastHit.bed'\n",
    "with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "\tfor line in infile:\n",
    "\t\tchromosome = line.split()[1].split(':')[0]\n",
    "\t\tstartPre = int(line.split()[1].split(':')[1].split('-')[0])\n",
    "\t\tstopPre = int(line.split()[1].split(':')[1].split('-')[1].split('(')[0])\n",
    "\t\tname = line.split()[1]\n",
    "\t\totherName = line.split()[0]\n",
    "\t\tstrand = line.split()[1].split(':')[1].split('(')[1][0]\n",
    "\t\tif strand == '+' or strand == '.':\n",
    "\t\t\tif int(line.split()[8]) < int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[8])\n",
    "\t\t\t\tshiftEnd = int(line.split()[9])\n",
    "\t\t\telif int(line.split()[8]) > int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[9])\n",
    "\t\t\t\tshiftEnd = int(line.split()[8])\n",
    "\t\t\tstart = startPre + shiftStart - 1\n",
    "\t\t\tend = startPre + shiftEnd - 1\n",
    "\t\telif strand == '-':\n",
    "\t\t\tif int(line.split()[8]) < int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[8])\n",
    "\t\t\t\tshiftEnd = int(line.split()[9])\n",
    "\t\t\telif int(line.split()[8]) > int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[9])\n",
    "\t\t\t\tshiftEnd = int(line.split()[8])\n",
    "\t\t\tend = stopPre - shiftStart\n",
    "\t\t\tstart = stopPre - shiftEnd\n",
    "\t\tprint(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "aa01b597-caec-41cd-beeb-5a27c35210ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get out the TE similar EPI, save as ep_same_{te}.bedpe\n",
    "promoter_blast_df = pd.read_csv('EPI_promoterBlastHit.bed', sep='\\t', header=None)\n",
    "enhancer_blast_df = pd.read_csv('EPI_enhancerBlastHit.bed', sep='\\t', header=None)\n",
    "promoter_blast_df.columns = ['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end', 'promoter', 'eRNA', 'promoter_strand']\n",
    "enhancer_blast_df.columns = ['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end', 'eRNA', 'promoter', 'eRNA_strand']\n",
    "epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "epi_df.columns = ['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end',\n",
    "               'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']\n",
    "promoter_blast_df['promoter_chr'] = promoter_blast_df['promoter'].str.split(':').str[0]\n",
    "promoter_blast_df['promoter_start'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "promoter_blast_df['promoter_end'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "promoter_blast_df['eRNA_chr'] = promoter_blast_df['eRNA'].str.split(':').str[0]\n",
    "promoter_blast_df['eRNA_start'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "promoter_blast_df['eRNA_end'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "enhancer_blast_df['eRNA_chr'] = enhancer_blast_df['eRNA'].str.split(':').str[0]\n",
    "enhancer_blast_df['eRNA_start'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "enhancer_blast_df['eRNA_end'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "enhancer_blast_df['promoter_chr'] = enhancer_blast_df['promoter'].str.split(':').str[0]\n",
    "enhancer_blast_df['promoter_start'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "enhancer_blast_df['promoter_end'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "# Get out the intersection between transposon and blastHit\n",
    "transposonFile = 'galGal6.fa.bed'\n",
    "x = pd.merge(epi_df, promoter_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                            'promoter_start', 'promoter_end', 'promoter_strand'], how='inner')\n",
    "y = pd.merge(x, enhancer_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                       'promoter_start', 'promoter_end', 'promoter', 'eRNA', 'eRNA_strand'\n",
    "                                       ], how='inner')\n",
    "cmd = f\"bedtools intersect -a EPI_promoterBlastHit.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > EPI_promoterBlastHit_transposon_intersectWo.bed\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "cmd = f\"bedtools intersect -a EPI_enhancerBlastHit.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > EPI_enhancerBlastHit_transposon_intersectWo.bed\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "promoter_blast_te_intersectWo_df = pd.read_csv('EPI_promoterBlastHit_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "enhancer_blast_te_intersectWo_df = pd.read_csv('EPI_enhancerBlastHit_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "promoter_blast_te_intersectWo_df = promoter_blast_te_intersectWo_df.rename({\n",
    "    0:'promoter_blastHit_chr', 1:'promoter_blastHit_start', 2:'promoter_blastHit_end', \n",
    "    3:'promoter', 4:'eRNA', 5:'promoter_strand'}, axis=1)\n",
    "enhancer_blast_te_intersectWo_df = enhancer_blast_te_intersectWo_df.rename({\n",
    "    0:'eRNA_blastHit_chr', 1:'eRNA_blastHit_start', 2:'eRNA_blastHit_end',\n",
    "    3:'eRNA', 4:'promoter', 5:'eRNA_strand'}, axis=1)\n",
    "\n",
    "# merge the promoter and enhancer information together\n",
    "transposon_similar_EPI_df = pd.merge(\n",
    "    pd.merge(y, promoter_blast_te_intersectWo_df, on=['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end',\n",
    "                                                  'promoter', 'eRNA', 'promoter_strand'], how='inner'),\n",
    "    enhancer_blast_te_intersectWo_df, on=['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end',\n",
    "                                            'eRNA', 'promoter', 'eRNA_strand'], how='inner')\n",
    "\n",
    "transposon_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'] == transposon_similar_EPI_df['12_y']]\n",
    "SINE_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'].str.contains('SINE')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "d7eb6e52-ab55-4194-9056-67741228d3e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "378\n",
      "313\n",
      "0\n",
      "LINE/CR1 0.75\n",
      "Simple_repeat 0.17\n",
      "Low_complexity 0.03\n",
      "LTR/ERVL 0.0\n",
      "LTR/ERVK 0.05\n"
     ]
    }
   ],
   "source": [
    "# seuquence-similar EPI num\n",
    "print(y['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "# transposon-similar EPI num\n",
    "print(transposon_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "# SINE-similar EPI num\n",
    "print(SINE_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "# each transposon ratio in transposon-similar EPI (corrected by copy number)\n",
    "df = pd.read_csv(transposonFile, sep='\\t', header=None)\n",
    "transposon_num_dict = transposon_similar_EPI_df['12_x'].value_counts()\n",
    "transposon_copyNum_dict = df[6].value_counts().to_dict()\n",
    "transposon_num_correct_dict = dict()\n",
    "for key in transposon_num_dict.keys():\n",
    "    result = transposon_num_dict[key] / transposon_copyNum_dict[key]\n",
    "    transposon_num_correct_dict[key] = [result]\n",
    "for key, value in (pd.DataFrame.from_dict(transposon_num_correct_dict).T / pd.DataFrame.from_dict(transposon_num_correct_dict).T.sum())[0].to_dict().items():\n",
    "    print(key, round(value, 2))\n",
    "\n",
    "# result save\n",
    "for i, j in SINE_similar_EPI_df.groupby('12_x'):\n",
    "    j[['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end', 'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']\n",
    " ].drop_duplicates('epi_name').to_csv(f\"ep_same_{i.replace('/', '_')}.bedpe\", sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "9dfa0ffb-0d04-4926-9ad7-934a6ec13a77",
   "metadata": {},
   "outputs": [],
   "source": [
    "# random Control (50 shuffle)\n",
    "epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "epi_df[['eRNA_name', 'promoter_name']] = epi_df[6].str.split(';', expand=True)\n",
    "epi_df[[0, 1, 2, 'eRNA_name', 7, 8]].drop_duplicates('eRNA_name').to_csv('enhancer_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "epi_df[[3, 4, 5, 'promoter_name', 7, 9]].drop_duplicates('promoter_name').to_csv('promoter_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "for i in range(50):\n",
    "    epi_list = epi_df[['eRNA_name', 'promoter_name']].values.tolist()\n",
    "    df = pd.DataFrame(epi_list)\n",
    "    df[1] = df[1].sample(frac=1, replace=False).reset_index()[1]\n",
    "    random_epi_list = df.values\n",
    "    blast_bash_file = open(f'random_EPI_blast_{i}.sh', 'w')\n",
    "    if os.path.exists(f'random_EPI_blastResult_{i}.txt'):\n",
    "        os.remove(f'random_EPI_blastResult_{i}.txt')\n",
    "    for j in random_epi_list:\n",
    "        eRNA = j[0]\n",
    "        promoter = j[1]\n",
    "        print(f\"blastn -task blastn -query EPI_fasta/{eRNA}.fa -db EPI_fasta/{promoter}.fa -outfmt 6 -evalue 1e-4 >> random_EPI_blastResult_{i}.txt\", file=blast_bash_file)\n",
    "    blast_bash_file.close()\n",
    "\n",
    "outfile = open('random_control.sh', 'w')\n",
    "for i in [x for x in os.listdir() if x.startswith('random_EPI_blast')]:\n",
    "    print(f\"bash {i} &\", file=outfile)\n",
    "print('wait', file=outfile)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ce3c1e1-48c6-419c-935f-469f7b9b6350",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "subprocess.run('bash random_control.sh', shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "e4f70c1c-7c71-4917-98d3-64dd5743fdac",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████| 50/50 [01:03<00:00,  1.27s/it]\n"
     ]
    }
   ],
   "source": [
    "from tqdm import tqdm\n",
    "# random control result analysis\n",
    "random_seuquence_similar_EPI_num_list = []\n",
    "random_transposon_similar_EPI_num_list = []\n",
    "random_SINE_similar_EPI_num_list = []\n",
    "random_each_transposon_ratio_list = []\n",
    "transposon_num_df = pd.read_csv(transposonFile, sep='\\t', header=None)\n",
    "transposon_copyNum_dict = transposon_num_df[6].value_counts().to_dict()\n",
    "\n",
    "for i in tqdm(range(50)):\n",
    "    random_result_file = f\"random_EPI_blastResult_{i}.txt\"\n",
    "    # Get the sequence similar enhancer in EPI\n",
    "    blastResultFile = f\"random_EPI_blastResult_{i}.txt\"\n",
    "    blastResultBedFile = f'random_EPI_enhancerBlastHit_{i}.bed'\n",
    "    with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "        for line in infile:\n",
    "            chromosome = line.split()[0].split(':')[0]\n",
    "            startPre = int(line.split()[0].split(':')[1].split('-')[0])\n",
    "            stopPre = int(line.split()[0].split(':')[1].split('-')[1].split('(')[0])\n",
    "            name = line.split()[0]\n",
    "            otherName = line.split()[1]\n",
    "            strand = line.split()[0].split(':')[1].split('(')[1][0]\n",
    "            if strand == '+' or strand == '.':\n",
    "                if int(line.split()[6]) < int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[6])\n",
    "                    shiftEnd = int(line.split()[7])\n",
    "                elif int(line.split()[6]) > int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[7])\n",
    "                    shiftEnd = int(line.split()[6])\n",
    "                start = startPre + shiftStart - 1\n",
    "                end = startPre + shiftEnd - 1\n",
    "            elif strand == '-':\n",
    "                if int(line.split()[6]) < int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[6])\n",
    "                    shiftEnd = int(line.split()[7])\n",
    "                elif int(line.split()[6]) > int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[7])\n",
    "                    shiftEnd = int(line.split()[6])\n",
    "                end = stopPre - shiftStart\n",
    "                start = stopPre - shiftEnd\n",
    "            print(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)\n",
    "    # Get the sequence similar promoter in EPI\n",
    "    blastResultFile = f\"random_EPI_blastResult_{i}.txt\"\n",
    "    blastResultBedFile = f'random_EPI_promoterBlastHit_{i}.bed'\n",
    "    with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "        for line in infile:\n",
    "            chromosome = line.split()[1].split(':')[0]\n",
    "            startPre = int(line.split()[1].split(':')[1].split('-')[0])\n",
    "            stopPre = int(line.split()[1].split(':')[1].split('-')[1].split('(')[0])\n",
    "            name = line.split()[1]\n",
    "            otherName = line.split()[0]\n",
    "            strand = line.split()[1].split(':')[1].split('(')[1][0]\n",
    "            if strand == '+' or strand == '.':\n",
    "                if int(line.split()[8]) < int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[8])\n",
    "                    shiftEnd = int(line.split()[9])\n",
    "                elif int(line.split()[8]) > int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[9])\n",
    "                    shiftEnd = int(line.split()[8])\n",
    "                start = startPre + shiftStart - 1\n",
    "                end = startPre + shiftEnd - 1\n",
    "            elif strand == '-':\n",
    "                if int(line.split()[8]) < int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[8])\n",
    "                    shiftEnd = int(line.split()[9])\n",
    "                elif int(line.split()[8]) > int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[9])\n",
    "                    shiftEnd = int(line.split()[8])\n",
    "                end = stopPre - shiftStart\n",
    "                start = stopPre - shiftEnd\n",
    "            print(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)\n",
    "    # Get out the TE similar EPI, save as ep_same_{te}.bedpe\n",
    "    try:\n",
    "        promoter_blast_df = pd.read_csv(f'random_EPI_promoterBlastHit_{i}.bed', sep='\\t', header=None)\n",
    "        enhancer_blast_df = pd.read_csv(f'random_EPI_enhancerBlastHit_{i}.bed', sep='\\t', header=None)\n",
    "    except:\n",
    "        random_seuquence_similar_EPI_num_list.append(0)\n",
    "        random_transposon_similar_EPI_num_list.append(0)\n",
    "        random_SINE_similar_EPI_num_list.append(0)\n",
    "        random_each_transposon_ratio_list.append({})\n",
    "        continue\n",
    "    promoter_blast_df.columns = ['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end', 'promoter', 'eRNA', 'promoter_strand']\n",
    "    enhancer_blast_df.columns = ['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end', 'eRNA', 'promoter', 'eRNA_strand']\n",
    "    epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "    epi_df.columns = ['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end',\n",
    "                   'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']\n",
    "    promoter_blast_df['promoter_chr'] = promoter_blast_df['promoter'].str.split(':').str[0]\n",
    "    promoter_blast_df['promoter_start'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    promoter_blast_df['promoter_end'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "    promoter_blast_df['eRNA_chr'] = promoter_blast_df['eRNA'].str.split(':').str[0]\n",
    "    promoter_blast_df['eRNA_start'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    promoter_blast_df['eRNA_end'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "    enhancer_blast_df['eRNA_chr'] = enhancer_blast_df['eRNA'].str.split(':').str[0]\n",
    "    enhancer_blast_df['eRNA_start'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    enhancer_blast_df['eRNA_end'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "    enhancer_blast_df['promoter_chr'] = enhancer_blast_df['promoter'].str.split(':').str[0]\n",
    "    enhancer_blast_df['promoter_start'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    enhancer_blast_df['promoter_end'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "    # Get out the intersection between transposon and blastHit\n",
    "    x = pd.merge(epi_df, promoter_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                                'promoter_start', 'promoter_end', 'promoter_strand'], how='inner')\n",
    "    y = pd.merge(x, enhancer_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                           'promoter_start', 'promoter_end', 'promoter', 'eRNA', 'eRNA_strand'\n",
    "                                           ], how='inner')\n",
    "    cmd = f\"bedtools intersect -a random_EPI_promoterBlastHit_{i}.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > random_EPI_promoterBlastHit_{i}_transposon_intersectWo.bed\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    cmd = f\"bedtools intersect -a random_EPI_enhancerBlastHit_{i}.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > random_EPI_enhancerBlastHit_{i}_transposon_intersectWo.bed\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "\n",
    "    \n",
    "    promoter_blast_te_intersectWo_df = pd.read_csv(f'random_EPI_promoterBlastHit_{i}_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "    enhancer_blast_te_intersectWo_df = pd.read_csv(f'random_EPI_enhancerBlastHit_{i}_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "    promoter_blast_te_intersectWo_df = promoter_blast_te_intersectWo_df.rename({\n",
    "        0:'promoter_blastHit_chr', 1:'promoter_blastHit_start', 2:'promoter_blastHit_end', \n",
    "        3:'promoter', 4:'eRNA', 5:'promoter_strand'}, axis=1)\n",
    "    enhancer_blast_te_intersectWo_df = enhancer_blast_te_intersectWo_df.rename({\n",
    "        0:'eRNA_blastHit_chr', 1:'eRNA_blastHit_start', 2:'eRNA_blastHit_end',\n",
    "        3:'eRNA', 4:'promoter', 5:'eRNA_strand'}, axis=1)\n",
    "\n",
    "    # merge the promoter and enhancer information together\n",
    "    transposon_similar_EPI_df = pd.merge(\n",
    "        pd.merge(y, promoter_blast_te_intersectWo_df, on=['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end',\n",
    "                                                      'promoter', 'eRNA', 'promoter_strand'], how='inner'),\n",
    "        enhancer_blast_te_intersectWo_df, on=['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end',\n",
    "                                                'eRNA', 'promoter', 'eRNA_strand'], how='inner')\n",
    "\n",
    "    transposon_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'] == transposon_similar_EPI_df['12_y']]\n",
    "    SINE_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'].str.contains('SINE')]\n",
    "    \n",
    "    # seuquence-similar EPI num\n",
    "    random_seuquence_similar_EPI_num_list.append(y['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "    # transposon-similar EPI num\n",
    "    random_transposon_similar_EPI_num_list.append(transposon_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "    # SINE-similar EPI num\n",
    "    random_SINE_similar_EPI_num_list.append(SINE_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "    # each transposon ratio in transposon-similar EPI (corrected by copy number)\n",
    "    transposon_num_dict = transposon_similar_EPI_df['12_x'].value_counts()\n",
    "    if len(transposon_num_dict) == 0:\n",
    "        random_each_transposon_ratio_list.append({})\n",
    "        continue\n",
    "    transposon_num_correct_dict = dict()\n",
    "    transposon_ratio_correct_dict = dict()\n",
    "    for key in transposon_num_dict.keys():\n",
    "        result = transposon_num_dict[key] / transposon_copyNum_dict[key]\n",
    "        transposon_num_correct_dict[key] = [result]\n",
    "    for key, value in (pd.DataFrame.from_dict(transposon_num_correct_dict).T / pd.DataFrame.from_dict(transposon_num_correct_dict).T.sum())[0].to_dict().items():\n",
    "        transposon_ratio_correct_dict[key] = round(value, 2)\n",
    "    random_each_transposon_ratio_list.append(transposon_ratio_correct_dict)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "53f1c9ee-bc60-401a-828b-1fb9bffa9e93",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<matplotlib.lines.Line2D at 0x7f522dd6b4d0>,\n",
       " <matplotlib.lines.Line2D at 0x7f522b1e50d0>]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAB4AAAAPYCAYAAAA7MORPAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/NK7nSAAAACXBIWXMAAA9hAAAPYQGoP6dpAABHCElEQVR4nOzdf5DU9X348ddeaQ6b4ZbdVLg7PVEMmkocyiQKkvqDWhFsMKmk2qQRnHSSdmIzSRmH9DpxxvzDBZp0GMUmk45CHBo1GeRHStoqE4VqiCNVUttJLA5UiOHiJPdjA62Hifv9I19PLnAHp3us+7rHY+Yzw2f38/7w2pm9P3ae8/l8CtVqtRoAAAAAAAAANLymeg8AAAAAAAAAQG0IwAAAAAAAAABJCMAAAAAAAAAASQjAAAAAAAAAAEkIwAAAAAAAAABJCMAAAAAAAAAASQjAAAAAAAAAAEkIwAAAAAAAQFqVSiWeffbZeo8xak888US9RwAalAAMAAAAAACkVKlUYuHChbFy5cp6jzIqq1evjuuvvz66u7vrPQrQgARgAAAAAAAgndfib0tLS6xbt67e45yy1atXx6pVq2L79u3R2tpa73GABiQAAwAAAAAAqRwbfzdv3hwTJ06s90in5Nj4O3v27HqPAzQoARgAAAAAAEhD/AXGOwEYAAAAAABIQfwFEIABAAAAAIAExF+AXxGAAQAAAACAhib+ArxOAAYAAAAAABqW+AswlAAMAAAAAAA0JPEX4HgCMAAAAAAA0HDEX4ATE4ABAAAAAICGIv4CDE8ABgAAAAAAGob4CzAyARgAAAAAAGgI4i/AyQnAAAAAAADAW574C3BqBGAAAAAAAOAtTfwFOHUCMAAAAAAA8JYl/gKMTqFarVbrPUQtvPrqq/HjH/84Jk2aFIVCod7jAAAAAAAAb1KlUokbbrghSqVSbNmypWHi76pVq2LVqlWxZcuWmDVrVr3HAZKoVqvx85//PNrb26OpafjrfNME4B/96EfR0dFR7zEAAAAAAIAa+8lPfhJTpkyp9xinZPXq1fGFL3whent76z0KkNTBgwfj7LPPHvb9CadxljE1adKkiPjVB25paanzNAAAAAAAwBv12pW/LS0t8fWvfz3OPPPMeo90So697fM73/nOeo/DW920aRF9fa/vT54c8cIL9ZqGBlCpVKKjo2Owiw4nTQB+7bbPLS0tAjAAAAAAADSoSqUSN954Y5TLZc/8Jbdff6RpoRChcXEKTvY43OFvDg0AAAAAAHAaVSqVWLhwYbS0tIi/AG+QAAwAAAAAANSd+AtQGwIwAAAAAABQV+IvQO0IwAAAAAAAQN2IvwC1JQADAAAAAAB1If4C1J4ADAAAAAAAnHbiL8DYEIABAAAAAIDTSvwFGDsCMAAAAAAAcNqIvwBjSwAGAAAAAABOC/EXYOwJwAAAAAAAwJgTfwFODwEYAAAAAAAYU+IvwOkjAAMAAAAAAGNG/AU4vQRgAAAAAABgTIi/AKefAAwAAAAAANSc+AtQHwIwAAAAAABQc7t27YpyudxQ8ffIkSOxadMm8RdoaIVqtVqt9xC1UKlUolgsRn9/f7S0tNR7HAAAAAAAGPeq1WoUCoV6jzEqjTgzDapcjujtfX2/VIro6anfPLzlnWoPdQUwAAAAAAAwJhoxpDbizADHEoABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkphQ7wEAAMba9OnTo6+vr95jwJs2efLk2LdvX73HAAAAAOAtTAAGANLr6+uL3t7eeo8BAAAAADDm3AIaAAAAAAAAIAkBGAAAAAAAACAJt4AGAMalUqlU7xHgpNy6HAAAAIDREoABgHGnVCpFT09PvceAkyqXyyIwAAAAAKPiFtAAAAAAAAAASQjAAAAAAAAAAEkIwAAAAAAAAABJCMAAAAAAAAAASQjAAAAAAAAAAEkIwAAAAAAAAABJCMAAAAAAAAAASQjAAAAAAAAAAEkIwAAAAAAAAABJCMAAAAAAAAAASQjAAAAAAAAAAEkIwAAAAAAAAABJCMAAAAAAAAAASQjAAAAAAAAAAEkIwAAAAAAAAABJCMAAAAAAAAAASQjAAAAAAAAAAEmMOgDv3LkzFi9eHO3t7VEoFGLz5s1D3i8UCifc/vZv/3bYc65fv/6Ea15++eVRfyAAAAAAAACA8WrUAfjIkSMxa9asWLt27QnfP3To0JDt3nvvjUKhEEuWLBnxvC0tLcetnThx4mjHAwAAAAAAABi3Jox2waJFi2LRokXDvt/a2jpkf8uWLTF//vyYPn36iOctFArHrQUAAAAAAADg1I3pM4B/8pOfxLZt2+LP/uzPTnrs4cOHY9q0aXH22WfH+9///njmmWdGPH5gYCAqlcqQDQAAAAAAAGA8G9MA/LWvfS0mTZoUN9xww4jHvetd74r169fH1q1b4/7774+JEyfG+973vti7d++wa7q6uqJYLA5uHR0dtR4fAAAAAAAAoKGMaQC+995740//9E9P+izfuXPnxkc/+tGYNWtWXH755fGNb3wjLrjggrjrrruGXdPZ2Rn9/f2D28GDB2s9PgAAAAAAAEBDGfUzgE/Vv/3bv8Vzzz0XDz744KjXNjU1xSWXXDLiFcDNzc3R3Nz8ZkYEAAAAAAAASGXMrgC+55574j3veU/MmjVr1Gur1Wrs2bMn2traxmAyAAAAAAAAgJxGfQXw4cOH4/nnnx/c379/f+zZsyfK5XKcc845ERFRqVTim9/8ZnzpS1864TmWLl0aZ511VnR1dUVExOc///mYO3duzJgxIyqVStx5552xZ8+euPvuu9/IZwIAAAAAAAAYl0YdgHfv3h3z588f3F++fHlERCxbtizWr18fEREPPPBAVKvV+PCHP3zCcxw4cCCaml6/+Livry8+8YlPRHd3dxSLxZg9e3bs3LkzLr300tGOBwAAAAAAADBuFarVarXeQ9RCpVKJYrEY/f390dLSUu9xAIC3kHK5HL29vYP7pVIpenp66jgRnBrfXQAAAEisXI445nd/lEoRfvczglPtoWP2DGAAAAAAAAAATi8BGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgiVEH4J07d8bixYujvb09CoVCbN68ecj7t9xySxQKhSHb3LlzT3rejRs3xkUXXRTNzc1x0UUXxaZNm0Y7GgAAAAAAAMC4NuoAfOTIkZg1a1asXbt22GMWLlwYhw4dGty+/e1vj3jOXbt2xU033RQ333xzfP/734+bb745brzxxnjyySdHOx4AAAAAAADAuDVhtAsWLVoUixYtGvGY5ubmaG1tPeVzrlmzJq655pro7OyMiIjOzs7YsWNHrFmzJu6///7RjggAAAAAAAAwLo3JM4Afe+yxmDJlSlxwwQXx8Y9/PF566aURj9+1a1csWLBgyGvXXnttfPe73x12zcDAQFQqlSEbAAAAAAAAwHhW8wC8aNGi+Md//Mf4zne+E1/60pfiqaeeit///d+PgYGBYdd0d3fH1KlTh7w2derU6O7uHnZNV1dXFIvFwa2jo6NmnwEAAAAAAACgEY36FtAnc9NNNw3++93vfne8973vjWnTpsW2bdvihhtuGHZdoVAYsl+tVo977VidnZ2xfPnywf1KpSICAwAAAAAAAONazQPwr2tra4tp06bF3r17hz2mtbX1uKt9X3rppeOuCj5Wc3NzNDc312xOAAAAAAAAgEY3Js8APtbPfvazOHjwYLS1tQ17zGWXXRaPPPLIkNcefvjhmDdv3liPBwAAAAAAAJDGqK8APnz4cDz//POD+/v37489e/ZEuVyOcrkcd9xxRyxZsiTa2trif/7nf+Jv/uZv4rd/+7fjj/7ojwbXLF26NM4666zo6uqKiIhPf/rTccUVV8SqVaviAx/4QGzZsiW2b98ejz/+eA0+IgAAAAAAAMD4MOoAvHv37pg/f/7g/mvP4V22bFl8+ctfjmeffTbuu+++6Ovri7a2tpg/f348+OCDMWnSpME1Bw4ciKam1y8+njdvXjzwwAPxuc99Lm6//fY4//zz48EHH4w5c+a8mc8GAAAAAAAAMK4UqtVqtd5D1EKlUolisRj9/f3R0tJS73EAgLeQcrkcvb29g/ulUil6enrqOBGcGt9dAAAASKxcjjjmd3+UShF+9zOCU+2hY/4MYAAAAAAAAABODwEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACCJUQfgnTt3xuLFi6O9vT0KhUJs3rx58L1XXnklPvvZz8bFF18cb3/726O9vT2WLl0aP/7xj0c85/r166NQKBy3vfzyy6P+QAAAAAAAAADj1agD8JEjR2LWrFmxdu3a49773//933j66afj9ttvj6effjoeeuih+O///u+4/vrrT3relpaWOHTo0JBt4sSJox0PAAAAAAAAYNyaMNoFixYtikWLFp3wvWKxGI888siQ1+6666649NJL48CBA3HOOecMe95CoRCtra2jHQcAAAAAAACA/2/MnwHc398fhUIhJk+ePOJxhw8fjmnTpsXZZ58d73//++OZZ54Z8fiBgYGoVCpDNgAAAAAAAIDxbEwD8Msvvxx//dd/HR/5yEeipaVl2OPe9a53xfr162Pr1q1x//33x8SJE+N973tf7N27d9g1XV1dUSwWB7eOjo6x+AgAAAAAAAAADWPMAvArr7wSf/InfxKvvvpq/P3f//2Ix86dOzc++tGPxqxZs+Lyyy+Pb3zjG3HBBRfEXXfdNeyazs7O6O/vH9wOHjxY648AAAAAAAAA0FBG/QzgU/HKK6/EjTfeGPv374/vfOc7I179eyJNTU1xySWXjHgFcHNzczQ3N7/ZUQEAAAAAAADSqPkVwK/F371798b27dvjHe94x6jPUa1WY8+ePdHW1lbr8QAAAAAAAADSGvUVwIcPH47nn39+cH///v2xZ8+eKJfL0d7eHh/60Ifi6aefjn/6p3+KX/7yl9Hd3R0REeVyOd72trdFRMTSpUvjrLPOiq6uroiI+PznPx9z586NGTNmRKVSiTvvvDP27NkTd999dy0+IwAAAAAAAMC4MOoAvHv37pg/f/7g/vLlyyMiYtmyZXHHHXfE1q1bIyLid3/3d4ese/TRR+Oqq66KiIgDBw5EU9PrFx/39fXFJz7xieju7o5isRizZ8+OnTt3xqWXXjra8QAAAAAAAADGrUK1Wq3We4haqFQqUSwWo7+/f9TPHAYAciuXy9Hb2zu4XyqVoqenp44Twanx3QUAAIDEyuWIY373R6kU4Xc/IzjVHlrzZwADAAAAAAAAUB8CMAAAAAAAAEASAjAAAAAAAABAEgIwAAAAAAAAQBICMAAAAAAAAEASAjAAAAAAAABAEgIwAAAAAAAAQBICMAAAAAAAAEASAjAAAAAAAABAEgIwAAAAAAAAQBICMAAAAAAAAEASAjAAAAAAAABAEgIwAAAAAAAAQBICMAAAAAAAAEASAjAAAAAAAABAEgIwAAAAAAAAQBIT6j1ArU2bNi0KhUK9x4CT6u/vj2q1OrhfKBSiWCzWcSJ4YyZPnhz79u2r9xgAAAAAAEAkDMB9fX31HgHekGq1Gr29vfUeAwAAAAAAgAbmFtAAAAAAAAAASQjAAAAAAAAAAEmkuwX0a0qlUr1HgBGd6HbPvrc0ArcqBwAAAACAt66UAbhUKkVPT0+9x4ARNTU1RbVaHdz3vaVRlMtlERgAAAAAAN6i3AIaAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgiVEH4J07d8bixYujvb09CoVCbN68ecj71Wo17rjjjmhvb48zzjgjrrrqqviv//qvk55348aNcdFFF0Vzc3NcdNFFsWnTptGOBgAAAAAAADCujToAHzlyJGbNmhVr16494furV6+Ov/u7v4u1a9fGU089Fa2trXHNNdfEz3/+82HPuWvXrrjpppvi5ptvju9///tx8803x4033hhPPvnkaMcDAAAAAAAAGLcK1Wq1+oYXFwqxadOm+OAHPxgRv7r6t729PT7zmc/EZz/72YiIGBgYiKlTp8aqVaviz//8z094nptuuikqlUr88z//8+BrCxcujFKpFPfff/8J1wwMDMTAwMDgfqVSiY6OjoiIKJVK0dPT80Y/FpwWTU1Nceyfn+8tjaJcLkdvb+/gvu8ujcD3lkbluwsAAACJlcsRx/zuj1Ipwu9+RlCpVKJYLEZ/f3+0tLQMe1xNnwG8f//+6O7ujgULFgy+1tzcHFdeeWV897vfHXbdrl27hqyJiLj22mtHXNPV1RXFYnFwey3+AgAAAAAAAIxXNQ3A3d3dERExderUIa9PnTp18L3h1o12TWdnZ/T39w9uBw8efBOTAwAAAAAAADS+CWNx0kKhMGS/Wq0e99qbXdPc3BzNzc1vfEgAAAAAAACAZGp6BXBra2tExHFX7r700kvHXeH76+tGuwYAAAAAAACAoWoagM8777xobW2NRx55ZPC1o0ePxo4dO2LevHnDrrvsssuGrImIePjhh0dcAwAAAAAAAMBQo74F9OHDh+P5558f3N+/f3/s2bMnyuVynHPOOfGZz3wmVq5cGTNmzIgZM2bEypUr47d+67fiIx/5yOCapUuXxllnnRVdXV0REfHpT386rrjiili1alV84AMfiC1btsT27dvj8ccfr8FHBAAAAAAAABgfRh2Ad+/eHfPnzx/cX758eURELFu2LNavXx8rVqyI//u//4tPfvKT0dvbG3PmzImHH344Jk2aNLjmwIED0dT0+sXH8+bNiwceeCA+97nPxe233x7nn39+PPjggzFnzpw389kAAAAAAAAAxpVCtVqt1nuIWqhUKlEsFiMiolQqRU9PT50ngpE1NTXFsX9+vrc0inK5HL29vYP7vrs0At9bGpXvLgAAACRWLkcc87s/SqUIv/sZwWs9tL+/P1paWoY9rqbPAAYAAAAAAACgfgRgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkah6Azz333CgUCsdtt9566wmPf+yxx054/A9/+MNajwYAAAAAAACQ2oRan/Cpp56KX/7yl4P7//mf/xnXXHNN/PEf//GI65577rloaWkZ3D/zzDNrPRoAAAAAAABAajUPwL8ebr/whS/E+eefH1deeeWI66ZMmRKTJ0+u9TgAAAAAAAAA48aYPgP46NGjsWHDhvjYxz4WhUJhxGNnz54dbW1tcfXVV8ejjz560nMPDAxEpVIZsgEAAAAAAACMZ2MagDdv3hx9fX1xyy23DHtMW1tbfPWrX42NGzfGQw89FBdeeGFcffXVsXPnzhHP3dXVFcVicXDr6Oio8fQAAAAAAAAAjaVQrVarY3Xya6+9Nt72trfFt771rVGtW7x4cRQKhdi6deuwxwwMDMTAwMDgfqVSGYzApVIpenp63tjQcJo0NTXFsX9+vrc0inK5HL29vYP7vrs0At9bGpXvLgAAACRWLkcc87s/SqUIv/sZQaVSiWKxGP39/dHS0jLscTV/BvBrXnjhhdi+fXs89NBDo147d+7c2LBhw4jHNDc3R3Nz8xsdDwAAAAAAACCdMbsF9Lp162LKlCnxh3/4h6Ne+8wzz0RbW9sYTAUAAAAAAACQ15hcAfzqq6/GunXrYtmyZTFhwtD/orOzM1588cW47777IiJizZo1ce6558bMmTPj6NGjsWHDhti4cWNs3LhxLEYDAAAAAAAASGtMAvD27dvjwIED8bGPfey49w4dOhQHDhwY3D969Gjcdttt8eKLL8YZZ5wRM2fOjG3btsV11103FqMBAAAAAAAApDUmAXjBggVRrVZP+N769euH7K9YsSJWrFgxFmMAAAAAAAAAjCtj9gxgAAAAAAAAAE4vARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIImaB+A77rgjCoXCkK21tXXENTt27Ij3vOc9MXHixJg+fXp85StfqfVYAAAAAAAAAOlNGIuTzpw5M7Zv3z64/xu/8RvDHrt///647rrr4uMf/3hs2LAhnnjiifjkJz8ZZ555ZixZsmQsxgMAAAAAAABIaUwC8IQJE0561e9rvvKVr8Q555wTa9asiYiI3/md34ndu3fHF7/4RQEYAIBxrb+/f8h+X19flMvlOk0Db8zkyZNj37599R4DAAAAxo0xCcB79+6N9vb2aG5ujjlz5sTKlStj+vTpJzx2165dsWDBgiGvXXvttXHPPffEK6+8Er/5m795wnUDAwMxMDAwuF+pVGr3AQAA4C2gWq0et9/b21unaQAAAABoBDV/BvCcOXPivvvui3/913+Nf/iHf4ju7u6YN29e/OxnPzvh8d3d3TF16tQhr02dOjV+8YtfxE9/+tNh/5+urq4oFouDW0dHR00/BwAAAAAAAECjqXkAXrRoUSxZsiQuvvji+IM/+IPYtm1bRER87WtfG3ZNoVAYsv/alQ6//vqxOjs7o7+/f3A7ePBgDaYHAAAAAAAAaFxjcgvoY7397W+Piy++OPbu3XvC91tbW6O7u3vIay+99FJMmDAh3vGOdwx73ubm5mhubq7prAAA8FZXKpXqPQKMyG3KAQAAoL7GPAAPDAzED37wg7j88stP+P5ll10W3/rWt4a89vDDD8d73/veYZ//CwAA41GpVIqenp56jwEjKpfLIjAAAADUUc1vAX3bbbfFjh07Yv/+/fHkk0/Ghz70oahUKrFs2bKI+NWtm5cuXTp4/F/8xV/ECy+8EMuXL48f/OAHce+998Y999wTt912W61HAwAAAAAAAEit5lcA/+hHP4oPf/jD8dOf/jTOPPPMmDt3bnzve9+LadOmRUTEoUOH4sCBA4PHn3feefHtb387/uqv/iruvvvuaG9vjzvvvDOWLFlS69EAAAAAAAAAUqt5AH7ggQdGfH/9+vXHvXbllVfG008/XetRAAAAAAAAAMaVmt8CGgAAAAAAAID6EIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJIQgAEAAAAAAACSEIABAAAAAAAAkhCAAQAAAAAAAJKoeQDu6uqKSy65JCZNmhRTpkyJD37wg/Hcc8+NuOaxxx6LQqFw3PbDH/6w1uMBAAAAAAAApFXzALxjx4649dZb43vf+1488sgj8Ytf/CIWLFgQR44cOena5557Lg4dOjS4zZgxo9bjAQAAAAAAAKQ1odYn/Jd/+Zch++vWrYspU6bEv//7v8cVV1wx4topU6bE5MmTaz0SAAAAAAAAwLgw5s8A7u/vj4iIcrl80mNnz54dbW1tcfXVV8ejjz464rEDAwNRqVSGbAAAAAAAAADj2ZgG4Gq1GsuXL4/f+73fi3e/+93DHtfW1hZf/epXY+PGjfHQQw/FhRdeGFdffXXs3Llz2DVdXV1RLBYHt46OjrH4CAAAAAAAAAANo+a3gD7WX/7lX8Z//Md/xOOPPz7icRdeeGFceOGFg/uXXXZZHDx4ML74xS8Oe9vozs7OWL58+eB+pVIRgQEAAAAAAIBxbcyuAP7Upz4VW7dujUcffTTOPvvsUa+fO3du7N27d9j3m5ubo6WlZcgGAAAAAAAAMJ7V/ArgarUan/rUp2LTpk3x2GOPxXnnnfeGzvPMM89EW1tbjacDAAAAAAAAyKvmAfjWW2+Nr3/967Fly5aYNGlSdHd3R0REsViMM844IyJ+dfvmF198Me67776IiFizZk2ce+65MXPmzDh69Ghs2LAhNm7cGBs3bqz1eAAAAAAAAABp1TwAf/nLX46IiKuuumrI6+vWrYtbbrklIiIOHToUBw4cGHzv6NGjcdttt8WLL74YZ5xxRsycOTO2bdsW1113Xa3HAwAAAAAAAEhrTG4BfTLr168fsr9ixYpYsWJFrUcBAAAAAAAAGFea6j0AAAAAAAAAALUhAAMAAAAAAAAkIQADAAAAAAAAJCEAAwAAAAAAACQhAAMAAAAAAAAkIQADAAAAAAAAJCEAAwAAAAAAACQhAAMAAAAAAAAkIQADAAAAAAAAJCEAAwAAAAAAACQhAAMAAAAAAAAkIQADAAAAAAAAJCEAAwAAAAAAACQhAAMAAAAAAAAkIQADAAAAAAAAJCEAAwAAAAAAACQxod4DAAAAANTb9OnTo6+vr95jwJsyefLk2LdvX73HAACgzgRgAAAAYNzr6+uL3t7eeo8BAADwprkFNAAAAAAAAEASAjAAAAAAAABAEm4BDQAAAHACpVKp3iPAiNy2HACAExGAAQAAAH5NqVSKnp6eeo8BIyqXyyIwAADHcQtoAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJCbUewAAAAAAYPT6+/uH7Pf19UW5XK7TNPDGTZ48Ofbt21fvMQAgDQEYAAAAABpQtVo9br+3t7dO0wAA8FbhFtAAAAAAAAAASQjAAAAAAAAAAEm4BTQAAAAAJFEqleo9ApyUW5UDwNgSgAEAAAAggVKpFD09PfUeA06qXC6LwAAwhtwCGgAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAAACAJARgAAAAAAAAgCQEYAAAAAAAAIAkBGAAAAAAASKtSqcSzzz5b7zFG7Yknnqj3CECDEoABAAAAAICUKpVKLFy4MFauXFnvUUZl9erVcf3110d3d3e9RwEakAAMAAAAAACk81r8bWlpiXXr1tV7nFO2evXqWLVqVWzfvj1aW1vrPQ7QgARgAAAAAAAglWPj7+bNm2PixIn1HumUHBt/Z8+eXe9xgAYlAAMAAAAAAGmIv8B4JwADAAAAAAApiL8AAjAAAAAAAJCA+AvwKwIwAAAAAADQ0MRfgNcJwAAAAAAAQMMSfwGGEoABAAAAAICGJP4CHE8ABgAAAAAAGo74C3BiAjAAAAAAANBQxF+A4QnAAAAAAMD/a+/+Y6qqGziOf+6D8WMEN0B+yER2QysL0XFpiAXrh13C5mS5+WPOqPVHNKyQuX7oH5JjQ9Y/1gzKpVRrBn8k6hZZbC1gAzYgSGfkdFrXih+hwMW7icnO88czeR6ER3ge7+V4ru/Xxh/nyzn3fu52/HoOn33PBQDLoPwFgFujAAYAAAAAAAAAAJZA+QsAM/NbAVxZWSmHw6HQ0FA5nU41Nzffcv/GxkY5nU6Fhobq/vvv10cffeSvaAAAAAAAAAAAwGIofwFgdvxSANfW1qq4uFi7du1SV1eXsrOzlZeXJ7fbPe3+Fy5c0Jo1a5Sdna2uri7t3LlTr7/+ur766it/xAMAAAAAAAAAABZC+QsAs2czDMPw9YtmZmYqPT1dVVVVE2NLly5Vfn6+ysvLp+z/1ltv6fjx4+rp6ZkYKyws1E8//aTW1tZZvafH45Hdbp/YjoqKuo1PAPjf0NDQlDHOW1jBzeeuzWbTfffdZ04YYJaGh4d18yUPcy6sgOsFWBHXCrAqrhdgRVwrwKpuPnejoqJ0+fJlk9LACih/EbCio6X/nBOjoiTmQ9zCjT50ZGREkZGR/3W/eb5+42vXrqmzs1Nvv/32pHGXy6WWlpZpj2ltbZXL5Zo0lpubq4MHD+rvv//WPffcM+WYsbExjY2NTWyPjIxM+v10F8DAnY7zFlZkGAbnLiyJ8xZWxbkLq+FaAVbGuQsr4ryFFRmGIY/HY3YM3KE8Ho+ef/55RUVFWar8raioUEVFhY4dO6aUlBTOcUzv5jWahiFxruAWbswlM63v9XkBPDg4qPHxccXHx08aj4+PV19f37TH9PX1Tbv/9evXNTg4qAULFkw5pry8XO+++67vggMAAAAAAAAA5tzw8PCkpzsC0+nv77dM+Xtj5e/Q0JBycnLMjgMrGR6WmA8xC6Ojo7f8v9PnBfANNptt0rZhGFPGZtp/uvEb3nnnHZWUlExsDw8PKzk5WW63m4sFAD7j8XiUlJSkixcv3vJxCgDwv2BuAeAPzC0A/IG5BYC/ML9gJjdW/kZGRurw4cOKjY01O9Ks/OdjnxcvXmx2HFgA8yH+F4ZhaHR0VImJibfcz+cF8Pz58xUUFDRlte/AwMCUVb43JCQkTLv/vHnzFBMTM+0xISEhCgkJmTJut9v5BwLA5yIjI5lbAPgccwsAf2BuAeAPzC0A/IX5BdPxeDzasGGDoqOjLfXYZ77zF7eD+RCzNZuFsP/w9ZsGBwfL6XSqoaFh0nhDQ4NWrVo17TFZWVlT9v/uu++UkZEx7ff/AgAAAAAAAACAwOPxePTss88qMjKS8hcA/k8+L4AlqaSkRJ988okOHTqknp4ebd++XW63W4WFhZL+9fjmF154YWL/wsJC/fbbbyopKVFPT48OHTqkgwcPaseOHf6IBwAAAAAAAAAA7jCUvwDgG375DuCNGzfq0qVL2rNnj3p7e5Wamqr6+nolJydLknp7e+V2uyf2dzgcqq+v1/bt2/Xhhx8qMTFRH3zwgdavXz/r9wwJCdHu3bunfSw0APy/mFsA+ANzCwB/YG4B4A/MLQD8hfkFN6P8xd2K+RD+YDMMwzA7BAAAAAAAAAAAuDtR/gKAb/nlEdAAAAAAAAAAAAAzofwFAN+jAAYAAAAAAAAAAHOO8hcA/IMCGAAAAAAAAAAAzCnKXwDwHwpgAAAAAAAAAAAwZyh/AcC/KIABAAAAAAAAAMCcoPwFAP8LiAK4srJSDodDoaGhcjqdam5uNjsSAItramrS2rVrlZiYKJvNpqNHj5odCUAAKC8v16OPPqqIiAjFxcUpPz9fZ86cMTsWAIurqqpSWlqaIiMjFRkZqaysLH3zzTdmxwIQYMrLy2Wz2VRcXGx2FAAWVlpaKpvNNuknISHB7FiYQ5S/uFvMdJ925MgR5ebmav78+bLZbOru7p7yGq+88opSUlIUFham2NhYrVu3Tr/88sscfgpYmeUL4NraWhUXF2vXrl3q6upSdna28vLy5Ha7zY4GwMK8Xq+WL1+u/fv3mx0FQABpbGxUUVGR2tra1NDQoOvXr8vlcsnr9ZodDYCFLVy4UHv37lVHR4c6Ojr01FNPad26dTp9+rTZ0QAEiPb2dh04cEBpaWlmRwEQAB555BH19vZO/Jw6dcrsSJgjlL+4m8x0n+b1evXYY49p7969//U1nE6nqqur1dPTo2+//VaGYcjlcml8fHyuPgYszGYYhmF2iNuRmZmp9PR0VVVVTYwtXbpU+fn5Ki8vNzEZgEBhs9lUV1en/Px8s6MACDB//fWX4uLi1NjYqJycHLPjAAgg0dHReu+99/Tyyy+bHQWAxV25ckXp6emqrKxUWVmZVqxYoX379pkdC4BFlZaW6ujRo9OudENgo/wFpr9P+/XXX+VwONTV1aUVK1bc8viTJ09q+fLlOnfunFJSUvycFlZn6RXA165dU2dnp1wu16Rxl8ullpYWk1IBAADMzsjIiKR/3QAAgC+Mj4+rpqZGXq9XWVlZZscBEACKior03HPPafXq1WZHARAgzp49q8TERDkcDm3atEnnz583OxL8jPIXdztf3Kd5vV5VV1fL4XAoKSnJxwkRiOaZHeB2DA4Oanx8XPHx8ZPG4+Pj1dfXZ1IqAACAmRmGoZKSEj3++ONKTU01Ow4Aizt16pSysrJ09epV3Xvvvaqrq9PDDz9sdiwAFldTU6Mff/xR7e3tZkcBECAyMzP1+eef64EHHlB/f7/Kysq0atUqnT59WjExMWbHgx9Q/uJu5ov7tMrKSr355pvyer166KGH1NDQoODgYD8lRiCx9ArgG2w226RtwzCmjAEAANxJtm3bppMnT+rLL780OwqAAPDggw+qu7tbbW1tevXVV1VQUKCff/7Z7FgALOzixYt644039MUXX1jmj/UA7nx5eXlav369li1bptWrV+vrr7+WJH322WcmJ4O/tLa2Kjo62lLlr9frVV1dHeUvbpsv7tO2bNmirq4uNTY2asmSJdqwYYOuXr3qp8QIJJZeATx//nwFBQVNWe07MDAwZVUwAADAneK1117T8ePH1dTUpIULF5odB0AACA4O1uLFiyVJGRkZam9v1/vvv6+PP/7Y5GQArKqzs1MDAwNyOp0TY+Pj42pqatL+/fs1NjamoKAgExMCCATh4eFatmyZzp49a3YU+Elubq5cLpelFmyFh4erpaXFUplxZ/LFfZrdbpfdbteSJUu0cuVKRUVFqa6uTps3b/ZXbAQIS68ADg4OltPpVENDw6TxhoYGrVq1yqRUAAAA0zMMQ9u2bdORI0f0/fffy+FwmB0JQIAyDENjY2NmxwBgYU8//bROnTql7u7uiZ+MjAxt2bJF3d3dlL8AfGJsbEw9PT1asGCB2VHgR1YsUq2YGXc+X9ynca+H2bL0CmBJKikp0datW5WRkaGsrCwdOHBAbrdbhYWFZkcDYGFXrlzRuXPnJrYvXLig7u5uRUdHa9GiRSYmA2BlRUVFOnz4sI4dO6aIiIiJp5jY7XaFhYWZnA6AVe3cuVN5eXlKSkrS6Oioampq9MMPP+jEiRNmRwNgYREREUpNTZ00Fh4erpiYmCnjADBbO3bs0Nq1a7Vo0SINDAyorKxMHo9HBQUFZkcDAJ+a6T7t8uXLcrvd+vPPPyVJZ86ckSQlJCQoISFB58+fV21trVwul2JjY/XHH3+ooqJCYWFhWrNmjWmfC9Zh+QJ448aNunTpkvbs2aPe3l6lpqaqvr5eycnJZkcDYGEdHR168sknJ7ZLSkokSQUFBfr0009NSgXA6qqqqiRJTzzxxKTx6upqvfjii3MfCEBA6O/v19atW9Xb2yu73a60tDSdOHFCzzzzjNnRAAAAJvn999+1efNmDQ4OKjY2VitXrlRbWxt/ywUQcGa6Tzt+/Lheeumlif03bdokSdq9e7dKS0sVGhqq5uZm7du3T0NDQ4qPj1dOTo5aWloUFxdnymeCtdgMwzDMDgEAAAAAAAAAAAAAuH2W/g5gAAAAAAAAAAAAAMC/UQADAAAAAAAAAAAAQICgAAYAAAAAAAAAAACAAEEBDAAAAAAAAAAAAAABggIYAAAAAAAAAAAAAAIEBTAAAAAAAAAAAAAABAgKYAAAAAAAAAAAAAAIEBTAAAAAAAAAAAAAABAgKIABAAAAAAAAAAAAIEBQAAMAAAAAAAAAAABAgKAABgAAAAAAAAAAAIAA8U+jOPcQY/arKgAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 2400x1200 with 3 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# depict random result picture: transposon similar\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.gridspec as gridspec\n",
    "from brokenaxes import brokenaxes\n",
    "\n",
    "exp_result = 313\n",
    "control_min = min(random_transposon_similar_EPI_num_list)\n",
    "control_max = max(random_transposon_similar_EPI_num_list)\n",
    "\n",
    "fig = plt.figure(figsize=(24,12), linewidth=10)\n",
    "\n",
    "ax = brokenaxes(xlims=((control_min, control_max+0.5), (exp_result-0.5, exp_result+0.5)),despine=False, wspace=0.05,d=0.01)\n",
    "\n",
    "plt.rc('axes', linewidth=3.5)\n",
    "plt.rcParams['font.size'] = 20  # 14pt font size  \n",
    "\n",
    "\n",
    "ax.hist(random_transposon_similar_EPI_num_list, bins=8, edgecolor='black', color='white', linewidth=3.5)\n",
    "ax.axvline(x=exp_result, color='red', linewidth=3.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1974b89-e68e-471c-8c46-cc836302e87e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:chipseq]",
   "language": "python",
   "name": "conda-env-chipseq-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
