{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "30c1bbfe-30a6-47fa-8395-726b61ff4d9d",
   "metadata": {},
   "source": [
    "# Calculate enhancer、eRNA、EPI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b22f6316-2196-4121-9eec-12c6311e4dbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('display.max_columns', None)\n",
    "import subprocess\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import os\n",
    "import pyBigWig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "da5c224c-d4f4-4690-bdf1-9e1289a9d529",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fuction define. if eRNA intersect with promoter, return False. else return True\n",
    "def ifFilter(row):\n",
    "    try:\n",
    "        if (row['eRNA_start'] >= row['promoter_start']) & (row['eRNA_start'] <= row['promoter_end']):\n",
    "            return True\n",
    "        elif (row['eRNA_end'] >= row['promoter_start']) & (row['eRNA_end'] <= row['promoter_end']):\n",
    "            return True\n",
    "        else:\n",
    "            return False\n",
    "    except:\n",
    "        try:\n",
    "            if (row['enhancer_start'] >= row['promoter_start']) & (row['enhancer_start'] <= row['promoter_end']):\n",
    "                return True\n",
    "            elif (row['enhancer_end'] >= row['promoter_start']) & (row['enhancer_end'] <= row['promoter_end']):\n",
    "                return True\n",
    "            else:\n",
    "                return False\n",
    "        except:\n",
    "            raise KeyError"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "96ee92d6-3211-447c-860b-7fd2580da9d5",
   "metadata": {},
   "source": [
    "## enhancer identification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f1420f2-9122-4efd-834b-519621c1ca7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # sequence data download\n",
    "# cmd = f\"bash download.sh\"\n",
    "# subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "97202849-729d-4c76-90c4-660f4b1e3276",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:07:42,166 - SRR ids: ['DNase_r1_pair.sra']\n",
      "2023-11-21 19:07:42,166 - extra args: ['--gzip', '--split-3']\n",
      "2023-11-21 19:07:42,228 - tempdir: /tmp/pfd_fvbh5iji\n",
      "2023-11-21 19:07:42,228 - CMD: sra-stat --meta --quick DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,254 - DNase_r1_pair.sra spots: 58415047\n",
      "2023-11-21 19:07:43,255 - blocks: [[1, 5841504], [5841505, 11683008], [11683009, 17524512], [17524513, 23366016], [23366017, 29207520], [29207521, 35049024], [35049025, 40890528], [40890529, 46732032], [46732033, 52573536], [52573537, 58415047]]\n",
      "2023-11-21 19:07:43,256 - CMD: fastq-dump -N 1 -X 5841504 -O /tmp/pfd_fvbh5iji/0 --gzip --split-3 DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,284 - CMD: fastq-dump -N 5841505 -X 11683008 -O /tmp/pfd_fvbh5iji/1 --gzip --split-3 DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,293 - CMD: fastq-dump -N 11683009 -X 17524512 -O /tmp/pfd_fvbh5iji/2 --gzip --split-3 DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,296 - CMD: fastq-dump -N 17524513 -X 23366016 -O /tmp/pfd_fvbh5iji/3 --gzip --split-3 DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,307 - CMD: fastq-dump -N 23366017 -X 29207520 -O /tmp/pfd_fvbh5iji/4 --gzip --split-3 DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,326 - CMD: fastq-dump -N 29207521 -X 35049024 -O /tmp/pfd_fvbh5iji/5 --gzip --split-3 DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,328 - CMD: fastq-dump -N 35049025 -X 40890528 -O /tmp/pfd_fvbh5iji/6 --gzip --split-3 DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,330 - CMD: fastq-dump -N 40890529 -X 46732032 -O /tmp/pfd_fvbh5iji/7 --gzip --split-3 DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,333 - CMD: fastq-dump -N 46732033 -X 52573536 -O /tmp/pfd_fvbh5iji/8 --gzip --split-3 DNase_r1_pair.sra\n",
      "2023-11-21 19:07:43,335 - CMD: fastq-dump -N 52573537 -X 58415047 -O /tmp/pfd_fvbh5iji/9 --gzip --split-3 DNase_r1_pair.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 5841504 spots for DNase_r1_pair.sra\n",
      "Written 5841504 spots for DNase_r1_pair.sra\n",
      "Read 5841504 spots for DNase_r1_pair.sra\n",
      "Written 5841504 spots for DNase_r1_pair.sra\n",
      "Read 5841504 spots for DNase_r1_pair.sra\n",
      "Written 5841504 spots for DNase_r1_pair.sra\n",
      "Read 5841504 spots for DNase_r1_pair.sra\n",
      "Written 5841504 spots for DNase_r1_pair.sra\n",
      "Read 5841504 spots for DNase_r1_pair.sra\n",
      "Written 5841504 spots for DNase_r1_pair.sra\n",
      "Read 5841504 spots for DNase_r1_pair.sra\n",
      "Written 5841504 spots for DNase_r1_pair.sra\n",
      "Read 5841504 spots for DNase_r1_pair.sra\n",
      "Written 5841504 spots for DNase_r1_pair.sra\n",
      "Read 5841511 spots for DNase_r1_pair.sra\n",
      "Written 5841511 spots for DNase_r1_pair.sra\n",
      "Read 5841504 spots for DNase_r1_pair.sra\n",
      "Written 5841504 spots for DNase_r1_pair.sra\n",
      "Read 5841504 spots for DNase_r1_pair.sra\n",
      "Written 5841504 spots for DNase_r1_pair.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:17:10,428 - SRR ids: ['DNase_r2_pair.sra']\n",
      "2023-11-21 19:17:10,428 - extra args: ['--gzip', '--split-3']\n",
      "2023-11-21 19:17:10,520 - tempdir: /tmp/pfd_svpm0b4l\n",
      "2023-11-21 19:17:10,520 - CMD: sra-stat --meta --quick DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,839 - DNase_r2_pair.sra spots: 44471527\n",
      "2023-11-21 19:17:10,840 - blocks: [[1, 4447152], [4447153, 8894304], [8894305, 13341456], [13341457, 17788608], [17788609, 22235760], [22235761, 26682912], [26682913, 31130064], [31130065, 35577216], [35577217, 40024368], [40024369, 44471527]]\n",
      "2023-11-21 19:17:10,841 - CMD: fastq-dump -N 1 -X 4447152 -O /tmp/pfd_svpm0b4l/0 --gzip --split-3 DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,843 - CMD: fastq-dump -N 4447153 -X 8894304 -O /tmp/pfd_svpm0b4l/1 --gzip --split-3 DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,845 - CMD: fastq-dump -N 8894305 -X 13341456 -O /tmp/pfd_svpm0b4l/2 --gzip --split-3 DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,847 - CMD: fastq-dump -N 13341457 -X 17788608 -O /tmp/pfd_svpm0b4l/3 --gzip --split-3 DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,848 - CMD: fastq-dump -N 17788609 -X 22235760 -O /tmp/pfd_svpm0b4l/4 --gzip --split-3 DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,850 - CMD: fastq-dump -N 22235761 -X 26682912 -O /tmp/pfd_svpm0b4l/5 --gzip --split-3 DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,851 - CMD: fastq-dump -N 26682913 -X 31130064 -O /tmp/pfd_svpm0b4l/6 --gzip --split-3 DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,853 - CMD: fastq-dump -N 31130065 -X 35577216 -O /tmp/pfd_svpm0b4l/7 --gzip --split-3 DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,855 - CMD: fastq-dump -N 35577217 -X 40024368 -O /tmp/pfd_svpm0b4l/8 --gzip --split-3 DNase_r2_pair.sra\n",
      "2023-11-21 19:17:10,857 - CMD: fastq-dump -N 40024369 -X 44471527 -O /tmp/pfd_svpm0b4l/9 --gzip --split-3 DNase_r2_pair.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 4447152 spots for DNase_r2_pair.sra\n",
      "Written 4447152 spots for DNase_r2_pair.sra\n",
      "Read 4447152 spots for DNase_r2_pair.sra\n",
      "Written 4447152 spots for DNase_r2_pair.sra\n",
      "Read 4447152 spots for DNase_r2_pair.sra\n",
      "Written 4447152 spots for DNase_r2_pair.sra\n",
      "Read 4447152 spots for DNase_r2_pair.sra\n",
      "Written 4447152 spots for DNase_r2_pair.sra\n",
      "Read 4447159 spots for DNase_r2_pair.sra\n",
      "Written 4447159 spots for DNase_r2_pair.sra\n",
      "Read 4447152 spots for DNase_r2_pair.sra\n",
      "Written 4447152 spots for DNase_r2_pair.sra\n",
      "Read 4447152 spots for DNase_r2_pair.sra\n",
      "Written 4447152 spots for DNase_r2_pair.sra\n",
      "Read 4447152 spots for DNase_r2_pair.sra\n",
      "Written 4447152 spots for DNase_r2_pair.sra\n",
      "Read 4447152 spots for DNase_r2_pair.sra\n",
      "Written 4447152 spots for DNase_r2_pair.sra\n",
      "Read 4447152 spots for DNase_r2_pair.sra\n",
      "Written 4447152 spots for DNase_r2_pair.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:20:19,648 - SRR ids: ['H3K27ac_r1_single.sra']\n",
      "2023-11-21 19:20:19,648 - extra args: ['--gzip']\n",
      "2023-11-21 19:20:19,691 - tempdir: /tmp/pfd_f9gjoe5p\n",
      "2023-11-21 19:20:19,692 - CMD: sra-stat --meta --quick H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,764 - H3K27ac_r1_single.sra spots: 10825090\n",
      "2023-11-21 19:20:19,764 - blocks: [[1, 1082509], [1082510, 2165018], [2165019, 3247527], [3247528, 4330036], [4330037, 5412545], [5412546, 6495054], [6495055, 7577563], [7577564, 8660072], [8660073, 9742581], [9742582, 10825090]]\n",
      "2023-11-21 19:20:19,765 - CMD: fastq-dump -N 1 -X 1082509 -O /tmp/pfd_f9gjoe5p/0 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,768 - CMD: fastq-dump -N 1082510 -X 2165018 -O /tmp/pfd_f9gjoe5p/1 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,769 - CMD: fastq-dump -N 2165019 -X 3247527 -O /tmp/pfd_f9gjoe5p/2 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,771 - CMD: fastq-dump -N 3247528 -X 4330036 -O /tmp/pfd_f9gjoe5p/3 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,773 - CMD: fastq-dump -N 4330037 -X 5412545 -O /tmp/pfd_f9gjoe5p/4 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,775 - CMD: fastq-dump -N 5412546 -X 6495054 -O /tmp/pfd_f9gjoe5p/5 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,777 - CMD: fastq-dump -N 6495055 -X 7577563 -O /tmp/pfd_f9gjoe5p/6 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,778 - CMD: fastq-dump -N 7577564 -X 8660072 -O /tmp/pfd_f9gjoe5p/7 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,780 - CMD: fastq-dump -N 8660073 -X 9742581 -O /tmp/pfd_f9gjoe5p/8 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:20:19,781 - CMD: fastq-dump -N 9742582 -X 10825090 -O /tmp/pfd_f9gjoe5p/9 --gzip H3K27ac_r1_single.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n",
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n",
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n",
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n",
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n",
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n",
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n",
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n",
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n",
      "Read 1082509 spots for H3K27ac_r1_single.sra\n",
      "Written 1082509 spots for H3K27ac_r1_single.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:21:38,159 - SRR ids: ['H3K27ac_r2_single.sra']\n",
      "2023-11-21 19:21:38,160 - extra args: ['--gzip']\n",
      "2023-11-21 19:21:38,279 - tempdir: /tmp/pfd_boncglcu\n",
      "2023-11-21 19:21:38,280 - CMD: sra-stat --meta --quick H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,361 - H3K27ac_r2_single.sra spots: 7171554\n",
      "2023-11-21 19:21:38,361 - blocks: [[1, 717155], [717156, 1434310], [1434311, 2151465], [2151466, 2868620], [2868621, 3585775], [3585776, 4302930], [4302931, 5020085], [5020086, 5737240], [5737241, 6454395], [6454396, 7171554]]\n",
      "2023-11-21 19:21:38,362 - CMD: fastq-dump -N 1 -X 717155 -O /tmp/pfd_boncglcu/0 --gzip H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,364 - CMD: fastq-dump -N 717156 -X 1434310 -O /tmp/pfd_boncglcu/1 --gzip H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,366 - CMD: fastq-dump -N 1434311 -X 2151465 -O /tmp/pfd_boncglcu/2 --gzip H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,368 - CMD: fastq-dump -N 2151466 -X 2868620 -O /tmp/pfd_boncglcu/3 --gzip H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,370 - CMD: fastq-dump -N 2868621 -X 3585775 -O /tmp/pfd_boncglcu/4 --gzip H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,372 - CMD: fastq-dump -N 3585776 -X 4302930 -O /tmp/pfd_boncglcu/5 --gzip H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,373 - CMD: fastq-dump -N 4302931 -X 5020085 -O /tmp/pfd_boncglcu/6 --gzip H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,375 - CMD: fastq-dump -N 5020086 -X 5737240 -O /tmp/pfd_boncglcu/7 --gzip H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,377 - CMD: fastq-dump -N 5737241 -X 6454395 -O /tmp/pfd_boncglcu/8 --gzip H3K27ac_r2_single.sra\n",
      "2023-11-21 19:21:38,378 - CMD: fastq-dump -N 6454396 -X 7171554 -O /tmp/pfd_boncglcu/9 --gzip H3K27ac_r2_single.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 717155 spots for H3K27ac_r2_single.sra\n",
      "Written 717155 spots for H3K27ac_r2_single.sra\n",
      "Read 717155 spots for H3K27ac_r2_single.sra\n",
      "Written 717155 spots for H3K27ac_r2_single.sra\n",
      "Read 717159 spots for H3K27ac_r2_single.sra\n",
      "Written 717159 spots for H3K27ac_r2_single.sra\n",
      "Read 717155 spots for H3K27ac_r2_single.sra\n",
      "Written 717155 spots for H3K27ac_r2_single.sra\n",
      "Read 717155 spots for H3K27ac_r2_single.sra\n",
      "Written 717155 spots for H3K27ac_r2_single.sra\n",
      "Read 717155 spots for H3K27ac_r2_single.sra\n",
      "Written 717155 spots for H3K27ac_r2_single.sra\n",
      "Read 717155 spots for H3K27ac_r2_single.sra\n",
      "Written 717155 spots for H3K27ac_r2_single.sra\n",
      "Read 717155 spots for H3K27ac_r2_single.sra\n",
      "Written 717155 spots for H3K27ac_r2_single.sra\n",
      "Read 717155 spots for H3K27ac_r2_single.sra\n",
      "Written 717155 spots for H3K27ac_r2_single.sra\n",
      "Read 717155 spots for H3K27ac_r2_single.sra\n",
      "Written 717155 spots for H3K27ac_r2_single.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:22:40,699 - SRR ids: ['H3K4me1_r1_single.sra']\n",
      "2023-11-21 19:22:40,699 - extra args: ['--gzip']\n",
      "2023-11-21 19:22:42,014 - tempdir: /tmp/pfd_72hiz8st\n",
      "2023-11-21 19:22:42,014 - CMD: sra-stat --meta --quick H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,131 - H3K4me1_r1_single.sra spots: 9986477\n",
      "2023-11-21 19:22:42,131 - blocks: [[1, 998647], [998648, 1997294], [1997295, 2995941], [2995942, 3994588], [3994589, 4993235], [4993236, 5991882], [5991883, 6990529], [6990530, 7989176], [7989177, 8987823], [8987824, 9986477]]\n",
      "2023-11-21 19:22:42,132 - CMD: fastq-dump -N 1 -X 998647 -O /tmp/pfd_72hiz8st/0 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,134 - CMD: fastq-dump -N 998648 -X 1997294 -O /tmp/pfd_72hiz8st/1 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,136 - CMD: fastq-dump -N 1997295 -X 2995941 -O /tmp/pfd_72hiz8st/2 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,138 - CMD: fastq-dump -N 2995942 -X 3994588 -O /tmp/pfd_72hiz8st/3 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,140 - CMD: fastq-dump -N 3994589 -X 4993235 -O /tmp/pfd_72hiz8st/4 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,141 - CMD: fastq-dump -N 4993236 -X 5991882 -O /tmp/pfd_72hiz8st/5 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,143 - CMD: fastq-dump -N 5991883 -X 6990529 -O /tmp/pfd_72hiz8st/6 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,145 - CMD: fastq-dump -N 6990530 -X 7989176 -O /tmp/pfd_72hiz8st/7 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,147 - CMD: fastq-dump -N 7989177 -X 8987823 -O /tmp/pfd_72hiz8st/8 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:22:42,148 - CMD: fastq-dump -N 8987824 -X 9986477 -O /tmp/pfd_72hiz8st/9 --gzip H3K4me1_r1_single.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 998647 spots for H3K4me1_r1_single.sra\n",
      "Written 998647 spots for H3K4me1_r1_single.sra\n",
      "Read 998647 spots for H3K4me1_r1_single.sra\n",
      "Written 998647 spots for H3K4me1_r1_single.sra\n",
      "Read 998647 spots for H3K4me1_r1_single.sra\n",
      "Written 998647 spots for H3K4me1_r1_single.sra\n",
      "Read 998647 spots for H3K4me1_r1_single.sra\n",
      "Written 998647 spots for H3K4me1_r1_single.sra\n",
      "Read 998647 spots for H3K4me1_r1_single.sra\n",
      "Written 998647 spots for H3K4me1_r1_single.sra\n",
      "Read 998647 spots for H3K4me1_r1_single.sra\n",
      "Written 998647 spots for H3K4me1_r1_single.sra\n",
      "Read 998654 spots for H3K4me1_r1_single.sra\n",
      "Written 998654 spots for H3K4me1_r1_single.sra\n",
      "Read 998647 spots for H3K4me1_r1_single.sra\n",
      "Written 998647 spots for H3K4me1_r1_single.sra\n",
      "Read 998647 spots for H3K4me1_r1_single.sra\n",
      "Written 998647 spots for H3K4me1_r1_single.sra\n",
      "Read 998647 spots for H3K4me1_r1_single.sra\n",
      "Written 998647 spots for H3K4me1_r1_single.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:25:26,402 - SRR ids: ['H3K4me1_r2_single.sra']\n",
      "2023-11-21 19:25:26,403 - extra args: ['--gzip']\n",
      "2023-11-21 19:25:26,470 - tempdir: /tmp/pfd_eh8goqn5\n",
      "2023-11-21 19:25:26,470 - CMD: sra-stat --meta --quick H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,545 - H3K4me1_r2_single.sra spots: 9748012\n",
      "2023-11-21 19:25:26,545 - blocks: [[1, 974801], [974802, 1949602], [1949603, 2924403], [2924404, 3899204], [3899205, 4874005], [4874006, 5848806], [5848807, 6823607], [6823608, 7798408], [7798409, 8773209], [8773210, 9748012]]\n",
      "2023-11-21 19:25:26,546 - CMD: fastq-dump -N 1 -X 974801 -O /tmp/pfd_eh8goqn5/0 --gzip H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,548 - CMD: fastq-dump -N 974802 -X 1949602 -O /tmp/pfd_eh8goqn5/1 --gzip H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,550 - CMD: fastq-dump -N 1949603 -X 2924403 -O /tmp/pfd_eh8goqn5/2 --gzip H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,552 - CMD: fastq-dump -N 2924404 -X 3899204 -O /tmp/pfd_eh8goqn5/3 --gzip H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,554 - CMD: fastq-dump -N 3899205 -X 4874005 -O /tmp/pfd_eh8goqn5/4 --gzip H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,555 - CMD: fastq-dump -N 4874006 -X 5848806 -O /tmp/pfd_eh8goqn5/5 --gzip H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,557 - CMD: fastq-dump -N 5848807 -X 6823607 -O /tmp/pfd_eh8goqn5/6 --gzip H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,558 - CMD: fastq-dump -N 6823608 -X 7798408 -O /tmp/pfd_eh8goqn5/7 --gzip H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,559 - CMD: fastq-dump -N 7798409 -X 8773209 -O /tmp/pfd_eh8goqn5/8 --gzip H3K4me1_r2_single.sra\n",
      "2023-11-21 19:25:26,560 - CMD: fastq-dump -N 8773210 -X 9748012 -O /tmp/pfd_eh8goqn5/9 --gzip H3K4me1_r2_single.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 974801 spots for H3K4me1_r2_single.sra\n",
      "Written 974801 spots for H3K4me1_r2_single.sra\n",
      "Read 974801 spots for H3K4me1_r2_single.sra\n",
      "Written 974801 spots for H3K4me1_r2_single.sra\n",
      "Read 974801 spots for H3K4me1_r2_single.sra\n",
      "Written 974801 spots for H3K4me1_r2_single.sra\n",
      "Read 974801 spots for H3K4me1_r2_single.sra\n",
      "Written 974801 spots for H3K4me1_r2_single.sra\n",
      "Read 974801 spots for H3K4me1_r2_single.sra\n",
      "Written 974801 spots for H3K4me1_r2_single.sra\n",
      "Read 974801 spots for H3K4me1_r2_single.sra\n",
      "Written 974801 spots for H3K4me1_r2_single.sra\n",
      "Read 974801 spots for H3K4me1_r2_single.sra\n",
      "Written 974801 spots for H3K4me1_r2_single.sra\n",
      "Read 974801 spots for H3K4me1_r2_single.sra\n",
      "Written 974801 spots for H3K4me1_r2_single.sra\n",
      "Read 974803 spots for H3K4me1_r2_single.sra\n",
      "Written 974803 spots for H3K4me1_r2_single.sra\n",
      "Read 974801 spots for H3K4me1_r2_single.sra\n",
      "Written 974801 spots for H3K4me1_r2_single.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:26:41,417 - SRR ids: ['H3K4me3_r1_single.sra']\n",
      "2023-11-21 19:26:41,417 - extra args: ['--gzip']\n",
      "2023-11-21 19:26:41,493 - tempdir: /tmp/pfd_hbyla24e\n",
      "2023-11-21 19:26:41,493 - CMD: sra-stat --meta --quick H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,590 - H3K4me3_r1_single.sra spots: 3720668\n",
      "2023-11-21 19:26:41,590 - blocks: [[1, 372066], [372067, 744132], [744133, 1116198], [1116199, 1488264], [1488265, 1860330], [1860331, 2232396], [2232397, 2604462], [2604463, 2976528], [2976529, 3348594], [3348595, 3720668]]\n",
      "2023-11-21 19:26:41,592 - CMD: fastq-dump -N 1 -X 372066 -O /tmp/pfd_hbyla24e/0 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,594 - CMD: fastq-dump -N 372067 -X 744132 -O /tmp/pfd_hbyla24e/1 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,596 - CMD: fastq-dump -N 744133 -X 1116198 -O /tmp/pfd_hbyla24e/2 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,598 - CMD: fastq-dump -N 1116199 -X 1488264 -O /tmp/pfd_hbyla24e/3 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,600 - CMD: fastq-dump -N 1488265 -X 1860330 -O /tmp/pfd_hbyla24e/4 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,602 - CMD: fastq-dump -N 1860331 -X 2232396 -O /tmp/pfd_hbyla24e/5 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,604 - CMD: fastq-dump -N 2232397 -X 2604462 -O /tmp/pfd_hbyla24e/6 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,606 - CMD: fastq-dump -N 2604463 -X 2976528 -O /tmp/pfd_hbyla24e/7 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,608 - CMD: fastq-dump -N 2976529 -X 3348594 -O /tmp/pfd_hbyla24e/8 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:26:41,609 - CMD: fastq-dump -N 3348595 -X 3720668 -O /tmp/pfd_hbyla24e/9 --gzip H3K4me3_r1_single.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 372066 spots for H3K4me3_r1_single.sra\n",
      "Written 372066 spots for H3K4me3_r1_single.sra\n",
      "Read 372066 spots for H3K4me3_r1_single.sra\n",
      "Written 372066 spots for H3K4me3_r1_single.sra\n",
      "Read 372066 spots for H3K4me3_r1_single.sra\n",
      "Written 372066 spots for H3K4me3_r1_single.sra\n",
      "Read 372066 spots for H3K4me3_r1_single.sra\n",
      "Written 372066 spots for H3K4me3_r1_single.sra\n",
      "Read 372066 spots for H3K4me3_r1_single.sra\n",
      "Written 372066 spots for H3K4me3_r1_single.sra\n",
      "Read 372074 spots for H3K4me3_r1_single.sra\n",
      "Written 372074 spots for H3K4me3_r1_single.sra\n",
      "Read 372066 spots for H3K4me3_r1_single.sra\n",
      "Written 372066 spots for H3K4me3_r1_single.sra\n",
      "Read 372066 spots for H3K4me3_r1_single.sra\n",
      "Written 372066 spots for H3K4me3_r1_single.sra\n",
      "Read 372066 spots for H3K4me3_r1_single.sra\n",
      "Written 372066 spots for H3K4me3_r1_single.sra\n",
      "Read 372066 spots for H3K4me3_r1_single.sra\n",
      "Written 372066 spots for H3K4me3_r1_single.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:26:53,387 - SRR ids: ['H3K4me3_r2_single.sra']\n",
      "2023-11-21 19:26:53,387 - extra args: ['--gzip']\n",
      "2023-11-21 19:26:53,461 - tempdir: /tmp/pfd_nlncmtfc\n",
      "2023-11-21 19:26:53,462 - CMD: sra-stat --meta --quick H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,521 - H3K4me3_r2_single.sra spots: 10581396\n",
      "2023-11-21 19:26:53,521 - blocks: [[1, 1058139], [1058140, 2116278], [2116279, 3174417], [3174418, 4232556], [4232557, 5290695], [5290696, 6348834], [6348835, 7406973], [7406974, 8465112], [8465113, 9523251], [9523252, 10581396]]\n",
      "2023-11-21 19:26:53,522 - CMD: fastq-dump -N 1 -X 1058139 -O /tmp/pfd_nlncmtfc/0 --gzip H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,524 - CMD: fastq-dump -N 1058140 -X 2116278 -O /tmp/pfd_nlncmtfc/1 --gzip H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,526 - CMD: fastq-dump -N 2116279 -X 3174417 -O /tmp/pfd_nlncmtfc/2 --gzip H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,528 - CMD: fastq-dump -N 3174418 -X 4232556 -O /tmp/pfd_nlncmtfc/3 --gzip H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,530 - CMD: fastq-dump -N 4232557 -X 5290695 -O /tmp/pfd_nlncmtfc/4 --gzip H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,531 - CMD: fastq-dump -N 5290696 -X 6348834 -O /tmp/pfd_nlncmtfc/5 --gzip H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,533 - CMD: fastq-dump -N 6348835 -X 7406973 -O /tmp/pfd_nlncmtfc/6 --gzip H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,534 - CMD: fastq-dump -N 7406974 -X 8465112 -O /tmp/pfd_nlncmtfc/7 --gzip H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,537 - CMD: fastq-dump -N 8465113 -X 9523251 -O /tmp/pfd_nlncmtfc/8 --gzip H3K4me3_r2_single.sra\n",
      "2023-11-21 19:26:53,539 - CMD: fastq-dump -N 9523252 -X 10581396 -O /tmp/pfd_nlncmtfc/9 --gzip H3K4me3_r2_single.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 1058139 spots for H3K4me3_r2_single.sra\n",
      "Written 1058139 spots for H3K4me3_r2_single.sra\n",
      "Read 1058139 spots for H3K4me3_r2_single.sra\n",
      "Written 1058139 spots for H3K4me3_r2_single.sra\n",
      "Read 1058139 spots for H3K4me3_r2_single.sra\n",
      "Written 1058139 spots for H3K4me3_r2_single.sra\n",
      "Read 1058139 spots for H3K4me3_r2_single.sra\n",
      "Written 1058139 spots for H3K4me3_r2_single.sra\n",
      "Read 1058139 spots for H3K4me3_r2_single.sra\n",
      "Written 1058139 spots for H3K4me3_r2_single.sra\n",
      "Read 1058139 spots for H3K4me3_r2_single.sra\n",
      "Written 1058139 spots for H3K4me3_r2_single.sra\n",
      "Read 1058139 spots for H3K4me3_r2_single.sra\n",
      "Written 1058139 spots for H3K4me3_r2_single.sra\n",
      "Read 1058139 spots for H3K4me3_r2_single.sra\n",
      "Written 1058139 spots for H3K4me3_r2_single.sra\n",
      "Read 1058139 spots for H3K4me3_r2_single.sra\n",
      "Written 1058139 spots for H3K4me3_r2_single.sra\n",
      "Read 1058145 spots for H3K4me3_r2_single.sra\n",
      "Written 1058145 spots for H3K4me3_r2_single.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> DNase_r1_pair_1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t31\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t8\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "Using Nextera adapter for trimming (count: 31). Second best hit was smallRNA (count: 8)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/DNase_r1_pair_1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: DNase_r1_pair_1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to DNase_r1_pair_1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'CTGTCTCTTATA' from file DNase_r1_pair_1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a CTGTCTCTTATA DNase_r1_pair_1.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 503.563 s (8.620 µs/read; 6.96 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              58,415,047\n",
      "Reads with adapters:                 1,243,716 (2.1%)\n",
      "Reads written (passing filters):    58,415,047 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,869,281,504 bp\n",
      "Quality-trimmed:              31,241,966 bp (1.7%)\n",
      "Total written (filtered):  1,833,712,282 bp (98.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1243716 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 26.7%\n",
      "  C: 18.8%\n",
      "  G: 20.2%\n",
      "  T: 34.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t916751\t912735.1\t0\t916751\n",
      "4\t225866\t228183.8\t0\t225866\n",
      "5\t57210\t57045.9\t0\t57210\n",
      "6\t22165\t14261.5\t0\t22165\n",
      "7\t6605\t3565.4\t0\t6605\n",
      "8\t2769\t891.3\t0\t2769\n",
      "9\t2496\t222.8\t0\t2229 267\n",
      "10\t1472\t55.7\t1\t15 1457\n",
      "11\t1984\t13.9\t1\t1004 980\n",
      "12\t299\t3.5\t1\t7 292\n",
      "13\t913\t3.5\t1\t708 205\n",
      "14\t263\t3.5\t1\t6 257\n",
      "15\t503\t3.5\t1\t354 149\n",
      "16\t398\t3.5\t1\t2 396\n",
      "17\t431\t3.5\t1\t184 247\n",
      "18\t296\t3.5\t1\t96 200\n",
      "19\t304\t3.5\t1\t25 279\n",
      "20\t221\t3.5\t1\t9 212\n",
      "21\t205\t3.5\t1\t11 194\n",
      "22\t355\t3.5\t1\t22 333\n",
      "23\t261\t3.5\t1\t2 259\n",
      "24\t393\t3.5\t1\t1 392\n",
      "25\t271\t3.5\t1\t8 263\n",
      "26\t134\t3.5\t1\t2 132\n",
      "27\t183\t3.5\t1\t2 181\n",
      "28\t198\t3.5\t1\t0 198\n",
      "29\t390\t3.5\t1\t8 382\n",
      "30\t234\t3.5\t1\t34 200\n",
      "31\t95\t3.5\t1\t8 87\n",
      "32\t51\t3.5\t1\t0 51\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: DNase_r1_pair_1.fastq.gz\n",
      "=============================================\n",
      "58415047 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/DNase_r1_pair_2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: DNase_r1_pair_2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j -j 2\n",
      "Writing final adapter and quality trimmed output to DNase_r1_pair_2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'CTGTCTCTTATA' from file DNase_r1_pair_2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a CTGTCTCTTATA DNase_r1_pair_2.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 520.957 s (8.918 µs/read; 6.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              58,415,047\n",
      "Reads with adapters:                 1,226,041 (2.1%)\n",
      "Reads written (passing filters):    58,415,047 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,869,281,504 bp\n",
      "Quality-trimmed:              53,456,330 bp (2.9%)\n",
      "Total written (filtered):  1,811,548,316 bp (96.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 1226041 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 26.4%\n",
      "  C: 18.6%\n",
      "  G: 20.2%\n",
      "  T: 34.8%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t899808\t912735.1\t0\t899808\n",
      "4\t225709\t228183.8\t0\t225709\n",
      "5\t56857\t57045.9\t0\t56857\n",
      "6\t21793\t14261.5\t0\t21793\n",
      "7\t6500\t3565.4\t0\t6500\n",
      "8\t2913\t891.3\t0\t2913\n",
      "9\t2321\t222.8\t0\t2032 289\n",
      "10\t1493\t55.7\t1\t11 1482\n",
      "11\t1900\t13.9\t1\t906 994\n",
      "12\t329\t3.5\t1\t6 323\n",
      "13\t923\t3.5\t1\t674 249\n",
      "14\t247\t3.5\t1\t7 240\n",
      "15\t526\t3.5\t1\t341 185\n",
      "16\t405\t3.5\t1\t4 401\n",
      "17\t473\t3.5\t1\t166 307\n",
      "18\t317\t3.5\t1\t90 227\n",
      "19\t358\t3.5\t1\t22 336\n",
      "20\t249\t3.5\t1\t20 229\n",
      "21\t239\t3.5\t1\t1 238\n",
      "22\t367\t3.5\t1\t9 358\n",
      "23\t283\t3.5\t1\t12 271\n",
      "24\t470\t3.5\t1\t7 463\n",
      "25\t252\t3.5\t1\t4 248\n",
      "26\t174\t3.5\t1\t2 172\n",
      "27\t177\t3.5\t1\t2 175\n",
      "28\t210\t3.5\t1\t3 207\n",
      "29\t352\t3.5\t1\t5 347\n",
      "30\t209\t3.5\t1\t2 207\n",
      "31\t99\t3.5\t1\t5 94\n",
      "32\t88\t3.5\t1\t33 55\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: DNase_r1_pair_2.fastq.gz\n",
      "=============================================\n",
      "58415047 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files DNase_r1_pair_1_trimmed.fq.gz and DNase_r1_pair_2_trimmed.fq.gz\n",
      "file_1: DNase_r1_pair_1_trimmed.fq.gz, file_2: DNase_r1_pair_2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: DNase_r1_pair_1_trimmed.fq.gz and DNase_r1_pair_2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end Read 1 reads to DNase_r1_pair_1_val_1.fq.gz\n",
      "Writing validated paired-end Read 2 reads to DNase_r1_pair_2_val_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 58415047\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 792319 (1.36%)\n",
      "\n",
      "Deleting both intermediate output files DNase_r1_pair_1_trimmed.fq.gz and DNase_r1_pair_2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> DNase_r2_pair_1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t25\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t9\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Illumina\t1\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "Using Nextera adapter for trimming (count: 25). Second best hit was smallRNA (count: 9)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/DNase_r2_pair_1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: DNase_r2_pair_1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to DNase_r2_pair_1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'CTGTCTCTTATA' from file DNase_r2_pair_1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a CTGTCTCTTATA DNase_r2_pair_1.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 383.790 s (8.630 µs/read; 6.95 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              44,471,527\n",
      "Reads with adapters:                   923,381 (2.1%)\n",
      "Reads written (passing filters):    44,471,527 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,423,088,864 bp\n",
      "Quality-trimmed:              23,958,907 bp (1.7%)\n",
      "Total written (filtered):  1,395,896,569 bp (98.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 923381 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 26.5%\n",
      "  C: 19.2%\n",
      "  G: 19.5%\n",
      "  T: 34.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t676507\t694867.6\t0\t676507\n",
      "4\t169243\t173716.9\t0\t169243\n",
      "5\t41902\t43429.2\t0\t41902\n",
      "6\t17718\t10857.3\t0\t17718\n",
      "7\t5145\t2714.3\t0\t5145\n",
      "8\t2258\t678.6\t0\t2258\n",
      "9\t2154\t169.6\t0\t1956 198\n",
      "10\t1197\t42.4\t1\t16 1181\n",
      "11\t2070\t10.6\t1\t1274 796\n",
      "12\t273\t2.7\t1\t13 260\n",
      "13\t740\t2.7\t1\t577 163\n",
      "14\t244\t2.7\t1\t4 240\n",
      "15\t381\t2.7\t1\t252 129\n",
      "16\t331\t2.7\t1\t3 328\n",
      "17\t371\t2.7\t1\t128 243\n",
      "18\t184\t2.7\t1\t42 142\n",
      "19\t267\t2.7\t1\t16 251\n",
      "20\t141\t2.7\t1\t5 136\n",
      "21\t172\t2.7\t1\t5 167\n",
      "22\t269\t2.7\t1\t27 242\n",
      "23\t195\t2.7\t1\t2 193\n",
      "24\t340\t2.7\t1\t3 337\n",
      "25\t230\t2.7\t1\t5 225\n",
      "26\t128\t2.7\t1\t1 127\n",
      "27\t148\t2.7\t1\t3 145\n",
      "28\t139\t2.7\t1\t0 139\n",
      "29\t321\t2.7\t1\t17 304\n",
      "30\t174\t2.7\t1\t25 149\n",
      "31\t81\t2.7\t1\t2 79\n",
      "32\t58\t2.7\t1\t4 54\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: DNase_r2_pair_1.fastq.gz\n",
      "=============================================\n",
      "44471527 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/DNase_r2_pair_2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: DNase_r2_pair_2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j -j 2\n",
      "Writing final adapter and quality trimmed output to DNase_r2_pair_2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'CTGTCTCTTATA' from file DNase_r2_pair_2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a CTGTCTCTTATA DNase_r2_pair_2.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 379.296 s (8.529 µs/read; 7.03 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              44,471,527\n",
      "Reads with adapters:                   904,344 (2.0%)\n",
      "Reads written (passing filters):    44,471,527 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,423,088,864 bp\n",
      "Quality-trimmed:              41,002,421 bp (2.9%)\n",
      "Total written (filtered):  1,378,909,675 bp (96.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 904344 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 26.4%\n",
      "  C: 18.8%\n",
      "  G: 19.6%\n",
      "  T: 35.1%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t657700\t694867.6\t0\t657700\n",
      "4\t169558\t173716.9\t0\t169558\n",
      "5\t41471\t43429.2\t0\t41471\n",
      "6\t17593\t10857.3\t0\t17593\n",
      "7\t5149\t2714.3\t0\t5149\n",
      "8\t2335\t678.6\t0\t2335\n",
      "9\t2044\t169.6\t0\t1841 203\n",
      "10\t1100\t42.4\t1\t19 1081\n",
      "11\t2067\t10.6\t1\t1191 876\n",
      "12\t232\t2.7\t1\t6 226\n",
      "13\t775\t2.7\t1\t552 223\n",
      "14\t233\t2.7\t1\t1 232\n",
      "15\t397\t2.7\t1\t241 156\n",
      "16\t325\t2.7\t1\t5 320\n",
      "17\t360\t2.7\t1\t115 245\n",
      "18\t204\t2.7\t1\t45 159\n",
      "19\t316\t2.7\t1\t16 300\n",
      "20\t167\t2.7\t1\t9 158\n",
      "21\t221\t2.7\t1\t0 221\n",
      "22\t274\t2.7\t1\t6 268\n",
      "23\t215\t2.7\t1\t13 202\n",
      "24\t402\t2.7\t1\t11 391\n",
      "25\t179\t2.7\t1\t3 176\n",
      "26\t129\t2.7\t1\t1 128\n",
      "27\t132\t2.7\t1\t3 129\n",
      "28\t149\t2.7\t1\t2 147\n",
      "29\t308\t2.7\t1\t14 294\n",
      "30\t156\t2.7\t1\t1 155\n",
      "31\t80\t2.7\t1\t3 77\n",
      "32\t73\t2.7\t1\t27 46\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: DNase_r2_pair_2.fastq.gz\n",
      "=============================================\n",
      "44471527 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files DNase_r2_pair_1_trimmed.fq.gz and DNase_r2_pair_2_trimmed.fq.gz\n",
      "file_1: DNase_r2_pair_1_trimmed.fq.gz, file_2: DNase_r2_pair_2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: DNase_r2_pair_1_trimmed.fq.gz and DNase_r2_pair_2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end Read 1 reads to DNase_r2_pair_1_val_1.fq.gz\n",
      "Writing validated paired-end Read 2 reads to DNase_r2_pair_2_val_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 44471527\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 608482 (1.37%)\n",
      "\n",
      "Deleting both intermediate output files DNase_r2_pair_1_trimmed.fq.gz and DNase_r2_pair_2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> H3K27ac_r1_single.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t538\tAGATCGGAAGAGC\t1000000\t0.05\n",
      "Nextera\t3\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 538). Second best hit was Nextera (count: 3)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/H3K27ac_r1_single.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: H3K27ac_r1_single.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to H3K27ac_r1_single_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file H3K27ac_r1_single.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a AGATCGGAAGAGC H3K27ac_r1_single.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 150.725 s (13.924 µs/read; 4.31 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              10,825,090\n",
      "Reads with adapters:                   470,014 (4.3%)\n",
      "Reads written (passing filters):    10,825,090 (100.0%)\n",
      "\n",
      "Total basepairs processed:   552,079,590 bp\n",
      "Quality-trimmed:               1,291,763 bp (0.2%)\n",
      "Total written (filtered):    539,268,999 bp (97.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 470014 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.7%\n",
      "  C: 12.9%\n",
      "  G: 14.4%\n",
      "  T: 11.0%\n",
      "  none/other: 44.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t193251\t169142.0\t0\t193251\n",
      "4\t45747\t42285.5\t0\t45747\n",
      "5\t10586\t10571.4\t0\t10586\n",
      "6\t2468\t2642.8\t0\t2468\n",
      "7\t993\t660.7\t0\t993\n",
      "8\t598\t165.2\t0\t598\n",
      "9\t838\t41.3\t0\t543 295\n",
      "10\t942\t10.3\t1\t447 495\n",
      "11\t634\t2.6\t1\t429 205\n",
      "12\t534\t0.6\t1\t446 88\n",
      "13\t436\t0.2\t1\t387 49\n",
      "14\t378\t0.2\t1\t358 20\n",
      "15\t352\t0.2\t1\t318 34\n",
      "16\t338\t0.2\t1\t317 21\n",
      "17\t345\t0.2\t1\t321 24\n",
      "18\t325\t0.2\t1\t301 24\n",
      "19\t271\t0.2\t1\t246 25\n",
      "20\t265\t0.2\t1\t241 24\n",
      "21\t253\t0.2\t1\t231 22\n",
      "22\t250\t0.2\t1\t222 28\n",
      "23\t233\t0.2\t1\t204 29\n",
      "24\t222\t0.2\t1\t188 34\n",
      "25\t270\t0.2\t1\t188 82\n",
      "26\t220\t0.2\t1\t148 72\n",
      "27\t264\t0.2\t1\t161 103\n",
      "28\t224\t0.2\t1\t136 88\n",
      "29\t202\t0.2\t1\t138 64\n",
      "30\t148\t0.2\t1\t112 36\n",
      "31\t148\t0.2\t1\t90 58\n",
      "32\t161\t0.2\t1\t90 71\n",
      "33\t900\t0.2\t1\t122 778\n",
      "34\t616\t0.2\t1\t99 517\n",
      "35\t368\t0.2\t1\t93 275\n",
      "36\t284\t0.2\t1\t64 220\n",
      "37\t215\t0.2\t1\t94 121\n",
      "38\t193\t0.2\t1\t55 138\n",
      "39\t397\t0.2\t1\t90 307\n",
      "40\t236\t0.2\t1\t87 149\n",
      "41\t196\t0.2\t1\t70 126\n",
      "42\t179\t0.2\t1\t65 114\n",
      "43\t212\t0.2\t1\t83 129\n",
      "44\t923\t0.2\t1\t70 853\n",
      "45\t805\t0.2\t1\t83 722\n",
      "46\t498\t0.2\t1\t75 423\n",
      "47\t398\t0.2\t1\t87 311\n",
      "48\t750\t0.2\t1\t55 695\n",
      "49\t1186\t0.2\t1\t67 1119\n",
      "50\t4007\t0.2\t1\t58 3949\n",
      "51\t195755\t0.2\t1\t99 195656\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: H3K27ac_r1_single.fastq.gz\n",
      "=============================================\n",
      "10825090 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t209247 (1.9%)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> H3K27ac_r2_single.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t532\tAGATCGGAAGAGC\t1000000\t0.05\n",
      "smallRNA\t4\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 532). Second best hit was smallRNA (count: 4)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/H3K27ac_r2_single.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: H3K27ac_r2_single.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to H3K27ac_r2_single_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file H3K27ac_r2_single.fastq.gz <<< \n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a AGATCGGAAGAGC H3K27ac_r2_single.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 100.604 s (14.028 µs/read; 4.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,171,554\n",
      "Reads with adapters:                   238,407 (3.3%)\n",
      "Reads written (passing filters):     7,171,554 (100.0%)\n",
      "\n",
      "Total basepairs processed:   365,749,254 bp\n",
      "Quality-trimmed:                 829,303 bp (0.2%)\n",
      "Total written (filtered):    361,179,327 bp (98.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 238407 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 23.8%\n",
      "  C: 17.2%\n",
      "  G: 19.4%\n",
      "  T: 14.6%\n",
      "  none/other: 25.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t130773\t112055.5\t0\t130773\n",
      "4\t30987\t28013.9\t0\t30987\n",
      "5\t7301\t7003.5\t0\t7301\n",
      "6\t1774\t1750.9\t0\t1774\n",
      "7\t726\t437.7\t0\t726\n",
      "8\t499\t109.4\t0\t499\n",
      "9\t543\t27.4\t0\t371 172\n",
      "10\t635\t6.8\t1\t343 292\n",
      "11\t445\t1.7\t1\t316 129\n",
      "12\t374\t0.4\t1\t317 57\n",
      "13\t327\t0.1\t1\t311 16\n",
      "14\t280\t0.1\t1\t261 19\n",
      "15\t273\t0.1\t1\t255 18\n",
      "16\t239\t0.1\t1\t216 23\n",
      "17\t281\t0.1\t1\t255 26\n",
      "18\t224\t0.1\t1\t210 14\n",
      "19\t189\t0.1\t1\t170 19\n",
      "20\t197\t0.1\t1\t184 13\n",
      "21\t186\t0.1\t1\t172 14\n",
      "22\t163\t0.1\t1\t144 19\n",
      "23\t172\t0.1\t1\t152 20\n",
      "24\t117\t0.1\t1\t98 19\n",
      "25\t146\t0.1\t1\t123 23\n",
      "26\t114\t0.1\t1\t92 22\n",
      "27\t143\t0.1\t1\t111 32\n",
      "28\t117\t0.1\t1\t81 36\n",
      "29\t98\t0.1\t1\t72 26\n",
      "30\t91\t0.1\t1\t72 19\n",
      "31\t87\t0.1\t1\t60 27\n",
      "32\t92\t0.1\t1\t68 24\n",
      "33\t169\t0.1\t1\t55 114\n",
      "34\t156\t0.1\t1\t57 99\n",
      "35\t115\t0.1\t1\t66 49\n",
      "36\t123\t0.1\t1\t45 78\n",
      "37\t193\t0.1\t1\t67 126\n",
      "38\t116\t0.1\t1\t47 69\n",
      "39\t115\t0.1\t1\t51 64\n",
      "40\t95\t0.1\t1\t54 41\n",
      "41\t93\t0.1\t1\t58 35\n",
      "42\t72\t0.1\t1\t37 35\n",
      "43\t113\t0.1\t1\t61 52\n",
      "44\t184\t0.1\t1\t66 118\n",
      "45\t237\t0.1\t1\t63 174\n",
      "46\t183\t0.1\t1\t46 137\n",
      "47\t182\t0.1\t1\t54 128\n",
      "48\t222\t0.1\t1\t39 183\n",
      "49\t398\t0.1\t1\t40 358\n",
      "50\t1339\t0.1\t1\t47 1292\n",
      "51\t56709\t0.1\t1\t88 56621\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: H3K27ac_r2_single.fastq.gz\n",
      "=============================================\n",
      "7171554 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t61379 (0.9%)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> H3K4me1_r1_single.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t117\tAGATCGGAAGAGC\t1000000\t0.01\n",
      "smallRNA\t6\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t2\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 117). Second best hit was smallRNA (count: 6)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/H3K4me1_r1_single.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: H3K4me1_r1_single.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to H3K4me1_r1_single_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file H3K4me1_r1_single.fastq.gz <<< \n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a AGATCGGAAGAGC H3K4me1_r1_single.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 178.771 s (17.901 µs/read; 3.35 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               9,986,477\n",
      "Reads with adapters:                   260,609 (2.6%)\n",
      "Reads written (passing filters):     9,986,477 (100.0%)\n",
      "\n",
      "Total basepairs processed:   509,310,327 bp\n",
      "Quality-trimmed:               2,261,275 bp (0.4%)\n",
      "Total written (filtered):    505,392,242 bp (99.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 260609 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.4%\n",
      "  C: 21.3%\n",
      "  G: 23.9%\n",
      "  T: 18.3%\n",
      "  none/other: 6.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t186548\t156038.7\t0\t186548\n",
      "4\t43111\t39009.7\t0\t43111\n",
      "5\t9516\t9752.4\t0\t9516\n",
      "6\t1606\t2438.1\t0\t1606\n",
      "7\t502\t609.5\t0\t502\n",
      "8\t243\t152.4\t0\t243\n",
      "9\t431\t38.1\t0\t150 281\n",
      "10\t561\t9.5\t1\t104 457\n",
      "11\t306\t2.4\t1\t88 218\n",
      "12\t178\t0.6\t1\t100 78\n",
      "13\t96\t0.1\t1\t69 27\n",
      "14\t97\t0.1\t1\t80 17\n",
      "15\t88\t0.1\t1\t74 14\n",
      "16\t79\t0.1\t1\t58 21\n",
      "17\t73\t0.1\t1\t50 23\n",
      "18\t65\t0.1\t1\t48 17\n",
      "19\t74\t0.1\t1\t57 17\n",
      "20\t65\t0.1\t1\t50 15\n",
      "21\t70\t0.1\t1\t48 22\n",
      "22\t48\t0.1\t1\t33 15\n",
      "23\t56\t0.1\t1\t40 16\n",
      "24\t47\t0.1\t1\t29 18\n",
      "25\t48\t0.1\t1\t28 20\n",
      "26\t61\t0.1\t1\t33 28\n",
      "27\t49\t0.1\t1\t25 24\n",
      "28\t57\t0.1\t1\t29 28\n",
      "29\t46\t0.1\t1\t25 21\n",
      "30\t44\t0.1\t1\t19 25\n",
      "31\t41\t0.1\t1\t14 27\n",
      "32\t40\t0.1\t1\t13 27\n",
      "33\t77\t0.1\t1\t13 64\n",
      "34\t76\t0.1\t1\t23 53\n",
      "35\t45\t0.1\t1\t9 36\n",
      "36\t60\t0.1\t1\t16 44\n",
      "37\t48\t0.1\t1\t10 38\n",
      "38\t58\t0.1\t1\t13 45\n",
      "39\t55\t0.1\t1\t18 37\n",
      "40\t71\t0.1\t1\t17 54\n",
      "41\t59\t0.1\t1\t12 47\n",
      "42\t65\t0.1\t1\t11 54\n",
      "43\t101\t0.1\t1\t10 91\n",
      "44\t160\t0.1\t1\t21 139\n",
      "45\t168\t0.1\t1\t24 144\n",
      "46\t151\t0.1\t1\t19 132\n",
      "47\t135\t0.1\t1\t16 119\n",
      "48\t144\t0.1\t1\t16 128\n",
      "49\t181\t0.1\t1\t32 149\n",
      "50\t250\t0.1\t1\t20 230\n",
      "51\t14459\t0.1\t1\t62 14397\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: H3K4me1_r1_single.fastq.gz\n",
      "=============================================\n",
      "9986477 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t18601 (0.2%)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> H3K4me1_r2_single.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t7532\tAGATCGGAAGAGC\t1000000\t0.75\n",
      "smallRNA\t9\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 7532). Second best hit was smallRNA (count: 9)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/H3K4me1_r2_single.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: H3K4me1_r2_single.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to H3K4me1_r2_single_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file H3K4me1_r2_single.fastq.gz <<< \n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a AGATCGGAAGAGC H3K4me1_r2_single.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 205.357 s (21.067 µs/read; 2.85 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               9,748,012\n",
      "Reads with adapters:                   393,658 (4.0%)\n",
      "Reads written (passing filters):     9,748,012 (100.0%)\n",
      "\n",
      "Total basepairs processed:   740,848,912 bp\n",
      "Quality-trimmed:               1,531,243 bp (0.2%)\n",
      "Total written (filtered):    733,446,103 bp (99.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 393658 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.3%\n",
      "  C: 18.2%\n",
      "  G: 23.7%\n",
      "  T: 21.5%\n",
      "  none/other: 9.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t185919\t152312.7\t0\t185919\n",
      "4\t47609\t38078.2\t0\t47609\n",
      "5\t14309\t9519.5\t0\t14309\n",
      "6\t7207\t2379.9\t0\t7207\n",
      "7\t5879\t595.0\t0\t5879\n",
      "8\t5484\t148.7\t0\t5484\n",
      "9\t5732\t37.2\t0\t5500 232\n",
      "10\t5193\t9.3\t1\t4666 527\n",
      "11\t4974\t2.3\t1\t4614 360\n",
      "12\t4365\t0.6\t1\t4174 191\n",
      "13\t4432\t0.1\t1\t4228 204\n",
      "14\t4064\t0.1\t1\t3876 188\n",
      "15\t3835\t0.1\t1\t3667 168\n",
      "16\t3712\t0.1\t1\t3524 188\n",
      "17\t3629\t0.1\t1\t3448 181\n",
      "18\t3484\t0.1\t1\t3315 169\n",
      "19\t3057\t0.1\t1\t2911 146\n",
      "20\t2994\t0.1\t1\t2837 157\n",
      "21\t2839\t0.1\t1\t2701 138\n",
      "22\t2670\t0.1\t1\t2547 123\n",
      "23\t2436\t0.1\t1\t2317 119\n",
      "24\t2355\t0.1\t1\t2224 131\n",
      "25\t2218\t0.1\t1\t2105 113\n",
      "26\t2041\t0.1\t1\t1956 85\n",
      "27\t1919\t0.1\t1\t1831 88\n",
      "28\t1852\t0.1\t1\t1762 90\n",
      "29\t1592\t0.1\t1\t1519 73\n",
      "30\t1425\t0.1\t1\t1351 74\n",
      "31\t1278\t0.1\t1\t1214 64\n",
      "32\t1281\t0.1\t1\t1204 77\n",
      "33\t1179\t0.1\t1\t1100 79\n",
      "34\t1088\t0.1\t1\t1033 55\n",
      "35\t1016\t0.1\t1\t954 62\n",
      "36\t909\t0.1\t1\t850 59\n",
      "37\t759\t0.1\t1\t711 48\n",
      "38\t671\t0.1\t1\t633 38\n",
      "39\t633\t0.1\t1\t574 59\n",
      "40\t568\t0.1\t1\t524 44\n",
      "41\t524\t0.1\t1\t472 52\n",
      "42\t484\t0.1\t1\t442 42\n",
      "43\t584\t0.1\t1\t543 41\n",
      "44\t197\t0.1\t1\t157 40\n",
      "45\t231\t0.1\t1\t213 18\n",
      "46\t235\t0.1\t1\t203 32\n",
      "47\t235\t0.1\t1\t207 28\n",
      "48\t221\t0.1\t1\t190 31\n",
      "49\t178\t0.1\t1\t156 22\n",
      "50\t209\t0.1\t1\t178 31\n",
      "51\t197\t0.1\t1\t176 21\n",
      "52\t201\t0.1\t1\t177 24\n",
      "53\t245\t0.1\t1\t216 29\n",
      "54\t212\t0.1\t1\t178 34\n",
      "55\t246\t0.1\t1\t225 21\n",
      "56\t172\t0.1\t1\t150 22\n",
      "57\t193\t0.1\t1\t177 16\n",
      "58\t189\t0.1\t1\t173 16\n",
      "59\t209\t0.1\t1\t184 25\n",
      "60\t230\t0.1\t1\t210 20\n",
      "61\t198\t0.1\t1\t177 21\n",
      "62\t209\t0.1\t1\t189 20\n",
      "63\t229\t0.1\t1\t196 33\n",
      "64\t268\t0.1\t1\t241 27\n",
      "65\t325\t0.1\t1\t273 52\n",
      "66\t285\t0.1\t1\t252 33\n",
      "67\t337\t0.1\t1\t290 47\n",
      "68\t388\t0.1\t1\t330 58\n",
      "69\t421\t0.1\t1\t349 72\n",
      "70\t481\t0.1\t1\t377 104\n",
      "71\t577\t0.1\t1\t415 162\n",
      "72\t783\t0.1\t1\t487 296\n",
      "73\t1040\t0.1\t1\t417 623\n",
      "74\t1625\t0.1\t1\t479 1146\n",
      "75\t10862\t0.1\t1\t572 10290\n",
      "76\t23831\t0.1\t1\t980 22851\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: H3K4me1_r2_single.fastq.gz\n",
      "=============================================\n",
      "9748012 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t42796 (0.4%)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> H3K4me3_r1_single.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t350\tAGATCGGAAGAGC\t1000000\t0.03\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 350). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/H3K4me3_r1_single.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: H3K4me3_r1_single.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to H3K4me3_r1_single_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file H3K4me3_r1_single.fastq.gz <<< \n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a AGATCGGAAGAGC H3K4me3_r1_single.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 49.423 s (13.283 µs/read; 4.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               3,720,668\n",
      "Reads with adapters:                   269,216 (7.2%)\n",
      "Reads written (passing filters):     3,720,668 (100.0%)\n",
      "\n",
      "Total basepairs processed:   189,754,068 bp\n",
      "Quality-trimmed:                 424,653 bp (0.2%)\n",
      "Total written (filtered):    179,696,538 bp (94.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 269216 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 10.1%\n",
      "  C: 7.4%\n",
      "  G: 7.9%\n",
      "  T: 6.6%\n",
      "  none/other: 68.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t64959\t58135.4\t0\t64959\n",
      "4\t14849\t14533.9\t0\t14849\n",
      "5\t2957\t3633.5\t0\t2957\n",
      "6\t627\t908.4\t0\t627\n",
      "7\t267\t227.1\t0\t267\n",
      "8\t158\t56.8\t0\t158\n",
      "9\t245\t14.2\t0\t160 85\n",
      "10\t257\t3.5\t1\t133 124\n",
      "11\t178\t0.9\t1\t125 53\n",
      "12\t124\t0.2\t1\t107 17\n",
      "13\t136\t0.1\t1\t116 20\n",
      "14\t127\t0.1\t1\t120 7\n",
      "15\t123\t0.1\t1\t120 3\n",
      "16\t94\t0.1\t1\t88 6\n",
      "17\t95\t0.1\t1\t89 6\n",
      "18\t100\t0.1\t1\t89 11\n",
      "19\t76\t0.1\t1\t71 5\n",
      "20\t71\t0.1\t1\t66 5\n",
      "21\t108\t0.1\t1\t59 49\n",
      "22\t52\t0.1\t1\t43 9\n",
      "23\t58\t0.1\t1\t47 11\n",
      "24\t57\t0.1\t1\t47 10\n",
      "25\t66\t0.1\t1\t50 16\n",
      "26\t43\t0.1\t1\t23 20\n",
      "27\t43\t0.1\t1\t20 23\n",
      "28\t37\t0.1\t1\t24 13\n",
      "29\t45\t0.1\t1\t20 25\n",
      "30\t30\t0.1\t1\t20 10\n",
      "31\t38\t0.1\t1\t20 18\n",
      "32\t26\t0.1\t1\t13 13\n",
      "33\t136\t0.1\t1\t14 122\n",
      "34\t63\t0.1\t1\t9 54\n",
      "35\t39\t0.1\t1\t14 25\n",
      "36\t41\t0.1\t1\t7 34\n",
      "37\t68\t0.1\t1\t8 60\n",
      "38\t69\t0.1\t1\t11 58\n",
      "39\t30\t0.1\t1\t6 24\n",
      "40\t58\t0.1\t1\t12 46\n",
      "41\t86\t0.1\t1\t9 77\n",
      "42\t92\t0.1\t1\t9 83\n",
      "43\t119\t0.1\t1\t12 107\n",
      "44\t332\t0.1\t1\t11 321\n",
      "45\t356\t0.1\t1\t5 351\n",
      "46\t267\t0.1\t1\t7 260\n",
      "47\t430\t0.1\t1\t12 418\n",
      "48\t539\t0.1\t1\t11 528\n",
      "49\t793\t0.1\t1\t3 790\n",
      "50\t472\t0.1\t1\t3 469\n",
      "51\t179180\t0.1\t1\t14 179166\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: H3K4me3_r1_single.fastq.gz\n",
      "=============================================\n",
      "3720668 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t184485 (5.0%)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> H3K4me3_r2_single.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t185\tAGATCGGAAGAGC\t1000000\t0.02\n",
      "smallRNA\t7\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t2\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 185). Second best hit was smallRNA (count: 7)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/ce10/N2_mixed_embryo/H3K4me3_r2_single.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: H3K4me3_r2_single.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to H3K4me3_r2_single_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file H3K4me3_r2_single.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a AGATCGGAAGAGC H3K4me3_r2_single.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 147.071 s (13.899 µs/read; 4.32 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              10,581,396\n",
      "Reads with adapters:                   437,302 (4.1%)\n",
      "Reads written (passing filters):    10,581,396 (100.0%)\n",
      "\n",
      "Total basepairs processed:   539,651,196 bp\n",
      "Quality-trimmed:               2,333,756 bp (0.4%)\n",
      "Total written (filtered):    527,526,948 bp (97.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 437302 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 19.3%\n",
      "  C: 13.9%\n",
      "  G: 15.1%\n",
      "  T: 11.6%\n",
      "  none/other: 40.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t198268\t165334.3\t0\t198268\n",
      "4\t45907\t41333.6\t0\t45907\n",
      "5\t10124\t10333.4\t0\t10124\n",
      "6\t1808\t2583.3\t0\t1808\n",
      "7\t620\t645.8\t0\t620\n",
      "8\t438\t161.5\t0\t438\n",
      "9\t595\t40.4\t0\t278 317\n",
      "10\t727\t10.1\t1\t256 471\n",
      "11\t447\t2.5\t1\t222 225\n",
      "12\t362\t0.6\t1\t298 64\n",
      "13\t329\t0.2\t1\t297 32\n",
      "14\t219\t0.2\t1\t195 24\n",
      "15\t169\t0.2\t1\t154 15\n",
      "16\t130\t0.2\t1\t116 14\n",
      "17\t120\t0.2\t1\t105 15\n",
      "18\t95\t0.2\t1\t82 13\n",
      "19\t95\t0.2\t1\t80 15\n",
      "20\t83\t0.2\t1\t66 17\n",
      "21\t78\t0.2\t1\t50 28\n",
      "22\t66\t0.2\t1\t48 18\n",
      "23\t77\t0.2\t1\t50 27\n",
      "24\t72\t0.2\t1\t39 33\n",
      "25\t74\t0.2\t1\t30 44\n",
      "26\t82\t0.2\t1\t37 45\n",
      "27\t115\t0.2\t1\t30 85\n",
      "28\t70\t0.2\t1\t23 47\n",
      "29\t104\t0.2\t1\t26 78\n",
      "30\t76\t0.2\t1\t23 53\n",
      "31\t94\t0.2\t1\t24 70\n",
      "32\t81\t0.2\t1\t25 56\n",
      "33\t596\t0.2\t1\t25 571\n",
      "34\t1169\t0.2\t1\t13 1156\n",
      "35\t550\t0.2\t1\t16 534\n",
      "36\t240\t0.2\t1\t33 207\n",
      "37\t119\t0.2\t1\t11 108\n",
      "38\t135\t0.2\t1\t24 111\n",
      "39\t226\t0.2\t1\t28 198\n",
      "40\t182\t0.2\t1\t28 154\n",
      "41\t185\t0.2\t1\t29 156\n",
      "42\t278\t0.2\t1\t31 247\n",
      "43\t260\t0.2\t1\t20 240\n",
      "44\t493\t0.2\t1\t22 471\n",
      "45\t469\t0.2\t1\t26 443\n",
      "46\t1094\t0.2\t1\t39 1055\n",
      "47\t1626\t0.2\t1\t28 1598\n",
      "48\t2377\t0.2\t1\t26 2351\n",
      "49\t1825\t0.2\t1\t35 1790\n",
      "50\t2788\t0.2\t1\t39 2749\n",
      "51\t161165\t0.2\t1\t51 161114\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: H3K4me3_r2_single.fastq.gz\n",
      "=============================================\n",
      "10581396 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t178296 (1.7%)\n",
      "\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3184210 sequences (100000060 bp)...\n",
      "[M::process] read 3184872 sequences (100000028 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (65, 1337999, 22, 83)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (147, 324, 1548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4350)\n",
      "[M::mem_pestat] mean and std.dev: (694.77, 923.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5751)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 108, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 425)\n",
      "[M::mem_pestat] mean and std.dev: (132.68, 85.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (195, 344, 2499)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7107)\n",
      "[M::mem_pestat] mean and std.dev: (1260.67, 1618.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9411)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (175, 316, 1349)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3697)\n",
      "[M::mem_pestat] mean and std.dev: (601.64, 790.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4871)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3184210 reads in 182.599 CPU sec, 18.417 real sec\n",
      "[M::process] read 3185740 sequences (100000013 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (88, 1337758, 30, 66)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (126, 229, 637)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1659)\n",
      "[M::mem_pestat] mean and std.dev: (369.49, 389.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2170)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 106, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 415)\n",
      "[M::mem_pestat] mean and std.dev: (129.90, 82.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 531)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (378, 1649, 3221)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8907)\n",
      "[M::mem_pestat] mean and std.dev: (2532.33, 2688.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13287)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (167, 529, 1821)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5129)\n",
      "[M::mem_pestat] mean and std.dev: (1003.87, 1322.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6783)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3184872 reads in 186.958 CPU sec, 18.841 real sec\n",
      "[M::process] read 3186938 sequences (100000047 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (74, 1337389, 28, 89)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (137, 448, 1698)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4820)\n",
      "[M::mem_pestat] mean and std.dev: (857.00, 996.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6381)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 104, 178)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 400)\n",
      "[M::mem_pestat] mean and std.dev: (126.77, 79.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 511)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (462, 2063, 3221)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8739)\n",
      "[M::mem_pestat] mean and std.dev: (2312.93, 2120.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11498)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (103, 339, 950)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2644)\n",
      "[M::mem_pestat] mean and std.dev: (498.37, 523.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3491)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185740 reads in 193.664 CPU sec, 19.322 real sec\n",
      "[M::process] read 3186778 sequences (100000026 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (85, 1330914, 26, 82)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 276, 1100)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3070)\n",
      "[M::mem_pestat] mean and std.dev: (552.64, 683.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4055)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (66, 102, 172)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 384)\n",
      "[M::mem_pestat] mean and std.dev: (122.87, 75.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 490)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (570, 1988, 2878)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7494)\n",
      "[M::mem_pestat] mean and std.dev: (2029.35, 1730.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9802)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 370, 1010)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2790)\n",
      "[M::mem_pestat] mean and std.dev: (513.15, 585.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3680)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186938 reads in 188.077 CPU sec, 18.830 real sec\n",
      "[M::process] read 3184760 sequences (100000029 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (77, 1140117, 24, 78)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (155, 291, 1036)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2798)\n",
      "[M::mem_pestat] mean and std.dev: (592.99, 667.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3679)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 106, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 407)\n",
      "[M::mem_pestat] mean and std.dev: (128.84, 80.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 520)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 2004, 3479)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10037)\n",
      "[M::mem_pestat] mean and std.dev: (2298.29, 1981.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13316)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (156, 442, 1010)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2718)\n",
      "[M::mem_pestat] mean and std.dev: (499.18, 513.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3572)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186778 reads in 186.458 CPU sec, 18.572 real sec\n",
      "[M::process] read 3185462 sequences (100000061 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (71, 1338262, 19, 85)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (138, 298, 1055)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2889)\n",
      "[M::mem_pestat] mean and std.dev: (541.74, 699.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3806)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 107, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 419)\n",
      "[M::mem_pestat] mean and std.dev: (131.21, 83.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 536)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (900, 2889, 4541)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11823)\n",
      "[M::mem_pestat] mean and std.dev: (3163.32, 2410.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15464)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (149, 352, 1255)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3467)\n",
      "[M::mem_pestat] mean and std.dev: (696.96, 808.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4573)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3184760 reads in 188.892 CPU sec, 18.836 real sec\n",
      "[M::process] read 3186560 sequences (100000050 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (70, 1338197, 28, 87)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (152, 403, 1250)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3446)\n",
      "[M::mem_pestat] mean and std.dev: (657.76, 710.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4544)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 106, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 409)\n",
      "[M::mem_pestat] mean and std.dev: (128.74, 81.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 523)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (238, 975, 3078)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8758)\n",
      "[M::mem_pestat] mean and std.dev: (1926.82, 2336.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11598)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 309, 1013)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2803)\n",
      "[M::mem_pestat] mean and std.dev: (586.19, 663.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3698)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185462 reads in 186.504 CPU sec, 18.437 real sec\n",
      "[M::process] read 3190264 sequences (100000003 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (85, 1335985, 18, 65)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (130, 434, 1430)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4030)\n",
      "[M::mem_pestat] mean and std.dev: (802.78, 1024.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5330)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (66, 103, 175)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 393)\n",
      "[M::mem_pestat] mean and std.dev: (124.81, 76.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 502)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (517, 1223, 4015)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11011)\n",
      "[M::mem_pestat] mean and std.dev: (2561.78, 2880.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14509)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 265, 1008)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2786)\n",
      "[M::mem_pestat] mean and std.dev: (477.54, 514.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3675)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186560 reads in 179.940 CPU sec, 18.095 real sec\n",
      "[M::process] read 3185478 sequences (100000010 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (75, 1017547, 21, 72)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 360, 1935)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5585)\n",
      "[M::mem_pestat] mean and std.dev: (1054.07, 1413.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7410)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 102, 170)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 376)\n",
      "[M::mem_pestat] mean and std.dev: (121.94, 72.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 479)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (663, 1971, 3198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8268)\n",
      "[M::mem_pestat] mean and std.dev: (1651.53, 1185.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10803)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (103, 249, 771)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2107)\n",
      "[M::mem_pestat] mean and std.dev: (382.14, 435.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2775)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3190264 reads in 171.626 CPU sec, 16.890 real sec\n",
      "[M::process] read 3185160 sequences (100000057 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (73, 1248376, 23, 81)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (134, 376, 1031)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2825)\n",
      "[M::mem_pestat] mean and std.dev: (504.40, 488.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3722)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 109, 188)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 426)\n",
      "[M::mem_pestat] mean and std.dev: (133.47, 85.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 545)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (614, 1263, 2995)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7757)\n",
      "[M::mem_pestat] mean and std.dev: (1768.50, 1657.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10138)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (131, 343, 1448)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4082)\n",
      "[M::mem_pestat] mean and std.dev: (791.32, 1008.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5399)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185478 reads in 186.579 CPU sec, 18.404 real sec\n",
      "[M::process] read 3185554 sequences (100000010 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (81, 1338364, 24, 78)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (138, 278, 839)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2241)\n",
      "[M::mem_pestat] mean and std.dev: (425.56, 482.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2942)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 108, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 422)\n",
      "[M::mem_pestat] mean and std.dev: (131.94, 84.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 540)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 1536, 2201)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6193)\n",
      "[M::mem_pestat] mean and std.dev: (1202.14, 1032.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8189)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (168, 442, 1218)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3318)\n",
      "[M::mem_pestat] mean and std.dev: (568.12, 546.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4368)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185160 reads in 183.914 CPU sec, 18.171 real sec\n",
      "[M::process] read 3186860 sequences (100000006 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (76, 1337206, 22, 71)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (183, 470, 1577)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4365)\n",
      "[M::mem_pestat] mean and std.dev: (893.97, 1085.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5759)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 106, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 410)\n",
      "[M::mem_pestat] mean and std.dev: (129.45, 81.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 524)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (351, 1103, 4353)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12357)\n",
      "[M::mem_pestat] mean and std.dev: (2246.77, 2463.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16359)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 344, 1550)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4416)\n",
      "[M::mem_pestat] mean and std.dev: (768.74, 984.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5849)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185554 reads in 182.301 CPU sec, 18.018 real sec\n",
      "[M::process] read 3190522 sequences (100000042 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (110, 1315083, 26, 70)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 326, 1291)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3627)\n",
      "[M::mem_pestat] mean and std.dev: (699.01, 859.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4795)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 104, 176)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 394)\n",
      "[M::mem_pestat] mean and std.dev: (125.48, 77.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 503)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 2526, 4328)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12580)\n",
      "[M::mem_pestat] mean and std.dev: (2767.31, 2886.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16706)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (181, 447, 1392)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3814)\n",
      "[M::mem_pestat] mean and std.dev: (794.95, 897.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5025)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186860 reads in 186.319 CPU sec, 18.612 real sec\n",
      "[M::process] read 3184572 sequences (100000029 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (72, 979957, 26, 72)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 321, 1172)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3296)\n",
      "[M::mem_pestat] mean and std.dev: (553.14, 694.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4358)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 106, 179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 401)\n",
      "[M::mem_pestat] mean and std.dev: (128.13, 79.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 512)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (202, 984, 2884)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8248)\n",
      "[M::mem_pestat] mean and std.dev: (1574.42, 1602.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10930)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (163, 327, 1082)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2920)\n",
      "[M::mem_pestat] mean and std.dev: (648.72, 791.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3839)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3190522 reads in 180.128 CPU sec, 17.906 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (86, 1273464, 16, 74)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 259, 1151)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3273)\n",
      "[M::mem_pestat] mean and std.dev: (528.96, 723.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4334)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 109, 188)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 426)\n",
      "[M::mem_pestat] mean and std.dev: (133.33, 85.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 545)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (2486, 3266, 5261)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10811)\n",
      "[M::mem_pestat] mean and std.dev: (3542.56, 2321.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13586)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (92, 258, 912)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2552)\n",
      "[M::mem_pestat] mean and std.dev: (444.15, 581.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3372)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3184572 reads in 200.101 CPU sec, 20.366 real sec\n",
      "[M::process] read 3185220 sequences (100000026 bp)...\n",
      "[M::process] read 3186314 sequences (100000012 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (68, 1337412, 15, 103)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (131, 259, 1319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3695)\n",
      "[M::mem_pestat] mean and std.dev: (737.73, 942.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4883)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 107, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 419)\n",
      "[M::mem_pestat] mean and std.dev: (131.31, 83.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 536)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (343, 798, 2990)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8284)\n",
      "[M::mem_pestat] mean and std.dev: (1194.36, 1255.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10931)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (176, 430, 1317)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3599)\n",
      "[M::mem_pestat] mean and std.dev: (705.61, 754.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4740)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185220 reads in 179.599 CPU sec, 18.219 real sec\n",
      "[M::process] read 3197306 sequences (100000029 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (87, 1328465, 29, 77)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 330, 816)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2202)\n",
      "[M::mem_pestat] mean and std.dev: (454.06, 472.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2895)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 105, 179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 403)\n",
      "[M::mem_pestat] mean and std.dev: (127.99, 79.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 515)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 1208, 3000)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8604)\n",
      "[M::mem_pestat] mean and std.dev: (1604.37, 1953.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11406)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (106, 209, 747)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2029)\n",
      "[M::mem_pestat] mean and std.dev: (379.70, 442.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2670)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186314 reads in 184.340 CPU sec, 18.222 real sec\n",
      "[M::process] read 3185984 sequences (100000044 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (59, 837424, 15, 52)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (87, 264, 1229)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3513)\n",
      "[M::mem_pestat] mean and std.dev: (557.40, 724.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4655)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 104, 174)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 388)\n",
      "[M::mem_pestat] mean and std.dev: (124.74, 75.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 495)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (205, 1988, 5189)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15157)\n",
      "[M::mem_pestat] mean and std.dev: (2575.13, 2664.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 20141)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (192, 488, 1692)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4692)\n",
      "[M::mem_pestat] mean and std.dev: (837.38, 926.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6192)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3197306 reads in 168.831 CPU sec, 16.824 real sec\n",
      "[M::process] read 3185100 sequences (100000046 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (80, 1296587, 24, 82)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (109, 306, 806)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2200)\n",
      "[M::mem_pestat] mean and std.dev: (418.74, 424.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2897)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 109, 188)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 426)\n",
      "[M::mem_pestat] mean and std.dev: (133.00, 85.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 545)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (267, 1900, 3296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9354)\n",
      "[M::mem_pestat] mean and std.dev: (2112.88, 1950.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12383)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (150, 344, 1219)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3357)\n",
      "[M::mem_pestat] mean and std.dev: (666.97, 786.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4426)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185984 reads in 188.032 CPU sec, 18.817 real sec\n",
      "[M::process] read 3185836 sequences (100000012 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (68, 1338207, 19, 63)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (167, 358, 1143)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3095)\n",
      "[M::mem_pestat] mean and std.dev: (623.25, 750.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4071)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 107, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 413)\n",
      "[M::mem_pestat] mean and std.dev: (130.16, 82.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 528)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 1620, 4106)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12094)\n",
      "[M::mem_pestat] mean and std.dev: (2484.63, 2792.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16088)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (153, 363, 1281)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3537)\n",
      "[M::mem_pestat] mean and std.dev: (673.53, 820.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4665)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185100 reads in 190.189 CPU sec, 18.840 real sec\n",
      "[M::process] read 3187450 sequences (100000033 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (90, 1338102, 33, 78)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (108, 282, 1084)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3036)\n",
      "[M::mem_pestat] mean and std.dev: (520.16, 609.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4012)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 105, 179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 403)\n",
      "[M::mem_pestat] mean and std.dev: (127.64, 79.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 515)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (349, 1721, 4354)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12364)\n",
      "[M::mem_pestat] mean and std.dev: (2664.06, 2754.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16369)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 289, 919)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2523)\n",
      "[M::mem_pestat] mean and std.dev: (465.86, 517.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3325)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185836 reads in 186.436 CPU sec, 18.811 real sec\n",
      "[M::process] read 3190740 sequences (100000012 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (82, 1325213, 23, 76)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 287, 765)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2071)\n",
      "[M::mem_pestat] mean and std.dev: (437.90, 462.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2724)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (66, 103, 173)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 387)\n",
      "[M::mem_pestat] mean and std.dev: (123.79, 75.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 494)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (471, 2295, 5067)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14259)\n",
      "[M::mem_pestat] mean and std.dev: (3012.91, 2876.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18855)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 332, 1152)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3230)\n",
      "[M::mem_pestat] mean and std.dev: (688.26, 813.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4269)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3187450 reads in 183.326 CPU sec, 18.136 real sec\n",
      "[M::process] read 3184620 sequences (100000018 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (56, 921630, 24, 69)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (168, 317, 791)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2037)\n",
      "[M::mem_pestat] mean and std.dev: (411.96, 383.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2660)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 105, 178)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 398)\n",
      "[M::mem_pestat] mean and std.dev: (127.10, 78.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 508)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (286, 2813, 3790)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10798)\n",
      "[M::mem_pestat] mean and std.dev: (2567.62, 2523.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14302)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (151, 465, 1186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3256)\n",
      "[M::mem_pestat] mean and std.dev: (748.60, 794.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4291)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3190740 reads in 182.717 CPU sec, 18.201 real sec\n",
      "[M::process] read 3185248 sequences (100000058 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (95, 1339198, 26, 70)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 264, 918)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2528)\n",
      "[M::mem_pestat] mean and std.dev: (443.08, 517.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3333)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 108, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 422)\n",
      "[M::mem_pestat] mean and std.dev: (132.05, 84.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 540)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (154, 1536, 2298)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6586)\n",
      "[M::mem_pestat] mean and std.dev: (1140.91, 1081.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8730)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (142, 433, 1754)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4978)\n",
      "[M::mem_pestat] mean and std.dev: (862.85, 1054.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6590)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3184620 reads in 191.159 CPU sec, 19.022 real sec\n",
      "[M::process] read 3186304 sequences (100000032 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (84, 1339599, 30, 89)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (152, 297, 1467)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4097)\n",
      "[M::mem_pestat] mean and std.dev: (705.60, 931.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5412)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 106, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 410)\n",
      "[M::mem_pestat] mean and std.dev: (129.41, 81.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 524)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (293, 2953, 4177)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11945)\n",
      "[M::mem_pestat] mean and std.dev: (3007.70, 2860.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15829)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (171, 412, 1401)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3861)\n",
      "[M::mem_pestat] mean and std.dev: (777.00, 886.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5091)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185248 reads in 200.493 CPU sec, 19.856 real sec\n",
      "[M::process] read 3188808 sequences (100000035 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (72, 1338587, 19, 84)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 372, 1562)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4450)\n",
      "[M::mem_pestat] mean and std.dev: (756.27, 970.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5894)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 104, 177)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 397)\n",
      "[M::mem_pestat] mean and std.dev: (126.06, 78.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 507)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (41, 483, 3323)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9887)\n",
      "[M::mem_pestat] mean and std.dev: (1979.26, 2570.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13169)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (198, 460, 1614)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4446)\n",
      "[M::mem_pestat] mean and std.dev: (813.12, 880.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5862)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186304 reads in 205.437 CPU sec, 20.231 real sec\n",
      "[M::process] read 3186678 sequences (100000019 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (77, 1233281, 21, 69)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (133, 504, 2264)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6526)\n",
      "[M::mem_pestat] mean and std.dev: (1187.62, 1458.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8657)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (66, 102, 172)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 384)\n",
      "[M::mem_pestat] mean and std.dev: (122.99, 74.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 490)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (192, 1407, 3301)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9519)\n",
      "[M::mem_pestat] mean and std.dev: (2470.90, 2869.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13950)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (189, 578, 1704)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4734)\n",
      "[M::mem_pestat] mean and std.dev: (937.62, 1091.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6249)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3188808 reads in 206.148 CPU sec, 20.257 real sec\n",
      "[M::process] read 3184804 sequences (100000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (74, 1076821, 24, 68)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (134, 308, 677)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1763)\n",
      "[M::mem_pestat] mean and std.dev: (389.42, 381.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2306)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 109, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 423)\n",
      "[M::mem_pestat] mean and std.dev: (132.80, 84.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 541)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (247, 1236, 3294)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9388)\n",
      "[M::mem_pestat] mean and std.dev: (1752.13, 1983.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12435)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (154, 370, 1328)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3676)\n",
      "[M::mem_pestat] mean and std.dev: (582.37, 710.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4850)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186678 reads in 210.331 CPU sec, 20.612 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (79, 1337612, 15, 81)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (137, 568, 1325)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3701)\n",
      "[M::mem_pestat] mean and std.dev: (779.88, 858.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4889)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 109, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 423)\n",
      "[M::mem_pestat] mean and std.dev: (132.73, 84.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 541)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (572, 756, 2820)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7316)\n",
      "[M::mem_pestat] mean and std.dev: (1567.43, 1540.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9564)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (181, 504, 1825)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5113)\n",
      "[M::mem_pestat] mean and std.dev: (1050.12, 1335.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6757)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3184804 reads in 217.676 CPU sec, 22.232 real sec\n",
      "[M::process] read 3185470 sequences (100000052 bp)...\n",
      "[M::process] read 3186794 sequences (100000017 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (83, 1336784, 30, 89)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (133, 239, 752)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1990)\n",
      "[M::mem_pestat] mean and std.dev: (373.93, 377.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2609)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 107, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 416)\n",
      "[M::mem_pestat] mean and std.dev: (130.72, 82.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 532)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (622, 1382, 3243)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8485)\n",
      "[M::mem_pestat] mean and std.dev: (1681.96, 1438.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11106)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (138, 307, 1111)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3057)\n",
      "[M::mem_pestat] mean and std.dev: (590.14, 718.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4030)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185470 reads in 199.907 CPU sec, 20.086 real sec\n",
      "[M::process] read 3188320 sequences (100000060 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (81, 1313760, 29, 77)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 373, 1386)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3922)\n",
      "[M::mem_pestat] mean and std.dev: (723.78, 942.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5190)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 105, 178)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 400)\n",
      "[M::mem_pestat] mean and std.dev: (127.04, 78.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 511)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (346, 1206, 2950)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8158)\n",
      "[M::mem_pestat] mean and std.dev: (1552.50, 1320.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10762)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 400, 1263)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3375)\n",
      "[M::mem_pestat] mean and std.dev: (633.73, 722.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4431)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186794 reads in 205.256 CPU sec, 20.177 real sec\n",
      "[M::process] read 3184584 sequences (100000035 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (68, 971401, 26, 50)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (195, 346, 1215)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3255)\n",
      "[M::mem_pestat] mean and std.dev: (547.93, 573.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4275)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 106, 178)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 398)\n",
      "[M::mem_pestat] mean and std.dev: (127.73, 78.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 508)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (310, 1073, 6593)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 19159)\n",
      "[M::mem_pestat] mean and std.dev: (2903.62, 3229.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 25442)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (151, 388, 1023)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2767)\n",
      "[M::mem_pestat] mean and std.dev: (408.90, 329.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3639)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3188320 reads in 202.756 CPU sec, 19.904 real sec\n",
      "[M::process] read 3185084 sequences (100000021 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (75, 1334484, 23, 89)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (164, 388, 1151)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3125)\n",
      "[M::mem_pestat] mean and std.dev: (606.28, 706.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4112)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 110, 189)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 429)\n",
      "[M::mem_pestat] mean and std.dev: (133.89, 86.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (238, 1649, 3328)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9508)\n",
      "[M::mem_pestat] mean and std.dev: (2512.65, 2658.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13145)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (133, 281, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2500)\n",
      "[M::mem_pestat] mean and std.dev: (531.96, 631.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3289)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3184584 reads in 216.954 CPU sec, 21.309 real sec\n",
      "[M::process] read 3185962 sequences (100000006 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (84, 1338577, 23, 88)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 321, 919)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2519)\n",
      "[M::mem_pestat] mean and std.dev: (439.77, 432.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3319)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 108, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 422)\n",
      "[M::mem_pestat] mean and std.dev: (131.91, 84.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 540)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (203, 2795, 3408)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9818)\n",
      "[M::mem_pestat] mean and std.dev: (2529.00, 2382.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13023)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (176, 345, 940)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2468)\n",
      "[M::mem_pestat] mean and std.dev: (521.57, 512.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3232)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185084 reads in 222.483 CPU sec, 21.933 real sec\n",
      "[M::process] read 3190868 sequences (100000014 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (81, 1315165, 15, 65)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 230, 1002)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2802)\n",
      "[M::mem_pestat] mean and std.dev: (492.92, 571.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3702)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 106, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 407)\n",
      "[M::mem_pestat] mean and std.dev: (128.92, 80.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 520)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (548, 3086, 5869)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 16511)\n",
      "[M::mem_pestat] mean and std.dev: (3373.27, 3069.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 21832)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (134, 556, 1972)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5648)\n",
      "[M::mem_pestat] mean and std.dev: (1017.48, 1292.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7486)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185962 reads in 222.109 CPU sec, 22.056 real sec\n",
      "[M::process] read 524232 sequences (16359470 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (44, 734984, 17, 48)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (166, 439, 2066)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5866)\n",
      "[M::mem_pestat] mean and std.dev: (903.22, 994.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7766)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 104, 174)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 386)\n",
      "[M::mem_pestat] mean and std.dev: (124.87, 75.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 492)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1171, 1712, 4676)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11686)\n",
      "[M::mem_pestat] mean and std.dev: (2566.59, 2082.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15191)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (180, 332, 808)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2064)\n",
      "[M::mem_pestat] mean and std.dev: (461.48, 404.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2692)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3190868 reads in 202.626 CPU sec, 20.133 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 87534, 3, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 104, 169)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 367)\n",
      "[M::mem_pestat] mean and std.dev: (122.97, 69.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 466)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 524232 reads in 35.363 CPU sec, 3.466 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa DNase_r1_pair_1_val_1.fq.gz DNase_r1_pair_2_val_2.fq.gz\n",
      "[main] Real time: 716.910 sec; CPU: 7001.195 sec\n",
      "[bam_sort_core] merging from 20 files and 10 in-memory blocks...\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3184660 sequences (100000010 bp)...\n",
      "[M::process] read 3185598 sequences (100000057 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (86, 1365773, 109, 75)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (98, 295, 1006)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2822)\n",
      "[M::mem_pestat] mean and std.dev: (517.14, 650.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3730)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 114, 200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 460)\n",
      "[M::mem_pestat] mean and std.dev: (140.23, 93.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 590)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (331, 1788, 3410)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9568)\n",
      "[M::mem_pestat] mean and std.dev: (2238.35, 2258.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12647)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (157, 500, 1198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3280)\n",
      "[M::mem_pestat] mean and std.dev: (594.58, 675.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4321)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3184660 reads in 189.301 CPU sec, 19.024 real sec\n",
      "[M::process] read 3187098 sequences (100000024 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (70, 1366007, 111, 63)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (115, 236, 952)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2626)\n",
      "[M::mem_pestat] mean and std.dev: (375.46, 448.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3463)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 111, 192)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 438)\n",
      "[M::mem_pestat] mean and std.dev: (135.36, 88.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 561)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (469, 1546, 3903)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10771)\n",
      "[M::mem_pestat] mean and std.dev: (2460.84, 2462.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14205)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (158, 281, 821)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2147)\n",
      "[M::mem_pestat] mean and std.dev: (427.05, 431.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2810)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185598 reads in 200.631 CPU sec, 20.146 real sec\n",
      "[M::process] read 3186824 sequences (100000009 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (71, 1361249, 101, 67)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (86, 255, 1025)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2903)\n",
      "[M::mem_pestat] mean and std.dev: (462.49, 626.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3842)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 106, 182)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 412)\n",
      "[M::mem_pestat] mean and std.dev: (129.05, 81.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 527)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (377, 1727, 3120)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8606)\n",
      "[M::mem_pestat] mean and std.dev: (2090.38, 2017.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11349)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (139, 405, 1118)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3076)\n",
      "[M::mem_pestat] mean and std.dev: (554.72, 578.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4055)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3187098 reads in 202.602 CPU sec, 19.956 real sec\n",
      "[M::process] read 3185320 sequences (100000039 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (66, 1219349, 79, 84)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (83, 211, 1030)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2924)\n",
      "[M::mem_pestat] mean and std.dev: (444.42, 609.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3871)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 112, 192)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 436)\n",
      "[M::mem_pestat] mean and std.dev: (135.48, 87.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 558)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (518, 1996, 5050)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14114)\n",
      "[M::mem_pestat] mean and std.dev: (2820.39, 2631.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18646)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (77, 240, 697)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1937)\n",
      "[M::mem_pestat] mean and std.dev: (384.17, 467.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2557)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186824 reads in 186.458 CPU sec, 18.290 real sec\n",
      "[M::process] read 3186684 sequences (100000036 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (85, 1366048, 101, 80)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (91, 246, 1219)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3475)\n",
      "[M::mem_pestat] mean and std.dev: (668.59, 833.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4603)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 113, 195)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
      "[M::mem_pestat] mean and std.dev: (137.04, 89.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (424, 1443, 3412)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9388)\n",
      "[M::mem_pestat] mean and std.dev: (2127.34, 2021.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12376)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 285, 967)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2721)\n",
      "[M::mem_pestat] mean and std.dev: (415.33, 459.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3598)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185320 reads in 196.875 CPU sec, 19.271 real sec\n",
      "[M::process] read 3189974 sequences (100000031 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (66, 1364361, 84, 66)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (74, 234, 1248)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3596)\n",
      "[M::mem_pestat] mean and std.dev: (594.62, 775.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4770)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 109, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 422)\n",
      "[M::mem_pestat] mean and std.dev: (131.85, 84.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 540)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (709, 2031, 5169)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14089)\n",
      "[M::mem_pestat] mean and std.dev: (3067.00, 2909.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18549)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (91, 378, 636)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1726)\n",
      "[M::mem_pestat] mean and std.dev: (349.56, 330.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2271)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186684 reads in 188.181 CPU sec, 18.734 real sec\n",
      "[M::process] read 3184878 sequences (100000005 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (62, 1083051, 75, 48)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 349, 965)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2691)\n",
      "[M::mem_pestat] mean and std.dev: (544.46, 568.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3554)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 106, 179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 401)\n",
      "[M::mem_pestat] mean and std.dev: (127.91, 79.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 512)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (451, 1877, 3039)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8215)\n",
      "[M::mem_pestat] mean and std.dev: (2035.71, 1680.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10803)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 296, 1003)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2875)\n",
      "[M::mem_pestat] mean and std.dev: (388.88, 506.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3811)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3189974 reads in 189.320 CPU sec, 18.513 real sec\n",
      "[M::process] read 3185548 sequences (100000029 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (69, 1351466, 87, 71)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (74, 207, 502)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1358)\n",
      "[M::mem_pestat] mean and std.dev: (279.26, 293.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1786)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 116, 203)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 467)\n",
      "[M::mem_pestat] mean and std.dev: (142.09, 95.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 599)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1100, 2823, 4924)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12572)\n",
      "[M::mem_pestat] mean and std.dev: (3251.83, 2570.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16396)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (94, 222, 897)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2503)\n",
      "[M::mem_pestat] mean and std.dev: (443.83, 505.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3306)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3184878 reads in 196.738 CPU sec, 19.536 real sec\n",
      "[M::process] read 3187278 sequences (100000051 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (61, 1365008, 85, 68)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 362, 1179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3291)\n",
      "[M::mem_pestat] mean and std.dev: (573.14, 679.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4347)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 113, 195)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
      "[M::mem_pestat] mean and std.dev: (137.30, 90.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (447, 1565, 3198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8700)\n",
      "[M::mem_pestat] mean and std.dev: (2019.12, 1795.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11451)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 289, 1112)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3114)\n",
      "[M::mem_pestat] mean and std.dev: (519.02, 639.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4115)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185548 reads in 195.256 CPU sec, 19.396 real sec\n",
      "[M::process] read 3188626 sequences (100000056 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (66, 1303697, 82, 61)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (86, 265, 957)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2699)\n",
      "[M::mem_pestat] mean and std.dev: (454.66, 576.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3570)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 108, 185)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 419)\n",
      "[M::mem_pestat] mean and std.dev: (131.38, 83.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 536)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1071, 2011, 3559)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8535)\n",
      "[M::mem_pestat] mean and std.dev: (2367.59, 1933.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11023)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (130, 325, 1005)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2755)\n",
      "[M::mem_pestat] mean and std.dev: (491.00, 563.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3630)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3187278 reads in 196.576 CPU sec, 19.553 real sec\n",
      "[M::process] read 3185456 sequences (100000008 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (71, 1115878, 80, 63)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (97, 293, 932)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2602)\n",
      "[M::mem_pestat] mean and std.dev: (476.71, 586.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3437)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 114, 196)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 446)\n",
      "[M::mem_pestat] mean and std.dev: (138.00, 90.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 571)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (512, 1565, 3328)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8960)\n",
      "[M::mem_pestat] mean and std.dev: (1949.15, 1781.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11776)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (103, 282, 989)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2761)\n",
      "[M::mem_pestat] mean and std.dev: (482.59, 551.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3647)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3188626 reads in 191.578 CPU sec, 18.803 real sec\n",
      "[M::process] read 3186480 sequences (100000044 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (81, 1351379, 98, 74)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (74, 278, 941)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2675)\n",
      "[M::mem_pestat] mean and std.dev: (456.43, 565.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3542)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 114, 199)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 455)\n",
      "[M::mem_pestat] mean and std.dev: (139.50, 92.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 583)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (377, 1507, 3359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9323)\n",
      "[M::mem_pestat] mean and std.dev: (2386.94, 2544.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12565)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 257, 756)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2046)\n",
      "[M::mem_pestat] mean and std.dev: (409.63, 445.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2691)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185456 reads in 198.611 CPU sec, 19.482 real sec\n",
      "[M::process] read 3195120 sequences (100000055 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (71, 1350858, 99, 77)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 229, 917)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2505)\n",
      "[M::mem_pestat] mean and std.dev: (443.25, 524.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3299)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 111, 191)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 435)\n",
      "[M::mem_pestat] mean and std.dev: (134.74, 87.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 557)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (555, 1415, 4116)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11238)\n",
      "[M::mem_pestat] mean and std.dev: (2465.22, 2423.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14799)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 193, 913)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2595)\n",
      "[M::mem_pestat] mean and std.dev: (393.71, 543.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3436)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186480 reads in 194.983 CPU sec, 19.544 real sec\n",
      "[M::process] read 3185170 sequences (100000058 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (56, 965178, 67, 61)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (108, 329, 684)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1836)\n",
      "[M::mem_pestat] mean and std.dev: (362.44, 341.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2412)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 111, 189)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 427)\n",
      "[M::mem_pestat] mean and std.dev: (133.88, 85.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 546)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (580, 1822, 4181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11383)\n",
      "[M::mem_pestat] mean and std.dev: (2562.61, 2481.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14984)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (139, 436, 1157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3193)\n",
      "[M::mem_pestat] mean and std.dev: (486.69, 478.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4211)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3195120 reads in 180.755 CPU sec, 17.718 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (69, 1365897, 82, 78)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 454, 1346)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3806)\n",
      "[M::mem_pestat] mean and std.dev: (683.22, 728.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5036)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 114, 197)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 451)\n",
      "[M::mem_pestat] mean and std.dev: (138.57, 91.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 578)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (714, 1819, 3553)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9231)\n",
      "[M::mem_pestat] mean and std.dev: (2311.88, 2032.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12070)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (97, 195, 999)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2803)\n",
      "[M::mem_pestat] mean and std.dev: (418.46, 604.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3705)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185170 reads in 202.867 CPU sec, 20.576 real sec\n",
      "[M::process] read 3186044 sequences (100000059 bp)...\n",
      "[M::process] read 3187834 sequences (100000051 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (72, 1365798, 113, 68)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (63, 134, 330)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 864)\n",
      "[M::mem_pestat] mean and std.dev: (169.08, 147.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1131)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 111, 191)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 435)\n",
      "[M::mem_pestat] mean and std.dev: (134.70, 87.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 557)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (383, 2062, 4465)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12629)\n",
      "[M::mem_pestat] mean and std.dev: (2776.34, 2680.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16711)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (61, 198, 1139)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3295)\n",
      "[M::mem_pestat] mean and std.dev: (460.00, 651.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4373)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186044 reads in 181.271 CPU sec, 18.213 real sec\n",
      "[M::process] read 3188340 sequences (100000024 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (54, 1287918, 93, 69)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (87, 159, 1040)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2946)\n",
      "[M::mem_pestat] mean and std.dev: (437.96, 572.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3899)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 106, 179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 401)\n",
      "[M::mem_pestat] mean and std.dev: (127.66, 79.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 512)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (509, 1577, 3754)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10244)\n",
      "[M::mem_pestat] mean and std.dev: (2465.38, 2440.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (106, 272, 817)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2239)\n",
      "[M::mem_pestat] mean and std.dev: (339.80, 328.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2950)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3187834 reads in 184.668 CPU sec, 18.172 real sec\n",
      "[M::process] read 3185518 sequences (100000017 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (48, 1133125, 75, 65)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (184, 323, 1038)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2746)\n",
      "[M::mem_pestat] mean and std.dev: (499.31, 545.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3600)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 114, 198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 452)\n",
      "[M::mem_pestat] mean and std.dev: (138.83, 91.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 579)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (907, 2761, 5196)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13774)\n",
      "[M::mem_pestat] mean and std.dev: (3172.33, 2535.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18063)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 375, 1151)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3263)\n",
      "[M::mem_pestat] mean and std.dev: (567.61, 632.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4319)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3188340 reads in 190.226 CPU sec, 18.539 real sec\n",
      "[M::process] read 3186800 sequences (100000003 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (74, 1367498, 84, 83)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (104, 241, 1009)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2819)\n",
      "[M::mem_pestat] mean and std.dev: (522.46, 608.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3724)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 112, 195)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
      "[M::mem_pestat] mean and std.dev: (136.93, 89.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (1030, 2302, 3699)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9037)\n",
      "[M::mem_pestat] mean and std.dev: (2493.85, 1957.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11706)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (138, 255, 897)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2415)\n",
      "[M::mem_pestat] mean and std.dev: (410.71, 415.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3174)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185518 reads in 193.098 CPU sec, 19.139 real sec\n",
      "[M::process] read 3188800 sequences (100000009 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (87, 1366230, 98, 61)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (106, 199, 476)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1216)\n",
      "[M::mem_pestat] mean and std.dev: (246.25, 239.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1586)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 108, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 422)\n",
      "[M::mem_pestat] mean and std.dev: (131.57, 84.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 540)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (701, 1984, 4082)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10844)\n",
      "[M::mem_pestat] mean and std.dev: (2788.72, 2542.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14225)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (76, 225, 1018)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2902)\n",
      "[M::mem_pestat] mean and std.dev: (496.91, 656.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3844)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186800 reads in 190.592 CPU sec, 18.879 real sec\n",
      "[M::process] read 3185402 sequences (100000016 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (63, 1105139, 70, 59)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (60, 398, 895)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2565)\n",
      "[M::mem_pestat] mean and std.dev: (495.43, 497.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3400)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 108, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 414)\n",
      "[M::mem_pestat] mean and std.dev: (130.36, 82.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 529)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (804, 2046, 3769)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9699)\n",
      "[M::mem_pestat] mean and std.dev: (2588.27, 2370.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 12664)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (109, 264, 1017)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2833)\n",
      "[M::mem_pestat] mean and std.dev: (491.60, 591.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3741)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3188800 reads in 177.336 CPU sec, 17.553 real sec\n",
      "[M::process] read 3185818 sequences (100000048 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (75, 1362439, 80, 65)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (76, 162, 472)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1264)\n",
      "[M::mem_pestat] mean and std.dev: (279.52, 309.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1660)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 116, 202)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
      "[M::mem_pestat] mean and std.dev: (141.41, 94.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (624, 2191, 3752)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10008)\n",
      "[M::mem_pestat] mean and std.dev: (2536.88, 2113.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13136)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (79, 241, 853)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2401)\n",
      "[M::mem_pestat] mean and std.dev: (445.81, 511.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3175)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185402 reads in 183.638 CPU sec, 18.026 real sec\n",
      "[M::process] read 3187624 sequences (100000034 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (61, 1365585, 84, 69)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (98, 294, 1056)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2972)\n",
      "[M::mem_pestat] mean and std.dev: (508.09, 645.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3930)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 113, 195)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 445)\n",
      "[M::mem_pestat] mean and std.dev: (137.41, 89.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (622, 2587, 5107)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14077)\n",
      "[M::mem_pestat] mean and std.dev: (3084.12, 2667.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18562)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 451, 1194)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3354)\n",
      "[M::mem_pestat] mean and std.dev: (578.58, 647.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4434)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185818 reads in 178.262 CPU sec, 17.514 real sec\n",
      "[M::process] read 3186134 sequences (100000004 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (56, 1213926, 101, 49)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (185, 361, 1046)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2768)\n",
      "[M::mem_pestat] mean and std.dev: (562.35, 563.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3629)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 108, 184)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 414)\n",
      "[M::mem_pestat] mean and std.dev: (131.04, 82.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 529)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (561, 2057, 4373)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11997)\n",
      "[M::mem_pestat] mean and std.dev: (2735.50, 2542.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15809)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (77, 265, 1204)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3458)\n",
      "[M::mem_pestat] mean and std.dev: (405.39, 514.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4585)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3187624 reads in 172.267 CPU sec, 16.824 real sec\n",
      "[M::process] read 3185590 sequences (100000037 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (76, 1230023, 83, 80)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (136, 368, 1034)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2830)\n",
      "[M::mem_pestat] mean and std.dev: (541.32, 620.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3728)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 116, 202)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
      "[M::mem_pestat] mean and std.dev: (141.47, 93.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (522, 1465, 3673)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9975)\n",
      "[M::mem_pestat] mean and std.dev: (2350.35, 2273.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13126)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 293, 905)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2471)\n",
      "[M::mem_pestat] mean and std.dev: (467.65, 484.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3254)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186134 reads in 176.994 CPU sec, 17.342 real sec\n",
      "[M::process] read 3186678 sequences (100000038 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (80, 1366103, 100, 67)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (96, 366, 1332)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3804)\n",
      "[M::mem_pestat] mean and std.dev: (708.96, 880.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5040)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 114, 198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 452)\n",
      "[M::mem_pestat] mean and std.dev: (138.90, 91.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 579)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (495, 2075, 3900)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10710)\n",
      "[M::mem_pestat] mean and std.dev: (2683.81, 2468.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14115)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (128, 351, 1069)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2951)\n",
      "[M::mem_pestat] mean and std.dev: (464.33, 534.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3892)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3185590 reads in 178.955 CPU sec, 17.463 real sec\n",
      "[M::process] read 1680794 sequences (52589437 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (56, 1280403, 85, 69)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (97, 298, 689)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1873)\n",
      "[M::mem_pestat] mean and std.dev: (350.60, 348.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2465)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 110, 189)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 429)\n",
      "[M::mem_pestat] mean and std.dev: (133.71, 85.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 549)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (627, 2839, 4572)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12462)\n",
      "[M::mem_pestat] mean and std.dev: (3144.41, 2618.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16407)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (91, 415, 1124)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3190)\n",
      "[M::mem_pestat] mean and std.dev: (534.80, 551.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4223)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 3186678 reads in 176.446 CPU sec, 17.438 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (18, 251726, 28, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (290, 469, 1900)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5120)\n",
      "[M::mem_pestat] mean and std.dev: (935.50, 1215.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6730)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 108, 176)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 386)\n",
      "[M::mem_pestat] mean and std.dev: (127.21, 74.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 491)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (607, 1835, 5143)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 14215)\n",
      "[M::mem_pestat] mean and std.dev: (2971.71, 2704.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18751)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (121, 541, 2271)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6571)\n",
      "[M::mem_pestat] mean and std.dev: (1294.92, 1501.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8721)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1680794 reads in 86.345 CPU sec, 8.532 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa DNase_r2_pair_1_val_1.fq.gz DNase_r2_pair_2_val_2.fq.gz\n",
      "[main] Real time: 547.760 sec; CPU: 5190.214 sec\n",
      "[bam_sort_core] merging from 20 files and 10 in-memory blocks...\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1968766 sequences (100000089 bp)...\n",
      "[M::process] read 1968494 sequences (100000064 bp)...\n",
      "[M::mem_process_seqs] Processed 1968766 reads in 120.493 CPU sec, 11.908 real sec\n",
      "[M::process] read 1968372 sequences (100000089 bp)...\n",
      "[M::mem_process_seqs] Processed 1968494 reads in 131.311 CPU sec, 12.916 real sec\n",
      "[M::mem_process_seqs] Processed 1968372 reads in 121.958 CPU sec, 11.985 real sec\n",
      "[M::process] read 1969356 sequences (100000095 bp)...\n",
      "[M::process] read 1968616 sequences (100000022 bp)...\n",
      "[M::mem_process_seqs] Processed 1969356 reads in 120.349 CPU sec, 11.811 real sec\n",
      "[M::process] read 772239 sequences (39239945 bp)...\n",
      "[M::mem_process_seqs] Processed 1968616 reads in 116.194 CPU sec, 11.450 real sec\n",
      "[M::mem_process_seqs] Processed 772239 reads in 51.467 CPU sec, 5.031 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa H3K27ac_r1_single_trimmed.fq.gz\n",
      "[main] Real time: 76.773 sec; CPU: 667.972 sec\n",
      "[bam_sort_core] merging from 0 files and 10 in-memory blocks...\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1968718 sequences (100000015 bp)...\n",
      "[M::process] read 1968396 sequences (100000098 bp)...\n",
      "[M::mem_process_seqs] Processed 1968718 reads in 121.346 CPU sec, 12.054 real sec\n",
      "[M::process] read 1969194 sequences (100000033 bp)...\n",
      "[M::mem_process_seqs] Processed 1968396 reads in 134.795 CPU sec, 13.218 real sec\n",
      "[M::process] read 1203867 sequences (61160494 bp)...\n",
      "[M::mem_process_seqs] Processed 1969194 reads in 127.691 CPU sec, 12.527 real sec\n",
      "[M::mem_process_seqs] Processed 1203867 reads in 73.689 CPU sec, 7.268 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa H3K27ac_r2_single_trimmed.fq.gz\n",
      "[main] Real time: 49.105 sec; CPU: 461.229 sec\n",
      "[bam_sort_core] merging from 0 files and 10 in-memory blocks...\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1972902 sequences (100000017 bp)...\n",
      "[M::process] read 1972594 sequences (100000094 bp)...\n",
      "[M::mem_process_seqs] Processed 1972902 reads in 113.363 CPU sec, 11.101 real sec\n",
      "[M::process] read 1971402 sequences (100000066 bp)...\n",
      "[M::mem_process_seqs] Processed 1972594 reads in 136.081 CPU sec, 13.275 real sec\n",
      "[M::process] read 1973172 sequences (100000017 bp)...\n",
      "[M::mem_process_seqs] Processed 1971402 reads in 120.877 CPU sec, 11.551 real sec\n",
      "[M::process] read 1972294 sequences (100000080 bp)...\n",
      "[M::mem_process_seqs] Processed 1973172 reads in 123.324 CPU sec, 11.821 real sec\n",
      "[M::process] read 105512 sequences (5354267 bp)...\n",
      "[M::mem_process_seqs] Processed 1972294 reads in 113.558 CPU sec, 11.302 real sec\n",
      "[M::mem_process_seqs] Processed 105512 reads in 7.756 CPU sec, 0.761 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa H3K4me1_r1_single_trimmed.fq.gz\n",
      "[main] Real time: 76.008 sec; CPU: 618.946 sec\n",
      "[bam_sort_core] merging from 0 files and 10 in-memory blocks...\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1324324 sequences (100000005 bp)...\n",
      "[M::process] read 1323236 sequences (100000045 bp)...\n",
      "[M::mem_process_seqs] Processed 1324324 reads in 95.326 CPU sec, 9.354 real sec\n",
      "[M::process] read 1324134 sequences (100000044 bp)...\n",
      "[M::mem_process_seqs] Processed 1323236 reads in 112.887 CPU sec, 11.202 real sec\n",
      "[M::process] read 1323236 sequences (100000135 bp)...\n",
      "[M::mem_process_seqs] Processed 1324134 reads in 103.197 CPU sec, 10.130 real sec\n",
      "[M::process] read 1322930 sequences (100000089 bp)...\n",
      "[M::mem_process_seqs] Processed 1323236 reads in 111.103 CPU sec, 10.901 real sec\n",
      "[M::process] read 1322920 sequences (100000031 bp)...\n",
      "[M::mem_process_seqs] Processed 1322930 reads in 102.945 CPU sec, 10.057 real sec\n",
      "[M::process] read 1322790 sequences (100000139 bp)...\n",
      "[M::mem_process_seqs] Processed 1322920 reads in 106.044 CPU sec, 10.239 real sec\n",
      "[M::process] read 441646 sequences (33395523 bp)...\n",
      "[M::mem_process_seqs] Processed 1322790 reads in 104.031 CPU sec, 10.342 real sec\n",
      "[M::mem_process_seqs] Processed 441646 reads in 39.211 CPU sec, 3.824 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa H3K4me1_r2_single_trimmed.fq.gz\n",
      "[main] Real time: 78.896 sec; CPU: 777.285 sec\n",
      "[bam_sort_core] merging from 0 files and 10 in-memory blocks...\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1967844 sequences (100000006 bp)...\n",
      "[M::process] read 1568339 sequences (79680360 bp)...\n",
      "[M::mem_process_seqs] Processed 1967844 reads in 130.026 CPU sec, 12.959 real sec\n",
      "[M::mem_process_seqs] Processed 1568339 reads in 110.731 CPU sec, 11.107 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa H3K4me3_r1_single_trimmed.fq.gz\n",
      "[main] Real time: 28.304 sec; CPU: 244.554 sec\n",
      "[bam_sort_core] merging from 0 files and 10 in-memory blocks...\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1972028 sequences (100000027 bp)...\n",
      "[M::process] read 1971862 sequences (100000009 bp)...\n",
      "[M::mem_process_seqs] Processed 1972028 reads in 127.432 CPU sec, 12.531 real sec\n",
      "[M::process] read 1971408 sequences (100000017 bp)...\n",
      "[M::mem_process_seqs] Processed 1971862 reads in 139.300 CPU sec, 13.542 real sec\n",
      "[M::process] read 1972642 sequences (100000092 bp)...\n",
      "[M::mem_process_seqs] Processed 1971408 reads in 134.495 CPU sec, 12.921 real sec\n",
      "[M::process] read 1972660 sequences (100000087 bp)...\n",
      "[M::mem_process_seqs] Processed 1972642 reads in 147.393 CPU sec, 14.274 real sec\n",
      "[M::process] read 542500 sequences (27498112 bp)...\n",
      "[M::mem_process_seqs] Processed 1972660 reads in 145.658 CPU sec, 14.388 real sec\n",
      "[M::mem_process_seqs] Processed 542500 reads in 45.095 CPU sec, 4.371 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa H3K4me3_r2_single_trimmed.fq.gz\n",
      "[main] Real time: 76.102 sec; CPU: 742.955 sec\n",
      "[bam_sort_core] merging from 0 files and 10 in-memory blocks...\n",
      "[bam_rmdup_core] processing reference chrI...\n",
      "[bam_rmdup_core] 5451 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrII...\n",
      "[bam_rmdup_core] 2607 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrIII...\n",
      "[bam_rmdup_core] 4628 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrIV...\n",
      "[bam_rmdup_core] 5662 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrV...\n",
      "[bam_rmdup_core] 3905 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrX...\n",
      "[bam_rmdup_core] 4252 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrM...\n",
      "[bam_rmdup_core] 25374306 / 50650819 = 0.5010 in library '\t'\n",
      "[bam_rmdup_core] processing reference chrI...\n",
      "[bam_rmdup_core] 2684 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrII...\n",
      "[bam_rmdup_core] 777 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrIII...\n",
      "[bam_rmdup_core] 2149 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrIV...\n",
      "[bam_rmdup_core] 3236 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrV...\n",
      "[bam_rmdup_core] 1521 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrX...\n",
      "[bam_rmdup_core] 1987 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrM...\n",
      "[bam_rmdup_core] 17211050 / 38955053 = 0.4418 in library '\t'\n",
      "[bam_rmdupse_core] 532801 / 8854708 = 0.0602 in library '\t'\n",
      "[bam_rmdupse_core] 224318 / 5852767 = 0.0383 in library '\t'\n",
      "[bam_rmdupse_core] 424555 / 8744635 = 0.0486 in library '\t'\n",
      "[bam_rmdupse_core] 352054 / 8689985 = 0.0405 in library '\t'\n",
      "[bam_rmdupse_core] 223749 / 3038140 = 0.0736 in library '\t'\n",
      "[bam_rmdupse_core] 494659 / 8245394 = 0.0600 in library '\t'\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args='bash align.sh 10 /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa', returncode=0)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# # Align from original reads\n",
    "# cmd = f\"bash align.sh 10 /home/limh/Reference_Genome/C.elegan/BWA/ce10.fa\"\n",
    "# subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "600eff65-1eac-4dcf-823a-09d541cee4eb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args='bedtools intersect -a enhancer_primary.bed -b DNase_idr_sort_rmdup_filter.broadPeak -u -f 0.5 -F 0.5 -e > enhancer.bed', returncode=0)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Histone modification ChIP-seq call peak\n",
    "files = ['H3K27ac_r1_sort_rmdup.bam', 'H3K27ac_r2_sort_rmdup.bam', 'H3K4me1_r1_sort_rmdup.bam', 'H3K4me1_r2_sort_rmdup.bam']\n",
    "for file in files:\n",
    "    cmd = f\"sicer -t {file} -s ce10 -fdr 0.05 -cpu 20 -o .\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    \n",
    "# Get standard-format broadPeak file from sicer result\n",
    "files = [x for x in os.listdir() if (x.endswith('-W200-G600.scoreisland') and 'H3K' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df.insert(3, 'peakName', [f'peak{x}' for x in range(1, df.shape[0]+1)])\n",
    "    df['strand'] = '.'\n",
    "    df['signalValue'] = df[3]\n",
    "    df['pValue'] = df[3]\n",
    "    df['qValue'] = df[3]\n",
    "    df.to_csv(file.replace('-W200-G600.scoreisland', '.broadPeak'), sep='\\t', header=None, index=False)\n",
    "    \n",
    "# Use idr to combine different replicates\n",
    "files = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'H3K' in x and 'r1' in x)]\n",
    "for file in files:\n",
    "    file_rep2 = file.replace('r1', 'r2')\n",
    "    outfile = file.replace('r1',  'idr')\n",
    "    cmd = f'''idr --samples {file} {file_rep2} --input-file-type broadPeak \\\n",
    "        --rank q.value --output-file {outfile} --output-file-type broadPeak \\\n",
    "            --plot --log-output-file {outfile}.log'''\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    \n",
    "# Filter idr result with threshold as 0.1\n",
    "files = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'H3K' in x and 'idr' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df = df[df[4]> int(-125 *  np.log2(0.1))]\n",
    "    df.to_csv(file.replace('.broadPeak', '_filter.broadPeak'), sep='\\t', header=None, index=False)\n",
    "\n",
    "# H3K4me1 H3K27ac intersection calculation\n",
    "H3K4me1_file = [x for x in os.listdir() if (x.endswith('_filter.broadPeak') and 'H3K4me1' in x and 'idr' in x)][0]\n",
    "H3K27ac_file = [x for x in os.listdir() if (x.endswith('_filter.broadPeak') and 'H3K27ac' in x and 'idr' in x)][0]\n",
    "cmd = f\"bedtools intersect -a {H3K4me1_file} -b {H3K27ac_file} > H3K4me1_H3K27ac_intersect.bed\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "# Retain H3K4me1 H3K27ac intersection part which show lower H3K4me3 histone modification signal. Get primary enhancer.\n",
    "H3K4me1_bam = [x for x in os.listdir() if (x.endswith('rmdup.bam') and 'H3K4me1' in x)]\n",
    "for file in H3K4me1_bam:\n",
    "    if f\"{file}.bai\" in os.listdir():\n",
    "        continue\n",
    "    cmd = f\"samtools index -@ 20 {file}\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "cmd = f'''bamCompare -b1 {H3K4me1_bam[0]} -b2 {H3K4me1_bam[1]} --outFileName H3K4me1_all.bw \\\n",
    "          --normalizeUsing CPM --binSize 10 --operation add --numberOfProcessors 20 --scaleFactorsMethod None'''\n",
    "subprocess.run(cmd, shell=True)\n",
    "H3K4me3_bam = [x for x in os.listdir() if (x.endswith('rmdup.bam') and 'H3K4me3' in x)]\n",
    "for file in H3K4me3_bam:\n",
    "    if f\"{file}.bai\" in os.listdir():\n",
    "        continue\n",
    "    cmd = f\"samtools index -@ 20 {file}\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "cmd = f'''bamCompare -b1 {H3K4me3_bam[0]} -b2 {H3K4me3_bam[1]} --outFileName H3K4me3_all.bw \\\n",
    "            --normalizeUsing CPM --binSize 10 --operation add --numberOfProcessors 20 --scaleFactorsMethod None'''\n",
    "subprocess.run(cmd, shell=True)\n",
    "me1BW = pyBigWig.open('H3K4me1_all.bw')\n",
    "me3BW = pyBigWig.open('H3K4me3_all.bw')\n",
    "def ifRetain(row, me1BW, me3BW, me1ScoreAll=1, me3ScoreAll=1):\n",
    "    chr = row[0]\n",
    "    start = row[1]\n",
    "    end = row[2]\n",
    "    me1Score = np.array(me1BW.stats(chr, start, end, type='sum'))[0] / me1ScoreAll * pow(10, 6)\n",
    "    me3Score = np.array(me3BW.stats(chr, start, end, type='sum'))[0] / me3ScoreAll * pow(10, 6)\n",
    "    if me1Score > 1.5 * me3Score:\n",
    "        return True, me1Score, me3Score\n",
    "    else:\n",
    "        return False, me1Score, me3Score\n",
    "df = pd.read_csv('H3K4me1_H3K27ac_intersect.bed', sep='\\t', header=None)\n",
    "df['result'] = df.apply(ifRetain, axis=1, args=(me1BW, me3BW))\n",
    "df[['retain', 'me1Score', 'me3Score']] = df['result'].apply(pd.Series)\n",
    "me1BW.close()\n",
    "me3BW.close()\n",
    "df[df['retain']][[0,1,2]].to_csv('enhancer_primary.bed', sep='\\t', header=None, index=False)\n",
    "\n",
    "# DNase-seq data process to get broadPeak.\n",
    "files = [x for x in os.listdir() if ((x.endswith('_rmdup.bam')) and ('DNase' in x))]\n",
    "for file in files:\n",
    "    cmd = f\"sicer -t {file} -s ce10 -fdr 0.05 -cpu 20 -o .\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "files = [x for x in os.listdir() if (x.endswith('-W200-G600.scoreisland') and 'DNase' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df.insert(3, 'peakName', [f'peak{x}' for x in range(1, df.shape[0]+1)])\n",
    "    df['strand'] = '.'\n",
    "    df['signalValue'] = df[3]\n",
    "    df['pValue'] = df[3]\n",
    "    df['qValue'] = df[3]\n",
    "    df.to_csv(file.replace('-W200-G600.scoreisland', '.broadPeak'), sep='\\t', header=None, index=False)\n",
    "    \n",
    "# Use idr to combine different replicates\n",
    "files = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'DNase' in x and 'r1' in x)]\n",
    "for file in files:\n",
    "    file_rep2 = file.replace('r1', 'r2')\n",
    "    outfile = file.replace('r1',  'idr')\n",
    "    cmd = f'''idr --samples {file} {file_rep2} --input-file-type broadPeak \\\n",
    "        --rank q.value --output-file {outfile} --output-file-type broadPeak \\\n",
    "            --plot --log-output-file {outfile}.log'''\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    \n",
    "# Filter idr result with threshold as 0.1\n",
    "files = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'DNase' in x and 'idr' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df = df[df[4]> int(-125 *  np.log2(0.1))]\n",
    "    df.to_csv(file.replace('.broadPeak', '_filter.broadPeak'), sep='\\t', header=None, index=False)\n",
    "    \n",
    "\n",
    "# Filter primary enhancer with its DNase broadPeak. Obtain final enhancer.\n",
    "DNase_file = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'DNase' in x and 'idr' in x)][0]\n",
    "cmd = f\"bedtools intersect -a enhancer_primary.bed -b {DNase_file} -u -f 0.5 -F 0.5 -e > enhancer.bed\"\n",
    "subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3588caa1-5144-4e5e-a332-afc3d1d42c35",
   "metadata": {},
   "outputs": [],
   "source": [
    "# process enhancer file\n",
    "df = pd.read_csv('enhancer.bed', sep='\\t', header=None)\n",
    "df[3] = [f\"enhancer{i}\" for i in range(df.shape[0])]\n",
    "df[4] = 1\n",
    "df[5] = '.'\n",
    "df.to_csv('enhancer.bg', sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47d6400b-3f49-482d-b7dd-dc1883a84188",
   "metadata": {},
   "source": [
    "## enhancer promoter interaction construction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "5af5035d-a636-4ede-8eb0-2483d811861d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args='bedtools sort -i TAD.bed > TAD_sort.bed', returncode=0)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# # Use arrowhead to get TAD        \n",
    "# cmd = f\"juicer_tools arrowhead GSM5206715_DJ73_inter_30.hic -k KR ./arrowhead\"\n",
    "# subprocess.run(cmd, shell=True)\n",
    "        \n",
    "# get TAD bed file\n",
    "df = pd.read_csv('arrowhead/10000_blocks.bedpe', sep='\\t', header=None, skiprows=2)\n",
    "df[0] = 'chr' + df[0]\n",
    "df[[0,1,2]].to_csv('TAD.bed', sep='\\t', header=None, index=None)\n",
    "\n",
    "# Sort TAD, the result is: TAD_sort.bed\n",
    "cmd = f\"bedtools sort -i TAD.bed > TAD_sort.bed\"\n",
    "subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "aa7a4870-f2b6-4ebf-a1e3-636b1ada582e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter TAD, the result is: TAD_filter_sort.bed\n",
    "def ifIntersect(x:tuple, y:tuple, f=0., F=0.)->bool:\n",
    "    '''If two section overlap'''\n",
    "    if int(x[0])<int(y[0])<int(x[1]):\n",
    "        if int(y[1])<int(x[1]):\n",
    "            return 1\n",
    "        else:\n",
    "            if int(x[1])-int(y[0]) / int(x[1])-int(x[0]) > f or int(x[1])-int(y[0]) / int(y[1])-int(y[0]) > F:\n",
    "                return 1\n",
    "            else:\n",
    "                return 0\n",
    "    elif int(y[0])<int(x[0])<int(y[1]):\n",
    "        if int(y[1]) > int(x[1]):\n",
    "            return 1\n",
    "        else:\n",
    "            if int(y[1])-int(x[0]) / int(x[1])-int(x[0]) > f or int(y[1])-int(x[0]) / int(y[1])-int(y[0]) > F:\n",
    "                return 1\n",
    "            else:\n",
    "                return 0\n",
    "    return 0\n",
    "\n",
    "from collections import defaultdict as ddict\n",
    "tadInterval = ddict(list)\n",
    "with open('TAD_sort.bed', 'r') as infile:\n",
    "    for line in infile:\n",
    "        if len(tadInterval[line.split()[0]]) == 0:\n",
    "            tadInterval[line.split()[0]].append((int(line.split()[1]), int(line.split()[2])))\n",
    "        else:\n",
    "            tadInter = (int(line.split()[1]), int(line.split()[2]))\n",
    "            if ifIntersect(tadInterval[line.split()[0]][-1], tadInter):\n",
    "                lengthOld = tadInterval[line.split()[0]][-1][1] - tadInterval[line.split()[0]][-1][0]\n",
    "                lengthNew = tadInter[1] - tadInter[0]\n",
    "                if lengthNew > lengthOld:\n",
    "                    tadInterval[line.split()[0]].remove(tadInterval[line.split()[0]][-1])\n",
    "                    tadInterval[line.split()[0]].append(tadInter)\n",
    "            else:\n",
    "                tadInterval[line.split()[0]].append(tadInter)\n",
    "outfile = open('TAD_filter.bed', 'w')\n",
    "for chromosome in tadInterval.keys():\n",
    "    for i in tadInterval[chromosome]:\n",
    "        print(f'{chromosome}\\t{i[0]}\\t{i[1]}', file=outfile)\n",
    "outfile.close()\n",
    "!bedtools sort -i TAD_filter.bed > TAD_filter_sort.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "002a518e-58f1-462c-b77f-9e8b0c5c6fe3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/limh/miniconda3/envs/chipseq/lib/python3.7/site-packages/ipykernel_launcher.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    }
   ],
   "source": [
    "# build enhancer promoter interaction\n",
    "cmd = f\"bedtools intersect -a enhancer.bg -b TAD_filter_sort.bed -wo -f 0.8 -F 0.8 -e > enhancer_TAD_intersectWo.txt\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "cmd = f\"bedtools intersect -a promoter.bed -b TAD_filter_sort.bed -wo -f 0.8 -F 0.8  -e > promoter_TAD_intersectWo.txt\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "promoter_TAD_intersectWo_df = pd.read_csv('promoter_TAD_intersectWo.txt', sep='\\t', header=None)\n",
    "promoter_TAD_intersectWo_df.columns = ['promoter_chr', 'promoter_start', 'promoter_end', 'promoter_name', 'promoter_score', 'promoter_strand',\n",
    "                                       'TAD_chr', 'TAD_start', 'TAD_end', 'intersect']\n",
    "eRNA_TAD_intersectWo_df = pd.read_csv('enhancer_TAD_intersectWo.txt', sep='\\t', header=None)\n",
    "eRNA_TAD_intersectWo_df.columns = ['eRNA_chr', 'eRNA_start', 'eRNA_end', 'eRNA_name', 'eRNA_score', 'eRNA_strand',\n",
    "                                   'TAD_chr', 'TAD_start', 'TAD_end', 'intersect']\n",
    "epi_df = pd.merge(eRNA_TAD_intersectWo_df, promoter_TAD_intersectWo_df,\n",
    "                  on = ['TAD_chr', 'TAD_start', 'TAD_end'], how='inner')\n",
    "\n",
    "# EPI filter\n",
    "epi_df['filter'] = epi_df.apply(ifFilter, axis=1)\n",
    "epi_filter_df = epi_df[epi_df['filter']==False]\n",
    "epi_filter_df['epi_name'] = epi_filter_df['eRNA_name'].str.cat(epi_filter_df['promoter_name'], sep=';')\n",
    "\n",
    "# EPI result save\n",
    "epi_filter_df[['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end',\n",
    "               'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']].to_csv('EPI.bedpe', sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c5c7e37c-ed9b-4e72-badb-062f4c1751e5",
   "metadata": {},
   "source": [
    "# EP similarity analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "661c3c7e-19cb-41e8-9186-4d9e826b9126",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import subprocess\n",
    "import os\n",
    "import numpy as np\n",
    "import pyBigWig\n",
    "pd.set_option('display.max_columns', None)\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "1cd2bddc-7462-48d4-b236-513d35def4a2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "index file /home/limh/Reference_Genome/C.elegan/ce10.fa.fai not found, generating...\n"
     ]
    }
   ],
   "source": [
    "# get enhancer、promoter from EPI\n",
    "referenceGenome = '/home/limh/Reference_Genome/C.elegan/ce10.fa'\n",
    "epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "epi_df[['eRNA_name', 'promoter_name']] = epi_df[6].str.split(';', expand=True)\n",
    "epi_df[[0, 1, 2, 'eRNA_name', 7, 8]].drop_duplicates('eRNA_name').to_csv('enhancer_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "epi_df[[3, 4, 5, 'promoter_name', 7, 9]].drop_duplicates('promoter_name').to_csv('promoter_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "\n",
    "# Get fasta of eRNAinEPI,promoterInEPI\n",
    "cmd = f\"bedtools getfasta -fi {referenceGenome} -fo enhancer_inEPI.fa -name -bed enhancer_inEPI.bed -s\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "cmd = f\"bedtools getfasta -fi {referenceGenome} -fo promoter_inEPI.fa -name -bed promoter_inEPI.bed -s\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "# Split fasta\n",
    "os.makedirs('EPI_fasta', exist_ok=True)\n",
    "with open('enhancer_inEPI.fa', 'r') as f:\n",
    "\tfor line in f:\n",
    "\t\tif line.startswith('>'):\n",
    "\t\t\tname = line.strip().split('::')[0][1:]\n",
    "\t\t\tout = open(f\"EPI_fasta/{name}.fa\", 'w')\n",
    "\t\t\t# out.write(f\">{name}\\n\")\n",
    "\t\t\t# out.write(line)\n",
    "\t\t\tprint('>' + line.strip().split('::')[1], file=out)\n",
    "\t\telse:\n",
    "\t\t\tout.write(line)\n",
    "\tout.close()\n",
    "with open('promoter_inEPI.fa', 'r') as f:\n",
    "\tfor line in f:\n",
    "\t\tif line.startswith('>'):\n",
    "\t\t\tname = line.strip().split('::')[0][1:]\n",
    "\t\t\tout = open(f\"EPI_fasta/{name}.fa\", 'w')\n",
    "\t\t\t# out.write(f\">{name}\\n\")\n",
    "\t\t\t# out.write(line)\n",
    "\t\t\tprint('>' + line.strip().split('::')[1], file=out)\n",
    "\t\telse:\n",
    "\t\t\tout.write(line)\n",
    "\tout.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "e358e71c-2dc0-48c8-b370-047bdb9191ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "# enhancer\n",
    "# makeblastdb script\n",
    "genes = [x for x in os.listdir('EPI_fasta') if (x.endswith('.fa') and not x.startswith('enhancer'))]\n",
    "makedf_bash_file = open('EPI_makedb.sh', 'w')\n",
    "if os.path.exists('EPI_makedb.log'):\n",
    "    os.remove('EPI_makedb.log')\n",
    "for gene in enumerate(genes):\n",
    "    print(f\"makeblastdb -in EPI_fasta/{gene[1]} -dbtype nucl >> EPI_makedb.log\", file=makedf_bash_file)\n",
    "makedf_bash_file.close()\n",
    "\n",
    "# blast script\n",
    "epi_list = epi_df[['eRNA_name', 'promoter_name']].values.tolist()\n",
    "blast_bash_file = open('EPI_blast.sh', 'w')\n",
    "if os.path.exists('EPI_blastResult.txt'):\n",
    "    os.remove('EPI_blastResult.txt')\n",
    "for i in epi_list:\n",
    "    eRNA = i[0]\n",
    "    promoter = i[1]\n",
    "    print(f\"blastn -task blastn -query EPI_fasta/{eRNA}.fa -db EPI_fasta/{promoter}.fa -outfmt 6 -evalue 1e-4 >> EPI_blastResult.txt\", file=blast_bash_file)\n",
    "blast_bash_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e8f77fd-3c88-4dce-bc8f-bc3e8ba20263",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "subprocess.run('bash EPI_makedb.sh', shell=True)\n",
    "subprocess.run('bash EPI_blast.sh', shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "9f0025be-c666-41d2-8fe1-f3da95e6e5a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the sequence similar enhancer in EPI\n",
    "blastResultFile = 'EPI_blastResult.txt'\n",
    "blastResultBedFile = 'EPI_enhancerBlastHit.bed'\n",
    "with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "\tfor line in infile:\n",
    "\t\tchromosome = line.split()[0].split(':')[0]\n",
    "\t\tstartPre = int(line.split()[0].split(':')[1].split('-')[0])\n",
    "\t\tstopPre = int(line.split()[0].split(':')[1].split('-')[1].split('(')[0])\n",
    "\t\tname = line.split()[0]\n",
    "\t\totherName = line.split()[1]\n",
    "\t\tstrand = line.split()[0].split(':')[1].split('(')[1][0]\n",
    "\t\tif strand == '+' or strand == '.':\n",
    "\t\t\tif int(line.split()[6]) < int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[6])\n",
    "\t\t\t\tshiftEnd = int(line.split()[7])\n",
    "\t\t\telif int(line.split()[6]) > int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[7])\n",
    "\t\t\t\tshiftEnd = int(line.split()[6])\n",
    "\t\t\tstart = startPre + shiftStart - 1\n",
    "\t\t\tend = startPre + shiftEnd - 1\n",
    "\t\telif strand == '-':\n",
    "\t\t\tif int(line.split()[6]) < int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[6])\n",
    "\t\t\t\tshiftEnd = int(line.split()[7])\n",
    "\t\t\telif int(line.split()[6]) > int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[7])\n",
    "\t\t\t\tshiftEnd = int(line.split()[6])\n",
    "\t\t\tend = stopPre - shiftStart\n",
    "\t\t\tstart = stopPre - shiftEnd\n",
    "\t\tprint(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)\n",
    "        \n",
    "# Get the sequence similar promoter in EPI\n",
    "blastResultFile = 'EPI_blastResult.txt'\n",
    "blastResultBedFile = 'EPI_promoterBlastHit.bed'\n",
    "with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "\tfor line in infile:\n",
    "\t\tchromosome = line.split()[1].split(':')[0]\n",
    "\t\tstartPre = int(line.split()[1].split(':')[1].split('-')[0])\n",
    "\t\tstopPre = int(line.split()[1].split(':')[1].split('-')[1].split('(')[0])\n",
    "\t\tname = line.split()[1]\n",
    "\t\totherName = line.split()[0]\n",
    "\t\tstrand = line.split()[1].split(':')[1].split('(')[1][0]\n",
    "\t\tif strand == '+' or strand == '.':\n",
    "\t\t\tif int(line.split()[8]) < int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[8])\n",
    "\t\t\t\tshiftEnd = int(line.split()[9])\n",
    "\t\t\telif int(line.split()[8]) > int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[9])\n",
    "\t\t\t\tshiftEnd = int(line.split()[8])\n",
    "\t\t\tstart = startPre + shiftStart - 1\n",
    "\t\t\tend = startPre + shiftEnd - 1\n",
    "\t\telif strand == '-':\n",
    "\t\t\tif int(line.split()[8]) < int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[8])\n",
    "\t\t\t\tshiftEnd = int(line.split()[9])\n",
    "\t\t\telif int(line.split()[8]) > int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[9])\n",
    "\t\t\t\tshiftEnd = int(line.split()[8])\n",
    "\t\t\tend = stopPre - shiftStart\n",
    "\t\t\tstart = stopPre - shiftEnd\n",
    "\t\tprint(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "71fbfb7e-8e0d-4e6d-96b5-46ab2b2b4d87",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get out the TE similar EPI, save as ep_same_{te}.bedpe\n",
    "promoter_blast_df = pd.read_csv('EPI_promoterBlastHit.bed', sep='\\t', header=None)\n",
    "enhancer_blast_df = pd.read_csv('EPI_enhancerBlastHit.bed', sep='\\t', header=None)\n",
    "promoter_blast_df.columns = ['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end', 'promoter', 'eRNA', 'promoter_strand']\n",
    "enhancer_blast_df.columns = ['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end', 'eRNA', 'promoter', 'eRNA_strand']\n",
    "epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "epi_df.columns = ['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end',\n",
    "               'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']\n",
    "promoter_blast_df['promoter_chr'] = promoter_blast_df['promoter'].str.split(':').str[0]\n",
    "promoter_blast_df['promoter_start'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "promoter_blast_df['promoter_end'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "promoter_blast_df['eRNA_chr'] = promoter_blast_df['eRNA'].str.split(':').str[0]\n",
    "promoter_blast_df['eRNA_start'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "promoter_blast_df['eRNA_end'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "enhancer_blast_df['eRNA_chr'] = enhancer_blast_df['eRNA'].str.split(':').str[0]\n",
    "enhancer_blast_df['eRNA_start'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "enhancer_blast_df['eRNA_end'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "enhancer_blast_df['promoter_chr'] = enhancer_blast_df['promoter'].str.split(':').str[0]\n",
    "enhancer_blast_df['promoter_start'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "enhancer_blast_df['promoter_end'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "# Get out the intersection between transposon and blastHit\n",
    "transposonFile = 'ce10.fa.bed'\n",
    "x = pd.merge(epi_df, promoter_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                            'promoter_start', 'promoter_end', 'promoter_strand'], how='inner')\n",
    "y = pd.merge(x, enhancer_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                       'promoter_start', 'promoter_end', 'promoter', 'eRNA', 'eRNA_strand'\n",
    "                                       ], how='inner')\n",
    "cmd = f\"bedtools intersect -a EPI_promoterBlastHit.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > EPI_promoterBlastHit_transposon_intersectWo.bed\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "cmd = f\"bedtools intersect -a EPI_enhancerBlastHit.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > EPI_enhancerBlastHit_transposon_intersectWo.bed\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "promoter_blast_te_intersectWo_df = pd.read_csv('EPI_promoterBlastHit_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "enhancer_blast_te_intersectWo_df = pd.read_csv('EPI_enhancerBlastHit_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "promoter_blast_te_intersectWo_df = promoter_blast_te_intersectWo_df.rename({\n",
    "    0:'promoter_blastHit_chr', 1:'promoter_blastHit_start', 2:'promoter_blastHit_end', \n",
    "    3:'promoter', 4:'eRNA', 5:'promoter_strand'}, axis=1)\n",
    "enhancer_blast_te_intersectWo_df = enhancer_blast_te_intersectWo_df.rename({\n",
    "    0:'eRNA_blastHit_chr', 1:'eRNA_blastHit_start', 2:'eRNA_blastHit_end',\n",
    "    3:'eRNA', 4:'promoter', 5:'eRNA_strand'}, axis=1)\n",
    "\n",
    "# merge the promoter and enhancer information together\n",
    "transposon_similar_EPI_df = pd.merge(\n",
    "    pd.merge(y, promoter_blast_te_intersectWo_df, on=['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end',\n",
    "                                                  'promoter', 'eRNA', 'promoter_strand'], how='inner'),\n",
    "    enhancer_blast_te_intersectWo_df, on=['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end',\n",
    "                                            'eRNA', 'promoter', 'eRNA_strand'], how='inner')\n",
    "\n",
    "transposon_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'] == transposon_similar_EPI_df['12_y']]\n",
    "SINE_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'].str.contains('SINE')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "13128c1c-4517-45ee-9e24-1a87d94a55d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1182\n",
      "916\n",
      "3\n",
      "Unknown 0.5\n",
      "Satellite 0.48\n",
      "Simple_repeat 0.01\n",
      "DNA 0.0\n",
      "DNA/TcMar-Tc1? 0.01\n",
      "DNA/hAT 0.0\n",
      "DNA/TcMar-Tc1 0.0\n",
      "DNA/PiggyBac? 0.0\n",
      "SINE/tRNA-RTE 0.0\n",
      "RC/Helitron 0.0\n",
      "Low_complexity 0.0\n",
      "DNA/CMC-Chapaev 0.0\n"
     ]
    }
   ],
   "source": [
    "# seuquence-similar EPI num\n",
    "print(y['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "# transposon-similar EPI num\n",
    "print(transposon_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "# SINE-similar EPI num\n",
    "print(SINE_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "# each transposon ratio in transposon-similar EPI (corrected by copy number)\n",
    "df = pd.read_csv(transposonFile, sep='\\t', header=None)\n",
    "transposon_num_dict = transposon_similar_EPI_df['12_x'].value_counts()\n",
    "transposon_copyNum_dict = df[6].value_counts().to_dict()\n",
    "transposon_num_correct_dict = dict()\n",
    "for key in transposon_num_dict.keys():\n",
    "    result = transposon_num_dict[key] / transposon_copyNum_dict[key]\n",
    "    transposon_num_correct_dict[key] = [result]\n",
    "for key, value in (pd.DataFrame.from_dict(transposon_num_correct_dict).T / pd.DataFrame.from_dict(transposon_num_correct_dict).T.sum())[0].to_dict().items():\n",
    "    print(key, round(value, 2))\n",
    "\n",
    "# result save\n",
    "for i, j in SINE_similar_EPI_df.groupby('12_x'):\n",
    "    j[['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end', 'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']\n",
    " ].drop_duplicates('epi_name').to_csv(f\"ep_same_{i.replace('/', '_')}.bedpe\", sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "540367ae-a17e-4370-a130-da1950d6f517",
   "metadata": {},
   "outputs": [],
   "source": [
    "# random Control (50 shuffle)\n",
    "epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "epi_df[['eRNA_name', 'promoter_name']] = epi_df[6].str.split(';', expand=True)\n",
    "epi_df[[0, 1, 2, 'eRNA_name', 7, 8]].drop_duplicates('eRNA_name').to_csv('enhancer_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "epi_df[[3, 4, 5, 'promoter_name', 7, 9]].drop_duplicates('promoter_name').to_csv('promoter_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "for i in range(50):\n",
    "    epi_list = epi_df[['eRNA_name', 'promoter_name']].values.tolist()\n",
    "    df = pd.DataFrame(epi_list)\n",
    "    df[1] = df[1].sample(frac=1, replace=False).reset_index()[1]\n",
    "    random_epi_list = df.values\n",
    "    blast_bash_file = open(f'random_EPI_blast_{i}.sh', 'w')\n",
    "    if os.path.exists(f'random_EPI_blastResult_{i}.txt'):\n",
    "        os.remove(f'random_EPI_blastResult_{i}.txt')\n",
    "    for j in random_epi_list:\n",
    "        eRNA = j[0]\n",
    "        promoter = j[1]\n",
    "        print(f\"blastn -task blastn -query EPI_fasta/{eRNA}.fa -db EPI_fasta/{promoter}.fa -outfmt 6 -evalue 1e-4 >> random_EPI_blastResult_{i}.txt\", file=blast_bash_file)\n",
    "    blast_bash_file.close()\n",
    "\n",
    "outfile = open('random_control.sh', 'w')\n",
    "for i in [x for x in os.listdir() if x.startswith('random_EPI_blast')]:\n",
    "    print(f\"bash {i} &\", file=outfile)\n",
    "print('wait', file=outfile)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65cf5b82-172d-40d8-a3fe-e08fdfd86c9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "subprocess.run('bash random_control.sh', shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "b5e8b015-758a-4c62-96e2-30d315e5ba4e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████| 50/50 [00:46<00:00,  1.08it/s]\n"
     ]
    }
   ],
   "source": [
    "from tqdm import tqdm\n",
    "# random control result analysis\n",
    "random_seuquence_similar_EPI_num_list = []\n",
    "random_transposon_similar_EPI_num_list = []\n",
    "random_SINE_similar_EPI_num_list = []\n",
    "random_each_transposon_ratio_list = []\n",
    "transposon_num_df = pd.read_csv(transposonFile, sep='\\t', header=None)\n",
    "transposon_copyNum_dict = transposon_num_df[6].value_counts().to_dict()\n",
    "\n",
    "for i in tqdm(range(50)):\n",
    "    random_result_file = f\"random_EPI_blastResult_{i}.txt\"\n",
    "    # Get the sequence similar enhancer in EPI\n",
    "    blastResultFile = f\"random_EPI_blastResult_{i}.txt\"\n",
    "    blastResultBedFile = f'random_EPI_enhancerBlastHit_{i}.bed'\n",
    "    with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "        for line in infile:\n",
    "            chromosome = line.split()[0].split(':')[0]\n",
    "            startPre = int(line.split()[0].split(':')[1].split('-')[0])\n",
    "            stopPre = int(line.split()[0].split(':')[1].split('-')[1].split('(')[0])\n",
    "            name = line.split()[0]\n",
    "            otherName = line.split()[1]\n",
    "            strand = line.split()[0].split(':')[1].split('(')[1][0]\n",
    "            if strand == '+' or strand == '.':\n",
    "                if int(line.split()[6]) < int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[6])\n",
    "                    shiftEnd = int(line.split()[7])\n",
    "                elif int(line.split()[6]) > int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[7])\n",
    "                    shiftEnd = int(line.split()[6])\n",
    "                start = startPre + shiftStart - 1\n",
    "                end = startPre + shiftEnd - 1\n",
    "            elif strand == '-':\n",
    "                if int(line.split()[6]) < int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[6])\n",
    "                    shiftEnd = int(line.split()[7])\n",
    "                elif int(line.split()[6]) > int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[7])\n",
    "                    shiftEnd = int(line.split()[6])\n",
    "                end = stopPre - shiftStart\n",
    "                start = stopPre - shiftEnd\n",
    "            print(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)\n",
    "    # Get the sequence similar promoter in EPI\n",
    "    blastResultFile = f\"random_EPI_blastResult_{i}.txt\"\n",
    "    blastResultBedFile = f'random_EPI_promoterBlastHit_{i}.bed'\n",
    "    with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "        for line in infile:\n",
    "            chromosome = line.split()[1].split(':')[0]\n",
    "            startPre = int(line.split()[1].split(':')[1].split('-')[0])\n",
    "            stopPre = int(line.split()[1].split(':')[1].split('-')[1].split('(')[0])\n",
    "            name = line.split()[1]\n",
    "            otherName = line.split()[0]\n",
    "            strand = line.split()[1].split(':')[1].split('(')[1][0]\n",
    "            if strand == '+' or strand == '.':\n",
    "                if int(line.split()[8]) < int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[8])\n",
    "                    shiftEnd = int(line.split()[9])\n",
    "                elif int(line.split()[8]) > int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[9])\n",
    "                    shiftEnd = int(line.split()[8])\n",
    "                start = startPre + shiftStart - 1\n",
    "                end = startPre + shiftEnd - 1\n",
    "            elif strand == '-':\n",
    "                if int(line.split()[8]) < int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[8])\n",
    "                    shiftEnd = int(line.split()[9])\n",
    "                elif int(line.split()[8]) > int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[9])\n",
    "                    shiftEnd = int(line.split()[8])\n",
    "                end = stopPre - shiftStart\n",
    "                start = stopPre - shiftEnd\n",
    "            print(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)\n",
    "    # Get out the TE similar EPI, save as ep_same_{te}.bedpe\n",
    "    try:\n",
    "        promoter_blast_df = pd.read_csv(f'random_EPI_promoterBlastHit_{i}.bed', sep='\\t', header=None)\n",
    "        enhancer_blast_df = pd.read_csv(f'random_EPI_enhancerBlastHit_{i}.bed', sep='\\t', header=None)\n",
    "    except:\n",
    "        random_seuquence_similar_EPI_num_list.append(0)\n",
    "        random_transposon_similar_EPI_num_list.append(0)\n",
    "        random_SINE_similar_EPI_num_list.append(0)\n",
    "        random_each_transposon_ratio_list.append({})\n",
    "        continue\n",
    "    promoter_blast_df.columns = ['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end', 'promoter', 'eRNA', 'promoter_strand']\n",
    "    enhancer_blast_df.columns = ['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end', 'eRNA', 'promoter', 'eRNA_strand']\n",
    "    epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "    epi_df.columns = ['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end',\n",
    "                   'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']\n",
    "    promoter_blast_df['promoter_chr'] = promoter_blast_df['promoter'].str.split(':').str[0]\n",
    "    promoter_blast_df['promoter_start'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    promoter_blast_df['promoter_end'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "    promoter_blast_df['eRNA_chr'] = promoter_blast_df['eRNA'].str.split(':').str[0]\n",
    "    promoter_blast_df['eRNA_start'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    promoter_blast_df['eRNA_end'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "    enhancer_blast_df['eRNA_chr'] = enhancer_blast_df['eRNA'].str.split(':').str[0]\n",
    "    enhancer_blast_df['eRNA_start'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    enhancer_blast_df['eRNA_end'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "    enhancer_blast_df['promoter_chr'] = enhancer_blast_df['promoter'].str.split(':').str[0]\n",
    "    enhancer_blast_df['promoter_start'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    enhancer_blast_df['promoter_end'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "    # Get out the intersection between transposon and blastHit\n",
    "    x = pd.merge(epi_df, promoter_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                                'promoter_start', 'promoter_end', 'promoter_strand'], how='inner')\n",
    "    y = pd.merge(x, enhancer_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                           'promoter_start', 'promoter_end', 'promoter', 'eRNA', 'eRNA_strand'\n",
    "                                           ], how='inner')\n",
    "    cmd = f\"bedtools intersect -a random_EPI_promoterBlastHit_{i}.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > random_EPI_promoterBlastHit_{i}_transposon_intersectWo.bed\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    cmd = f\"bedtools intersect -a random_EPI_enhancerBlastHit_{i}.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > random_EPI_enhancerBlastHit_{i}_transposon_intersectWo.bed\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "\n",
    "    \n",
    "    promoter_blast_te_intersectWo_df = pd.read_csv(f'random_EPI_promoterBlastHit_{i}_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "    enhancer_blast_te_intersectWo_df = pd.read_csv(f'random_EPI_enhancerBlastHit_{i}_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "    promoter_blast_te_intersectWo_df = promoter_blast_te_intersectWo_df.rename({\n",
    "        0:'promoter_blastHit_chr', 1:'promoter_blastHit_start', 2:'promoter_blastHit_end', \n",
    "        3:'promoter', 4:'eRNA', 5:'promoter_strand'}, axis=1)\n",
    "    enhancer_blast_te_intersectWo_df = enhancer_blast_te_intersectWo_df.rename({\n",
    "        0:'eRNA_blastHit_chr', 1:'eRNA_blastHit_start', 2:'eRNA_blastHit_end',\n",
    "        3:'eRNA', 4:'promoter', 5:'eRNA_strand'}, axis=1)\n",
    "\n",
    "    # merge the promoter and enhancer information together\n",
    "    transposon_similar_EPI_df = pd.merge(\n",
    "        pd.merge(y, promoter_blast_te_intersectWo_df, on=['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end',\n",
    "                                                      'promoter', 'eRNA', 'promoter_strand'], how='inner'),\n",
    "        enhancer_blast_te_intersectWo_df, on=['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end',\n",
    "                                                'eRNA', 'promoter', 'eRNA_strand'], how='inner')\n",
    "\n",
    "    transposon_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'] == transposon_similar_EPI_df['12_y']]\n",
    "    SINE_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'].str.contains('SINE')]\n",
    "    \n",
    "    # seuquence-similar EPI num\n",
    "    random_seuquence_similar_EPI_num_list.append(y['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "    # transposon-similar EPI num\n",
    "    random_transposon_similar_EPI_num_list.append(transposon_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "    # SINE-similar EPI num\n",
    "    random_SINE_similar_EPI_num_list.append(SINE_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "    # each transposon ratio in transposon-similar EPI (corrected by copy number)\n",
    "    transposon_num_dict = transposon_similar_EPI_df['12_x'].value_counts()\n",
    "    if len(transposon_num_dict) == 0:\n",
    "        random_each_transposon_ratio_list.append({})\n",
    "        continue\n",
    "    transposon_num_correct_dict = dict()\n",
    "    transposon_ratio_correct_dict = dict()\n",
    "    for key in transposon_num_dict.keys():\n",
    "        result = transposon_num_dict[key] / transposon_copyNum_dict[key]\n",
    "        transposon_num_correct_dict[key] = [result]\n",
    "    for key, value in (pd.DataFrame.from_dict(transposon_num_correct_dict).T / pd.DataFrame.from_dict(transposon_num_correct_dict).T.sum())[0].to_dict().items():\n",
    "        transposon_ratio_correct_dict[key] = round(value, 2)\n",
    "    random_each_transposon_ratio_list.append(transposon_ratio_correct_dict)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "f1f8faec-f979-462e-9f13-a0a1a79a28b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "xxx = pd.read_csv(transposonFile, sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "bfa5461d-96ab-4123-8196-46b5c2987b36",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrI</td>\n",
       "      <td>596</td>\n",
       "      <td>676</td>\n",
       "      <td>RCS5</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>1307;1387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>chrI</td>\n",
       "      <td>55528</td>\n",
       "      <td>55642</td>\n",
       "      <td>CeRep52</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>2;112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>chrI</td>\n",
       "      <td>60410</td>\n",
       "      <td>60496</td>\n",
       "      <td>MINISAT1_CE</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>1;163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>chrI</td>\n",
       "      <td>60624</td>\n",
       "      <td>60657</td>\n",
       "      <td>MINISAT1_CE</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>164;231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>chrI</td>\n",
       "      <td>121683</td>\n",
       "      <td>121694</td>\n",
       "      <td>MINISAT1_CE</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>57;195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84807</th>\n",
       "      <td>chrX</td>\n",
       "      <td>17574180</td>\n",
       "      <td>17574264</td>\n",
       "      <td>RCS5</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>1305;1387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84849</th>\n",
       "      <td>chrX</td>\n",
       "      <td>17643559</td>\n",
       "      <td>17643648</td>\n",
       "      <td>MINISAT1_CE</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>81;184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84857</th>\n",
       "      <td>chrX</td>\n",
       "      <td>17655652</td>\n",
       "      <td>17655758</td>\n",
       "      <td>MINISAT1_CE</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>15;124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84859</th>\n",
       "      <td>chrX</td>\n",
       "      <td>17655960</td>\n",
       "      <td>17656060</td>\n",
       "      <td>MINISAT1_CE</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>88;193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84877</th>\n",
       "      <td>chrX</td>\n",
       "      <td>17686716</td>\n",
       "      <td>17686913</td>\n",
       "      <td>MINISAT1_CE</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>Satellite</td>\n",
       "      <td>45;238</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5495 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          0         1         2            3  4  5          6          7\n",
       "2      chrI       596       676         RCS5  1  -  Satellite  1307;1387\n",
       "75     chrI     55528     55642      CeRep52  1  +  Satellite      2;112\n",
       "79     chrI     60410     60496  MINISAT1_CE  1  +  Satellite      1;163\n",
       "81     chrI     60624     60657  MINISAT1_CE  1  +  Satellite    164;231\n",
       "132    chrI    121683    121694  MINISAT1_CE  1  -  Satellite     57;195\n",
       "...     ...       ...       ...          ... .. ..        ...        ...\n",
       "84807  chrX  17574180  17574264         RCS5  1  +  Satellite  1305;1387\n",
       "84849  chrX  17643559  17643648  MINISAT1_CE  1  -  Satellite     81;184\n",
       "84857  chrX  17655652  17655758  MINISAT1_CE  1  -  Satellite     15;124\n",
       "84859  chrX  17655960  17656060  MINISAT1_CE  1  -  Satellite     88;193\n",
       "84877  chrX  17686716  17686913  MINISAT1_CE  1  -  Satellite     45;238\n",
       "\n",
       "[5495 rows x 8 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xxx[xxx[6].str.contains('Satellite')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "ca2ca6ae-98ec-44f1-89be-0f6b95fb82d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "te_dict = dict()\n",
    "with open('/home/limh/download/repeat_20181026.fasta', 'r') as infile:\n",
    "    for id, line in enumerate(infile):\n",
    "        if id%2 == 0:\n",
    "            name = line.split(';')[0].strip('>')\n",
    "        else:\n",
    "            sequence = line.strip()\n",
    "            te_dict[name] = sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "0fd0f8bc-e2de-4b8e-a51e-19c83aca9a53",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'gttttgaaaatttccggcaatcggcaatttcggcaattgccggttttggaaattttcggcaatttcggcaattgccggttttggaaatttccggcaatcggcaatttcggcaattgccggttttggaaatttccggcaatcggcaatttcggcaattgccggttttggaaattttcggcaatttcggcaattgccggttttgaaaatttccggcaatcggcaatttcggcaattgccg'"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "te_dict['MINISAT1_CE']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "442ea117-ef83-4a7b-9402-5b65574f2fce",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:chipseq]",
   "language": "python",
   "name": "conda-env-chipseq-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
