{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5d6bdb36-ab40-468f-b143-d7ca147884fe",
   "metadata": {},
   "source": [
    "# Calculate enhancer、eRNA、EPI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0cfd3213-c781-45bb-af2c-875946ccc192",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('display.max_columns', None)\n",
    "import subprocess\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import os\n",
    "import pyBigWig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2765aed2-32cf-4ecc-acaa-a5c80cdd3763",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fuction define. if eRNA intersect with promoter, return False. else return True\n",
    "def ifFilter(row):\n",
    "    try:\n",
    "        if (row['eRNA_start'] >= row['promoter_start']) & (row['eRNA_start'] <= row['promoter_end']):\n",
    "            return True\n",
    "        elif (row['eRNA_end'] >= row['promoter_start']) & (row['eRNA_end'] <= row['promoter_end']):\n",
    "            return True\n",
    "        else:\n",
    "            return False\n",
    "    except:\n",
    "        try:\n",
    "            if (row['enhancer_start'] >= row['promoter_start']) & (row['enhancer_start'] <= row['promoter_end']):\n",
    "                return True\n",
    "            elif (row['enhancer_end'] >= row['promoter_start']) & (row['enhancer_end'] <= row['promoter_end']):\n",
    "                return True\n",
    "            else:\n",
    "                return False\n",
    "        except:\n",
    "            raise KeyError"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "630b2185-8eb0-498c-b866-3e49ac6a939b",
   "metadata": {},
   "source": [
    "## enhancer identification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "dd0c962a-bfe9-445e-92b1-753167512044",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # sequence data download\n",
    "# cmd = f\"bash download.sh\"\n",
    "# subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "363e8789-4f4d-458c-918e-a916cf480f99",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:42:50,401 - SRR ids: ['ATAC_r1_pair.sra']\n",
      "2023-11-21 19:42:50,401 - extra args: ['--gzip', '--split-3']\n",
      "2023-11-21 19:42:50,516 - tempdir: /tmp/pfd_rla0a8ta\n",
      "2023-11-21 19:42:50,516 - CMD: sra-stat --meta --quick ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,550 - ATAC_r1_pair.sra spots: 38189723\n",
      "2023-11-21 19:42:50,550 - blocks: [[1, 3818972], [3818973, 7637944], [7637945, 11456916], [11456917, 15275888], [15275889, 19094860], [19094861, 22913832], [22913833, 26732804], [26732805, 30551776], [30551777, 34370748], [34370749, 38189723]]\n",
      "2023-11-21 19:42:50,552 - CMD: fastq-dump -N 1 -X 3818972 -O /tmp/pfd_rla0a8ta/0 --gzip --split-3 ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,553 - CMD: fastq-dump -N 3818973 -X 7637944 -O /tmp/pfd_rla0a8ta/1 --gzip --split-3 ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,556 - CMD: fastq-dump -N 7637945 -X 11456916 -O /tmp/pfd_rla0a8ta/2 --gzip --split-3 ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,581 - CMD: fastq-dump -N 11456917 -X 15275888 -O /tmp/pfd_rla0a8ta/3 --gzip --split-3 ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,584 - CMD: fastq-dump -N 15275889 -X 19094860 -O /tmp/pfd_rla0a8ta/4 --gzip --split-3 ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,586 - CMD: fastq-dump -N 19094861 -X 22913832 -O /tmp/pfd_rla0a8ta/5 --gzip --split-3 ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,589 - CMD: fastq-dump -N 22913833 -X 26732804 -O /tmp/pfd_rla0a8ta/6 --gzip --split-3 ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,591 - CMD: fastq-dump -N 26732805 -X 30551776 -O /tmp/pfd_rla0a8ta/7 --gzip --split-3 ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,592 - CMD: fastq-dump -N 30551777 -X 34370748 -O /tmp/pfd_rla0a8ta/8 --gzip --split-3 ATAC_r1_pair.sra\n",
      "2023-11-21 19:42:50,594 - CMD: fastq-dump -N 34370749 -X 38189723 -O /tmp/pfd_rla0a8ta/9 --gzip --split-3 ATAC_r1_pair.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 3818972 spots for ATAC_r1_pair.sra\n",
      "Written 3818972 spots for ATAC_r1_pair.sra\n",
      "Read 3818972 spots for ATAC_r1_pair.sra\n",
      "Written 3818972 spots for ATAC_r1_pair.sra\n",
      "Read 3818975 spots for ATAC_r1_pair.sra\n",
      "Written 3818975 spots for ATAC_r1_pair.sra\n",
      "Read 3818972 spots for ATAC_r1_pair.sra\n",
      "Written 3818972 spots for ATAC_r1_pair.sra\n",
      "Read 3818972 spots for ATAC_r1_pair.sra\n",
      "Written 3818972 spots for ATAC_r1_pair.sra\n",
      "Read 3818972 spots for ATAC_r1_pair.sra\n",
      "Written 3818972 spots for ATAC_r1_pair.sra\n",
      "Read 3818972 spots for ATAC_r1_pair.sra\n",
      "Written 3818972 spots for ATAC_r1_pair.sra\n",
      "Read 3818972 spots for ATAC_r1_pair.sra\n",
      "Written 3818972 spots for ATAC_r1_pair.sra\n",
      "Read 3818972 spots for ATAC_r1_pair.sra\n",
      "Written 3818972 spots for ATAC_r1_pair.sra\n",
      "Read 3818972 spots for ATAC_r1_pair.sra\n",
      "Written 3818972 spots for ATAC_r1_pair.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:47:27,873 - SRR ids: ['H3K27ac_r1_single.sra']\n",
      "2023-11-21 19:47:27,873 - extra args: ['--gzip']\n",
      "2023-11-21 19:47:27,903 - tempdir: /tmp/pfd_exxmys2y\n",
      "2023-11-21 19:47:27,903 - CMD: sra-stat --meta --quick H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,957 - H3K27ac_r1_single.sra spots: 11440822\n",
      "2023-11-21 19:47:27,957 - blocks: [[1, 1144082], [1144083, 2288164], [2288165, 3432246], [3432247, 4576328], [4576329, 5720410], [5720411, 6864492], [6864493, 8008574], [8008575, 9152656], [9152657, 10296738], [10296739, 11440822]]\n",
      "2023-11-21 19:47:27,958 - CMD: fastq-dump -N 1 -X 1144082 -O /tmp/pfd_exxmys2y/0 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,960 - CMD: fastq-dump -N 1144083 -X 2288164 -O /tmp/pfd_exxmys2y/1 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,962 - CMD: fastq-dump -N 2288165 -X 3432246 -O /tmp/pfd_exxmys2y/2 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,964 - CMD: fastq-dump -N 3432247 -X 4576328 -O /tmp/pfd_exxmys2y/3 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,966 - CMD: fastq-dump -N 4576329 -X 5720410 -O /tmp/pfd_exxmys2y/4 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,968 - CMD: fastq-dump -N 5720411 -X 6864492 -O /tmp/pfd_exxmys2y/5 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,970 - CMD: fastq-dump -N 6864493 -X 8008574 -O /tmp/pfd_exxmys2y/6 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,972 - CMD: fastq-dump -N 8008575 -X 9152656 -O /tmp/pfd_exxmys2y/7 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,974 - CMD: fastq-dump -N 9152657 -X 10296738 -O /tmp/pfd_exxmys2y/8 --gzip H3K27ac_r1_single.sra\n",
      "2023-11-21 19:47:27,976 - CMD: fastq-dump -N 10296739 -X 11440822 -O /tmp/pfd_exxmys2y/9 --gzip H3K27ac_r1_single.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 1144082 spots for H3K27ac_r1_single.sra\n",
      "Written 1144082 spots for H3K27ac_r1_single.sra\n",
      "Read 1144084 spots for H3K27ac_r1_single.sra\n",
      "Written 1144084 spots for H3K27ac_r1_single.sra\n",
      "Read 1144082 spots for H3K27ac_r1_single.sra\n",
      "Written 1144082 spots for H3K27ac_r1_single.sra\n",
      "Read 1144082 spots for H3K27ac_r1_single.sra\n",
      "Written 1144082 spots for H3K27ac_r1_single.sra\n",
      "Read 1144082 spots for H3K27ac_r1_single.sra\n",
      "Written 1144082 spots for H3K27ac_r1_single.sra\n",
      "Read 1144082 spots for H3K27ac_r1_single.sra\n",
      "Written 1144082 spots for H3K27ac_r1_single.sra\n",
      "Read 1144082 spots for H3K27ac_r1_single.sra\n",
      "Written 1144082 spots for H3K27ac_r1_single.sra\n",
      "Read 1144082 spots for H3K27ac_r1_single.sra\n",
      "Written 1144082 spots for H3K27ac_r1_single.sra\n",
      "Read 1144082 spots for H3K27ac_r1_single.sra\n",
      "Written 1144082 spots for H3K27ac_r1_single.sra\n",
      "Read 1144082 spots for H3K27ac_r1_single.sra\n",
      "Written 1144082 spots for H3K27ac_r1_single.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:49:03,983 - SRR ids: ['H3K4me1_r1_single.sra']\n",
      "2023-11-21 19:49:03,983 - extra args: ['--gzip']\n",
      "2023-11-21 19:49:04,256 - tempdir: /tmp/pfd_p1hog9nb\n",
      "2023-11-21 19:49:04,256 - CMD: sra-stat --meta --quick H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,338 - H3K4me1_r1_single.sra spots: 20429006\n",
      "2023-11-21 19:49:04,338 - blocks: [[1, 2042900], [2042901, 4085800], [4085801, 6128700], [6128701, 8171600], [8171601, 10214500], [10214501, 12257400], [12257401, 14300300], [14300301, 16343200], [16343201, 18386100], [18386101, 20429006]]\n",
      "2023-11-21 19:49:04,339 - CMD: fastq-dump -N 1 -X 2042900 -O /tmp/pfd_p1hog9nb/0 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,341 - CMD: fastq-dump -N 2042901 -X 4085800 -O /tmp/pfd_p1hog9nb/1 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,343 - CMD: fastq-dump -N 4085801 -X 6128700 -O /tmp/pfd_p1hog9nb/2 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,345 - CMD: fastq-dump -N 6128701 -X 8171600 -O /tmp/pfd_p1hog9nb/3 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,347 - CMD: fastq-dump -N 8171601 -X 10214500 -O /tmp/pfd_p1hog9nb/4 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,349 - CMD: fastq-dump -N 10214501 -X 12257400 -O /tmp/pfd_p1hog9nb/5 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,350 - CMD: fastq-dump -N 12257401 -X 14300300 -O /tmp/pfd_p1hog9nb/6 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,351 - CMD: fastq-dump -N 14300301 -X 16343200 -O /tmp/pfd_p1hog9nb/7 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,353 - CMD: fastq-dump -N 16343201 -X 18386100 -O /tmp/pfd_p1hog9nb/8 --gzip H3K4me1_r1_single.sra\n",
      "2023-11-21 19:49:04,354 - CMD: fastq-dump -N 18386101 -X 20429006 -O /tmp/pfd_p1hog9nb/9 --gzip H3K4me1_r1_single.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 2042900 spots for H3K4me1_r1_single.sra\n",
      "Written 2042900 spots for H3K4me1_r1_single.sra\n",
      "Read 2042900 spots for H3K4me1_r1_single.sra\n",
      "Written 2042900 spots for H3K4me1_r1_single.sra\n",
      "Read 2042900 spots for H3K4me1_r1_single.sra\n",
      "Written 2042900 spots for H3K4me1_r1_single.sra\n",
      "Read 2042900 spots for H3K4me1_r1_single.sra\n",
      "Written 2042900 spots for H3K4me1_r1_single.sra\n",
      "Read 2042900 spots for H3K4me1_r1_single.sra\n",
      "Written 2042900 spots for H3K4me1_r1_single.sra\n",
      "Read 2042900 spots for H3K4me1_r1_single.sra\n",
      "Written 2042900 spots for H3K4me1_r1_single.sra\n",
      "Read 2042900 spots for H3K4me1_r1_single.sra\n",
      "Written 2042900 spots for H3K4me1_r1_single.sra\n",
      "Read 2042900 spots for H3K4me1_r1_single.sra\n",
      "Written 2042900 spots for H3K4me1_r1_single.sra\n",
      "Read 2042900 spots for H3K4me1_r1_single.sra\n",
      "Written 2042900 spots for H3K4me1_r1_single.sra\n",
      "Read 2042906 spots for H3K4me1_r1_single.sra\n",
      "Written 2042906 spots for H3K4me1_r1_single.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-21 19:50:36,149 - SRR ids: ['H3K4me3_r1_single.sra']\n",
      "2023-11-21 19:50:36,149 - extra args: ['--gzip']\n",
      "2023-11-21 19:50:36,298 - tempdir: /tmp/pfd_x3nuwx2c\n",
      "2023-11-21 19:50:36,299 - CMD: sra-stat --meta --quick H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,358 - H3K4me3_r1_single.sra spots: 24215389\n",
      "2023-11-21 19:50:36,358 - blocks: [[1, 2421538], [2421539, 4843076], [4843077, 7264614], [7264615, 9686152], [9686153, 12107690], [12107691, 14529228], [14529229, 16950766], [16950767, 19372304], [19372305, 21793842], [21793843, 24215389]]\n",
      "2023-11-21 19:50:36,360 - CMD: fastq-dump -N 1 -X 2421538 -O /tmp/pfd_x3nuwx2c/0 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,362 - CMD: fastq-dump -N 2421539 -X 4843076 -O /tmp/pfd_x3nuwx2c/1 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,363 - CMD: fastq-dump -N 4843077 -X 7264614 -O /tmp/pfd_x3nuwx2c/2 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,365 - CMD: fastq-dump -N 7264615 -X 9686152 -O /tmp/pfd_x3nuwx2c/3 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,367 - CMD: fastq-dump -N 9686153 -X 12107690 -O /tmp/pfd_x3nuwx2c/4 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,369 - CMD: fastq-dump -N 12107691 -X 14529228 -O /tmp/pfd_x3nuwx2c/5 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,371 - CMD: fastq-dump -N 14529229 -X 16950766 -O /tmp/pfd_x3nuwx2c/6 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,373 - CMD: fastq-dump -N 16950767 -X 19372304 -O /tmp/pfd_x3nuwx2c/7 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,375 - CMD: fastq-dump -N 19372305 -X 21793842 -O /tmp/pfd_x3nuwx2c/8 --gzip H3K4me3_r1_single.sra\n",
      "2023-11-21 19:50:36,377 - CMD: fastq-dump -N 21793843 -X 24215389 -O /tmp/pfd_x3nuwx2c/9 --gzip H3K4me3_r1_single.sra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 2421538 spots for H3K4me3_r1_single.sra\n",
      "Written 2421538 spots for H3K4me3_r1_single.sra\n",
      "Read 2421538 spots for H3K4me3_r1_single.sra\n",
      "Written 2421538 spots for H3K4me3_r1_single.sra\n",
      "Read 2421538 spots for H3K4me3_r1_single.sra\n",
      "Written 2421538 spots for H3K4me3_r1_single.sra\n",
      "Read 2421538 spots for H3K4me3_r1_single.sra\n",
      "Written 2421538 spots for H3K4me3_r1_single.sra\n",
      "Read 2421538 spots for H3K4me3_r1_single.sra\n",
      "Written 2421538 spots for H3K4me3_r1_single.sra\n",
      "Read 2421547 spots for H3K4me3_r1_single.sra\n",
      "Written 2421547 spots for H3K4me3_r1_single.sra\n",
      "Read 2421538 spots for H3K4me3_r1_single.sra\n",
      "Written 2421538 spots for H3K4me3_r1_single.sra\n",
      "Read 2421538 spots for H3K4me3_r1_single.sra\n",
      "Written 2421538 spots for H3K4me3_r1_single.sra\n",
      "Read 2421538 spots for H3K4me3_r1_single.sra\n",
      "Written 2421538 spots for H3K4me3_r1_single.sra\n",
      "Read 2421538 spots for H3K4me3_r1_single.sra\n",
      "Written 2421538 spots for H3K4me3_r1_single.sra\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/danRer10/whole_embryo_24hpf/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ATAC_r1_pair_1.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t92990\tCTGTCTCTTATA\t1000000\t9.30\n",
      "Illumina\t1\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Nextera adapter for trimming (count: 92990). Second best hit was Illumina (count: 1)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/danRer10/whole_embryo_24hpf/ATAC_r1_pair_1.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ATAC_r1_pair_1.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to ATAC_r1_pair_1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'CTGTCTCTTATA' from file ATAC_r1_pair_1.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a CTGTCTCTTATA ATAC_r1_pair_1.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 638.240 s (16.712 µs/read; 3.59 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              38,189,723\n",
      "Reads with adapters:                 9,706,231 (25.4%)\n",
      "Reads written (passing filters):    38,189,723 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,864,229,225 bp\n",
      "Quality-trimmed:               3,419,290 bp (0.1%)\n",
      "Total written (filtered):  2,767,363,996 bp (96.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 9706231 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 12.5%\n",
      "  C: 38.8%\n",
      "  G: 26.5%\n",
      "  T: 22.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1131615\t596714.4\t0\t1131615\n",
      "4\t666332\t149178.6\t0\t666332\n",
      "5\t510526\t37294.7\t0\t510526\n",
      "6\t479877\t9323.7\t0\t479877\n",
      "7\t502863\t2330.9\t0\t502863\n",
      "8\t561486\t582.7\t0\t561486\n",
      "9\t641116\t145.7\t0\t640092 1024\n",
      "10\t733215\t36.4\t1\t715249 17966\n",
      "11\t775612\t9.1\t1\t755401 20211\n",
      "12\t815765\t2.3\t1\t792730 23035\n",
      "13\t776676\t2.3\t1\t757709 18967\n",
      "14\t743264\t2.3\t1\t723233 20031\n",
      "15\t679048\t2.3\t1\t662147 16901\n",
      "16\t641160\t2.3\t1\t625082 16078\n",
      "17\t1393\t2.3\t1\t217 1176\n",
      "18\t1151\t2.3\t1\t173 978\n",
      "19\t1913\t2.3\t1\t380 1533\n",
      "20\t2095\t2.3\t1\t407 1688\n",
      "21\t1933\t2.3\t1\t417 1516\n",
      "22\t2422\t2.3\t1\t580 1842\n",
      "23\t1681\t2.3\t1\t376 1305\n",
      "24\t2308\t2.3\t1\t576 1732\n",
      "25\t1857\t2.3\t1\t510 1347\n",
      "26\t1641\t2.3\t1\t458 1183\n",
      "27\t1803\t2.3\t1\t509 1294\n",
      "28\t1773\t2.3\t1\t508 1265\n",
      "29\t1775\t2.3\t1\t514 1261\n",
      "30\t2353\t2.3\t1\t703 1650\n",
      "31\t1699\t2.3\t1\t529 1170\n",
      "32\t1940\t2.3\t1\t583 1357\n",
      "33\t1532\t2.3\t1\t443 1089\n",
      "34\t1416\t2.3\t1\t359 1057\n",
      "35\t1033\t2.3\t1\t332 701\n",
      "36\t1103\t2.3\t1\t275 828\n",
      "37\t733\t2.3\t1\t151 582\n",
      "38\t851\t2.3\t1\t163 688\n",
      "39\t798\t2.3\t1\t163 635\n",
      "40\t582\t2.3\t1\t99 483\n",
      "41\t485\t2.3\t1\t93 392\n",
      "42\t602\t2.3\t1\t99 503\n",
      "43\t383\t2.3\t1\t41 342\n",
      "44\t398\t2.3\t1\t42 356\n",
      "45\t355\t2.3\t1\t20 335\n",
      "46\t442\t2.3\t1\t49 393\n",
      "47\t353\t2.3\t1\t34 319\n",
      "48\t311\t2.3\t1\t40 271\n",
      "49\t317\t2.3\t1\t20 297\n",
      "50\t323\t2.3\t1\t11 312\n",
      "51\t240\t2.3\t1\t13 227\n",
      "52\t260\t2.3\t1\t10 250\n",
      "53\t223\t2.3\t1\t5 218\n",
      "54\t252\t2.3\t1\t7 245\n",
      "55\t270\t2.3\t1\t2 268\n",
      "56\t236\t2.3\t1\t2 234\n",
      "57\t207\t2.3\t1\t5 202\n",
      "58\t195\t2.3\t1\t3 192\n",
      "59\t226\t2.3\t1\t1 225\n",
      "60\t241\t2.3\t1\t4 237\n",
      "61\t247\t2.3\t1\t6 241\n",
      "62\t283\t2.3\t1\t1 282\n",
      "63\t299\t2.3\t1\t1 298\n",
      "64\t204\t2.3\t1\t2 202\n",
      "65\t296\t2.3\t1\t1 295\n",
      "66\t225\t2.3\t1\t0 225\n",
      "67\t648\t2.3\t1\t7 641\n",
      "68\t145\t2.3\t1\t0 145\n",
      "69\t215\t2.3\t1\t4 211\n",
      "70\t73\t2.3\t1\t0 73\n",
      "71\t133\t2.3\t1\t0 133\n",
      "72\t350\t2.3\t1\t1 349\n",
      "73\t175\t2.3\t1\t1 174\n",
      "74\t132\t2.3\t1\t1 131\n",
      "75\t147\t2.3\t1\t0 147\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ATAC_r1_pair_1.fastq.gz\n",
      "=============================================\n",
      "38189723 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/danRer10/whole_embryo_24hpf/ATAC_r1_pair_2.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ATAC_r1_pair_2.fastq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j -j 2\n",
      "Writing final adapter and quality trimmed output to ATAC_r1_pair_2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'CTGTCTCTTATA' from file ATAC_r1_pair_2.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a CTGTCTCTTATA ATAC_r1_pair_2.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 681.180 s (17.837 µs/read; 3.36 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              38,189,723\n",
      "Reads with adapters:                 9,640,664 (25.2%)\n",
      "Reads written (passing filters):    38,189,723 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,864,229,225 bp\n",
      "Quality-trimmed:              10,310,954 bp (0.4%)\n",
      "Total written (filtered):  2,761,088,426 bp (96.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 9640664 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 12.7%\n",
      "  C: 39.1%\n",
      "  G: 26.2%\n",
      "  T: 22.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1115730\t596714.4\t0\t1115730\n",
      "4\t665731\t149178.6\t0\t665731\n",
      "5\t509465\t37294.7\t0\t509465\n",
      "6\t476659\t9323.7\t0\t476659\n",
      "7\t498060\t2330.9\t0\t498060\n",
      "8\t555588\t582.7\t0\t555588\n",
      "9\t636497\t145.7\t0\t635514 983\n",
      "10\t728752\t36.4\t1\t708659 20093\n",
      "11\t773618\t9.1\t1\t751848 21770\n",
      "12\t815634\t2.3\t1\t790042 25592\n",
      "13\t767102\t2.3\t1\t747407 19695\n",
      "14\t741868\t2.3\t1\t720977 20891\n",
      "15\t674348\t2.3\t1\t657228 17120\n",
      "16\t637355\t2.3\t1\t620990 16365\n",
      "17\t1402\t2.3\t1\t222 1180\n",
      "18\t1030\t2.3\t1\t142 888\n",
      "19\t1661\t2.3\t1\t304 1357\n",
      "20\t1717\t2.3\t1\t365 1352\n",
      "21\t1806\t2.3\t1\t450 1356\n",
      "22\t1586\t2.3\t1\t330 1256\n",
      "23\t1724\t2.3\t1\t420 1304\n",
      "24\t1697\t2.3\t1\t419 1278\n",
      "25\t1589\t2.3\t1\t371 1218\n",
      "26\t2091\t2.3\t1\t563 1528\n",
      "27\t1525\t2.3\t1\t426 1099\n",
      "28\t1385\t2.3\t1\t391 994\n",
      "29\t1777\t2.3\t1\t566 1211\n",
      "30\t1766\t2.3\t1\t530 1236\n",
      "31\t1313\t2.3\t1\t357 956\n",
      "32\t1666\t2.3\t1\t532 1134\n",
      "33\t1123\t2.3\t1\t325 798\n",
      "34\t1464\t2.3\t1\t401 1063\n",
      "35\t891\t2.3\t1\t262 629\n",
      "36\t1159\t2.3\t1\t282 877\n",
      "37\t959\t2.3\t1\t233 726\n",
      "38\t926\t2.3\t1\t176 750\n",
      "39\t686\t2.3\t1\t146 540\n",
      "40\t771\t2.3\t1\t166 605\n",
      "41\t584\t2.3\t1\t91 493\n",
      "42\t661\t2.3\t1\t93 568\n",
      "43\t487\t2.3\t1\t71 416\n",
      "44\t648\t2.3\t1\t130 518\n",
      "45\t379\t2.3\t1\t44 335\n",
      "46\t526\t2.3\t1\t85 441\n",
      "47\t338\t2.3\t1\t41 297\n",
      "48\t463\t2.3\t1\t85 378\n",
      "49\t335\t2.3\t1\t31 304\n",
      "50\t320\t2.3\t1\t29 291\n",
      "51\t246\t2.3\t1\t8 238\n",
      "52\t262\t2.3\t1\t16 246\n",
      "53\t225\t2.3\t1\t10 215\n",
      "54\t258\t2.3\t1\t14 244\n",
      "55\t259\t2.3\t1\t4 255\n",
      "56\t206\t2.3\t1\t2 204\n",
      "57\t217\t2.3\t1\t1 216\n",
      "58\t208\t2.3\t1\t2 206\n",
      "59\t230\t2.3\t1\t2 228\n",
      "60\t265\t2.3\t1\t4 261\n",
      "61\t210\t2.3\t1\t2 208\n",
      "62\t294\t2.3\t1\t1 293\n",
      "63\t295\t2.3\t1\t3 292\n",
      "64\t208\t2.3\t1\t1 207\n",
      "65\t292\t2.3\t1\t0 292\n",
      "66\t222\t2.3\t1\t0 222\n",
      "67\t667\t2.3\t1\t5 662\n",
      "68\t125\t2.3\t1\t0 125\n",
      "69\t175\t2.3\t1\t3 172\n",
      "70\t67\t2.3\t1\t1 66\n",
      "71\t143\t2.3\t1\t1 142\n",
      "72\t337\t2.3\t1\t1 336\n",
      "73\t149\t2.3\t1\t1 148\n",
      "74\t114\t2.3\t1\t3 111\n",
      "75\t128\t2.3\t1\t0 128\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ATAC_r1_pair_2.fastq.gz\n",
      "=============================================\n",
      "38189723 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files ATAC_r1_pair_1_trimmed.fq.gz and ATAC_r1_pair_2_trimmed.fq.gz\n",
      "file_1: ATAC_r1_pair_1_trimmed.fq.gz, file_2: ATAC_r1_pair_2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: ATAC_r1_pair_1_trimmed.fq.gz and ATAC_r1_pair_2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end Read 1 reads to ATAC_r1_pair_1_val_1.fq.gz\n",
      "Writing validated paired-end Read 2 reads to ATAC_r1_pair_2_val_2.fq.gz\n",
      "\n",
      "Total number of sequences analysed: 38189723\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 10678 (0.03%)\n",
      "\n",
      "Deleting both intermediate output files ATAC_r1_pair_1_trimmed.fq.gz and ATAC_r1_pair_2_trimmed.fq.gz\n",
      "\n",
      "====================================================================================================\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/danRer10/whole_embryo_24hpf/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> H3K27ac_r1_single.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t2\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Nextera adapter for trimming (count: 2). Second best hit was Illumina (count: 0)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/danRer10/whole_embryo_24hpf/H3K27ac_r1_single.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: H3K27ac_r1_single.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to H3K27ac_r1_single_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'CTGTCTCTTATA' from file H3K27ac_r1_single.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a CTGTCTCTTATA H3K27ac_r1_single.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 137.934 s (12.056 µs/read; 4.98 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              11,440,822\n",
      "Reads with adapters:                   318,499 (2.8%)\n",
      "Reads written (passing filters):    11,440,822 (100.0%)\n",
      "\n",
      "Total basepairs processed:   560,600,278 bp\n",
      "Quality-trimmed:               5,590,676 bp (1.0%)\n",
      "Total written (filtered):    553,840,362 bp (98.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 318499 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 26.1%\n",
      "  C: 17.3%\n",
      "  G: 27.2%\n",
      "  T: 29.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t230025\t178762.8\t0\t230025\n",
      "4\t64434\t44690.7\t0\t64434\n",
      "5\t12996\t11172.7\t0\t12996\n",
      "6\t5025\t2793.2\t0\t5025\n",
      "7\t872\t698.3\t0\t872\n",
      "8\t453\t174.6\t0\t453\n",
      "9\t541\t43.6\t0\t102 439\n",
      "10\t545\t10.9\t1\t14 531\n",
      "11\t268\t2.7\t1\t1 267\n",
      "12\t90\t0.7\t1\t0 90\n",
      "13\t85\t0.7\t1\t0 85\n",
      "14\t90\t0.7\t1\t1 89\n",
      "15\t82\t0.7\t1\t1 81\n",
      "16\t68\t0.7\t1\t1 67\n",
      "17\t82\t0.7\t1\t0 82\n",
      "18\t82\t0.7\t1\t0 82\n",
      "19\t78\t0.7\t1\t1 77\n",
      "20\t77\t0.7\t1\t0 77\n",
      "21\t101\t0.7\t1\t1 100\n",
      "22\t73\t0.7\t1\t0 73\n",
      "23\t78\t0.7\t1\t0 78\n",
      "24\t98\t0.7\t1\t1 97\n",
      "25\t100\t0.7\t1\t0 100\n",
      "26\t68\t0.7\t1\t0 68\n",
      "27\t82\t0.7\t1\t1 81\n",
      "28\t104\t0.7\t1\t1 103\n",
      "29\t84\t0.7\t1\t1 83\n",
      "30\t98\t0.7\t1\t0 98\n",
      "31\t82\t0.7\t1\t1 81\n",
      "32\t91\t0.7\t1\t0 91\n",
      "33\t94\t0.7\t1\t0 94\n",
      "34\t100\t0.7\t1\t4 96\n",
      "35\t105\t0.7\t1\t0 105\n",
      "36\t90\t0.7\t1\t0 90\n",
      "37\t87\t0.7\t1\t0 87\n",
      "38\t77\t0.7\t1\t1 76\n",
      "39\t67\t0.7\t1\t0 67\n",
      "40\t87\t0.7\t1\t1 86\n",
      "41\t99\t0.7\t1\t1 98\n",
      "42\t88\t0.7\t1\t3 85\n",
      "43\t70\t0.7\t1\t0 70\n",
      "44\t85\t0.7\t1\t2 83\n",
      "45\t107\t0.7\t1\t0 107\n",
      "46\t105\t0.7\t1\t0 105\n",
      "47\t96\t0.7\t1\t0 96\n",
      "48\t75\t0.7\t1\t0 75\n",
      "49\t115\t0.7\t1\t1 114\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: H3K27ac_r1_single.fastq.gz\n",
      "=============================================\n",
      "11440822 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t33623 (0.3%)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/danRer10/whole_embryo_24hpf/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> H3K4me1_r1_single.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t46\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "smallRNA\t2\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t2\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 46). Second best hit was smallRNA (count: 2)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/danRer10/whole_embryo_24hpf/H3K4me1_r1_single.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: H3K4me1_r1_single.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to H3K4me1_r1_single_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file H3K4me1_r1_single.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a AGATCGGAAGAGC H3K4me1_r1_single.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 183.968 s (9.005 µs/read; 6.66 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              20,429,006\n",
      "Reads with adapters:                   454,437 (2.2%)\n",
      "Reads written (passing filters):    20,429,006 (100.0%)\n",
      "\n",
      "Total basepairs processed:   715,015,210 bp\n",
      "Quality-trimmed:              45,294,695 bp (6.3%)\n",
      "Total written (filtered):    668,177,329 bp (93.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 454437 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.4%\n",
      "  C: 30.8%\n",
      "  G: 23.3%\n",
      "  T: 18.2%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t348209\t319203.2\t0\t348209\n",
      "4\t82484\t79800.8\t0\t82484\n",
      "5\t16125\t19950.2\t0\t16125\n",
      "6\t2479\t4987.6\t0\t2479\n",
      "7\t951\t1246.9\t0\t951\n",
      "8\t503\t311.7\t0\t503\n",
      "9\t661\t77.9\t0\t461 200\n",
      "10\t650\t19.5\t1\t382 268\n",
      "11\t457\t4.9\t1\t342 115\n",
      "12\t321\t1.2\t1\t267 54\n",
      "13\t221\t0.3\t1\t194 27\n",
      "14\t157\t0.3\t1\t134 23\n",
      "15\t82\t0.3\t1\t71 11\n",
      "16\t40\t0.3\t1\t34 6\n",
      "17\t45\t0.3\t1\t33 12\n",
      "18\t24\t0.3\t1\t13 11\n",
      "19\t22\t0.3\t1\t15 7\n",
      "20\t13\t0.3\t1\t5 8\n",
      "21\t43\t0.3\t1\t28 15\n",
      "22\t24\t0.3\t1\t12 12\n",
      "23\t15\t0.3\t1\t3 12\n",
      "24\t16\t0.3\t1\t8 8\n",
      "25\t18\t0.3\t1\t8 10\n",
      "26\t20\t0.3\t1\t3 17\n",
      "27\t15\t0.3\t1\t2 13\n",
      "28\t13\t0.3\t1\t5 8\n",
      "29\t29\t0.3\t1\t9 20\n",
      "30\t13\t0.3\t1\t4 9\n",
      "31\t14\t0.3\t1\t0 14\n",
      "32\t46\t0.3\t1\t34 12\n",
      "33\t17\t0.3\t1\t9 8\n",
      "34\t148\t0.3\t1\t118 30\n",
      "35\t562\t0.3\t1\t8 554\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: H3K4me1_r1_single.fastq.gz\n",
      "=============================================\n",
      "20429006 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t1326463 (6.5%)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.4\n",
      "Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 2 cores\n",
      "Proceeding with 'gzip' for compression. PLEASE NOTE: Using multi-cores for trimming with 'gzip' only has only very limited effect! (see here: https://github.com/FelixKrueger/TrimGalore/issues/16#issuecomment-458557103)\n",
      "To increase performance, please install 'pigz' and run again\n",
      "\n",
      "Output will be written into the directory: /home/limh/lmh_project_code/1EPI_construction/danRer10/whole_embryo_24hpf/\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> H3K4me3_r1_single.fastq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t12\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 12). Second best hit was Nextera (count: 1)\n",
      "\n",
      "Writing report to '/home/limh/lmh_project_code/1EPI_construction/danRer10/whole_embryo_24hpf/H3K4me3_r1_single.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: H3K4me3_r1_single.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.4\n",
      "Python version: 3.10.6\n",
      "Number of cores used for trimming: 2\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 3 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 2\n",
      "Writing final adapter and quality trimmed output to H3K4me3_r1_single_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file H3K4me3_r1_single.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 4.4 with Python 3.10.6\n",
      "Command line parameters: -j 2 -e 0.1 -q 20 -O 3 -a AGATCGGAAGAGC H3K4me3_r1_single.fastq.gz\n",
      "Processing single-end reads on 2 cores ...\n",
      "Finished in 250.898 s (10.361 µs/read; 5.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              24,215,389\n",
      "Reads with adapters:                   532,929 (2.2%)\n",
      "Reads written (passing filters):    24,215,389 (100.0%)\n",
      "\n",
      "Total basepairs processed:   847,538,615 bp\n",
      "Quality-trimmed:               3,588,510 bp (0.4%)\n",
      "Total written (filtered):    842,169,181 bp (99.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 532929 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.7%\n",
      "  C: 30.9%\n",
      "  G: 23.7%\n",
      "  T: 17.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t409460\t378365.5\t0\t409460\n",
      "4\t97526\t94591.4\t0\t97526\n",
      "5\t19306\t23647.8\t0\t19306\n",
      "6\t3069\t5912.0\t0\t3069\n",
      "7\t853\t1478.0\t0\t853\n",
      "8\t294\t369.5\t0\t294\n",
      "9\t470\t92.4\t0\t165 305\n",
      "10\t514\t23.1\t1\t135 379\n",
      "11\t307\t5.8\t1\t93 214\n",
      "12\t136\t1.4\t1\t71 65\n",
      "13\t75\t0.4\t1\t54 21\n",
      "14\t64\t0.4\t1\t51 13\n",
      "15\t63\t0.4\t1\t39 24\n",
      "16\t39\t0.4\t1\t23 16\n",
      "17\t45\t0.4\t1\t23 22\n",
      "18\t36\t0.4\t1\t15 21\n",
      "19\t44\t0.4\t1\t18 26\n",
      "20\t32\t0.4\t1\t5 27\n",
      "21\t19\t0.4\t1\t5 14\n",
      "22\t20\t0.4\t1\t2 18\n",
      "23\t28\t0.4\t1\t4 24\n",
      "24\t22\t0.4\t1\t4 18\n",
      "25\t13\t0.4\t1\t3 10\n",
      "26\t34\t0.4\t1\t5 29\n",
      "27\t24\t0.4\t1\t4 20\n",
      "28\t25\t0.4\t1\t2 23\n",
      "29\t23\t0.4\t1\t3 20\n",
      "30\t19\t0.4\t1\t1 18\n",
      "31\t4\t0.4\t1\t2 2\n",
      "32\t37\t0.4\t1\t18 19\n",
      "33\t27\t0.4\t1\t3 24\n",
      "34\t74\t0.4\t1\t51 23\n",
      "35\t227\t0.4\t1\t8 219\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: H3K4me3_r1_single.fastq.gz\n",
      "=============================================\n",
      "24215389 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t61000 (0.3%)\n",
      "\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1382344 sequences (100000042 bp)...\n",
      "[M::process] read 1381412 sequences (100000062 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 469006, 74, 16)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (127, 130, 373)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 865)\n",
      "[M::mem_pestat] mean and std.dev: (159.91, 119.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1111)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (114.14, 57.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (74, 272, 1089)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3119)\n",
      "[M::mem_pestat] mean and std.dev: (497.86, 655.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4134)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 346, 447)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1109)\n",
      "[M::mem_pestat] mean and std.dev: (250.86, 176.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1440)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1382344 reads in 544.309 CPU sec, 54.737 real sec\n",
      "[M::process] read 1381110 sequences (100000045 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 468275, 56, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (105, 186, 638)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1704)\n",
      "[M::mem_pestat] mean and std.dev: (375.38, 414.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2237)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (114.50, 57.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 400)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (168, 456, 1269)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3471)\n",
      "[M::mem_pestat] mean and std.dev: (651.08, 650.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4572)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (48, 157, 3394)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10086)\n",
      "[M::mem_pestat] mean and std.dev: (1430.54, 1899.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13432)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381412 reads in 551.053 CPU sec, 55.415 real sec\n",
      "[M::process] read 1381608 sequences (100000080 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 468140, 75, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (114.91, 58.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 404)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (143, 284, 1103)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3023)\n",
      "[M::mem_pestat] mean and std.dev: (542.49, 700.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3983)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1381110 reads in 538.269 CPU sec, 53.781 real sec\n",
      "[M::process] read 1381312 sequences (100000064 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 468265, 70, 9)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (178, 1024, 3961)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11527)\n",
      "[M::mem_pestat] mean and std.dev: (2140.75, 2451.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15310)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (114.54, 57.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 403)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (180, 452, 1287)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3501)\n",
      "[M::mem_pestat] mean and std.dev: (665.97, 629.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4608)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1381608 reads in 519.114 CPU sec, 51.969 real sec\n",
      "[M::process] read 1381356 sequences (100000121 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 468295, 76, 13)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (114.89, 58.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 407)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (155, 392, 1423)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3959)\n",
      "[M::mem_pestat] mean and std.dev: (674.19, 907.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5227)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (443, 1181, 2794)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7496)\n",
      "[M::mem_pestat] mean and std.dev: (1791.77, 1905.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9847)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381312 reads in 482.866 CPU sec, 48.329 real sec\n",
      "[M::process] read 1381926 sequences (100000060 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 468438, 83, 14)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (114.71, 58.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 407)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (109, 343, 928)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2566)\n",
      "[M::mem_pestat] mean and std.dev: (493.74, 566.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3385)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 199, 500)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1310)\n",
      "[M::mem_pestat] mean and std.dev: (248.67, 263.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1715)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381356 reads in 466.387 CPU sec, 46.675 real sec\n",
      "[M::process] read 1382250 sequences (100000106 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 468332, 63, 10)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 324, 3899)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11507)\n",
      "[M::mem_pestat] mean and std.dev: (1797.58, 2678.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 15311)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (114.31, 57.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 403)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (93, 297, 1388)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3978)\n",
      "[M::mem_pestat] mean and std.dev: (643.56, 798.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5273)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (152, 2078, 3045)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8831)\n",
      "[M::mem_pestat] mean and std.dev: (1591.11, 1871.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11724)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381926 reads in 467.530 CPU sec, 46.768 real sec\n",
      "[M::process] read 1381744 sequences (100000075 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 467679, 62, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (137, 148, 2508)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7250)\n",
      "[M::mem_pestat] mean and std.dev: (401.33, 747.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9621)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 93, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (113.77, 57.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 352, 848)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2300)\n",
      "[M::mem_pestat] mean and std.dev: (415.64, 424.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3026)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (147, 3012, 3691)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10779)\n",
      "[M::mem_pestat] mean and std.dev: (2329.92, 2681.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14323)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1382250 reads in 457.430 CPU sec, 45.772 real sec\n",
      "[M::process] read 1381366 sequences (100000059 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 467613, 70, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 297, 2391)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6935)\n",
      "[M::mem_pestat] mean and std.dev: (1066.31, 1472.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9207)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (114.62, 58.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 407)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (81, 311, 535)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1443)\n",
      "[M::mem_pestat] mean and std.dev: (297.25, 287.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1897)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (80, 295, 4562)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13526)\n",
      "[M::mem_pestat] mean and std.dev: (1761.92, 2547.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 18008)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381744 reads in 461.270 CPU sec, 46.132 real sec\n",
      "[M::process] read 1381678 sequences (100000046 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 468303, 66, 11)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (153, 205, 307)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 615)\n",
      "[M::mem_pestat] mean and std.dev: (159.78, 82.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 769)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (114.77, 58.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 407)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (163, 475, 987)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2635)\n",
      "[M::mem_pestat] mean and std.dev: (513.45, 549.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3459)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (323, 609, 1174)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2876)\n",
      "[M::mem_pestat] mean and std.dev: (777.70, 703.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3727)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381366 reads in 458.911 CPU sec, 45.855 real sec\n",
      "[M::process] read 1381524 sequences (100000134 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 467833, 65, 16)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 326)\n",
      "[M::mem_pestat] mean and std.dev: (115.07, 58.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 411)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (131, 379, 939)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2555)\n",
      "[M::mem_pestat] mean and std.dev: (532.38, 575.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3363)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (94, 253, 1183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3361)\n",
      "[M::mem_pestat] mean and std.dev: (593.60, 801.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4450)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381678 reads in 488.942 CPU sec, 48.928 real sec\n",
      "[M::process] read 1381438 sequences (100000021 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 467350, 65, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (81, 295, 1416)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4086)\n",
      "[M::mem_pestat] mean and std.dev: (546.83, 938.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5421)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 326)\n",
      "[M::mem_pestat] mean and std.dev: (115.13, 58.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 411)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (96, 395, 1483)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4257)\n",
      "[M::mem_pestat] mean and std.dev: (700.81, 998.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5644)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1381524 reads in 504.924 CPU sec, 50.439 real sec\n",
      "[M::process] read 1382094 sequences (100000141 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 466608, 59, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (153, 718, 4130)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12084)\n",
      "[M::mem_pestat] mean and std.dev: (2446.40, 3544.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16626)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 326)\n",
      "[M::mem_pestat] mean and std.dev: (115.14, 59.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 411)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 368, 1076)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3008)\n",
      "[M::mem_pestat] mean and std.dev: (590.52, 647.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3974)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (83, 136, 1742)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5060)\n",
      "[M::mem_pestat] mean and std.dev: (693.83, 1247.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6719)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381438 reads in 489.913 CPU sec, 48.881 real sec\n",
      "[M::process] read 1384396 sequences (100000039 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 466819, 53, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (150, 173, 564)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1392)\n",
      "[M::mem_pestat] mean and std.dev: (191.22, 157.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1806)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (114.69, 58.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 407)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (87, 283, 1360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3906)\n",
      "[M::mem_pestat] mean and std.dev: (595.25, 775.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5179)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (60, 116, 370)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 990)\n",
      "[M::mem_pestat] mean and std.dev: (151.00, 150.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1300)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1382094 reads in 476.031 CPU sec, 47.492 real sec\n",
      "[M::process] read 1382344 sequences (100000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 466917, 61, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 93, 151)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 311)\n",
      "[M::mem_pestat] mean and std.dev: (113.31, 56.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 391)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (128, 355, 732)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1940)\n",
      "[M::mem_pestat] mean and std.dev: (396.79, 351.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2544)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (79, 256, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1393)\n",
      "[M::mem_pestat] mean and std.dev: (297.27, 314.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1831)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1384396 reads in 480.578 CPU sec, 48.018 real sec\n",
      "[M::process] read 1382380 sequences (100000016 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 469478, 58, 15)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (194, 315, 497)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1103)\n",
      "[M::mem_pestat] mean and std.dev: (330.36, 268.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1406)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (114.34, 57.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (127, 264, 512)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1282)\n",
      "[M::mem_pestat] mean and std.dev: (298.75, 274.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1667)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 165, 295)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 705)\n",
      "[M::mem_pestat] mean and std.dev: (203.47, 153.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 910)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1382344 reads in 473.181 CPU sec, 47.304 real sec\n",
      "[M::process] read 1382008 sequences (100000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 466978, 42, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (106, 158, 1017)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2839)\n",
      "[M::mem_pestat] mean and std.dev: (343.91, 367.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3750)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (114.12, 57.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 278, 592)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1572)\n",
      "[M::mem_pestat] mean and std.dev: (260.17, 221.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2062)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (217, 788, 2330)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6556)\n",
      "[M::mem_pestat] mean and std.dev: (1067.25, 1075.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8669)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1382380 reads in 492.345 CPU sec, 49.208 real sec\n",
      "[M::process] read 1382066 sequences (100000145 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 467333, 58, 18)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (134, 166, 506)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1250)\n",
      "[M::mem_pestat] mean and std.dev: (230.10, 149.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1622)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (113.98, 57.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (74, 285, 638)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1766)\n",
      "[M::mem_pestat] mean and std.dev: (358.66, 371.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2330)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 196, 250)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 608)\n",
      "[M::mem_pestat] mean and std.dev: (149.20, 86.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 787)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1382008 reads in 472.212 CPU sec, 47.080 real sec\n",
      "[M::process] read 1381740 sequences (100000053 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 468526, 57, 16)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (126, 314, 1894)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5430)\n",
      "[M::mem_pestat] mean and std.dev: (762.90, 921.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7198)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (113.95, 57.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (168, 358, 1015)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2709)\n",
      "[M::mem_pestat] mean and std.dev: (607.43, 639.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3556)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 236, 1310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3794)\n",
      "[M::mem_pestat] mean and std.dev: (650.12, 850.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5036)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1382066 reads in 466.619 CPU sec, 46.568 real sec\n",
      "[M::process] read 1381952 sequences (100000043 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 468236, 55, 19)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (114.04, 57.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (144, 436, 2176)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6240)\n",
      "[M::mem_pestat] mean and std.dev: (914.90, 1158.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8272)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 177, 349)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 809)\n",
      "[M::mem_pestat] mean and std.dev: (157.47, 84.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1039)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381740 reads in 480.885 CPU sec, 47.990 real sec\n",
      "[M::process] read 1381572 sequences (100000078 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 468299, 48, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (78, 195, 663)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1833)\n",
      "[M::mem_pestat] mean and std.dev: (250.23, 253.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2418)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 93, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (113.86, 57.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (229, 423, 1616)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4390)\n",
      "[M::mem_pestat] mean and std.dev: (860.00, 1069.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5777)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1381952 reads in 472.067 CPU sec, 47.033 real sec\n",
      "[M::process] read 1381884 sequences (100000105 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 468806, 55, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (66, 212, 2756)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8136)\n",
      "[M::mem_pestat] mean and std.dev: (1203.38, 1780.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10826)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (114.05, 57.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (135, 404, 817)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2181)\n",
      "[M::mem_pestat] mean and std.dev: (410.45, 419.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2863)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1381572 reads in 465.455 CPU sec, 46.533 real sec\n",
      "[M::process] read 1381662 sequences (100000099 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (18, 468622, 58, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (267, 648, 1635)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4371)\n",
      "[M::mem_pestat] mean and std.dev: (782.93, 755.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5739)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 93, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (113.88, 57.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 399)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (179, 352, 1195)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3227)\n",
      "[M::mem_pestat] mean and std.dev: (455.22, 571.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4243)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (196, 241, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 688)\n",
      "[M::mem_pestat] mean and std.dev: (234.67, 166.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 899)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381884 reads in 464.151 CPU sec, 46.457 real sec\n",
      "[M::process] read 1381376 sequences (100000056 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 467967, 64, 14)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (101, 332, 832)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2294)\n",
      "[M::mem_pestat] mean and std.dev: (497.60, 620.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3025)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (114.24, 57.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 403)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (80, 252, 521)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1403)\n",
      "[M::mem_pestat] mean and std.dev: (243.13, 253.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1844)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (88, 251, 831)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2317)\n",
      "[M::mem_pestat] mean and std.dev: (217.00, 212.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3060)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381662 reads in 463.380 CPU sec, 46.393 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 468516, 52, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (114.86, 58.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 407)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 288, 696)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1868)\n",
      "[M::mem_pestat] mean and std.dev: (376.94, 371.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2454)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (63, 372, 1337)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3885)\n",
      "[M::mem_pestat] mean and std.dev: (684.00, 899.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5159)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::process] read 1381668 sequences (100000029 bp)...\n",
      "[M::mem_process_seqs] Processed 1381376 reads in 477.394 CPU sec, 47.647 real sec\n",
      "[M::process] read 1381966 sequences (100000117 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 468609, 52, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (114.27, 57.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 403)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 251, 737)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1991)\n",
      "[M::mem_pestat] mean and std.dev: (348.98, 347.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2618)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (352, 630, 4012)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11332)\n",
      "[M::mem_pestat] mean and std.dev: (2067.33, 2377.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14992)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381668 reads in 469.000 CPU sec, 46.963 real sec\n",
      "[M::process] read 1381868 sequences (100000076 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 467174, 53, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 182, 3073)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8993)\n",
      "[M::mem_pestat] mean and std.dev: (1863.79, 3015.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13925)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (114.21, 57.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 403)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (105, 375, 1390)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3960)\n",
      "[M::mem_pestat] mean and std.dev: (665.54, 824.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5245)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (109, 324, 549)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1429)\n",
      "[M::mem_pestat] mean and std.dev: (306.09, 245.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1869)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381966 reads in 490.765 CPU sec, 49.137 real sec\n",
      "[M::process] read 1381198 sequences (100000035 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 467398, 60, 14)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 234, 490)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1234)\n",
      "[M::mem_pestat] mean and std.dev: (209.50, 135.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1606)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 320)\n",
      "[M::mem_pestat] mean and std.dev: (114.42, 57.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 403)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (98, 399, 999)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2801)\n",
      "[M::mem_pestat] mean and std.dev: (521.58, 566.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3702)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (172, 311, 1671)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4669)\n",
      "[M::mem_pestat] mean and std.dev: (986.46, 1412.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6635)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381868 reads in 486.752 CPU sec, 48.584 real sec\n",
      "[M::process] read 1380628 sequences (100000061 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 468445, 74, 17)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (103, 160, 2419)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7051)\n",
      "[M::mem_pestat] mean and std.dev: (1240.77, 1854.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9367)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (114.64, 57.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 400)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (224, 569, 1406)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3770)\n",
      "[M::mem_pestat] mean and std.dev: (758.23, 779.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4952)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 333, 1310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3796)\n",
      "[M::mem_pestat] mean and std.dev: (830.65, 1110.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5271)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381198 reads in 489.841 CPU sec, 49.237 real sec\n",
      "[M::process] read 1380854 sequences (100000043 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 468403, 54, 11)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (115.02, 57.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 404)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (143, 445, 1186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3272)\n",
      "[M::mem_pestat] mean and std.dev: (586.90, 610.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4315)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (162, 200, 1298)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3570)\n",
      "[M::mem_pestat] mean and std.dev: (337.33, 376.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4706)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380628 reads in 488.726 CPU sec, 48.879 real sec\n",
      "[M::process] read 1380490 sequences (100000001 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 469671, 61, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (114.72, 57.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 400)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (84, 268, 2092)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6108)\n",
      "[M::mem_pestat] mean and std.dev: (912.09, 1596.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8116)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (75, 171, 2667)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7851)\n",
      "[M::mem_pestat] mean and std.dev: (1163.17, 1882.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10443)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380854 reads in 494.482 CPU sec, 49.442 real sec\n",
      "[M::process] read 1380626 sequences (100000038 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 468882, 76, 14)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (176, 240, 563)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1337)\n",
      "[M::mem_pestat] mean and std.dev: (233.91, 130.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1724)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 95, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (115.46, 58.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 408)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (121, 293, 1310)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3688)\n",
      "[M::mem_pestat] mean and std.dev: (617.31, 807.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4877)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (186, 369, 1164)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3120)\n",
      "[M::mem_pestat] mean and std.dev: (654.00, 754.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4098)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380490 reads in 465.165 CPU sec, 46.436 real sec\n",
      "[M::process] read 1380392 sequences (100000089 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 468452, 71, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (115.39, 58.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 408)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (91, 357, 961)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2701)\n",
      "[M::mem_pestat] mean and std.dev: (553.94, 591.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3571)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 267, 1876)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5406)\n",
      "[M::mem_pestat] mean and std.dev: (881.08, 1238.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7171)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380626 reads in 492.785 CPU sec, 49.344 real sec\n",
      "[M::process] read 1380884 sequences (100000059 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 468090, 60, 19)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 95, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (115.69, 58.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 408)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (131, 412, 1069)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2945)\n",
      "[M::mem_pestat] mean and std.dev: (596.40, 625.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3883)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (103, 205, 2241)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6517)\n",
      "[M::mem_pestat] mean and std.dev: (742.76, 1049.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8655)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380392 reads in 495.177 CPU sec, 49.465 real sec\n",
      "[M::process] read 1380902 sequences (100000052 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (18, 469032, 55, 17)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (231, 315, 834)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2040)\n",
      "[M::mem_pestat] mean and std.dev: (404.12, 312.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2643)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (115.46, 58.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 408)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (139, 416, 1164)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3214)\n",
      "[M::mem_pestat] mean and std.dev: (652.22, 756.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4239)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (43, 235, 680)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1954)\n",
      "[M::mem_pestat] mean and std.dev: (299.12, 326.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2591)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380884 reads in 489.678 CPU sec, 49.010 real sec\n",
      "[M::process] read 1380992 sequences (100000102 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 467778, 69, 15)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (76, 359, 1598)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4642)\n",
      "[M::mem_pestat] mean and std.dev: (893.21, 1040.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6164)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (115.27, 58.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 408)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (227, 494, 1616)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4394)\n",
      "[M::mem_pestat] mean and std.dev: (630.08, 611.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5783)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 166, 1821)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5323)\n",
      "[M::mem_pestat] mean and std.dev: (881.33, 1319.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7074)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380902 reads in 521.771 CPU sec, 52.162 real sec\n",
      "[M::process] read 1380604 sequences (100000109 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 467524, 68, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (88, 151, 403)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1033)\n",
      "[M::mem_pestat] mean and std.dev: (183.17, 139.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1348)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 326)\n",
      "[M::mem_pestat] mean and std.dev: (115.30, 58.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 411)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (124, 341, 1391)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3925)\n",
      "[M::mem_pestat] mean and std.dev: (714.47, 952.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5192)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1380992 reads in 516.549 CPU sec, 51.604 real sec\n",
      "[M::process] read 1380764 sequences (100000034 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 467638, 83, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 239, 1548)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4404)\n",
      "[M::mem_pestat] mean and std.dev: (461.14, 508.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5832)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 95, 157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 327)\n",
      "[M::mem_pestat] mean and std.dev: (115.74, 59.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 412)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (192, 433, 1279)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3453)\n",
      "[M::mem_pestat] mean and std.dev: (647.78, 776.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4540)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (142, 232, 397)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 907)\n",
      "[M::mem_pestat] mean and std.dev: (293.77, 212.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1162)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380604 reads in 471.114 CPU sec, 47.092 real sec\n",
      "[M::process] read 1381306 sequences (100000118 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 467682, 63, 16)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 600, 2296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6508)\n",
      "[M::mem_pestat] mean and std.dev: (728.38, 766.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8614)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 95, 157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 327)\n",
      "[M::mem_pestat] mean and std.dev: (115.87, 59.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 412)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (130, 390, 824)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2212)\n",
      "[M::mem_pestat] mean and std.dev: (409.96, 383.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2906)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (153, 222, 857)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2265)\n",
      "[M::mem_pestat] mean and std.dev: (364.00, 331.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2969)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380764 reads in 465.841 CPU sec, 46.702 real sec\n",
      "[M::process] read 1381226 sequences (100000134 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 466927, 63, 18)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (101, 169, 572)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1514)\n",
      "[M::mem_pestat] mean and std.dev: (287.40, 358.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1985)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 329)\n",
      "[M::mem_pestat] mean and std.dev: (115.72, 59.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 415)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (209, 524, 1230)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3272)\n",
      "[M::mem_pestat] mean and std.dev: (669.77, 664.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4293)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 186, 905)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2491)\n",
      "[M::mem_pestat] mean and std.dev: (463.88, 612.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3284)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381306 reads in 465.583 CPU sec, 46.563 real sec\n",
      "[M::process] read 1381472 sequences (100000097 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (18, 467416, 69, 15)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (106, 202, 1509)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4315)\n",
      "[M::mem_pestat] mean and std.dev: (873.28, 1136.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5718)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 327)\n",
      "[M::mem_pestat] mean and std.dev: (115.52, 59.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 412)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (145, 445, 1359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3787)\n",
      "[M::mem_pestat] mean and std.dev: (728.69, 892.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5001)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 332, 3488)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10056)\n",
      "[M::mem_pestat] mean and std.dev: (1896.20, 2802.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13340)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381226 reads in 461.018 CPU sec, 46.140 real sec\n",
      "[M::process] read 1381214 sequences (100000105 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 468014, 66, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (94, 182, 336)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 820)\n",
      "[M::mem_pestat] mean and std.dev: (189.10, 130.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1062)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (115.05, 58.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 407)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (190, 419, 979)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2557)\n",
      "[M::mem_pestat] mean and std.dev: (579.16, 522.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3346)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (130, 245, 832)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2236)\n",
      "[M::mem_pestat] mean and std.dev: (415.55, 430.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2938)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381472 reads in 456.780 CPU sec, 45.700 real sec\n",
      "[M::process] read 1381018 sequences (100000069 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 468620, 56, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 153)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 315)\n",
      "[M::mem_pestat] mean and std.dev: (114.20, 56.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 396)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 468, 1179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3297)\n",
      "[M::mem_pestat] mean and std.dev: (576.63, 637.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4356)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1381214 reads in 455.700 CPU sec, 45.597 real sec\n",
      "[M::process] read 1380846 sequences (100000130 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 469263, 59, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (171, 253, 1539)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4275)\n",
      "[M::mem_pestat] mean and std.dev: (637.42, 650.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5643)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (114.46, 57.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 400)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (138, 382, 908)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2448)\n",
      "[M::mem_pestat] mean and std.dev: (459.54, 497.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3218)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (63, 190, 752)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2130)\n",
      "[M::mem_pestat] mean and std.dev: (309.17, 316.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2819)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381018 reads in 448.802 CPU sec, 44.923 real sec\n",
      "[M::process] read 1381128 sequences (100000096 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (23, 468860, 64, 14)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (156, 218, 679)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1725)\n",
      "[M::mem_pestat] mean and std.dev: (354.71, 385.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2248)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (114.76, 57.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 400)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (131, 498, 1170)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3248)\n",
      "[M::mem_pestat] mean and std.dev: (645.21, 703.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4287)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (126, 223, 2287)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6609)\n",
      "[M::mem_pestat] mean and std.dev: (870.17, 1594.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 8770)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380846 reads in 460.395 CPU sec, 46.092 real sec\n",
      "[M::process] read 1380858 sequences (100000018 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 467869, 61, 17)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (65, 111, 555)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1535)\n",
      "[M::mem_pestat] mean and std.dev: (249.38, 331.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2025)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (114.60, 57.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 400)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 279, 1396)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4008)\n",
      "[M::mem_pestat] mean and std.dev: (557.80, 768.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5314)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (92, 179, 2968)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8720)\n",
      "[M::mem_pestat] mean and std.dev: (978.94, 1418.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11596)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381128 reads in 467.552 CPU sec, 46.804 real sec\n",
      "[M::process] read 1380748 sequences (100000066 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 468093, 73, 18)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (162, 242, 2387)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6837)\n",
      "[M::mem_pestat] mean and std.dev: (731.33, 1041.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9062)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (114.92, 57.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 404)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (248, 621, 1390)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3674)\n",
      "[M::mem_pestat] mean and std.dev: (753.24, 726.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4816)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (80, 210, 525)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1415)\n",
      "[M::mem_pestat] mean and std.dev: (317.41, 307.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1860)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380858 reads in 460.297 CPU sec, 46.042 real sec\n",
      "[M::process] read 1381064 sequences (100000038 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 468298, 80, 21)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (208, 626, 2754)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7846)\n",
      "[M::mem_pestat] mean and std.dev: (1155.18, 1177.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10392)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (114.91, 57.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 404)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 484, 1866)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5352)\n",
      "[M::mem_pestat] mean and std.dev: (891.85, 1240.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7095)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (76, 138, 643)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1777)\n",
      "[M::mem_pestat] mean and std.dev: (213.11, 273.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2344)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380748 reads in 478.585 CPU sec, 47.892 real sec\n",
      "[M::process] read 1380784 sequences (100000043 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 468255, 55, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 318)\n",
      "[M::mem_pestat] mean and std.dev: (114.60, 57.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 400)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (98, 489, 1149)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3251)\n",
      "[M::mem_pestat] mean and std.dev: (611.76, 714.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4302)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1381064 reads in 458.061 CPU sec, 45.760 real sec\n",
      "[M::process] read 1381134 sequences (100000131 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 468540, 55, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (144, 258, 812)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2148)\n",
      "[M::mem_pestat] mean and std.dev: (403.73, 482.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2816)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (115.05, 58.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 404)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (158, 458, 1719)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4841)\n",
      "[M::mem_pestat] mean and std.dev: (887.46, 1067.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6402)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1380784 reads in 457.954 CPU sec, 45.748 real sec\n",
      "[M::process] read 1381188 sequences (100000034 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 469232, 71, 9)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (224, 332, 2066)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5750)\n",
      "[M::mem_pestat] mean and std.dev: (903.71, 1160.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7592)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 155)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 323)\n",
      "[M::mem_pestat] mean and std.dev: (114.92, 58.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 407)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 377, 722)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1928)\n",
      "[M::mem_pestat] mean and std.dev: (474.92, 490.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2531)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1381134 reads in 452.995 CPU sec, 45.461 real sec\n",
      "[M::process] read 1380986 sequences (100000138 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 468412, 57, 11)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (76, 181, 305)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 763)\n",
      "[M::mem_pestat] mean and std.dev: (188.00, 162.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 992)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 94, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 326)\n",
      "[M::mem_pestat] mean and std.dev: (115.32, 58.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 411)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 352, 742)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2036)\n",
      "[M::mem_pestat] mean and std.dev: (394.49, 373.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2683)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (128, 232, 2689)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7811)\n",
      "[M::mem_pestat] mean and std.dev: (1336.82, 1903.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10372)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1381188 reads in 451.590 CPU sec, 45.141 real sec\n",
      "[M::process] read 1381078 sequences (100000131 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 468644, 69, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 94, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 324)\n",
      "[M::mem_pestat] mean and std.dev: (115.21, 58.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 408)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 248, 692)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1872)\n",
      "[M::mem_pestat] mean and std.dev: (333.82, 354.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2462)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (139, 317, 1239)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3439)\n",
      "[M::mem_pestat] mean and std.dev: (655.50, 781.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4539)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1380986 reads in 457.675 CPU sec, 45.735 real sec\n",
      "[M::mem_process_seqs] Processed 1102875 reads in 241.641 CPU sec, 24.140 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/Danio_rerio/danRer10/BWA/danRer10.fa H3K27ac_r1_single_trimmed.fq.gz\n",
      "[main] Real time: 249.260 sec; CPU: 2430.648 sec\n",
      "[bam_sort_core] merging from 0 files and 10 in-memory blocks...\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2873554 sequences (100000042 bp)...\n",
      "[M::process] read 2872508 sequences (100000057 bp)...\n",
      "[M::mem_process_seqs] Processed 2873554 reads in 473.698 CPU sec, 48.036 real sec\n",
      "[M::process] read 2871580 sequences (100000044 bp)...\n",
      "[M::mem_process_seqs] Processed 2872508 reads in 464.727 CPU sec, 46.485 real sec\n",
      "[M::process] read 2873804 sequences (100000025 bp)...\n",
      "[M::mem_process_seqs] Processed 2871580 reads in 470.237 CPU sec, 46.987 real sec\n",
      "[M::process] read 2872420 sequences (100000050 bp)...\n",
      "[M::mem_process_seqs] Processed 2873804 reads in 466.903 CPU sec, 46.617 real sec\n",
      "[M::process] read 2880078 sequences (100000046 bp)...\n",
      "[M::mem_process_seqs] Processed 2872420 reads in 474.771 CPU sec, 47.476 real sec\n",
      "[M::process] read 1858599 sequences (64547189 bp)...\n",
      "[M::mem_process_seqs] Processed 2880078 reads in 460.933 CPU sec, 46.082 real sec\n",
      "[M::mem_process_seqs] Processed 1858599 reads in 305.181 CPU sec, 30.459 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/Danio_rerio/danRer10/BWA/danRer10.fa H3K4me1_r1_single_trimmed.fq.gz\n",
      "[main] Real time: 318.201 sec; CPU: 3122.213 sec\n",
      "[bam_sort_core] merging from 0 files and 10 in-memory blocks...\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 2869282 sequences (100000015 bp)...\n",
      "[M::process] read 2868204 sequences (100000030 bp)...\n",
      "[M::mem_process_seqs] Processed 2869282 reads in 314.272 CPU sec, 31.729 real sec\n",
      "[M::process] read 2867970 sequences (100000017 bp)...\n",
      "[M::mem_process_seqs] Processed 2868204 reads in 306.562 CPU sec, 30.582 real sec\n",
      "[M::process] read 2868442 sequences (100000011 bp)...\n",
      "[M::mem_process_seqs] Processed 2867970 reads in 309.340 CPU sec, 30.917 real sec\n",
      "[M::process] read 2869860 sequences (100000055 bp)...\n",
      "[M::mem_process_seqs] Processed 2868442 reads in 315.424 CPU sec, 31.366 real sec\n",
      "[M::process] read 2869938 sequences (100000051 bp)...\n",
      "[M::mem_process_seqs] Processed 2869860 reads in 313.317 CPU sec, 31.225 real sec\n",
      "[M::process] read 2869814 sequences (100000008 bp)...\n",
      "[M::mem_process_seqs] Processed 2869938 reads in 312.807 CPU sec, 31.146 real sec\n",
      "[M::process] read 2870316 sequences (100000059 bp)...\n",
      "[M::mem_process_seqs] Processed 2869814 reads in 311.996 CPU sec, 31.080 real sec\n",
      "[M::process] read 1200563 sequences (41687834 bp)...\n",
      "[M::mem_process_seqs] Processed 2870316 reads in 306.049 CPU sec, 30.578 real sec\n",
      "[M::mem_process_seqs] Processed 1200563 reads in 135.438 CPU sec, 13.431 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 10 -M /home/limh/Reference_Genome/Danio_rerio/danRer10/BWA/danRer10.fa H3K4me3_r1_single_trimmed.fq.gz\n",
      "[main] Real time: 267.441 sec; CPU: 2630.559 sec\n",
      "[bam_sort_core] merging from 0 files and 10 in-memory blocks...\n",
      "[bam_rmdup_core] processing reference chr1...\n",
      "[bam_rmdup_core] 2926 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr10...\n",
      "[bam_rmdup_core] 2715 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr11...\n",
      "[bam_rmdup_core] 2405 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr12...\n",
      "[bam_rmdup_core] 2302 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr13...\n",
      "[bam_rmdup_core] 2619 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr14...\n",
      "[bam_rmdup_core] 2780 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr15...\n",
      "[bam_rmdup_core] 2546 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr16...\n",
      "[bam_rmdup_core] 3100 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr17...\n",
      "[bam_rmdup_core] 2764 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr18...\n",
      "[bam_rmdup_core] 2562 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr19...\n",
      "[bam_rmdup_core] 2628 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr2...\n",
      "[bam_rmdup_core] 2917 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr20...\n",
      "[bam_rmdup_core] 2931 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr21...\n",
      "[bam_rmdup_core] 2443 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr22...\n",
      "[bam_rmdup_core] 2154 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr23...\n",
      "[bam_rmdup_core] 2432 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr24...\n",
      "[bam_rmdup_core] 2101 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr25...\n",
      "[bam_rmdup_core] 1964 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr3...\n",
      "[bam_rmdup_core] 3462 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr4...\n",
      "[bam_rmdup_core] 2373 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr5...\n",
      "[bam_rmdup_core] 3655 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr6...\n",
      "[bam_rmdup_core] 2920 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr7...\n",
      "[bam_rmdup_core] 4092 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr8...\n",
      "[bam_rmdup_core] 2820 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chr9...\n",
      "[bam_rmdup_core] 2786 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149679v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149680v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149681v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149682v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149683v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149684v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149685v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149686v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149687v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149688v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149689v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149690v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149691v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149692v1...\n",
      "[bam_rmdup_core] 36 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149693v1...\n",
      "[bam_rmdup_core] 65 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149694v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149695v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149696v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149697v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149698v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149699v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149700v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149701v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149702v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149703v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149704v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149705v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149706v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149707v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149708v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149709v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149710v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149711v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149712v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149713v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149714v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149715v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149716v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149717v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149718v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149719v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149720v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149721v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149722v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149723v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149724v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149725v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149726v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149727v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149728v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149729v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149730v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149731v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149732v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149733v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149734v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149735v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149736v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149737v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149738v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149739v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149740v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149741v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149742v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149743v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149744v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149745v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149746v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149747v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149748v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149749v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149750v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149751v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149752v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149753v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149754v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149755v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149756v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149757v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149758v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149759v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149760v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149761v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149762v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149763v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149764v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149765v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149766v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149767v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149768v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149769v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149770v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149771v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149772v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149773v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149774v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149775v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149776v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149777v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149778v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149779v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149780v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149781v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149782v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149783v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149784v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149785v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149786v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149787v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149788v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149789v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149790v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149791v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149792v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149793v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149794v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149795v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149796v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149797v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149798v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149799v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149800v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149801v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149802v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149803v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149804v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149805v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149806v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149807v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149808v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149809v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149810v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149811v1...\n",
      "[bam_rmdup_core] 8 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149812v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149813v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149814v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149815v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149816v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149817v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149818v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149819v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149820v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149821v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149822v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149823v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149824v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149825v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149826v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149827v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149828v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149829v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149830v1...\n",
      "[bam_rmdup_core] 11 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149831v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149832v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149833v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149834v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149835v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149836v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149837v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149838v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149839v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149840v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149841v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149842v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149843v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149844v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149845v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149846v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149847v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149848v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149849v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149850v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149851v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149852v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149853v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149854v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149855v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149856v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149857v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149858v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149859v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149860v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149861v1...\n",
      "[bam_rmdup_core] 23 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149862v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149863v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149864v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149865v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149866v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149867v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149868v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149869v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149870v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149871v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149872v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149873v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149874v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149875v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149876v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149877v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149878v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149879v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149880v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149881v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149882v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149883v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149884v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149885v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149886v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149887v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149888v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149889v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149890v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149891v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149892v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149893v1...\n",
      "[bam_rmdup_core] 8 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149894v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149895v1...\n",
      "[bam_rmdup_core] 7 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149896v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149897v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149898v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149899v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149900v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149901v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149902v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149903v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149904v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149905v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149906v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149907v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149908v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149909v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149910v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149911v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149912v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149913v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149914v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149915v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149916v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149917v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149918v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149919v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149920v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149921v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149922v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149923v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149924v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149925v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149926v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149927v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149928v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149929v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149930v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149931v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149932v1...\n",
      "[bam_rmdup_core] 17 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149933v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149934v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149935v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149936v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149937v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149938v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149939v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149940v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149941v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149942v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149943v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149944v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149945v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149946v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149947v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149948v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149949v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149950v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149951v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149952v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149953v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149954v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149955v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149956v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149957v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149958v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149959v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149960v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149961v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149962v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149963v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149964v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149965v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149966v1...\n",
      "[bam_rmdup_core] 19 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149967v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149968v1...\n",
      "[bam_rmdup_core] 8 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149969v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149970v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149971v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149972v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149973v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149974v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149975v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149976v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149977v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149978v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149979v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149980v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149981v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149982v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149983v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149984v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149985v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149986v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149987v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149988v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149989v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149990v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149991v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149992v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149993v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149994v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149995v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149996v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149997v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN149998v1...\n",
      "[bam_rmdup_core] 8 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN149999v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150000v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150001v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150002v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150003v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150004v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150005v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150006v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150007v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150008v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150009v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150010v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150011v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150012v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150013v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150014v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150015v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150016v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150017v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150019v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150020v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150021v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150022v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150023v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150024v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150025v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150026v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150027v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150028v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150029v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150030v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150031v1...\n",
      "[bam_rmdup_core] 19 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150032v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150033v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150034v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150035v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150036v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150037v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150038v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150039v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150040v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150041v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150042v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150043v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150044v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150045v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150046v1...\n",
      "[bam_rmdup_core] 18 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150047v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150048v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150049v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150050v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150051v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150052v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150053v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150054v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150055v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150056v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150057v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150058v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150059v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150060v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150061v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150062v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150063v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150064v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150065v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150066v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150067v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150068v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150069v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150070v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150071v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150072v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150073v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150074v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150075v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150076v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150077v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150078v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150079v1...\n",
      "[bam_rmdup_core] 7 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150080v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150081v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150082v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150083v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150084v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150085v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150086v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150087v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150088v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150089v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150090v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150091v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150092v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150093v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150094v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150095v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150096v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150097v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150098v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150099v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150100v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150101v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150102v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150103v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150104v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150105v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150106v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150107v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150108v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150109v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150110v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150111v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150112v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150113v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150114v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150115v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150116v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150117v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150118v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150119v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150120v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150121v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150122v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150123v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150124v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150125v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150126v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150127v1...\n",
      "[bam_rmdup_core] 34 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150128v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150129v1...\n",
      "[bam_rmdup_core] 8 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150130v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150131v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150132v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150133v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150134v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150135v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150136v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150137v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150138v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150139v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150140v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150141v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150142v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150143v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150144v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150145v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150146v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150147v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150148v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150149v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150150v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150151v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150152v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150153v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150154v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150155v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150156v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150157v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150158v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150159v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150160v1...\n",
      "[bam_rmdup_core] 46 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150161v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150162v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150163v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150164v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150165v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150166v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150167v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150168v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150169v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150170v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150171v1...\n",
      "[bam_rmdup_core] 10 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150172v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150173v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150174v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150175v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150176v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150177v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150178v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150179v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150180v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150181v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150182v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150183v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150184v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150185v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150186v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150187v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150188v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150189v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150190v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150191v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150192v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150193v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150194v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150195v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150196v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150197v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150198v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150199v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150200v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150201v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150202v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150203v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150204v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150205v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150206v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150207v1...\n",
      "[bam_rmdup_core] 10 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150208v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150209v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150210v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150211v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150212v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150213v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150214v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150215v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150216v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150217v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150218v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150219v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150220v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150221v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150222v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150223v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150224v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150225v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150226v1...\n",
      "[bam_rmdup_core] 7 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150227v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150228v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150229v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150230v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150231v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150232v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150233v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150234v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150235v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150236v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150237v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150238v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150239v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150240v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150241v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150242v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150243v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150244v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150245v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150246v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150247v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150248v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150249v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150250v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150251v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150252v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150253v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150254v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150255v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150256v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150257v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150258v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150259v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150260v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150261v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150262v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150263v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150264v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150265v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150266v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150267v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150268v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150269v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150270v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150271v1...\n",
      "[bam_rmdup_core] 408 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150272v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150273v1...\n",
      "[bam_rmdup_core] 11 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150274v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150275v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150276v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150277v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150278v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150279v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150280v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150281v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150282v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150283v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150284v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150285v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150286v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150287v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150288v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150289v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150290v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150291v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150293v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150294v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150295v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150296v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150297v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150298v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150299v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150300v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150301v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150302v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150303v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150304v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150305v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150306v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150307v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150308v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150309v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150310v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150311v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150312v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150313v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150314v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150315v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150316v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150317v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150318v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150319v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150320v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150321v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150322v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150323v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150324v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150325v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150326v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150327v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150328v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150329v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150330v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150331v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150332v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150333v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150334v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150335v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150336v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150337v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150338v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150339v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150340v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150341v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150342v1...\n",
      "[bam_rmdup_core] 8 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150343v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150344v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150345v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150346v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150347v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150348v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150349v1...\n",
      "[bam_rmdup_core] 10 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150350v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150351v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150352v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150353v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150354v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150355v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150356v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150357v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150358v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150359v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150360v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150361v1...\n",
      "[bam_rmdup_core] 11 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150362v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150363v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150364v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150365v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150366v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150367v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150368v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150369v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150370v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150371v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150372v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150373v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150374v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150375v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150376v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150377v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150378v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150379v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150380v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150381v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150382v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150383v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150384v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150385v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150386v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150387v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150388v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150389v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150390v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150391v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150392v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150393v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150394v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150395v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150396v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150397v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150398v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150399v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150400v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150401v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150402v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150403v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150404v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150405v1...\n",
      "[bam_rmdup_core] 7 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150406v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150407v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150408v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150409v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150410v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150411v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150412v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150413v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150414v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150415v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150416v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150417v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150418v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150419v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150420v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150421v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150422v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150423v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150424v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150425v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150426v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150427v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150428v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150429v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150430v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150431v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150432v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150433v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150434v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150435v1...\n",
      "[bam_rmdup_core] 14 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150436v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150437v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150438v1...\n",
      "[bam_rmdup_core] 13 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150439v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150440v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150441v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150442v1...\n",
      "[bam_rmdup_core] 16 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150443v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150444v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150445v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150446v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150447v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150448v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150449v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150450v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150451v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150452v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150453v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150454v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150455v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150456v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150457v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150458v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150459v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150460v1...\n",
      "[bam_rmdup_core] 13 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150461v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150462v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150463v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150464v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150465v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150466v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150467v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150468v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150469v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150470v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150471v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150472v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150473v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150474v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150475v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150476v1...\n",
      "[bam_rmdup_core] 11 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150477v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150478v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150479v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150480v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150481v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150482v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150483v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150484v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150485v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150486v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150487v1...\n",
      "[bam_rmdup_core] 14 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150488v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150489v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150490v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150491v1...\n",
      "[bam_rmdup_core] 8 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150492v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150493v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150494v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150495v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150496v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150497v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150498v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150499v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150500v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150501v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150502v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150503v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150504v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150505v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150506v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150507v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150508v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150509v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150510v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150511v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150512v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150513v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150514v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150515v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150516v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150517v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150518v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150519v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150520v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150521v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150522v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150523v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150524v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150525v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150526v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150527v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150528v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150529v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150530v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150531v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150532v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150533v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150534v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150535v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150536v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150537v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150538v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150539v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150540v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150541v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150542v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150543v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150544v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150545v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150546v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150547v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150548v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150549v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150550v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150551v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150552v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150553v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150554v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150555v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150556v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150557v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150558v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150559v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150560v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150561v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150562v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150563v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150564v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150565v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150566v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150567v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150568v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150569v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150570v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150571v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150572v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150573v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150574v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150575v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150576v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150577v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150578v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150579v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150580v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150581v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150582v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150583v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150584v1...\n",
      "[bam_rmdup_core] 6 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150585v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150586v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150587v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150588v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150589v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150590v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150591v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150592v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150593v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150594v1...\n",
      "[bam_rmdup_core] 12 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150595v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150596v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150597v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150598v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150599v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150600v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150601v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150602v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150603v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150604v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150605v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150606v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150607v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150608v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150609v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150610v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150611v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150612v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150613v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150614v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150615v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150616v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150617v1...\n",
      "[bam_rmdup_core] 14 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150618v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150619v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150620v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150621v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150622v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150623v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150624v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150625v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150626v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150627v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150628v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150629v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150630v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150631v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150632v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150633v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150634v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150635v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150636v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150637v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150638v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150639v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150640v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150641v1...\n",
      "[bam_rmdup_core] 9 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150642v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150643v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150644v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150645v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150646v1...\n",
      "[bam_rmdup_core] 7 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150647v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150648v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150649v1...\n",
      "[bam_rmdup_core] 3 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150650v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150651v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150652v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150653v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150654v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150655v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150656v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150657v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150658v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150659v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150660v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150661v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150662v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150663v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150664v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150665v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150666v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150667v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150668v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150669v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150670v1...\n",
      "[bam_rmdup_core] 10 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150671v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150672v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150673v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150674v1...\n",
      "[bam_rmdup_core] 5 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150675v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150676v1...\n",
      "[bam_rmdup_core] 4 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150677v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150678v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150679v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150680v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150681v1...\n",
      "[bam_rmdup_core] 8 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150682v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150683v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150684v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150685v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150686v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150687v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150688v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150689v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150690v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150691v1...\n",
      "[bam_rmdup_core] 7 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150692v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150693v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150694v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150695v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150696v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150697v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150698v1...\n",
      "[bam_rmdup_core] 7 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150699v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150700v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150701v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150702v1...\n",
      "[bam_rmdup_core] 16 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150703v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150704v1...\n",
      "[bam_rmdup_core] 1 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150705v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150706v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150707v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150708v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150709v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150710v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150711v1...\n",
      "[bam_rmdup_core] processing reference chrUn_KN150712v1...\n",
      "[bam_rmdup_core] 2 unmatched pairs\n",
      "[bam_rmdup_core] processing reference chrUn_KN150713v1...\n",
      "[bam_rmdup_core] processing reference chrM...\n",
      "[bam_rmdup_core] 5482351 / 31018945 = 0.1767 in library '\t'\n",
      "[bam_rmdupse_core] 680292 / 8657850 = 0.0786 in library '\t'\n",
      "[bam_rmdupse_core] 5474117 / 10466181 = 0.5230 in library '\t'\n",
      "[bam_rmdupse_core] 8014786 / 16562803 = 0.4839 in library '\t'\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args='bash align.sh 10 /home/limh/Reference_Genome/Danio_rerio/danRer10/BWA/danRer10.fa', returncode=0)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Align from original reads\n",
    "cmd = f\"bash align.sh 10 /home/limh/Reference_Genome/Danio_rerio/danRer10/BWA/danRer10.fa\"\n",
    "subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "5791ce2a-61a2-4116-bd12-fae1527adba1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running SICER with given arguments \n",
      "\n",
      "Preprocess the ATAC_r2_sort_rmdup.bed file to remove redundancy with threshold of 1 \n",
      "\n",
      "---------------------------------------------------------------------------------------------------------\n",
      "chrom    Total plus reads        Retained plus reads       Total minus reads      Retained minus reads   \n",
      "---------------------------------------------------------------------------------------------------------\n",
      "chr4          1149917                  1092501                  1153719                  1092574         \n",
      "chr7          1903730                  1793047                  1909450                  1793617         \n",
      "chr5          1843955                  1738932                  1849253                  1739821         \n",
      "chr3          1562501                  1468418                  1567575                  1469242         \n",
      "chr6          1526584                  1445678                  1531156                  1446016         \n",
      "chr2          1486128                  1406355                  1490445                  1406967         \n",
      "chr1          1490803                  1411902                  1495098                  1412143         \n",
      "chr9          1425391                  1356935                  1429403                  1356627         \n",
      "chr16         1418567                  1338617                  1423042                  1339157         \n",
      "chr20         1393929                  1318224                  1398075                  1318774         \n",
      "chr8          1364395                  1289813                  1368772                  1289862         \n",
      "chr17         1338877                  1268973                  1342989                  1269400         \n",
      "chr14         1335685                  1265958                  1339510                  1266421         \n",
      "chr13         1351471                  1278252                  1355406                  1278704         \n",
      "chr18         1336314                  1268693                  1339992                  1268989         \n",
      "chr12         1195495                  1135136                  1199106                  1135945         \n",
      "chr19         1299540                  1225473                  1303486                  1226540         \n",
      "chr15         1239472                  1176732                  1243064                  1177130         \n",
      "chr23         1203948                  1137878                  1207553                  1138223         \n",
      "chr21         1153232                  1090365                  1156625                  1090758         \n",
      "chr10         1197324                  1124715                  1201086                  1124808         \n",
      "chr11         1178906                  1113969                  1182336                  1113998         \n",
      "chr24         1050930                  995419                   1054640                  995155          \n",
      "chr22         981635                   924185                   984522                   924520          \n",
      "chr25         939964                   884129                   942831                   884548          \n",
      "\n",
      "\n",
      "Partition the genome in windows and generate summary files... \n",
      "\n",
      "Total count of chr4 tags: 2185074\n",
      "Total count of chr7 tags: 3586664\n",
      "Total count of chr5 tags: 3478753\n",
      "Total count of chr3 tags: 2937660\n",
      "Total count of chr6 tags: 2891694\n",
      "Total count of chr2 tags: 2813322\n",
      "Total count of chr1 tags: 2824045\n",
      "Total count of chr9 tags: 2713554\n",
      "Total count of chr16 tags: 2677772\n",
      "Total count of chr20 tags: 2636998\n",
      "Total count of chr8 tags: 2579675\n",
      "Total count of chr17 tags: 2538373\n",
      "Total count of chr14 tags: 2532379\n",
      "Total count of chr13 tags: 2556950\n",
      "Total count of chr18 tags: 2537680\n",
      "Total count of chr12 tags: 2271080\n",
      "Total count of chr19 tags: 2452012\n",
      "Total count of chr15 tags: 2353862\n",
      "Total count of chr23 tags: 2276066\n",
      "Total count of chr21 tags: 2181123\n",
      "Total count of chr10 tags: 2249521\n",
      "Total count of chr11 tags: 2227966\n",
      "Total count of chr24 tags: 1990574\n",
      "Total count of chr22 tags: 1848705\n",
      "Total count of chr25 tags: 1768676\n",
      "\n",
      "\n",
      "Normalizing graphs by total island filitered reads per million and generating summary WIG file...\n",
      "\n",
      "Finding candidate islands exhibiting clustering...\n",
      "\n",
      "Species:  danRer10\n",
      "Window_size:  200\n",
      "Gap size:  600\n",
      "E value is: 1000\n",
      "Total read count: 63110017\n",
      "Genome Length:  1340430591\n",
      "Effective genome Length:  991918637\n",
      "Window average: 12.724837430390977\n",
      "Window pvalue: 0.2\n",
      "Minimum num of tags in a qualified window:  17\n",
      "\n",
      "Determining the score threshold from random background...\n",
      "The score threshold is: 31.079\n",
      "Generating the enriched probscore summary graph and filtering the summary graph to eliminate ineligible windows... \n",
      "Total number of islands:  84899\n",
      "\n",
      "\n",
      "End of SICER\n",
      "Removing temporary directory and all files in it.\n",
      "\n",
      "Program Finished Running\n",
      "Running SICER with given arguments \n",
      "\n",
      "Preprocess the ATAC_r1_sort_rmdup.bed file to remove redundancy with threshold of 1 \n",
      "\n",
      "---------------------------------------------------------------------------------------------------------\n",
      "chrom    Total plus reads        Retained plus reads       Total minus reads      Retained minus reads   \n",
      "---------------------------------------------------------------------------------------------------------\n",
      "chr4          869289                   810994                   871764                   810800          \n",
      "chr7          1411656                  1291632                  1415691                  1292483         \n",
      "chr5          1352752                  1245472                  1356451                  1245445         \n",
      "chr3          1135625                  1039457                  1139194                  1039500         \n",
      "chr6          1113243                  1032049                  1116202                  1031934         \n",
      "chr2          1093883                  1011466                  1096827                  1011130         \n",
      "chr1          1085820                  1005456                  1088970                  1005967         \n",
      "chr9          1050845                  979555                   1053657                  980063          \n",
      "chr16         1037511                  954400                   1040616                  954762          \n",
      "chr20         1023749                  945195                   1026598                  945552          \n",
      "chr8          1002771                  925632                   1005600                  925254          \n",
      "chr17         981057                   907765                   984026                   908009          \n",
      "chr14         983296                   910512                   986035                   910659          \n",
      "chr13         981309                   905679                   984030                   906395          \n",
      "chr18         974452                   905390                   976930                   905667          \n",
      "chr12         867863                   807298                   870259                   807500          \n",
      "chr19         946578                   871318                   949229                   872167          \n",
      "chr15         905673                   840742                   908307                   840561          \n",
      "chr23         879882                   811378                   882303                   811405          \n",
      "chr21         839772                   775546                   842002                   775664          \n",
      "chr10         880543                   802044                   883184                   802215          \n",
      "chr11         861808                   795579                   864160                   796159          \n",
      "chr24         769591                   712823                   771763                   712426          \n",
      "chr22         719673                   660335                   721719                   660518          \n",
      "chr25         683563                   627741                   685457                   627233          \n",
      "\n",
      "\n",
      "Partition the genome in windows and generate summary files... \n",
      "\n",
      "Total count of chr4 tags: 1621789\n",
      "Total count of chr7 tags: 2584114\n",
      "Total count of chr5 tags: 2490917\n",
      "Total count of chr3 tags: 2078957\n",
      "Total count of chr6 tags: 2063983\n",
      "Total count of chr2 tags: 2022596\n",
      "Total count of chr1 tags: 2011423\n",
      "Total count of chr9 tags: 1959613\n",
      "Total count of chr16 tags: 1909159\n",
      "Total count of chr20 tags: 1890747\n",
      "Total count of chr8 tags: 1850885\n",
      "Total count of chr17 tags: 1815774\n",
      "Total count of chr14 tags: 1821171\n",
      "Total count of chr13 tags: 1812071\n",
      "Total count of chr18 tags: 1811056\n",
      "Total count of chr12 tags: 1614796\n",
      "Total count of chr19 tags: 1743485\n",
      "Total count of chr15 tags: 1681303\n",
      "Total count of chr23 tags: 1622743\n",
      "Total count of chr21 tags: 1551210\n",
      "Total count of chr10 tags: 1604255\n",
      "Total count of chr11 tags: 1591735\n",
      "Total count of chr24 tags: 1425249\n",
      "Total count of chr22 tags: 1320853\n",
      "Total count of chr25 tags: 1254972\n",
      "\n",
      "\n",
      "Normalizing graphs by total island filitered reads per million and generating summary WIG file...\n",
      "\n",
      "Finding candidate islands exhibiting clustering...\n",
      "\n",
      "Species:  danRer10\n",
      "Window_size:  200\n",
      "Gap size:  600\n",
      "E value is: 1000\n",
      "Total read count: 45154751\n",
      "Genome Length:  1340430591\n",
      "Effective genome Length:  991918637\n",
      "Window average: 9.104527189158963\n",
      "Window pvalue: 0.2\n",
      "Minimum num of tags in a qualified window:  13\n",
      "\n",
      "Determining the score threshold from random background...\n",
      "The score threshold is: 27.88\n",
      "Generating the enriched probscore summary graph and filtering the summary graph to eliminate ineligible windows... \n",
      "Total number of islands:  78980\n",
      "\n",
      "\n",
      "End of SICER\n",
      "Removing temporary directory and all files in it.\n",
      "\n",
      "Program Finished Running\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args='bedtools intersect -a enhancer_primary.bed -b ATAC_r2_sort_rmdup.broadPeak -u -f 0.5 -F 0.5 -e > enhancer.bed', returncode=0)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Histone modification ChIP-seq call peak\n",
    "files = ['H3K27ac_r1_sort_rmdup.bam', 'H3K4me1_r1_sort_rmdup.bam']\n",
    "for file in files:\n",
    "    inputFile = 'input_r1_sort_rmdup.bam' if 'r1' in file else 'input_r2_sort_rmdup.bam'\n",
    "    cmd = f\"sicer -t {file} -s danRer10 -fdr 0.05 -cpu 20 -o .\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    \n",
    "# Get standard-format broadPeak file from sicer result\n",
    "files = [x for x in os.listdir() if (x.endswith('-W200-G600.scoreisland') and 'H3K' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df.insert(3, 'peakName', [f'peak{x}' for x in range(1, df.shape[0]+1)])\n",
    "    df['strand'] = '.'\n",
    "    df['signalValue'] = df[3]\n",
    "    df['pValue'] = df[3]\n",
    "    df['qValue'] = df[3]\n",
    "    df.to_csv(file.replace('-W200-G600.scoreisland', '.broadPeak'), sep='\\t', header=None, index=False)\n",
    "    \n",
    "\n",
    "# H3K4me1 H3K27ac intersection calculation\n",
    "H3K4me1_file = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'H3K4me1' in x)][0]\n",
    "H3K27ac_file = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'H3K27ac' in x)][0]\n",
    "cmd = f\"bedtools intersect -a {H3K4me1_file} -b {H3K27ac_file} > H3K4me1_H3K27ac_intersect.bed\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "# Retain H3K4me1 H3K27ac intersection part which show lower H3K4me3 histone modification signal. Get primary enhancer.\n",
    "H3K4me1_bam = [x for x in os.listdir() if (x.endswith('rmdup.bam') and 'H3K4me1' in x)]\n",
    "for file in H3K4me1_bam:\n",
    "    if f\"{file}.bai\" in os.listdir():\n",
    "        continue\n",
    "    cmd = f\"samtools index -@ 20 {file}\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "cmd = f'''bamCompare -b1 {H3K4me1_bam[0]} -b2 {H3K4me1_bam[0]} --outFileName H3K4me1_all.bw \\\n",
    "          --normalizeUsing CPM --binSize 10 --operation add --numberOfProcessors 20 --scaleFactorsMethod None'''\n",
    "subprocess.run(cmd, shell=True)\n",
    "H3K4me3_bam = [x for x in os.listdir() if (x.endswith('rmdup.bam') and 'H3K4me3' in x)]\n",
    "for file in H3K4me3_bam:\n",
    "    if f\"{file}.bai\" in os.listdir():\n",
    "        continue\n",
    "    cmd = f\"samtools index -@ 20 {file}\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "cmd = f'''bamCompare -b1 {H3K4me3_bam[0]} -b2 {H3K4me3_bam[0]} --outFileName H3K4me3_all.bw \\\n",
    "            --normalizeUsing CPM --binSize 10 --operation add --numberOfProcessors 20 --scaleFactorsMethod None'''\n",
    "subprocess.run(cmd, shell=True)\n",
    "me1BW = pyBigWig.open('H3K4me1_all.bw')\n",
    "me3BW = pyBigWig.open('H3K4me3_all.bw')\n",
    "def ifRetain(row, me1BW, me3BW, me1ScoreAll=1, me3ScoreAll=1):\n",
    "    chr = row[0]\n",
    "    start = row[1]\n",
    "    end = row[2]\n",
    "    me1Score = np.array(me1BW.stats(chr, start, end, type='sum'))[0] / me1ScoreAll * pow(10, 6)\n",
    "    me3Score = np.array(me3BW.stats(chr, start, end, type='sum'))[0] / me3ScoreAll * pow(10, 6)\n",
    "    if me1Score > 1.5 * me3Score:\n",
    "        return True, me1Score, me3Score\n",
    "    else:\n",
    "        return False, me1Score, me3Score\n",
    "df = pd.read_csv('H3K4me1_H3K27ac_intersect.bed', sep='\\t', header=None)\n",
    "df['result'] = df.apply(ifRetain, axis=1, args=(me1BW, me3BW))\n",
    "df[['retain', 'me1Score', 'me3Score']] = df['result'].apply(pd.Series)\n",
    "me1BW.close()\n",
    "me3BW.close()\n",
    "df[df['retain']][[0,1,2]].to_csv('enhancer_primary.bed', sep='\\t', header=None, index=False)\n",
    "\n",
    "# ATAC-seq data process to get broadPeak.\n",
    "files = [x for x in os.listdir() if ((x.endswith('rmdup.bam')) and ('ATAC' in x))]\n",
    "for file in files:\n",
    "    cmd = f\"sicer -t {file} -s danRer10 -fdr 0.05 -cpu 20 -o .\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "files = [x for x in os.listdir() if (x.endswith('-W200-G600.scoreisland') and 'ATAC' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df.insert(3, 'peakName', [f'peak{x}' for x in range(1, df.shape[0]+1)])\n",
    "    df['strand'] = '.'\n",
    "    df['signalValue'] = df[3]\n",
    "    df['pValue'] = df[3]\n",
    "    df['qValue'] = df[3]\n",
    "    df.to_csv(file.replace('-W200-G600.scoreisland', '.broadPeak'), sep='\\t', header=None, index=False)\n",
    "    \n",
    "# Use idr to combine different replicates\n",
    "files = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'ATAC' in x and 'r1' in x)]\n",
    "for file in files:\n",
    "    file_rep2 = file.replace('r1', 'r2')\n",
    "    outfile = file.replace('r1',  'idr')\n",
    "    cmd = f'''idr --samples {file} {file_rep2} --input-file-type broadPeak \\\n",
    "        --rank q.value --output-file {outfile} --output-file-type broadPeak \\\n",
    "            --plot --log-output-file {outfile}.log'''\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    \n",
    "# Filter idr result with threshold as 0.1\n",
    "files = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'ATAC' in x and 'idr' in x)]\n",
    "for file in files:\n",
    "    df = pd.read_csv(file, sep='\\t', header=None)\n",
    "    df = df[df[4]> int(-125 *  np.log2(0.1))]\n",
    "    df.to_csv(file.replace('.broadPeak', '_filter.broadPeak'), sep='\\t', header=None, index=False)\n",
    "\n",
    "# Filter primary enhancer with its DNase broadPeak. Obtain final enhancer.\n",
    "DNase_file = [x for x in os.listdir() if (x.endswith('.broadPeak') and 'ATAC' in x and 'idr' in x)][0]\n",
    "cmd = f\"bedtools intersect -a enhancer_primary.bed -b {DNase_file} -u -f 0.5 -F 0.5 -e > enhancer.bed\"\n",
    "subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e5755a5c-bd67-426f-8a6f-2f9a8e3ebbe5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# process enhancer file\n",
    "df = pd.read_csv('enhancer.bed', sep='\\t', header=None)\n",
    "df[3] = [f\"enhancer{i}\" for i in range(df.shape[0])]\n",
    "df[4] = 1\n",
    "df[5] = '.'\n",
    "df.to_csv('enhancer.bg', sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7d879c7-4102-4bdf-99e0-159377d55bdb",
   "metadata": {},
   "source": [
    "## enhancer promoter interaction construction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "be34f224-8ec4-4efe-8361-fc6ede5a0231",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args='bedtools sort -i TAD.bed > TAD_sort.bed', returncode=0)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# # Use arrowhead to get TAD        \n",
    "# cmd = f\"juicer_tools arrowhead GSE156097_HiC_24h_wt_comb.hic -k KR ./arrowhead\"\n",
    "# subprocess.run(cmd, shell=True)\n",
    "\n",
    "# get TAD bed file\n",
    "df = pd.read_csv('arrowhead/10000_blocks.bedpe', sep='\\t', header=None, skiprows=2)\n",
    "df[0] = df[0]\n",
    "df[[0,1,2]].to_csv('TAD.bed', sep='\\t', header=None, index=None)\n",
    "\n",
    "# Sort TAD, the result is: TAD_sort.bed\n",
    "cmd = f\"bedtools sort -i TAD.bed > TAD_sort.bed\"\n",
    "subprocess.run(cmd, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "f42e8693-4065-4f6f-98d3-285d02e3c021",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter TAD, the result is: TAD_filter_sort.bed\n",
    "def ifIntersect(x:tuple, y:tuple, f=0., F=0.)->bool:\n",
    "    '''If two section overlap'''\n",
    "    if int(x[0])<int(y[0])<int(x[1]):\n",
    "        if int(y[1])<int(x[1]):\n",
    "            return 1\n",
    "        else:\n",
    "            if int(x[1])-int(y[0]) / int(x[1])-int(x[0]) > f or int(x[1])-int(y[0]) / int(y[1])-int(y[0]) > F:\n",
    "                return 1\n",
    "            else:\n",
    "                return 0\n",
    "    elif int(y[0])<int(x[0])<int(y[1]):\n",
    "        if int(y[1]) > int(x[1]):\n",
    "            return 1\n",
    "        else:\n",
    "            if int(y[1])-int(x[0]) / int(x[1])-int(x[0]) > f or int(y[1])-int(x[0]) / int(y[1])-int(y[0]) > F:\n",
    "                return 1\n",
    "            else:\n",
    "                return 0\n",
    "    return 0\n",
    "\n",
    "from collections import defaultdict as ddict\n",
    "tadInterval = ddict(list)\n",
    "with open('TAD_sort.bed', 'r') as infile:\n",
    "    for line in infile:\n",
    "        if len(tadInterval[line.split()[0]]) == 0:\n",
    "            tadInterval[line.split()[0]].append((int(line.split()[1]), int(line.split()[2])))\n",
    "        else:\n",
    "            tadInter = (int(line.split()[1]), int(line.split()[2]))\n",
    "            if ifIntersect(tadInterval[line.split()[0]][-1], tadInter):\n",
    "                lengthOld = tadInterval[line.split()[0]][-1][1] - tadInterval[line.split()[0]][-1][0]\n",
    "                lengthNew = tadInter[1] - tadInter[0]\n",
    "                if lengthNew > lengthOld:\n",
    "                    tadInterval[line.split()[0]].remove(tadInterval[line.split()[0]][-1])\n",
    "                    tadInterval[line.split()[0]].append(tadInter)\n",
    "            else:\n",
    "                tadInterval[line.split()[0]].append(tadInter)\n",
    "outfile = open('TAD_filter.bed', 'w')\n",
    "for chromosome in tadInterval.keys():\n",
    "    for i in tadInterval[chromosome]:\n",
    "        print(f'{chromosome}\\t{i[0]}\\t{i[1]}', file=outfile)\n",
    "outfile.close()\n",
    "!bedtools sort -i TAD_filter.bed > TAD_filter_sort.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "fa78302f-17e8-47cf-b546-fb320bfe0469",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/limh/miniconda3/envs/chipseq/lib/python3.7/site-packages/ipykernel_launcher.py:19: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    }
   ],
   "source": [
    "# build enhancer promoter interaction\n",
    "cmd = f\"bedtools intersect -a enhancer.bg -b TAD_filter_sort.bed -wo -f 0.8 -F 0.8 -e > enhancer_TAD_intersectWo.txt\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "cmd = f\"bedtools intersect -a promoter.bed -b TAD_filter_sort.bed -wo -f 0.8 -F 0.8  -e > promoter_TAD_intersectWo.txt\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "promoter_TAD_intersectWo_df = pd.read_csv('promoter_TAD_intersectWo.txt', sep='\\t', header=None)\n",
    "promoter_TAD_intersectWo_df.columns = ['promoter_chr', 'promoter_start', 'promoter_end', 'promoter_name', 'promoter_score', 'promoter_strand',\n",
    "                                       'TAD_chr', 'TAD_start', 'TAD_end', 'intersect']\n",
    "eRNA_TAD_intersectWo_df = pd.read_csv('enhancer_TAD_intersectWo.txt', sep='\\t', header=None)\n",
    "eRNA_TAD_intersectWo_df.columns = ['eRNA_chr', 'eRNA_start', 'eRNA_end', 'eRNA_name', 'eRNA_score', 'eRNA_strand',\n",
    "                                   'TAD_chr', 'TAD_start', 'TAD_end', 'intersect']\n",
    "epi_df = pd.merge(eRNA_TAD_intersectWo_df, promoter_TAD_intersectWo_df,\n",
    "                  on = ['TAD_chr', 'TAD_start', 'TAD_end'], how='inner')\n",
    "\n",
    "# EPI filter\n",
    "epi_df['filter'] = epi_df.apply(ifFilter, axis=1)\n",
    "epi_filter_df = epi_df[epi_df['filter']==False]\n",
    "epi_filter_df['epi_name'] = epi_filter_df['eRNA_name'].str.cat(epi_filter_df['promoter_name'], sep=';')\n",
    "\n",
    "# EPI result save\n",
    "epi_filter_df[['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end',\n",
    "               'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']].to_csv('EPI.bedpe', sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "847d54d0-eee6-4fbe-9add-1e926ddc5a8b",
   "metadata": {},
   "source": [
    "# EP similarity analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9d1aebc1-e14c-4054-a21c-2c45902fa330",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import subprocess\n",
    "import os\n",
    "import numpy as np\n",
    "import pyBigWig\n",
    "pd.set_option('display.max_columns', None)\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "d05851fe-67b8-44a5-aadf-153a5b8d7ab9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "index file /home/limh/Reference_Genome/Danio_rerio/danRer10/danRer10.fa.fai not found, generating...\n"
     ]
    }
   ],
   "source": [
    "# get enhancer、promoter from EPI\n",
    "referenceGenome = '/home/limh/Reference_Genome/Danio_rerio/danRer10/danRer10.fa'\n",
    "epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "epi_df[['eRNA_name', 'promoter_name']] = epi_df[6].str.split(';', expand=True)\n",
    "epi_df[[0, 1, 2, 'eRNA_name', 7, 8]].drop_duplicates('eRNA_name').to_csv('enhancer_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "epi_df[[3, 4, 5, 'promoter_name', 7, 9]].drop_duplicates('promoter_name').to_csv('promoter_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "\n",
    "# Get fasta of eRNAinEPI,promoterInEPI\n",
    "cmd = f\"bedtools getfasta -fi {referenceGenome} -fo enhancer_inEPI.fa -name -bed enhancer_inEPI.bed -s\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "cmd = f\"bedtools getfasta -fi {referenceGenome} -fo promoter_inEPI.fa -name -bed promoter_inEPI.bed -s\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "# Split fasta\n",
    "os.makedirs('EPI_fasta', exist_ok=True)\n",
    "with open('enhancer_inEPI.fa', 'r') as f:\n",
    "\tfor line in f:\n",
    "\t\tif line.startswith('>'):\n",
    "\t\t\tname = line.strip().split('::')[0][1:]\n",
    "\t\t\tout = open(f\"EPI_fasta/{name}.fa\", 'w')\n",
    "\t\t\t# out.write(f\">{name}\\n\")\n",
    "\t\t\t# out.write(line)\n",
    "\t\t\tprint('>' + line.strip().split('::')[1], file=out)\n",
    "\t\telse:\n",
    "\t\t\tout.write(line)\n",
    "\tout.close()\n",
    "with open('promoter_inEPI.fa', 'r') as f:\n",
    "\tfor line in f:\n",
    "\t\tif line.startswith('>'):\n",
    "\t\t\tname = line.strip().split('::')[0][1:]\n",
    "\t\t\tout = open(f\"EPI_fasta/{name}.fa\", 'w')\n",
    "\t\t\t# out.write(f\">{name}\\n\")\n",
    "\t\t\t# out.write(line)\n",
    "\t\t\tprint('>' + line.strip().split('::')[1], file=out)\n",
    "\t\telse:\n",
    "\t\t\tout.write(line)\n",
    "\tout.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "c0dd6328-f655-48f4-bc00-671591136368",
   "metadata": {},
   "outputs": [],
   "source": [
    "# enhancer\n",
    "# makeblastdb script\n",
    "genes = [x for x in os.listdir('EPI_fasta') if (x.endswith('.fa') and not x.startswith('enhancer'))]\n",
    "makedf_bash_file = open('EPI_makedb.sh', 'w')\n",
    "if os.path.exists('EPI_makedb.log'):\n",
    "    os.remove('EPI_makedb.log')\n",
    "for gene in enumerate(genes):\n",
    "    print(f\"makeblastdb -in EPI_fasta/{gene[1]} -dbtype nucl >> EPI_makedb.log\", file=makedf_bash_file)\n",
    "makedf_bash_file.close()\n",
    "\n",
    "# blast script\n",
    "epi_list = epi_df[['eRNA_name', 'promoter_name']].values.tolist()\n",
    "blast_bash_file = open('EPI_blast.sh', 'w')\n",
    "if os.path.exists('EPI_blastResult.txt'):\n",
    "    os.remove('EPI_blastResult.txt')\n",
    "for i in epi_list:\n",
    "    eRNA = i[0]\n",
    "    promoter = i[1]\n",
    "    print(f\"blastn -task blastn -query EPI_fasta/{eRNA}.fa -db EPI_fasta/{promoter}.fa -outfmt 6 -evalue 1e-4 >> EPI_blastResult.txt\", file=blast_bash_file)\n",
    "blast_bash_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3365dc3-647d-49a1-8a8b-5ded446ffa8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "subprocess.run('bash EPI_makedb.sh', shell=True)\n",
    "subprocess.run('bash EPI_blast.sh', shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "6d2295ee-2a79-42a4-a667-acc648b0966e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the sequence similar enhancer in EPI\n",
    "blastResultFile = 'EPI_blastResult.txt'\n",
    "blastResultBedFile = 'EPI_enhancerBlastHit.bed'\n",
    "with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "\tfor line in infile:\n",
    "\t\tchromosome = line.split()[0].split(':')[0]\n",
    "\t\tstartPre = int(line.split()[0].split(':')[1].split('-')[0])\n",
    "\t\tstopPre = int(line.split()[0].split(':')[1].split('-')[1].split('(')[0])\n",
    "\t\tname = line.split()[0]\n",
    "\t\totherName = line.split()[1]\n",
    "\t\tstrand = line.split()[0].split(':')[1].split('(')[1][0]\n",
    "\t\tif strand == '+' or strand == '.':\n",
    "\t\t\tif int(line.split()[6]) < int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[6])\n",
    "\t\t\t\tshiftEnd = int(line.split()[7])\n",
    "\t\t\telif int(line.split()[6]) > int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[7])\n",
    "\t\t\t\tshiftEnd = int(line.split()[6])\n",
    "\t\t\tstart = startPre + shiftStart - 1\n",
    "\t\t\tend = startPre + shiftEnd - 1\n",
    "\t\telif strand == '-':\n",
    "\t\t\tif int(line.split()[6]) < int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[6])\n",
    "\t\t\t\tshiftEnd = int(line.split()[7])\n",
    "\t\t\telif int(line.split()[6]) > int(line.split()[7]):\n",
    "\t\t\t\tshiftStart = int(line.split()[7])\n",
    "\t\t\t\tshiftEnd = int(line.split()[6])\n",
    "\t\t\tend = stopPre - shiftStart\n",
    "\t\t\tstart = stopPre - shiftEnd\n",
    "\t\tprint(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)\n",
    "        \n",
    "# Get the sequence similar promoter in EPI\n",
    "blastResultFile = 'EPI_blastResult.txt'\n",
    "blastResultBedFile = 'EPI_promoterBlastHit.bed'\n",
    "with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "\tfor line in infile:\n",
    "\t\tchromosome = line.split()[1].split(':')[0]\n",
    "\t\tstartPre = int(line.split()[1].split(':')[1].split('-')[0])\n",
    "\t\tstopPre = int(line.split()[1].split(':')[1].split('-')[1].split('(')[0])\n",
    "\t\tname = line.split()[1]\n",
    "\t\totherName = line.split()[0]\n",
    "\t\tstrand = line.split()[1].split(':')[1].split('(')[1][0]\n",
    "\t\tif strand == '+' or strand == '.':\n",
    "\t\t\tif int(line.split()[8]) < int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[8])\n",
    "\t\t\t\tshiftEnd = int(line.split()[9])\n",
    "\t\t\telif int(line.split()[8]) > int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[9])\n",
    "\t\t\t\tshiftEnd = int(line.split()[8])\n",
    "\t\t\tstart = startPre + shiftStart - 1\n",
    "\t\t\tend = startPre + shiftEnd - 1\n",
    "\t\telif strand == '-':\n",
    "\t\t\tif int(line.split()[8]) < int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[8])\n",
    "\t\t\t\tshiftEnd = int(line.split()[9])\n",
    "\t\t\telif int(line.split()[8]) > int(line.split()[9]):\n",
    "\t\t\t\tshiftStart = int(line.split()[9])\n",
    "\t\t\t\tshiftEnd = int(line.split()[8])\n",
    "\t\t\tend = stopPre - shiftStart\n",
    "\t\t\tstart = stopPre - shiftEnd\n",
    "\t\tprint(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "f7ba5398-a266-45e8-becf-6bf1ac5f2f66",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get out the TE similar EPI, save as ep_same_{te}.bedpe\n",
    "promoter_blast_df = pd.read_csv('EPI_promoterBlastHit.bed', sep='\\t', header=None)\n",
    "enhancer_blast_df = pd.read_csv('EPI_enhancerBlastHit.bed', sep='\\t', header=None)\n",
    "promoter_blast_df.columns = ['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end', 'promoter', 'eRNA', 'promoter_strand']\n",
    "enhancer_blast_df.columns = ['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end', 'eRNA', 'promoter', 'eRNA_strand']\n",
    "epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "epi_df.columns = ['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end',\n",
    "               'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']\n",
    "promoter_blast_df['promoter_chr'] = promoter_blast_df['promoter'].str.split(':').str[0]\n",
    "promoter_blast_df['promoter_start'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "promoter_blast_df['promoter_end'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "promoter_blast_df['eRNA_chr'] = promoter_blast_df['eRNA'].str.split(':').str[0]\n",
    "promoter_blast_df['eRNA_start'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "promoter_blast_df['eRNA_end'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "enhancer_blast_df['eRNA_chr'] = enhancer_blast_df['eRNA'].str.split(':').str[0]\n",
    "enhancer_blast_df['eRNA_start'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "enhancer_blast_df['eRNA_end'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "enhancer_blast_df['promoter_chr'] = enhancer_blast_df['promoter'].str.split(':').str[0]\n",
    "enhancer_blast_df['promoter_start'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "enhancer_blast_df['promoter_end'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "# Get out the intersection between transposon and blastHit\n",
    "transposonFile = 'danRer10.fa.bed'\n",
    "x = pd.merge(epi_df, promoter_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                            'promoter_start', 'promoter_end', 'promoter_strand'], how='inner')\n",
    "y = pd.merge(x, enhancer_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                       'promoter_start', 'promoter_end', 'promoter', 'eRNA', 'eRNA_strand'\n",
    "                                       ], how='inner')\n",
    "cmd = f\"bedtools intersect -a EPI_promoterBlastHit.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > EPI_promoterBlastHit_transposon_intersectWo.bed\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "cmd = f\"bedtools intersect -a EPI_enhancerBlastHit.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > EPI_enhancerBlastHit_transposon_intersectWo.bed\"\n",
    "subprocess.run(cmd, shell=True)\n",
    "\n",
    "promoter_blast_te_intersectWo_df = pd.read_csv('EPI_promoterBlastHit_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "enhancer_blast_te_intersectWo_df = pd.read_csv('EPI_enhancerBlastHit_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "promoter_blast_te_intersectWo_df = promoter_blast_te_intersectWo_df.rename({\n",
    "    0:'promoter_blastHit_chr', 1:'promoter_blastHit_start', 2:'promoter_blastHit_end', \n",
    "    3:'promoter', 4:'eRNA', 5:'promoter_strand'}, axis=1)\n",
    "enhancer_blast_te_intersectWo_df = enhancer_blast_te_intersectWo_df.rename({\n",
    "    0:'eRNA_blastHit_chr', 1:'eRNA_blastHit_start', 2:'eRNA_blastHit_end',\n",
    "    3:'eRNA', 4:'promoter', 5:'eRNA_strand'}, axis=1)\n",
    "\n",
    "# merge the promoter and enhancer information together\n",
    "transposon_similar_EPI_df = pd.merge(\n",
    "    pd.merge(y, promoter_blast_te_intersectWo_df, on=['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end',\n",
    "                                                  'promoter', 'eRNA', 'promoter_strand'], how='inner'),\n",
    "    enhancer_blast_te_intersectWo_df, on=['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end',\n",
    "                                            'eRNA', 'promoter', 'eRNA_strand'], how='inner')\n",
    "\n",
    "transposon_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'] == transposon_similar_EPI_df['12_y']]\n",
    "SINE_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'].str.contains('SINE')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "b88bd9c5-f766-43b5-8520-47595632c0ae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "697\n",
      "399\n",
      "5\n",
      "Simple_repeat 0.36\n",
      "DNA 0.0\n",
      "DNA/Merlin 0.02\n",
      "DNA/Kolobok? 0.03\n",
      "DNA/CMC-EnSpm 0.01\n",
      "DNA/P? 0.5\n",
      "RC/Helitron 0.01\n",
      "DNA/TcMar? 0.03\n",
      "DNA/hAT-Ac 0.0\n",
      "Low_complexity 0.01\n",
      "DNA/hAT-Charlie 0.0\n",
      "DNA/TcMar-Tc1 0.0\n",
      "DNA/CMC-EnSpm? 0.0\n",
      "DNA/hAT-hAT5 0.0\n",
      "SINE/tRNA-V 0.0\n",
      "DNA/hAT 0.0\n",
      "LTR/Ngaro 0.0\n",
      "LINE?/Penelope? 0.0\n",
      "LINE/L2 0.0\n",
      "DNA/IS3EU 0.0\n",
      "DNA/hAT-Ac? 0.0\n",
      "DNA/Kolobok-T2? 0.0\n",
      "Unknown 0.0\n",
      "DNA/hAT-Tip100 0.0\n",
      "DNA/Kolobok-T2 0.0\n",
      "DNA/Sola 0.0\n",
      "DNA/hAT-Tol2 0.0\n",
      "DNA/Maverick 0.0\n",
      "DNA/hAT? 0.0\n",
      "DNA/PIF-Harbinger 0.0\n",
      "SINE/5S-Deu-L2 0.0\n",
      "DNA/TcMar 0.0\n",
      "tRNA 0.0\n",
      "Satellite 0.0\n"
     ]
    }
   ],
   "source": [
    "# seuquence-similar EPI num\n",
    "print(y['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "# transposon-similar EPI num\n",
    "print(transposon_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "# SINE-similar EPI num\n",
    "print(SINE_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "# each transposon ratio in transposon-similar EPI (corrected by copy number)\n",
    "df = pd.read_csv(transposonFile, sep='\\t', header=None)\n",
    "transposon_num_dict = transposon_similar_EPI_df['12_x'].value_counts()\n",
    "transposon_copyNum_dict = df[6].value_counts().to_dict()\n",
    "transposon_num_correct_dict = dict()\n",
    "for key in transposon_num_dict.keys():\n",
    "    result = transposon_num_dict[key] / transposon_copyNum_dict[key]\n",
    "    transposon_num_correct_dict[key] = [result]\n",
    "for key, value in (pd.DataFrame.from_dict(transposon_num_correct_dict).T / pd.DataFrame.from_dict(transposon_num_correct_dict).T.sum())[0].to_dict().items():\n",
    "    print(key, round(value, 2))\n",
    "\n",
    "# result save\n",
    "for i, j in SINE_similar_EPI_df.groupby('12_x'):\n",
    "    j[['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end', 'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']\n",
    " ].drop_duplicates('epi_name').to_csv(f\"ep_same_{i.replace('/', '_')}.bedpe\", sep='\\t', header=None, index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "9383630c-9369-4406-b803-5356133bd566",
   "metadata": {},
   "outputs": [],
   "source": [
    "# random Control (50 shuffle)\n",
    "epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "epi_df[['eRNA_name', 'promoter_name']] = epi_df[6].str.split(';', expand=True)\n",
    "epi_df[[0, 1, 2, 'eRNA_name', 7, 8]].drop_duplicates('eRNA_name').to_csv('enhancer_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "epi_df[[3, 4, 5, 'promoter_name', 7, 9]].drop_duplicates('promoter_name').to_csv('promoter_inEPI.bed', sep='\\t', header=None, index=None)\n",
    "for i in range(50):\n",
    "    epi_list = epi_df[['eRNA_name', 'promoter_name']].values.tolist()\n",
    "    df = pd.DataFrame(epi_list)\n",
    "    df[1] = df[1].sample(frac=1, replace=False).reset_index()[1]\n",
    "    random_epi_list = df.values\n",
    "    blast_bash_file = open(f'random_EPI_blast_{i}.sh', 'w')\n",
    "    if os.path.exists(f'random_EPI_blastResult_{i}.txt'):\n",
    "        os.remove(f'random_EPI_blastResult_{i}.txt')\n",
    "    for j in random_epi_list:\n",
    "        eRNA = j[0]\n",
    "        promoter = j[1]\n",
    "        print(f\"blastn -task blastn -query EPI_fasta/{eRNA}.fa -db EPI_fasta/{promoter}.fa -outfmt 6 -evalue 1e-4 >> random_EPI_blastResult_{i}.txt\", file=blast_bash_file)\n",
    "    blast_bash_file.close()\n",
    "\n",
    "outfile = open('random_control.sh', 'w')\n",
    "for i in [x for x in os.listdir() if x.startswith('random_EPI_blast')]:\n",
    "    print(f\"bash {i} &\", file=outfile)\n",
    "print('wait', file=outfile)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1900631-f5d4-44eb-bda5-5ad7659fd1fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "subprocess.run('bash random_control.sh', shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "98b3301e-c63b-47ad-a625-85ad10db01c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████| 50/50 [06:08<00:00,  7.38s/it]\n"
     ]
    }
   ],
   "source": [
    "from tqdm import tqdm\n",
    "# random control result analysis\n",
    "random_seuquence_similar_EPI_num_list = []\n",
    "random_transposon_similar_EPI_num_list = []\n",
    "random_SINE_similar_EPI_num_list = []\n",
    "random_each_transposon_ratio_list = []\n",
    "transposon_num_df = pd.read_csv(transposonFile, sep='\\t', header=None)\n",
    "transposon_copyNum_dict = transposon_num_df[6].value_counts().to_dict()\n",
    "\n",
    "for i in tqdm(range(50)):\n",
    "    random_result_file = f\"random_EPI_blastResult_{i}.txt\"\n",
    "    # Get the sequence similar enhancer in EPI\n",
    "    blastResultFile = f\"random_EPI_blastResult_{i}.txt\"\n",
    "    blastResultBedFile = f'random_EPI_enhancerBlastHit_{i}.bed'\n",
    "    with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "        for line in infile:\n",
    "            chromosome = line.split()[0].split(':')[0]\n",
    "            startPre = int(line.split()[0].split(':')[1].split('-')[0])\n",
    "            stopPre = int(line.split()[0].split(':')[1].split('-')[1].split('(')[0])\n",
    "            name = line.split()[0]\n",
    "            otherName = line.split()[1]\n",
    "            strand = line.split()[0].split(':')[1].split('(')[1][0]\n",
    "            if strand == '+' or strand == '.':\n",
    "                if int(line.split()[6]) < int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[6])\n",
    "                    shiftEnd = int(line.split()[7])\n",
    "                elif int(line.split()[6]) > int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[7])\n",
    "                    shiftEnd = int(line.split()[6])\n",
    "                start = startPre + shiftStart - 1\n",
    "                end = startPre + shiftEnd - 1\n",
    "            elif strand == '-':\n",
    "                if int(line.split()[6]) < int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[6])\n",
    "                    shiftEnd = int(line.split()[7])\n",
    "                elif int(line.split()[6]) > int(line.split()[7]):\n",
    "                    shiftStart = int(line.split()[7])\n",
    "                    shiftEnd = int(line.split()[6])\n",
    "                end = stopPre - shiftStart\n",
    "                start = stopPre - shiftEnd\n",
    "            print(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)\n",
    "    # Get the sequence similar promoter in EPI\n",
    "    blastResultFile = f\"random_EPI_blastResult_{i}.txt\"\n",
    "    blastResultBedFile = f'random_EPI_promoterBlastHit_{i}.bed'\n",
    "    with open(blastResultFile, 'r') as infile, open(blastResultBedFile, 'w') as outfile:\n",
    "        for line in infile:\n",
    "            chromosome = line.split()[1].split(':')[0]\n",
    "            startPre = int(line.split()[1].split(':')[1].split('-')[0])\n",
    "            stopPre = int(line.split()[1].split(':')[1].split('-')[1].split('(')[0])\n",
    "            name = line.split()[1]\n",
    "            otherName = line.split()[0]\n",
    "            strand = line.split()[1].split(':')[1].split('(')[1][0]\n",
    "            if strand == '+' or strand == '.':\n",
    "                if int(line.split()[8]) < int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[8])\n",
    "                    shiftEnd = int(line.split()[9])\n",
    "                elif int(line.split()[8]) > int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[9])\n",
    "                    shiftEnd = int(line.split()[8])\n",
    "                start = startPre + shiftStart - 1\n",
    "                end = startPre + shiftEnd - 1\n",
    "            elif strand == '-':\n",
    "                if int(line.split()[8]) < int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[8])\n",
    "                    shiftEnd = int(line.split()[9])\n",
    "                elif int(line.split()[8]) > int(line.split()[9]):\n",
    "                    shiftStart = int(line.split()[9])\n",
    "                    shiftEnd = int(line.split()[8])\n",
    "                end = stopPre - shiftStart\n",
    "                start = stopPre - shiftEnd\n",
    "            print(f'{chromosome}\\t{start}\\t{end}\\t{name}\\t{otherName}\\t{strand}', file=outfile)\n",
    "    # Get out the TE similar EPI, save as ep_same_{te}.bedpe\n",
    "    try:\n",
    "        promoter_blast_df = pd.read_csv(f'random_EPI_promoterBlastHit_{i}.bed', sep='\\t', header=None)\n",
    "        enhancer_blast_df = pd.read_csv(f'random_EPI_enhancerBlastHit_{i}.bed', sep='\\t', header=None)\n",
    "    except:\n",
    "        random_seuquence_similar_EPI_num_list.append(0)\n",
    "        random_transposon_similar_EPI_num_list.append(0)\n",
    "        random_SINE_similar_EPI_num_list.append(0)\n",
    "        random_each_transposon_ratio_list.append({})\n",
    "        continue\n",
    "    promoter_blast_df.columns = ['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end', 'promoter', 'eRNA', 'promoter_strand']\n",
    "    enhancer_blast_df.columns = ['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end', 'eRNA', 'promoter', 'eRNA_strand']\n",
    "    epi_df = pd.read_csv('EPI.bedpe', sep='\\t', header=None)\n",
    "    epi_df.columns = ['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', 'promoter_start', 'promoter_end',\n",
    "                   'epi_name', 'eRNA_score', 'eRNA_strand', 'promoter_strand']\n",
    "    promoter_blast_df['promoter_chr'] = promoter_blast_df['promoter'].str.split(':').str[0]\n",
    "    promoter_blast_df['promoter_start'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    promoter_blast_df['promoter_end'] = promoter_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "    promoter_blast_df['eRNA_chr'] = promoter_blast_df['eRNA'].str.split(':').str[0]\n",
    "    promoter_blast_df['eRNA_start'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    promoter_blast_df['eRNA_end'] = promoter_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "    enhancer_blast_df['eRNA_chr'] = enhancer_blast_df['eRNA'].str.split(':').str[0]\n",
    "    enhancer_blast_df['eRNA_start'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    enhancer_blast_df['eRNA_end'] = enhancer_blast_df['eRNA'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "    enhancer_blast_df['promoter_chr'] = enhancer_blast_df['promoter'].str.split(':').str[0]\n",
    "    enhancer_blast_df['promoter_start'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[0].str.split('(').str[0].astype(int)\n",
    "    enhancer_blast_df['promoter_end'] = enhancer_blast_df['promoter'].str.split(':').str[1].str.split('-').str[1].str.split('(').str[0].astype(int)\n",
    "\n",
    "    # Get out the intersection between transposon and blastHit\n",
    "    x = pd.merge(epi_df, promoter_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                                'promoter_start', 'promoter_end', 'promoter_strand'], how='inner')\n",
    "    y = pd.merge(x, enhancer_blast_df, on=['eRNA_chr', 'eRNA_start', 'eRNA_end', 'promoter_chr', \n",
    "                                           'promoter_start', 'promoter_end', 'promoter', 'eRNA', 'eRNA_strand'\n",
    "                                           ], how='inner')\n",
    "    cmd = f\"bedtools intersect -a random_EPI_promoterBlastHit_{i}.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > random_EPI_promoterBlastHit_{i}_transposon_intersectWo.bed\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "    cmd = f\"bedtools intersect -a random_EPI_enhancerBlastHit_{i}.bed -b {transposonFile} -wo -f 0.8 -F 0.8 -e > random_EPI_enhancerBlastHit_{i}_transposon_intersectWo.bed\"\n",
    "    subprocess.run(cmd, shell=True)\n",
    "\n",
    "    \n",
    "    promoter_blast_te_intersectWo_df = pd.read_csv(f'random_EPI_promoterBlastHit_{i}_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "    enhancer_blast_te_intersectWo_df = pd.read_csv(f'random_EPI_enhancerBlastHit_{i}_transposon_intersectWo.bed', sep='\\t', header=None)\n",
    "    promoter_blast_te_intersectWo_df = promoter_blast_te_intersectWo_df.rename({\n",
    "        0:'promoter_blastHit_chr', 1:'promoter_blastHit_start', 2:'promoter_blastHit_end', \n",
    "        3:'promoter', 4:'eRNA', 5:'promoter_strand'}, axis=1)\n",
    "    enhancer_blast_te_intersectWo_df = enhancer_blast_te_intersectWo_df.rename({\n",
    "        0:'eRNA_blastHit_chr', 1:'eRNA_blastHit_start', 2:'eRNA_blastHit_end',\n",
    "        3:'eRNA', 4:'promoter', 5:'eRNA_strand'}, axis=1)\n",
    "\n",
    "    # merge the promoter and enhancer information together\n",
    "    transposon_similar_EPI_df = pd.merge(\n",
    "        pd.merge(y, promoter_blast_te_intersectWo_df, on=['promoter_blastHit_chr', 'promoter_blastHit_start', 'promoter_blastHit_end',\n",
    "                                                      'promoter', 'eRNA', 'promoter_strand'], how='inner'),\n",
    "        enhancer_blast_te_intersectWo_df, on=['eRNA_blastHit_chr', 'eRNA_blastHit_start', 'eRNA_blastHit_end',\n",
    "                                                'eRNA', 'promoter', 'eRNA_strand'], how='inner')\n",
    "\n",
    "    transposon_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'] == transposon_similar_EPI_df['12_y']]\n",
    "    SINE_similar_EPI_df = transposon_similar_EPI_df[transposon_similar_EPI_df['12_x'].str.contains('SINE')]\n",
    "    \n",
    "    # seuquence-similar EPI num\n",
    "    random_seuquence_similar_EPI_num_list.append(y['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "    # transposon-similar EPI num\n",
    "    random_transposon_similar_EPI_num_list.append(transposon_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "    # SINE-similar EPI num\n",
    "    random_SINE_similar_EPI_num_list.append(SINE_similar_EPI_df['epi_name'].drop_duplicates().shape[0])\n",
    "\n",
    "    # each transposon ratio in transposon-similar EPI (corrected by copy number)\n",
    "    transposon_num_dict = transposon_similar_EPI_df['12_x'].value_counts()\n",
    "    if len(transposon_num_dict) == 0:\n",
    "        random_each_transposon_ratio_list.append({})\n",
    "        continue\n",
    "    transposon_num_correct_dict = dict()\n",
    "    transposon_ratio_correct_dict = dict()\n",
    "    for key in transposon_num_dict.keys():\n",
    "        result = transposon_num_dict[key] / transposon_copyNum_dict[key]\n",
    "        transposon_num_correct_dict[key] = [result]\n",
    "    for key, value in (pd.DataFrame.from_dict(transposon_num_correct_dict).T / pd.DataFrame.from_dict(transposon_num_correct_dict).T.sum())[0].to_dict().items():\n",
    "        transposon_ratio_correct_dict[key] = round(value, 2)\n",
    "    random_each_transposon_ratio_list.append(transposon_ratio_correct_dict)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "5c456967-0d47-4883-bad8-aac66e897b9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<matplotlib.lines.Line2D at 0x7f1c56918e50>,\n",
       " <matplotlib.lines.Line2D at 0x7f1c5726ded0>]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAB4QAAAPmCAYAAAAMqyLoAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/NK7nSAAAACXBIWXMAAA9hAAAPYQGoP6dpAABX/0lEQVR4nOzde5DVhX3///dZVjYWZXePCNGAUWwQNbV1hHqvJFVSRgPBjrRmaiRqTEzDRDtGS9sgTBLRWDWpycQ6oGinIRNbI4rRGi9BJTJqEmNNZOTmaLyE6l6878Lu+f6RH+cnARZcWD7nvPfxmGHmc/Z8zjkvMutMhud8PqdUqVQqAQAAAAAAAEA6DUUPAAAAAAAAAGBgCMIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAF27BhQ9ET3reenp7o7e0tegawHYIwAAAAAABAgdatWxcf/ehH4+677y56yg7r6emJc845J8477zxRGGpcY9EDAAAAAAAABqt169bFpEmT4vnnn49PfepTcfvtt8eUKVOKntWnTTH4lltuqf5swYIF0dDgOkSoRf7LBAAAAAAAKMB7Y3BERHd3d3zqU5+q6SuFtxaDb7rpJlcKQw0rVSqVStEjAAAAAAAABpM/jMHvdeyxx8by5cujVCoVsGzbthaDN2lqaorHHnssjjjiiAKWAX1xhTAAAAAAAMBu1FcMPvLII2Pp0qV1F4PvuOMOMRhqlCAMAAAAAACwm2wvBt93331RLpcLWLZtOxKDJ0+eXMAyYEcIwgAAAAAAALuBGAwUQRAGAAAAAAAYYGIwUBRBGAAAAAAAYACJwUCRBGEAAAAAAIABIgYDRROEAQAAAAAABoAYDNQCQRgAAAAAAGAXE4OBWiEIAwAAAAAA7EJiMFBLBGEAAAAAAIBdRAwGao0gDAAAAAAAsAuIwUAtEoQBAAAAAAB2khgM1CpBGAAAAAAAYCeIwUAtE4QBAAAAAAD6SQwGal2pUqlUih6xq+2xxx7R09NTfVwqlaK5ubnARQAAAAAAQDa9vb3xxhtvRG9v7xbP1XIMbmlpiTfffHOrz++1116xxx577OZVwPvV2dkZ7828Q4YMiQ0bNmz13MbdNWp36unp2ex/gEqlEu3t7QUuAgAAAAAABotajsHnnHPONmNwRPT5HFC73nux7B9KeYVwQ0NDJPxrAQAAAAAANW7IkCGxfv36mo3BW7tNNFD/SqXSVu9WEOE7hAEAAAAAAHaZvffeWwwGakrKW0aXSqXNrhAulUrR0tJS3CAAAAAAAKDu9fWdwUOGDIm99947WltbC1i2bduLwb4zOKmOjoj33k23VIrQylLp6OjYooduS8og3NzcvNl3Bre0tERbW1uBiwAAAAAAgHq2bt26mDRpUnR2dm7xXK1/Z/DWYnBTU1PccccdMXny5AKWMeDK5Yj3tLJoaYnQylIpl8ub9dDm5uZtnuuW0QAAAAAAAH3YFIOff/75LZ4Tg4FaJwgDAAAAAABsgxgM1DtBGAAAAAAAYCvEYCADQRgAAAAAAOAPiMFAFoIwAAAAAADAe4jBQCaCMAAAAAAAwP9HDAayEYQBAAAAAABCDAZyEoQBAAAAAIBBTwwGshKEAQAAAACAQU0MBjIThAEAAAAAgEFLDAayE4QBAAAAAIBBSQwGBgNBGAAAAAAAGHTEYGCwEIQBAAAAAIBBRQwGBhNBGAAAAAAAGFRuv/32uorBEb+P2EuWLNni52IwsD2CMAAAAAAAMKhceOGFMWfOnM1+VssxOCLij//4j+O+++6L5ubm6s/EYGBHCMIAAAAAAMCgUiqVYu7cudUoXOsxeJMJEyZUo7AYDOyoxqIHAAAAAAAA7G6bovB+++0XM2bMqPkYvMmmKNzW1iYGAztEEAYAAAAAAAalUqkUX/jCF4qe8b5NmDCh6AlAHXHLaAAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApBqLHgBs39ixY6Ojo6PoGbBLdXZ2RqVSqT4ulUrR3Nxc4CLYtVpaWmLt2rVFzwAAAAAABjlBGOpAR0dHtLe3Fz0DBlSlUvF7DgAAAAAAu5hbRgMAAAAAAAAkJQgDAAAAAAAAJOWW0VCnWltbi54AO2Vrt4f2e009c8tzAAAAAKAWCcJQh1pbW6Otra3oGbBTGhoaolKpVB/7vabelctlURgAAAAAqDluGQ0AAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQVL+C8Pr162Pp0qUxZ86cmDJlSowYMSJKpVKUSqWYOXPmDr3Hu+++G0uWLIlZs2bF0UcfHeVyOfbYY48ol8tx7LHHxty5c+Pll1/uzzwAAAAAAAAAIqKxPy8aNWrUTn3oU089FSeccEK88cYbWzzX3t4eK1asiBUrVsQ111wTCxYsiBkzZuzU5wEAAAAAAAAMRv0Kwu81ZsyYOPTQQ+Pee+/d4de8/vrr1Rh8/PHHx2mnnRYTJkyIffbZJ/7v//4vbrvttliwYEG88cYb8elPfzr23nvvmDJlys5OBQAAAAAAABhU+hWE58yZExMnToyJEyfGqFGj4rnnnouDDjpoh1/f0NAQM2bMiMsuuywOO+ywLZ6fPHlyTJkyJaZPnx49PT0xa9asWLVqVZRKpf7MBQAAAAAAABiU+hWE582bt1Mfetxxx8Vxxx3X5znTpk2L008/Pf77v/871qxZE08++WQceeSRO/W5AAAAAAAAAINJQ9ED+vKxj32serxmzZoClwAAAAAAAADUn5oOwl1dXdXjhoaangoAAAAAAABQc2q6si5btqx6PH78+AKXAAAAAAAAANSffn2H8O7wq1/9Ku66666IiDj88MPjsMMO6/P8rq6u6hXFlUplwPcBAAAAAAAA1LqavEK4q6srzjvvvOjp6YmIiMsvv3y7r5k/f340NzdHc3NzdHR0DPBCAAAAAAAAgNpXk0H4S1/6UjzxxBMREXH22WfH1KlTt/ua2bNnR2dnZ3R2dkZLS8sALwQAAAAAAACofTV3y+j58+fHggULIiLiqKOOiu9+97s79LqmpqZoamqKiIhSqTRg+wAAAAAAAADqRU1dIfzv//7v8U//9E8REXHIIYfE3XffHcOGDSt4FQAAAAAAAEB9qpkgvHjx4vjiF78YEREf/vCH47777ot999234FUAAAAAAAAA9asmgvAdd9wRn/nMZ6K3tzf222+/uP/++2P06NFFzwIAAAAAAACoa4UH4fvvvz9mzJgRGzdujH322Sd+8pOfxMEHH1z0LAAAAAAAAIC6V2gQ/tnPfhbTpk2Lrq6uGD58ePzP//xPHH744UVOAgAAAAAAAEijsCD85JNPxqmnnhpvvfVWDBs2LH784x/HUUcdVdQcAAAAAAAAgHQa+/OiRx55JFavXl19/Oqrr1aPV69eHYsWLdrs/JkzZ272eM2aNfGJT3wiOjo6IiLi61//ejQ3N8fTTz+9zc8cOXJkjBw5sj9zAQAAAAAAAAalfgXhBQsWxM0337zV55YvXx7Lly/f7Gd/GIQffvjhWL9+ffXxRRddtN3PvOyyy2Lu3LnveysAAAAAAADAYFXodwgDAAAAAAAAMHD6FYQXLVoUlUplh//8oZkzZ76v11cqFVcHAwAAAAAAALxPrhAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEhKEAYAAAAAAABIShAGAAAAAAAASEoQBgAAAAAAAEiqsegBu0NHR0eUy+WiZ0C/dXR0FD0BAAAAAACAOjQognClUon29vaiZwAAAAAAAADsVm4ZDQAAAAAAAJCUIAwAAAAAAACQ1KC4ZXRERGtra9EToN/c8hwAAAAAAID+GBRBuFQqRVtbW9EzoN8aGhqiUqkUPQMAAAAAAIA645bRAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASfUrCK9fvz6WLl0ac+bMiSlTpsSIESOiVCpFqVSKmTNnvu/3u+eee+L000+P0aNHR1NTU4wePTpOP/30uOeee/ozDwAAAAAAAICIaOzPi0aNGrVLPrxSqcQXvvCFuOGGGzb7+Ysvvhg/+tGP4kc/+lGcf/75cf3110epVNolnwkAAAAAAAAwWOz0LaPHjBkTkydP7tdr/+Vf/qUag4888shYvHhxPPbYY7F48eI48sgjIyLihhtuiK9+9as7OxMAAAAAAABg0OnXFcJz5syJiRMnxsSJE2PUqFHx3HPPxUEHHfS+3mP16tXxzW9+MyIiJkyYEA899FDsueeeERExceLEmDp1apx00knxxBNPxJVXXhmf/exn4+CDD+7PXAAAAAAAAIBBqV9XCM+bNy9OO+20nbp19LXXXhsbN26MiIjrrruuGoM3+aM/+qO47rrrIiJi48aN8a1vfavfnwUAAAAAAAAwGO30LaP7o1KpxJIlSyIiYvz48XHMMcds9bxjjjkmDjnkkIiIuP3226NSqey2jQAAAAAAAAD1rpAgvG7dunjxxRcjIuKkk07q89xNz//2t7+N5557bqCnAQAAAAAAAKRRSBB+5plnqsfjx4/v89z3Pv/e1wEAAAAAAADQt8YiPvSFF16oHo8ePbrPc8eMGbPV1/2hrq6u6Orqiohwa2kAAAAAAACAKOgK4TfeeKN6vNdee/V57rBhw6rHb7755jbPmz9/fjQ3N0dzc3N0dHTs9EYAAAAAAACAeldIEH733Xerx0OHDu3z3KampurxO++8s83zZs+eHZ2dndHZ2RktLS07vREAAAAAAACg3hVyy+gPfOAD1ePu7u4+z910G+iIiD333HOb5zU1NVXjcalU2smFAAAAAAAAAPWvkCuE99577+pxX7eBjoh46623qsfbu700AAAAAAAAAP+/QoLw6NGjq8e//e1v+zz3hRdeqB6PGTNmwDYBAAAAAAAAZFNIED7ssMOqxytXruzz3Pc+f+ihhw7YJgAAAAAAAIBsCgnCBx10UOy///4REbFs2bI+z33ooYciIuJDH/pQHHjggQM9DQAAAAAAACCNQoJwqVSKadOmRcTvrwBesWLFVs9bsWJF9QrhadOmRalU2m0bAQAAAAAAAOpdIUE4IuLCCy+MxsbGiIiYNWtWvPPOO5s9/84778SsWbMiIqKxsTEuvPDC3T0RAAAAAAAAoK419udFjzzySKxevbr6+NVXX60er169OhYtWrTZ+TNnztziPcaNGxcXX3xxXHHFFfHEE0/E8ccfH5deemkcfPDBsWbNmrjyyivjl7/8ZUREfOUrX4mPfOQj/ZkKAAAAAAAAMGiVKpVK5f2+aObMmXHzzTfv8Pnb+oje3t743Oc+FzfeeOM2X3vuuefGDTfcEA0NO34xc7lcjvb29urjUqkUvb29O/x6qDUNDQ2b/XfU2toabW1tBS6Cnef3mmz+8P9/+J0GAAAAoDDlcsR7/q0qWlsj/FtVKu/n3yMLu2V0xO9jwMKFC+Ouu+6KadOmxf777x9Dhw6N/fffP6ZNmxY//vGPY8GCBe8rBgMAAAAAAADwe/26QrjWuUKYbFxJSUZ+r8nGFcIAAAAA1AxXCKdXN1cIAwAAAAAAADBwBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApARhAAAAAAAAgKQEYQAAAAAAAICkBGEAAAAAAACApGoiCHd3d8fChQvjr/7qr2K//faLpqam2GuvveKQQw6Jc845J1asWFH0RAAAAAAAAIC601j0gBdeeCFOPfXU+N///d/Nft7d3R3PPvtsPPvss3HTTTfFRRddFFdffXWUSqWClgIAAAAAAADUl0KvEN64ceNmMfiII46IRYsWxaOPPhr33ntvzJkzJ4YNGxYREddee23867/+a5FzAQAAAAAAAOpKoVcIL1mypBqDjz322Hj44YdjyJAh1edPOeWUmDp1ahx77LGxYcOGmD9/flx00UXR2Fj4hc0AAAAAAAAANa/QK4SXL19ePZ49e/ZmMXiTo446Kk477bSIiGhvb4+VK1futn0AAAAAAAAA9azQINzd3V09Hjt27DbPO/jgg6vHXV1dA7oJAAAAAAAAIItCg/C4ceOqx2vXrt3meWvWrImIiFKpFB/5yEcGfBcAAAAAAABABoUG4TPPPDOGDx8eERFXXnll9PT0bHHOL3/5y7jrrrsiIuJv//Zvq+cDAAAAAAAA0LdCg/C+++4bixYtij333DOWL18eEydOjFtuuSVWrFgR9913X8ybNy9OOumk6O7ujj/7sz+La665psi5AAAAAAAAAHWlsegB06dPjyeeeCKuueaauPHGG+Pss8/e7PlRo0bFvHnz4vzzz49hw4Zt8326urqq3y9cqVQGdDMAAAAAAABAPSj0CuGIiA0bNsT3v//9uPPOO7cacn/3u9/F4sWL46c//Wmf7zN//vxobm6O5ubm6OjoGJixAAAAAAAAAHWk0CD81ltvxcknnxzf+MY34rXXXotLLrkknnnmmejq6orOzs64995744QTTojHH388PvnJT8a3v/3tbb7X7Nmzo7OzMzo7O6OlpWX3/SUAAAAAAAAAalShQfiyyy6Lhx56KCIiFi5cGFdeeWWMHz8+hg4dGsOHD49TTjklHnzwwfjYxz4WlUol/uEf/iGeeuqprb5XU1NTDB8+PIYPHx6lUml3/jUAAAAAAAAAalJhQbhSqcRNN90UERHjxo3b4ruDN2lsbIyvfe1rERHR29tbfQ0AAAAAAAAAfSssCP/ud7+Ltra2iIg48sgj+zz3qKOOqh6vXLlyQHcBAAAAAAAAZFFYEG5sbKweb9y4sc9zN2zYsNXXAQAAAAAAALBthQXhcrkcw4cPj4iIRx99tM8ovGzZsurxQQcdNODbAAAAAAAAADIoLAg3NDTEqaeeGhERL730UnzjG9/Y6nnt7e1x6aWXVh+fdtppu2UfAAAAAAAAQL0r9P7Lc+bMiSVLlsTbb78dc+fOjZ///Odx9tlnx9ixY+Pdd9+NFStWxLe+9a14/vnnIyLiL//yL2Py5MlFTgYAAAAAAACoG4UG4fHjx8eSJUvizDPPjFdffTXuvPPOuPPOO7d67sc//vG49dZbd/NCAAAAAAAAgPpVaBCOiDj55JNj5cqVsXDhwrj77rvj17/+dXR0dERjY2N88IMfjIkTJ8anP/3pmDp1apRKpaLnAgAAAAAAANSNwoNwRMQ+++wTl1xySVxyySVFTwEAAAAAAABIo6HoAQAAAAAAAAAMDEEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIKnGogcAAAC1aezYsdHR0VH0DBgwLS0tsXbt2qJnAAAAwIAShAEAgK3q6OiI9vb2omcAAAAAsBPcMhoAAAAAAAAgKUEYAAAAAAAAICm3jAYAAHZYa2tr0ROg39wCHQAAgMFIEAYAAHZIa2trtLW1FT0D+q1cLovCAAAADDpuGQ0AAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQVM0E4VdffTW++c1vxvHHHx8f/OAHo6mpKfbff/84+uij4ytf+Uo8+uijRU8EAAAAAAAAqCuNRQ+IiLj11lvjggsuiNdee22zn7/88svx8ssvx2OPPRarVq2K22+/vZiBAAAAAAAAAHWo8CB8yy23xGc/+9no7e2NkSNHxgUXXBAnnHBClMvleOWVV2LNmjVx5513xh577FH0VAAAAAAAAIC6UmgQfuaZZ+L888+P3t7eOPHEE+POO++M5ubmLc6bNWtWdHd3F7AQAAAAAAAAoH4V+h3Cs2bNiq6urhgxYkTcdtttW43BmwwdOnQ3LgMAAAAAAACof4UF4ZUrV8b9998fERFf+tKXYsSIEUVNAQAAAAAAAEipsCB86623Vo/POOOM6nF7e3usWrUqXnvttSJmAQAAAAAAAKRRWBBesWJFREQ0NzfHoYceGv/5n/8Zf/qnfxrlcjnGjRsXI0aMiLFjx8a8efPizTffLGomAAAAAAAAQN0qLAj/5je/iYiIAw88MGbNmhV/93d/F0899dRm56xbty7mzp0bxx57bLz00kt9vl9XV1e8/vrr8frrr0elUhmw3QAAAAAAAAD1orAg3NbWFhG//y7h7373u9HS0hLXX399rF+/Pt599914/PHHY8qUKRER8fTTT8cZZ5wRvb2923y/+fPnR3NzczQ3N0dHR8fu+CsAAAAAAAAA1LTCgvBbb70VEb+/snfIkCFx9913x+c///nYd999o6mpKSZMmBBLly6tRuGf/exncdttt23z/WbPnh2dnZ3R2dkZLS0tu+OvAAAAAAAAAFDTCgvCH/jAB6rHZ5xxRhxzzDFbnNPQ0BBXXXVV9fHixYu3+X5NTU0xfPjwGD58eJRKpV07FgAAAAAAAKAOFRaE99577+rxpquAt+bwww+PD33oQxER8fjjjw/4LgAAAAAAAIAsCgvCY8aMqR6PHj16h85dv379gG4CAAAAAAAAyKSwIHz44YdXj3t6evo8d9PzjY2NA7oJAAAAAAAAIJPCgvBf/MVfVI/XrFnT57lr166NiKjeOhoAAAAAAACA7SssCE+dOjX22GOPiIi47bbbtnnesmXL4rXXXouIiBNPPHG3bAMAAAAAAADIoLAgvM8++8R5550XERE/+clP4gc/+MEW57zxxhtx4YUXVh9//vOf313zAAAAAAAAAOpeYUE4ImLevHlxwAEHRETEWWedFbNmzYoHH3wwfv7zn8eiRYviz//8z+PJJ5+MiIgLLrggJk6cWOBaAAAAAAAAgPrSWOSH77vvvnHPPffE1KlTY/Xq1fGd73wnvvOd72xx3jnnnBPf/va3C1gIAAAAAAAAUL8KvUI4IuLQQw+NJ598Mq666qo4+uijo1wux9ChQ2P06NHxN3/zN/HAAw/EwoULq983DAAAAAAAAMCOKfQK4U2GDRsWF198cVx88cVFTwEAAAAAAABIo/ArhAEAAAAAAAAYGIIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFI1GYQvueSSKJVK1T8//elPi54EAAAAAAAAUHdqLgj/6le/imuvvbboGQAAAAAAAAB1r6aCcG9vb3zuc5+LjRs3xsiRI4ueAwAAAAAAAFDXaioI/9u//Vs8/vjjMX78+Dj33HOLngMAAAAAAABQ12omCL/wwgvx1a9+NSIivve978XQoUMLXgQAAAAAAABQ32omCH/xi1+MN998M84+++yYNGlS0XMAAAAAAAAA6l5NBOEf/vCHsXTp0iiXy3HVVVcVPQcAAAAAAAAghcKDcEdHR3z5y1+OiIgrr7wy9t1334IXAQAAAAAAAORQeBC+5JJL4pVXXonjjjsuzj333KLnAAAAAAAAAKTRWOSHP/LII7FgwYJobGyM66+/PkqlUr/fq6urK7q6uiIiolKp7KqJAAAAAAAAAHWrsCuEu7u74/zzz49KpRIXXXRR/Mmf/MlOvd/8+fOjubk5mpubo6OjY9eMBAAAAAAAAKhjhQXhyy+/PJ555pk44IAD4rLLLtvp95s9e3Z0dnZGZ2dntLS07PxAAAAAAAAAgDpXyC2jV65cGfPnz4+IiOuuuy6GDRu20+/Z1NQUTU1NERE7detpAAAAAAAAgCwKCcLXXnttdHd3x9ixY+Ptt9+OH/zgB1uc8/TTT1ePH3jggXjllVciIuKTn/zkLgnIAAAAAAAAANkVEoS7uroiImLt2rVx5plnbvf8r33ta9XjdevWCcIAAAAAAAAAO6Cw7xAGAAAAAAAAYGAVEoQXLVoUlUqlzz+XXXZZ9fwHH3yw+vMDDzywiMkAAAAAAAAAdccVwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJ1WwQnjt3blQqlahUKjFp0qSi5wAAAAAAAADUnZoNwgAAAAAAAADsHEEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIClBGAAAAAAAACApQRgAAAAAAAAgKUEYAAAAAAAAIKnGogcAAADA7tDZ2bnZ446OjiiXywWtgYHR0tISa9euLXoGAABQQwRhAAAABoVKpbLF4/b29oLWAAAAwO7hltEAAAAAAAAASQnCAAAAAAAAAEm5ZTQAAACDVmtra9ETYKe47TkAALA9gjAAAACDUmtra7S1tRU9A3ZKuVwWhQEAgD65ZTQAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoIwAAAAAAAAQFKCMAAAAAAAAEBSgjAAAAAAAABAUoUG4V/84hdx+eWXx5QpU2LMmDHR1NQUe+21V4wbNy5mzpwZDz/8cJHzAAAAAAAAAOpaY1EffNJJJ8VDDz20xc+7u7tj1apVsWrVqrj55pvjrLPOigULFsTQoUMLWAkAAAAAAABQvwoLwi+++GJEROy///5xxhlnxIknnhgHHHBA9PT0xKOPPhpXX311vPjii/Ef//EfsXHjxvj+979f1FQAAAAAAACAulRYEB4/fnxcfvnl8dd//dcxZMiQzZ475phj4qyzzorjjz8+nn322Vi8eHFccMEFceKJJxa0FgAAAAAAAKD+FPYdwkuXLo0ZM2ZsEYM3GTFiRFx99dXVx//1X/+1u6YBAAAAAAAApFBYEN4RkyZNqh6vWbOmuCEAAAAAAAAAdaimg3B3d3f1uKGhpqcCAAAAAAAA1JzCvkN4Ryxbtqx6PH78+AKXAAAAANSezs7OzR53dHREuVwuaA3sei0tLbF27dqiZwAA1LWaDcK9vb1xxRVXVB/PmDGjz/O7urqiq6srIiIqlcqAbgMAAACoBX/4byCVSiXa29sLWgMAANSimr0P87XXXhuPPfZYRERMnz49JkyY0Of58+fPj+bm5mhubo6Ojo7dsBAAAAAAAACgttVkEF62bFn84z/+Y0REjBw5Mr73ve9t9zWzZ8+Ozs7O6OzsjJaWlgFeCAAAAAAAAFD7au6W0b/+9a9j+vTpsXHjxmhqaoof/vCHMWrUqO2+rqmpKZqamiIiolQqDfRMAAAAgJrU2tpa9AToN7c8BwDY9WoqCK9bty4mT54c7e3tMWTIkFi8eHGcdNJJRc8CAAAAqAutra3R1tZW9Azot3K5LAoDAOxiNXPL6JdeeilOPvnkeOmll6JUKsWNN94Y06dPL3oWAAAAAAAAQN2qiSD86quvximnnBJr166NiIjrrrsuPvOZzxS8CgAAAAAAAKC+FR6EOzs74xOf+ET85je/iYiIK664Iv7+7/++4FUAAAAAAAAA9a/QIPz222/HqaeeGr/4xS8iIuKf//mf49JLLy1yEgAAAAAAAEAahQXh7u7umD59eixfvjwiIr785S/H17/+9aLmAAAAAAAAAKTTWNQHn3nmmXHvvfdGRMTHP/7xOPfcc+Ppp5/e5vlDhw6NcePG7a55AAAAAAAAAHWvsCB82223VY8feOCBOOKII/o8/8Mf/nA899xzA7wKAAAAAAAAII9Cv0MYAAAAAAAAgIFT2BXClUqlqI8GAAAAAAAAGBRcIQwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkJQgDAAAAAAAAJCUIAwAAAAAAACQlCAMAAAAAAAAkFRj0QMAAAAAACIiOjs7N3vc0dER5XK5oDUwMFpaWmLt2rVFzwBgEBGEAQAAAICaUKlUtnjc3t5e0BoAgBzcMhoAAAAAAAAgKUEYAAAAAAAAICm3jAYAAAAAalZra2vRE2CnuO05AEUThAEAAACAmtTa2hptbW1Fz4CdUi6XRWEACuWW0QAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAABJCcIAAAAAAAAASQnCAAAAAAAAAEkJwgAAAAAAAAXbsGFD0RPet56enujt7S16BrAdgjAAAAAAAECB1q1bFx/96Efj7rvvLnrKDuvp6Ylz/l97dx8VVZ2AcfwZwBkMECTAcqWIlKLsxSNZhqa0almmkMLWWoqma5Rudixbazf11GZlvlTramZp9iJHWY8p6FqWGm640aZmBSVpopYShrwIOPKyf7DMoswAIcydGb+fczznMvfH5cFz7zB3nrm/O368JkyYQCkMuDgfowMAAAAAAAAAAACcrw4cOKCBAwcqPz9f8fHxWrdunYYOHWp0rCbVl8ErV660PbZs2TJ5eXEdIuCKODIBAAAAAAAAAAAM0LAMliSr1ar4+HiXvlLYXhm8fPlyrhQGXBiFMAAAAAAAAAAAgJOdXQbXs1qteuaZZ1RbW2tQMsfslcH13nvvPX311VcGpALQHAphAAAAAAAAAAAAJ3JUBktSr169lJ6eLpPJZEAyx5oqgy0Wi9avX69rr73WgGQAmkMhDAAAAAAAAAAA4CTNlcFbtmxRcHCwAckca0kZPGTIEAOSAWgJCmEAAAAAAAAAAAAnoAwGYAQKYQAAAAAAAAAAgHZGGQzAKBTCAAAAAAAAAAAA7YgyGICRKIQBAAAAAAAAAADaCWUwAKNRCAMAAAAAAAAAALQDymAArsBlCuH8/Hw99thjio6Olp+fn4KDg9WnTx+99NJLKi8vNzoeAAAAAAAAAABAi1EGA3AVPkYHkKSMjAyNHj1axcXFtsfKy8uVnZ2t7OxsLVu2TBs3blRkZKSBKQEAAAAAAAAAAJpHGQzAlRh+hfCePXuUlJSk4uJi+fv7669//as+/fRTffTRR5o4caIk6dtvv9Wdd96psrIyg9MCAAAAAAAAAAA4RhkMwNUYfoXw1KlTVV5eLh8fH33wwQfq27evbd2tt96qHj16aPr06crNzdX8+fP19NNPG5gWAAAAAAAAAADAPspgAK7I0CuEs7OztW3bNknSAw88cEYZXG/atGmKjo6WJC1cuFCnT592ZkQAAAAAAAAAAIBmUQYDcFWGFsLr1q2zLY8bN87uGC8vL40ZM0aSVFRUZCuQAQAAAAAAAAAAXAFlMABXZmghnJmZKUny8/NT7969HY4bMGCAbXnHjh3tngsAAAAAAAAAAKAlKIMBuDpTbW1trVE/PDQ0VIWFhbruuuu0e/duh+OKiopsT5aJiYlavXp1k9v19vZWTU3NGY+ZTKZzzgsYxd5hyj4Nd8d+DU/DPg1PxH4NT8M+DU/Efg1Pwz4NT3T2fm0ymRQUFGRMGLS5mpoalZaWNuokJNcug4OCglRWVmZ3vb+/vzp06ODkVGhzJ05IDZ9/TCaJ5x6PcuLEiTP+xnh5eam6utruWB9nhTpbZWWlCgsLJUndunVrcmznzp3l5+enkydP6tChQ3bHnDp1SqdOnZJk/4Wjgb030C7Yp+GJ2K/hadin4YnYr+Fp2Kfhidiv4WnYp+FpamtrVVRUZHQMtDNXLoPHjx/vsAyW1OQ6uLHaWonnHo/W1Gsmw6aMLi0ttS37+/s3O97Pz0+S4yeiOXPmKDAwUIGBgbxIBAAAAAAAAAAAhvD29nbpMtjeNNEAPJthhXBlZaVt2Ww2NzveYrFIkioqKuyunzFjhoqLi1VcXMw0MgAAAAAAAAAAwBABAQGUwQBcimFTRvv6+tqWrVZrs+Prp4Pu2LGj3fUWi8VWGnt7e6u6ulq1tbUymUwymUwKDAxsg9QAnKm2tlYnTpxQUFAQH/QA3BTHMeD+OI4B98dxDLg/jmPA/XEce46m7hns7e2tgIAAde7c2YBkjjVXBnPPYM/Fc49nKy4uPmPWZG9vb4djDSuEAwICbMstmY/+5MmTklo2vfTp06dVUlKiwMBAnThxQp06dWp9UACGqT+ODx48yHEMuCmOY8D9cRwD7o/jGHB/HMeA++M49gwHDhzQwIEDVVxc3Gidq98z2F4ZbLFYtH79eg0ZMsSAZHAGnntQz7Apo319fRUSEiJJOnz4cJNji4qKbIVweHh4u2cDAAAAAAAAAACoV18G5+fnN1pHGQzA1RlWCEtSdHS0JCkvL09VVVUOx+Xm5jb6nuZYLBbNnDnTNo00APfDcQy4P45jwP1xHAPuj+MYcH8cx4D74zh2b5TBcFc896Ceqbbh5NJO9uSTT2rOnDmSpJ07d+rGG2+0O+7555/XjBkzJEmbN2/mSQoAAAAAAAAAALQ7ymAAnsDQK4Tj4+Nty8uXL7c7pqamxvakFRQUpLi4OGdEAwAAAAAAAAAA5zHKYACewtBCuE+fPurfv78k6Y033lBWVlajMfPmzVNOTo4k6ZFHHlGHDh2cmhEAAAAAAAAAAJxfKIMBeBJDp4yWpF27dik2NlYVFRXy9/fXk08+qbi4OFVUVCg1NVVLly6VJEVFRenzzz9XQECAkXEBAAAAAAAAAIAHowwG4GkML4QlacOGDbrvvvtUUlJid31UVJQyMjLUvXt3JycDAAAAAAAAAADnC8pgAJ7IJQphSTp48KBefvllZWRk6PDhwzKbzerevbsSExM1efJkXXDBBUZHBAAAAAAAAAAAHooyGICnMvQewg1deumlmj9/vr799ludPHlSRUVFys7O1vTp039VGZyfn6/HHntM0dHR8vPzU3BwsPr06aOXXnpJ5eXl7fgbADgXBQUFSk9P19NPP62hQ4cqJCREJpNJJpNJycnJRscD0AJffPGFnnvuOQ0dOlTh4eGyWCzy9/dXVFSUkpOTlZmZaXREAE0oKSlRamqqpk2bpgEDBqh79+4KDAyU2WxWWFiYBg4cqBdffFHHjx83OiqAVpg+fbrt9bXJZNK2bduMjgTAgYbHalP/Bg4caHRUAC1QWFioF198UbGxsbroootksVjUtWtX3XjjjXr88ceVlZVldET8D2UwjNKW5+MHDhzQo48+qp49eyogIEB+fn6KiorSww8/rK+//rrFmdLT0zVq1Ch169ZNFotFISEh6tu3r+bNm0fX5qZc5grhtpCRkaHRo0eruLjY7vorrrhCGzduVGRkpJOTAWiOyWRyuG7s2LFasWKF88IA+NUGDBigTz75pNlx999/v5YtWyaz2eyEVAB+jS1btmjw4MHNjgsJCdE777yj2267zQmpALSFPXv2KCYmRlVVVbbHtm7dSpkEuKimzo8bGjBgAB/uAFzcmjVrlJKS0mSJM2LECK1bt855oWAXZTCM1Fbn40uXLtWUKVNktVrtrjebzVq4cKFSUlIc/ozS0lKNHj1aGzZscDimR48eWr9+va688spmM8N1+BgdoK3s2bNHSUlJKi8vl7+/v2bMmKG4uDhVVFQoNTVVr7/+ur799lvdeeedys7Olr+/v9GRATgQHh6u6OhoffDBB0ZHAdBCR44ckSR17dpViYmJ6t+/vy655BJVV1crKytL8+bN05EjR/T222+rqqpK7733nsGJAdgTHh6uuLg49e7dW+Hh4br44otVU1Ojw4cPKy0tTWvXrlVhYaGGDx+u7OxsXXvttUZHBtCMmpoaTZw4UVVVVQoLC1NBQYHRkQC0UEpKih566CGH6/38/JyYBsCvtXLlSo0bN041NTUKCwtTSkqK+vXrp+DgYB09elTff/+9NmzYoA4dOhgd9bxHGQxXcK7n46mpqZo0aZIkKTAwUNOmTdOtt94qi8WiXbt26cUXX1ReXp4efvhhhYaGatSoUY0y1NbW6ne/+502bdokSerdu7ceffRRXXnllSotLVVGRoZeffVV7du3T0OHDtXnn3+uCy+8sP3/c9AmPOYK4bi4OG3btk0+Pj765JNP1Ldv3zPWz507V9OnT5ckzZ49W08//bQRMQE4MHPmTN1www264YYb1KVLF/3www+67LLLJHGFMOAOhg0bpjFjxmjkyJHy9vZutL6wsFCxsbH67rvvJEmffPKJ+vfv7+yYAJpQXV1t9/htaN26dUpISJAk3X333frHP/7hjGgAzsHChQttb+IkJCRozpw5krhCGHBl9VcIz5w5U7NmzTI2DIBWycnJUa9evXTq1Cn1799fGzZsUGBgoN2xVquVWbQMRBkMV3Cu5+Pl5eW67LLLVFBQIH9/f2VlZalnz55nfH9JSYn69eunvXv36qKLLlJeXl6jD5elpaUpMTFRkjR48GClp6c3en7asmWLbr/9dlVXV2vKlCl65ZVXWv17w7lc5h7C5yI7O9s2Rc4DDzzQqAyWpGnTpik6OlpS3Qnx6dOnnRkRQDNmz56tYcOGqUuXLkZHAdAK6enpSkpKcvjiNSQkRPPmzbN9nZaW5qxoAFqouZNPSYqPj7dNCdWSaeIBGOvQoUP6y1/+IklavHgxbzYDAOAkU6ZM0alTpxQSEqK1a9c6LIMl8ffZQJTBcBXnej6+adMm20xAjzzySKMyWJI6deqk+fPnS5KOHj1q9wKst956y7a8aNEiu89PgwYN0j333CNJeu2111RUVNRsdrgGjyiEG95jYdy4cXbHeHl5acyYMZKkoqIi7rECAICTNbwK6fvvvzcuCIBzUv8J4srKSoOTAGjOQw89pLKyMo0dO5argQEAcJLc3Fx99NFHkqTJkycrJCTE4ERwZN26dW5VBkt1Jfb777/f6HHK4PODo/Px7Oxs2/LQoUMdfv/AgQPl6+sryf7FGvXb6d69u3r06OFwO7fffrukuhkO1q9f38L0MJpHFMKZmZmS6g6G3r17Oxw3YMAA2/KOHTvaPRcAAPg/q9VqW/by8oiXIMB5JycnR7t375Yk2yeTAbim1atXKz09XcHBwZo7d67RcQAAOG+sWbPGtlw/9apUd5HSvn37dPz4cSNiwY6pU6c2urWkK5fBUl1Rt2XLljOuOqcMPj80dT7+yy+/2JabmoHTx8fHtm9/+umnqqqqsrud5mbxbLh++/btzYeHS/CId2NzcnIk1T0Z+vj4OBzX8CCp/x4AAOAcDV8gUiQB7qO8vFz79u3T/PnzFRcXp+rqakl101ABcE0nTpywHaMvvPCCQkNDDU4EoDXWrFmjK664Qh07dlRAQIB69OihsWPHauvWrUZHA9CEnTt3SpICAwMVHR2td999V9ddd52Cg4MVFRWlkJAQRUZGavbs2SorKzM47fnNZDJp1qxZtlLY1cvgejExMbZSmDLYs7X0fLzhvYCLi4sdbq+2tlYlJSWS6i7cyMvLs7udprZx9vpvvvmmBb8JXIHj9tRNVFZWqrCwUJLUrVu3Jsd27txZfn5+OnnypA4dOuSMeAAAQFJNTY2ef/5529dJSUkGpgHQnBUrVji8FYskPfbYYxo9erQTEwH4NaZPn66jR4/q5ptv1gMPPGB0HACtdPYbrHl5ecrLy9PKlSsVHx+vFStWNHlfUgDGqD92IyIiNGXKFC1atKjRmAMHDmjWrFlKS0vT5s2b1bVrV2fHxP/Ul8IXX3yxkpKSXL4MrldfCv/yyy+UwR6mNefj0dHRtuXt27c7nEl3165dZ3wQJT8//4yLNqKjo5WVlaWcnBz9/PPPDj9Y2vAexvamXYdrcvsrhEtLS23L/v7+zY6v/4QDn74CAMB5FixYoM8++0ySlJCQoJiYGIMTAWiN66+/Xjt37tTcuXNlMpmMjgPAjh07dmjZsmXy8fHRkiVLOFYBN3TBBRfonnvu0euvv67MzEzt2rVLH3zwgZ566ildeOGFkuruezlixAidPn3a4LQAzlY/5Wpubq4WLVqkoKAgLVmyRAUFBaqsrFR2drbtHp9fffWVEhMTVVNTY2Tk857JZNKDDz7oNmVwvZiYGMrg80hT5+N33HGHOnToIEmaP3++7SLKhmpqavTUU0+d8VjDfk2SRowYIUmqrq7Wn//8Z7s59u3bp+XLlzvcBlyX2xfCDW+ebTabmx1vsVgkSRUVFe2WCQAA/N/27dv1pz/9SZIUFhamxYsXG5wIQHPi4+O1d+9e7d27V5999plWrVqlhIQE7d69W6NHj1Z6errREQHYYbVa9Yc//EG1tbV69NFHdc011xgdCUArHDlyRKtWrdKECRPUr18/XX/99Ro8eLCeffZZff311+rVq5ekutfZvLYGXM/JkyclSadOnZK3t7c2bdqkSZMmKTQ0VBaLRTExMUpPT7eVwp9++qnWrl1rZGQALqQ15+PdunVTSkqKpLrXEbGxsXr//fdVUlKiyspK7dy5U3fccYf++c9/ntGjnd2TpaSk2GbiXbp0qe6//359+eWXslqtOn78uN5++23dcsstKisrs22Hrs19uH0h7Ovra1u2Wq3Njj916pQkqWPHju2WCQAA1Pn666+VkJCgqqoqWSwWrV69Wl26dDE6FoBmBAUFqWfPnurZs6duuOEG3XPPPVq7dq1Wrlyp/fv3a8SIEVqxYoXRMQGc5bnnnlNOTo4uueQSzZw50+g4AFopKCjI4bouXbooLS3N9ibsq6++6qRUAFqq4fvViYmJuummmxqN8fLy0ty5c21fr1q1yinZALi+1p6Pz507V3fddZck6bvvvlN8fLwCAwPVsWNH9e3bV5s3b1ZkZKT++Mc/2r4nICDgjG106tRJ69ev10UXXSRJeuedd3TdddfJYrEoJCREY8aM0dGjR/XQQw8pKirK7jbguty+EG64s7VkGuj6T2i1ZHppAADQegcOHNCQIUNUVFQkb29vrVq1SgMGDDA6FoBzcP/999umtJs8ebKKioqMjgTgf3JzczVnzhxJdQVR/e2SAHieyMhIDR48WFLdfYV//PFHgxMBaKjh+9X1VwHbc/XVV+s3v/mNJCk7O7vdcwFwb82dj5vNZr3//vtavny5evfuLS+v/9d/QUFBmjJlir744gvV1tbaHu/cuXOjn9OrVy/t2bNHU6dO1cUXX3zGumuuuUYrV67UokWL9NNPPzncBlyTj9EBzpWvr69CQkJUWFiow4cPNzm2qKjIVgiHh4c7Ix4AAOelH3/8UYMGDdKPP/4ok8mkN998UwkJCUbHAtAGRowYodWrV+vkyZPatGmTfv/73xsdCYCkBQsWyGq1KjIyUuXl5UpNTW005quvvrItf/zxxzp69Kgk6a677qJABtzMVVddpYyMDEl1U0N27drV4EQA6oWHh9v+xtZPvdrU2CNHjqigoMAZ0QC4uebOx00mk5KTk5WcnKyysjIdO3ZMZrNZXbt2lbe3tyTpyy+/tI2/6qqr7P6csLAwLViwQAsWLFBBQYFOnDih0NBQW/n7008/6fjx401uA67H7QthSYqOjlZmZqby8vJUVVUlHx/7v1Zubu4Z3wMAANpeYWGhBg8erP3790uqu0ppzJgxBqcC0FZCQ0NtywcPHjQwCYCG6m+PtH//ft17773Njn/mmWdsywcOHKAQBtxMw6t7ALiWq6++2nbFb3V1dZNj69c7ej8bABr6Nefj/v7+jWbKtVqt+uyzzyTVzTgSEhLS7M8MCwtTWFjYGY9lZmbalm+88cZmtwHX4PZTRktSv379JNVNB/2f//zH4bjt27fblmNjY9s9FwAA55vi4mLddttt+uabbyRJzz//vB5++GGDUwFoS0eOHLEtcxsWAACMUf96WxJXBwMu5pZbbrEtf//9902Orf8gdf3U0QDQlHM9H9+4caOKi4slSUlJSa3O8d5779mWExMTW70dOJdHFMLx8fG25eXLl9sdU1NTo5UrV0qqmy89Li7OGdEAADhvlJeX684779QXX3whSXrqqaf0xBNPGJwKQFtbs2aNbfmaa64xMAmAhlasWKHa2tom/82cOdM2fuvWrbbHIyIijAsO4Ffbv3+/PvzwQ0l1V/dQJAGuZfjw4erQoYMkae3atQ7Hbd++3Tblav/+/Z2SDYB7O5fz8aqqKtv5QIcOHTRx4sRWZfj3v/+t9evXS5J++9vf6sorr2zVduB8HlEI9+nTx/ZH84033lBWVlajMfPmzVNOTo4k6ZFHHrH9UQYAAOfOarUqISFB//rXvyTV/a199tlnDU4F4NdYsWKFKisrmxyzYMECbdy4UZIUERFhm6kHAAC0jQ0bNqiqqsrh+mPHjmnUqFE6ffq0JDEbD+CCLrzwQk2YMEGS9OGHHyo1NbXRmNLSUk2dOtX29aRJk5wVD4ALaovz8cLCQpWXl9v9XqvVqvHjx9vuH/zEE08oMjLS7tj8/HyHGfLy8jRq1CjV1tbKbDbrlVdeaTIzXIup1kNuOrJr1y7FxsaqoqJC/v7+evLJJxUXF6eKigqlpqZq6dKlkqSoqCh9/vnnCggIMDgxgIZ27NihvLw829eFhYV6/PHHJdVN8V7/QrpecnKyM+MBaMbIkSNtn3y+9dZbtXDhQplMJofjzWazoqKinBUPQAtERESotLRUI0eOVL9+/XT55ZfL399fpaWl2rt3r959913bhz7MZrMyMjI0aNAgg1MD+DVmzZql2bNnS6q7QnjgwIHGBgLQSEREhE6fPq2RI0eqb9++ioiIUMeOHVVYWKht27ZpyZIltisK+/Xrpy1btshisRicGsDZfv75Z8XExCg/P18+Pj568MEHdffdd6tTp07au3evXnjhBeXm5kqSUlJS9Pe//93gxACM1Bbn42lpaZo4caJGjx6tQYMG6ZJLLlF5ebl27dqlJUuW2G43MWTIEG3YsEFms9lulmHDhungwYMaM2aMYmJiFBQUpIKCAm3evFmvvfaaysvLZTKZ9MYbb2jcuHHt+x+DNuUxhbBU9ynK++67TyUlJXbXR0VFKSMjQ927d3dyMgDNSU5O1ltvvdXi8R701AV4hKbKX3suvfRS/fDDD+0TBkCrRERE6ODBg82O69atm958800NHjzYCakAtCUKYcD1tfTv8ciRI7Vs2TIFBQW1fygArZKTk6Phw4efcQHE2caPH68lS5YwmyVwnmuL8/G0tLRm7+ebnJysxYsXy9fX1+GYYcOGKSMjw+H64OBg/e1vf9O9997bbF64Fh+jA7Slu+66S19++aVefvllZWRk6PDhwzKbzerevbsSExM1efJkXXDBBUbHBAAAAFzORx99pC1btmjr1q3KycnRsWPHdPz4cfn6+qpLly66/vrrNWzYMCUlJfGaGgCAdvLWW29p+/btysrK0v79+1VYWKiSkhL5+/srPDxcN998s8aOHau+ffsaHRVAM6Kjo7V7924tXrxYaWlp2rdvn8rKyhQWFqbY2FhNmjRJcXFxRscE4ALa4ny8f//+mjt3rj7++GPl5ubq2LFj8vLyUteuXRUXF6fk5GTddNNNzWaZMWOGrrjiCmVmZurQoUM6fvy4goKCdPnll2v48OGaMGGCQkND2/q/AE7gUVcIAwAAAAAAAAAAAAD+z8voAAAAAAAAAAAAAACA9kEhDAAAAAAAAAAAAAAeikIYAAAAAAAAAAAAADwUhTAAAAAAAAAAAAAAeCgKYQAAAAAAAAAAAADwUBTCAAAAAAAAAAAAAOChKIQBAAAAAAAAAAAAwENRCAMAAAAAAAAAAACAh6IQBgAAAAAAAAAAAAAPRSEMAAAAAAAAAAAAAB6KQhgAAAAAAAAAAAAAPBSFMAAAAAAAAAAAAAB4KAphAAAAAAAAAAAAAPBQFMIAAAAAAAAAAAAA4KH+C4Y5oGnzsi9mAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 2400x1200 with 3 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# depict random result picture: transposon similar\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.gridspec as gridspec\n",
    "from brokenaxes import brokenaxes\n",
    "\n",
    "exp_result = 399\n",
    "control_min = min(random_transposon_similar_EPI_num_list)\n",
    "control_max = max(random_transposon_similar_EPI_num_list)\n",
    "\n",
    "fig = plt.figure(figsize=(24,12), linewidth=10)\n",
    "\n",
    "ax = brokenaxes(xlims=((control_min, control_max+0.5), (exp_result-0.5, exp_result+0.5)),despine=False, wspace=0.05,d=0.01)\n",
    "\n",
    "plt.rc('axes', linewidth=3.5)\n",
    "plt.rcParams['font.size'] = 20  # 14pt font size  \n",
    "\n",
    "\n",
    "ax.hist(random_transposon_similar_EPI_num_list, bins=8, edgecolor='black', color='white', linewidth=3.5)\n",
    "ax.axvline(x=exp_result, color='red', linewidth=3.5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f254e043-338f-403f-a4c0-5e220485cb67",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:chipseq]",
   "language": "python",
   "name": "conda-env-chipseq-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
